---
title: "genes_in_QTL"
author: "Enoch Ng'oma"
date: "10/18/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
```

# Function

```{r}
DEunderPeak <- function(DEgenes, arm, upper, lower)
{
  #DEgenes is a data frame with differentially expressed genes including chromosome (chr)
  # gene start (startp) and gene stop (stopp)
  
  #upper is the upper bound of the QTL CI in release 6 coordinates
  #lower is the lower bound of the QTL CI in release 6 coordinates
  
  lw <- (which(DEgenes$chr==arm & 
                 ((DEgenes$startp <= upper & DEgenes$stopp >= lower))))  
  
  return(DEgenes[lw,])
}
```

# Data

```{r}

load(file="../../processed/DESEQ/all_fc_dat.rda")

# gene map table
gmtable <- read.csv("../../processed/DESEQ/DEG_QTL/gene_map_table.csv", 
                 stringsAsFactors = FALSE)
which(gmtable$gname == "d")
gmtable$gname[gmtable$gname == "d"] <- "CG42840" 

# treatment effect DEGs
DEGs_treat <- read.csv("../../processed/DESEQ/DEGs_lrt.treatment_0.05.csv")
names(DEGs_treat)[1]<-"FBgn"
DEGs_treat$FBgn <- as.character(DEGs_treat$FBgn)

#Interaction effect DEGs
DEGs_int <- read.csv("../../processed/DESEQ/DEGs_lrt.int_0.05.csv")
names(DEGs_int)[1]<-"FBgn"
DEGs_int$FBgn <- as.character(DEGs_int$FBgn)

# merge
merge_treat <- inner_join(DEGs_treat,gmtable, by="FBgn")
merge_int <- inner_join(DEGs_int,gmtable, by="FBgn")

# 3 genes drop out of DEGs_treat
which(!(DEGs_treat$FBgn %in% gmtable$FBgn))
DEGs_treat[c(214,1257,1668),]

# These genes are ignored for now:
# 214  FBgn0030737 maps to FBgn0286508 Had1, CG9914
# 1257 FBgn0028336 maps to FBgn0286222 Fum1, CG4094
# 1668 FBgn0037744 maps to FBgn0286506 Mpi, CG8417
# all these have alternative mapping to transcription start sites FBsf... numbers

# 3 genes drop out of DEGs_int

```

# DEGs in iis QTL peaks for the effect of treatment
## Focus on peaks for difference between treatments

```{r}
# trans-eQTL
# to do: extract BCIs programmatically from iis 

load("../../processed/DESEQ/DEG_QTL/all_iisPeak.rda")

#peak pos 9412495
qtl_degs <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 9472495, lower =7712495)

qtl_degs_int <- DEunderPeak(merge_int,  arm="2R", 
                   upper = 9472495, lower =7712495)

qtl_degs$effect <- "treat"
qtl_degs_int$effect <- "int"
all_qtl_degs <- rbind(qtl_degs,qtl_degs_int)

#module membership

load(file="../../processed/DESEQ/Coexpression/Obs_modMembership.csv")
drop.g <- read.csv(file="../../processed/DESEQ/Coexpression/Resamp_droppedGenes.csv",header=TRUE, stringsAsFactors = FALSE)
obs.dat[obs.dat$FBgn %in% drop.g$FBgn, "ModID"] <- "grey"

all_qtl_degs <- merge(all_qtl_degs, obs.dat, by="FBgn")

#gene list from IIS

tt <- read.table(file="../../processed/DESEQ/DEG_QTL/iis_genelist_clean.txt", header=TRUE, stringsAsFactors = FALSE, sep="\t")
tt[tt$id=="tribbles",'id'] <- "trbl"
tt[tt$id=="PP2A-B'",'id'] <- "wrd"
tt[tt$id=="eIF-4B",'id'] <- "eIF4B"

tt1 <- merge(tt, gmtable[,c("gname","FBgn")], by.x="id", by.y="gname")

tt1<-merge(tt1, obs.dat, by="FBgn")

write.csv(all_qtl_degs, file = "../../processed/DESEQ/DEG_QTL/all_diffExpr_underQTL.csv")

```

## Visualization







# Gene summaries

```{r}
# view few lines in the terminal: head -n 3 automated_gene_summaries.tsv | tail -n 3


gsum <- read_lines(file = "../../processed/DESEQ/DEG_QTL/automated_gene_summaries.tsv",
                   skip = 1)%>%
  str_split(pattern = "\t", simplify = TRUE) %>% 
  as_tibble() 
gsum <- gsum[,c(1,2)]
colnames(gsum) <- c('FBgn', 'summary')


fullset <- inner_join(distt, gsum, by="FBgn")

fullset[grep('life', fullset$summary),]

fullset[grep('transcription factor', fullset$summary),]


ahcy <- grep("Ahcy", gsum$summary)
gsum[ahcy[1],'summary']
pull(gsum[ahcy[1],'summary'])

CG40486 <- grep("CG40486", gsum$summary)
gsum[CG40486[1],'summary']
pull(gsum[CG40486[1],'summary'])

Cda4 <- grep("Cda4", gsum$summary)
gsum[Cda4[1],'summary']
pull(gsum[Cda4[1],'summary'])

# Get DEGs <100 KB to QTL peak
near_genes <- subset(fullset, fullset$qdist <= 1e5)
unique(near_genes$gname)

write.csv(unique(near_genes$gname), 
          file = "../../processed/DESEQ/DEG_QTL/unique_nearGenes.csv")

# Some descriptives
max(fullset[,"log2FC"])
min(fullset[,"log2FC"])
fullset$gname
unique(fullset$gname)

write.csv(unique(fullset$gname), 
          file = "../../processed/DESEQ/DEG_QTL/unique_geneList.csv")

```


# Gene association table with GO IDs

```{r}
GO_fly <- read_lines(file = "../../processed/gene_association.fb",
                                       skip = 5) %>% 
  str_split(pattern = "\t", simplify = TRUE) %>% 
  as_tibble()

colnames(GO_fly) <- c("DB","gname","g_symbol",
                      "Qualifier","GOid","reference",
                      "Evidence","sourceE","Aspect","full_gname",
                      "altname","type","taxon","date","sourceA")
GO_fly[,c(16:17)] <- NULL
GO_fly <- select(GO_fly, gname, GOid)
colnames(GO_fly)[1] <- "FBgn"

allDEGs.GO <- inner_join(all_qtl_degs,GO_fly, by="FBgn")

```
