---
title: "genes_in_QTL"
author: "Enoch Ng'oma"
date: "10/18/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
```

# Function

```{r}
DEunderPeak <- function(DEgenes, arm, upper, lower)
{
  #DEgenes is a data frame with differentially expressed genes including chromosome (chr)
  # gene start (startp) and gene stop (stopp)
  
  #upper is the upper bound of the QTL CI in release 6 coordinates
  #lower is the lower bound of the QTL CI in release 6 coordinates
  
  lw <- (which(DEgenes$chr==arm & 
                 ((DEgenes$startp <= upper & DEgenes$stopp >= lower))))  
  
  return(DEgenes[lw,])
}
```

# Data

```{r}
# gene map table
gmtable <- read.csv("../../processed/DESEQ/DEG_QTL/gene_map_table.csv", 
                 stringsAsFactors = FALSE)
which(gmtable$gname == "d")
gmtable$gname[gmtable$gname == "d"] <- "CG42840" 


# treatment effect DEGs
DEGs_treat <- read.csv("../../processed/DESEQ/DEGs_lrt.treatment_0.05.csv")
names(DEGs_treat)[1]<-"FBgn"
names(DEGs_treat)[3] <- "log2FC"
DEGs_treat$FBgn <- as.character(DEGs_treat$FBgn)

#Interaction effect DEGs
DEGs_int <- read.csv("../../processed/DESEQ/DEGs_lrt.int_0.05.csv")
names(DEGs_int)[1]<-"FBgn"
names(DEGs_int)[3] <- "log2FC"
DEGs_int$FBgn <- as.character(DEGs_int$FBgn)

# merge
merge_treat <- inner_join(DEGs_treat,gmtable, by="FBgn")
merge_int <- inner_join(DEGs_int,gmtable, by="FBgn")

# 3 genes drop out of DEGs_treat
which(!(DEGs_treat$FBgn %in% gmtable$FBgn))
DEGs_treat[c(214,1257,1668),]

# These genes are ignored for now:
# 214  FBgn0030737 maps to FBgn0286508 Had1, CG9914
# 1257 FBgn0028336 maps to FBgn0286222 Fum1, CG4094
# 1668 FBgn0037744 maps to FBgn0286506 Mpi, CG8417
# all these have alternative mapping to transcription start sites FBsf... numbers

# 3 genes drop out of DEGs_int

```

# DEGs in iis QTL peaks for the effect of treatment

```{r}
# trans-eQTL
# to do: extract BCIs programmatically from iis 

load("../../processed/DESEQ/DEG_QTL/all_iisPeak.rda")



step_2L <- DEunderPeak(merge_treat,  arm="X", 
                   upper = allPeak[allPeak$id=='step','Up'], lower =allPeak[allPeak$id=='step','Lp'])

Ilp5_3L <- DEunderPeak(merge_treat,  arm="X", 
                   upper = allPeak[allPeak$id=='Ilp5','Up'], lower = allPeak[allPeak$id=='Ilp5','Lp'])

# Lifespan QTL

#from IIS_mapping_DFA_byTreat.R
LinDiscrDR_2R <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 9632495, lower =6122495)
#peak pos 9412495
LinDiscrDRC_2R <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 9472495, lower =7712495)
#peak pos 7762495

#from lifespan.R
MedLifeDR_3L <-   DEunderPeak(merge_treat,  arm="3L", 
                   upper = 3840000, lower =3230000)
#peak 3650000
```

# Table of all DEGs under peaks

```{r}
# add identifier colum to each list
step_2L$QTL <- "step"
Ilp5_3L$QTL <- "Ilp5"
LinDiscrDR_2R$QTL <- "LinDiscrA"
LinDiscrDRC_2R$QTL <- "LinDiscrB"
MedLifeDR_3L$QTL <- "MedLife" 


qtl_degs <- rbind(step_2L,Ilp5_3L,
                  LinDiscrDR_2R,LinDiscrDRC_2R,MedLifeDR_3L)
write.csv(qtl_degs, file = "../../processed/DESEQ/DEG_QTL/diffExpr_trt_underQTL.csv")
```

# DEGs in iis QTL peaks for the interaction

```{r}
# trans-eQTL
# to do: extract BCIs programmatically from iis 
step_2L.int <- DEunderPeak(merge_int,  arm="X", 
                   upper = allPeak[allPeak$id=='step','Up'], lower =allPeak[allPeak$id=='step','Lp'])
Ilp5_3L.int <- DEunderPeak(merge_int,  arm="X", 
                   upper = allPeak[allPeak$id=='Ilp5','Up'], lower =allPeak[allPeak$id=='Ilp5','Lp'])

# Lifespan QTL
LinDiscrDR_2R.int <- DEunderPeak(merge_int,  arm="2R", 
                   upper = 9632495, lower =6122495)
LinDiscrDRC_2R.int <- DEunderPeak(merge_int,  arm="2R", 
                   upper = 9472495, lower =7712495)
MedLifeDR_3L.int <-   DEunderPeak(merge_int,  arm="3L", 
                   upper = 3840000, lower =3230000)
```

# Table of interaction DEGs under peaks

```{r}
# add identifier colum to each list
step_2L.int$QTL <- "step"
#Ilp5_3L.int$QTL <- "Ilp5" # no genes
LinDiscrDR_2R.int$QTL <- "LinDiscrA"
LinDiscrDRC_2R.int$QTL <- "LinDiscrB"
MedLifeDR_3L.int$QTL <- "MedLife" 


qtl_degs_int <- rbind(step_2L.int,
                    LinDiscrDR_2R.int,LinDiscrDRC_2R.int,
                    MedLifeDR_3L.int)

qtl_degs_int <- qtl_degs_int[order(qtl_degs_int$QTL), ]

write.csv(qtl_degs_int, file = "../../processed/DESEQ/DEG_QTL/diffExpr_int_underQTL.csv")

```

# Interaction and treatment tables together

```{r}
colnames(qtl_degs)
colnames(qtl_degs_int)

qtl_degs$effect <- "treat"
qtl_degs_int$effect <- "int"
all_qtl_degs <- rbind(qtl_degs,qtl_degs_int)
all.qtl.degs <- dplyr::select(all_qtl_degs,-c(lfcSE,v3,cyt,stat,pvalue,spp))

write.csv(all_qtl_degs, file = "../../processed/DESEQ/DEG_QTL/all_diffExpr_underQTL.csv")

```

# Calculate gene distance to center of peak

```{r}
#R6pos = peak position in v6

transQTL <- subset(allPeak, allPeak$id==c('Ilp5','step'))

transQTL <- select(transQTL, id, R6chr,R6pos)
colnames(transQTL)[1] <- "QTL"
transQTL$R6pos <- as.numeric(transQTL$R6pos)

addQTL <- data.frame('QTL'=c("LinDiscrA","LinDiscrB","MedLife"),
                     'R6chr'=c("2R","2R","3L"),
                     'R6pos'=c(9412495, 7762495, 3650000),
                     stringsAsFactors=FALSE)

addQTL <- rbind(transQTL, addQTL)

# Find distance of DEGs to qtl position
distt <- inner_join(all_qtl_degs,addQTL, by="QTL")

distt$qdist <- abs(distt$startp-distt$R6pos)

# Save for next script
write.csv(distt, file = "../../processed/DESEQ/DEG_QTL/distt.csv")
```

Summary of IIS QTL interval genes: 
215 DEGs identified under BCIs that are in both treatment effect DEGs and interaction effect DEGs (LRT, qval<0.05) (all_diffExpr_underQTL.csv). 
This set had 139 unique genes (unique_geneList.csv). 
12 of these are located within 100 Kb of eQTL peak markers: Ahcy, CG40486, Non1, prel, ced-6, CG33774, Wnt2, CG1882, Dgat2, mus205, CG12822, CG10357. Converted coordinates to v6


# Gene summaries

```{r}
# view few lines in the terminal: head -n 3 automated_gene_summaries.tsv | tail -n 3


gsum <- read_lines(file = "../../../../automated_gene_summaries.tsv",
                   skip = 1)%>%
  str_split(pattern = "\t", simplify = TRUE) %>% 
  as_tibble() 
gsum <- gsum[,c(1,2)]
colnames(gsum) <- c('FBgn', 'summary')


fullset <- inner_join(distt, gsum, by="FBgn")

fullset[grep('life', fullset$summary),]

fullset[grep('transcription factor', fullset$summary),]


ahcy <- grep("Ahcy", gsum$summary)
gsum[ahcy[1],'summary']
pull(gsum[ahcy[1],'summary'])

CG40486 <- grep("CG40486", gsum$summary)
gsum[CG40486[1],'summary']
pull(gsum[CG40486[1],'summary'])

Cda4 <- grep("Cda4", gsum$summary)
gsum[Cda4[1],'summary']
pull(gsum[Cda4[1],'summary'])

# Get DEGs <100 KB to QTL peak
near_genes <- subset(fullset, fullset$qdist <= 1e5)
unique(near_genes$gname)

write.csv(unique(near_genes$gname), 
          file = "../../processed/DESEQ/DEG_QTL/unique_nearGenes.csv")

# Some descriptives
max(fullset[,"log2FC"])
min(fullset[,"log2FC"])
fullset$gname
unique(fullset$gname)

write.csv(unique(fullset$gname), 
          file = "../../processed/DESEQ/DEG_QTL/unique_geneList.csv")

```


# Look for slif in DEG lists

```{r}
k <- gmtable[which(gmtable$gname=="slif"),]
kk <- which(DEGs_treat$FBgn %in% k$FBgn)
ki <- which(DEGs_int$FBgn %in% k$FBgn)
# slif not among DEGs!
```

# Gene association table with GO IDs

```{r}
GO_fly <- read_lines(file = "../../processed/gene_association.fb",
                                       skip = 5) %>% 
  str_split(pattern = "\t", simplify = TRUE) %>% 
  as_tibble()

colnames(GO_fly) <- c("DB","gname","g_symbol",
                      "Qualifier","GOid","reference",
                      "Evidence","sourceE","Aspect","full_gname",
                      "altname","type","taxon","date","sourceA")
GO_fly[,c(16:17)] <- NULL
GO_fly <- select(GO_fly, gname, GOid)
colnames(GO_fly)[1] <- "FBgn"

allDEGs.GO <- inner_join(all_qtl_degs,GO_fly, by="FBgn")

```

# To Delete
```{r}

# iis data
iis <- read.table("../../processed/DESEQ/DEG_QTL/iis_table1.txt",
                  header = TRUE, sep = "\t", stringsAsFactors = FALSE)

iis$chr <- str_split(iis$gene_location, ":",simplify=TRUE)[,1]
iis$startp <- str_split(iis$bci, "-", simplify = TRUE)[,1]
iis$stopp <- str_split(iis$bci, "-", simplify = TRUE)[,2]

iis$startp <- as.numeric(iis$startp)

```

