---
title: "genes_in_QTL"
author: "Enoch Ng'oma"
date: "10/18/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
```

# Function

```{r}
DEunderPeak <- function(DEgenes, arm, upper, lower)
{
  #DEgenes is a data frame with differentially expressed genes including chromosome (chr)
  # gene start (startp) and gene stop (stopp)
  
  #upper is the upper bound of the QTL CI in release 6 coordinates
  #lower is the lower bound of the QTL CI in release 6 coordinates
  
  lw <- (which(DEgenes$chr==arm & 
                 ((DEgenes$startp <= upper & DEgenes$stopp >= lower))))  
  
  return(DEgenes[lw,])
}
```

# Data

```{r}
# gene map table
gmtable <- read.csv("../../processed/DESEQ/DEG_QTL/gene_map_table.csv", 
                 stringsAsFactors = FALSE)
which(gmtable$gname == "d")
gmtable$gname[gmtable$gname == "d"] <- "CG42840" 


# treatment effect DEGs
DEGs_treat <- read.csv("../../processed/DESEQ/DEGs_lrt.treatment_0.05.csv")
names(DEGs_treat)[1]<-"FBgn"
names(DEGs_treat)[3] <- "log2FC"
DEGs_treat$FBgn <- as.character(DEGs_treat$FBgn)


# merge
merge_treat <- inner_join(DEGs_treat,gmtable, by="FBgn")

# # 3 genes drop out of DEGs_treat
which(!(DEGs_treat$FBgn %in% gmtable$FBgn))
DEGs_treat[c(214,1257,1668),]

# These genes are ignored for now:
# 214  FBgn0030737 maps to FBgn0286508 Had1, CG9914
# 1257 FBgn0028336 maps to FBgn0286222 Fum1, CG4094
# 1668 FBgn0037744 maps to FBgn0286506 Mpi, CG8417
# all these have alternative mapping to transcription start sites FBsf... numbers

# iis data
iis <- read.table("../../processed/DESEQ/DEG_QTL/iis_table1.txt",
                  header = TRUE, sep = "\t", stringsAsFactors = FALSE)

iis$chr <- str_split(iis$gene_location, ":",simplify=TRUE)[,1]
iis$startp <- str_split(iis$bci, "-", simplify = TRUE)[,1]
iis$stopp <- str_split(iis$bci, "-", simplify = TRUE)[,2]

iis$startp <- as.numeric(iis$startp)

# For now, exclude Slif (cross-centromere)
which(iis$trait=="slif")
iis <- iis[-19, ]

```

# Filter merge_treat using iis

```{r}
#qtl_iis <- merge_treat %>%
      #filter(FBgn %in% iis$FBgn)
```

# DEGs in iis QTL peaks

```{r}
# trans-eQTL
# to do: extract BCIs programmatically from iis 
Lip4_2L <- DEunderPeak(merge_treat,  arm="2L", 
                  upper = 10710000, lower = 10250000)
step_2L <- DEunderPeak(merge_treat,  arm="X", 
                   upper = 15470000, lower =15390000)
PRAS40_2R <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 14780000, lower =13980000)
PRAS40_2Rb <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 14610000, lower =14270000)
Ilp5_3L <- DEunderPeak(merge_treat,  arm="X", 
                   upper = 22980000, lower =22600000)

# Lifespan QTL
LinDiscrDR_2R <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 9630000, lower =6120000)
LinDiscrDRC_2R <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 9470000, lower =7710000)
#MedLifeDRC_3R <- # BCI cross-centromere
MedLifeDR_3L <-   DEunderPeak(merge_treat,  arm="3L", 
                   upper = 3840000, lower =3230000)
```

# The cross-centromeric QTL (Slif)

```{r}
# get genes on the 3L side of QTL
c <- subset(gmtable, gmtable$chr=="3L")
# In FlyBase covert 3L:3L:20,930,000 from v5 to v6 (http://flybase.org/convert/coordinates) 3L:20936900
cc <- subset(c, c$startp>=20936900) # left BCI slif in v6
cc <- inner_join(cc, DEGs_treat, by="FBgn")

# search DEGs on the 3L side
cx <- max(cc$stopp, na.rm = FALSE)
Slif_3L <- DEunderPeak(cc,  arm="3L", 
                  upper = max(cx, na.rm = FALSE), 
                  lower = 20936900)
slif <- subset(Slif_3L, Slif_3L$gname=="slif")

# get genes on the 3R side of QTL
d <- subset(gmtable, gmtable$chr=="3R")
# In FlyBase covert 3R:3,460,000 from v5 to v6 3R:7634278
dd <- subset(d,d$stopp<=7634278)
dd <- inner_join(dd, DEGs_treat, by="FBgn")

# search DEGs on the 3R side
Slif_3R <- DEunderPeak(dd,  arm="3R", 
                  upper = max(dd$stopp, na.rm = FALSE), 
                  lower = min(dd$startp, na.rm = FALSE))
```

# Table of all DEGs under peaks

```{r}
# add identifier colum to each list
Lip4_2L$QTL <- "Lip4"
step_2L$QTL <- "step"
PRAS40_2R$QTL <- "PRAS40a"
PRAS40_2Rb$QTL <- "PRAS40b"
Ilp5_3L$QTL <- "Ilp5"
LinDiscrDR_2R$QTL <- "LinDiscrA"
LinDiscrDRC_2R$QTL <- "LinDiscrB"
MedLifeDR_3L$QTL <- "MedLife" 
#MedLifeDRC_3R$QTL <- "MedlifeB"
Slif_3L$QTL <- "slifA"
Slif_3R$QTL <- "slifB"

qtl_degs <- rbind(Lip4_2L,step_2L,PRAS40_2R,PRAS40_2Rb,Ilp5_3L,
                  LinDiscrDR_2R,LinDiscrDRC_2R,MedLifeDR_3L,
                  Slif_3L,Slif_3R)

pp <- list(Lip4_2L,step_2L,PRAS40_2R,PRAS40_2Rb,Ilp5_3L,
                  LinDiscrDR_2R,LinDiscrDRC_2R,MedLifeDR_3L,
                  Slif_3L,Slif_3R)
str(pp)

qtl_degs <- qtl_degs[order(qtl_degs$QTL), ]
subset(qtl_degs,qtl_degs$QTL=="Lip4")

```








