---
title: "genes_in_QTL"
author: "Enoch Ng'oma"
date: "10/18/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
```

# Function

```{r}
DEunderPeak <- function(DEgenes, arm, upper, lower)
{
  #DEgenes is a data frame with differentially expressed genes including chromosome (chr)
  # gene start (startp) and gene stop (stopp)
  
  #upper is the upper bound of the QTL CI in release 6 coordinates
  #lower is the lower bound of the QTL CI in release 6 coordinates
  
  lw <- (which(DEgenes$chr==arm & 
                 ((DEgenes$startp <= upper & DEgenes$stopp >= lower))))  
  
  return(DEgenes[lw,])
}
```

# Data

```{r}
# gene map table
gmtable <- read.csv("../../processed/DESEQ/DEG_QTL/gene_map_table.csv", 
                 stringsAsFactors = FALSE)
which(gmtable$gname == "d")
gmtable$gname[gmtable$gname == "d"] <- "CG42840" 


# treatment effect DEGs
DEGs_treat <- read.csv("../../processed/DESEQ/DEGs_lrt.treatment_0.05.csv")
names(DEGs_treat)[1]<-"FBgn"
names(DEGs_treat)[3] <- "log2FC"
DEGs_treat$FBgn <- as.character(DEGs_treat$FBgn)

#Interaction effect DEGs
DEGs_int <- read.csv("../../processed/DESEQ/DEGs_lrt.int_0.05.csv")
names(DEGs_int)[1]<-"FBgn"
names(DEGs_int)[3] <- "log2FC"
DEGs_int$FBgn <- as.character(DEGs_int$FBgn)

# merge
merge_treat <- inner_join(DEGs_treat,gmtable, by="FBgn")
merge_int <- inner_join(DEGs_int,gmtable, by="FBgn")

# 3 genes drop out of DEGs_treat
which(!(DEGs_treat$FBgn %in% gmtable$FBgn))
DEGs_treat[c(214,1257,1668),]

# These genes are ignored for now:
# 214  FBgn0030737 maps to FBgn0286508 Had1, CG9914
# 1257 FBgn0028336 maps to FBgn0286222 Fum1, CG4094
# 1668 FBgn0037744 maps to FBgn0286506 Mpi, CG8417
# all these have alternative mapping to transcription start sites FBsf... numbers

# 3 genes drop out of DEGs_int

# iis data
iis <- read.table("../../processed/DESEQ/DEG_QTL/iis_table1.txt",
                  header = TRUE, sep = "\t", stringsAsFactors = FALSE)

iis$chr <- str_split(iis$gene_location, ":",simplify=TRUE)[,1]
iis$startp <- str_split(iis$bci, "-", simplify = TRUE)[,1]
iis$stopp <- str_split(iis$bci, "-", simplify = TRUE)[,2]

iis$startp <- as.numeric(iis$startp)

# For now, exclude Slif (cross-centromere)
which(iis$trait=="slif")
iis <- iis[-19, ]
```

# Filter merge_treat using iis

```{r}
#qtl_iis <- merge_treat %>%
      #filter(FBgn %in% iis$FBgn)
```

# DEGs in iis QTL peaks for the effect of treatment

```{r}
# trans-eQTL
# to do: extract BCIs programmatically from iis 
step_2L <- DEunderPeak(merge_treat,  arm="X", 
                   upper = 15470000, lower =15390000)
Ilp5_3L <- DEunderPeak(merge_treat,  arm="X", 
                   upper = 22980000, lower =22600000)

# Lifespan QTL
LinDiscrDR_2R <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 9630000, lower =6120000)
LinDiscrDRC_2R <- DEunderPeak(merge_treat,  arm="2R", 
                   upper = 9470000, lower =7710000)
MedLifeDR_3L <-   DEunderPeak(merge_treat,  arm="3L", 
                   upper = 3840000, lower =3230000)
```

# Table of all DEGs under peaks

```{r}
# add identifier colum to each list
step_2L$QTL <- "step"
Ilp5_3L$QTL <- "Ilp5"
LinDiscrDR_2R$QTL <- "LinDiscrA"
LinDiscrDRC_2R$QTL <- "LinDiscrB"
MedLifeDR_3L$QTL <- "MedLife" 


qtl_degs <- rbind(step_2L,Ilp5_3L,
                  LinDiscrDR_2R,LinDiscrDRC_2R,MedLifeDR_3L)

pp <- list(step_2L,Ilp5_3L,
                  LinDiscrDR_2R,LinDiscrDRC_2R,MedLifeDR_3L)
str(pp)

qtl_degs <- qtl_degs[order(qtl_degs$QTL), ]
subset(qtl_degs,qtl_degs$QTL=="slif")

```

# DEGs in iis QTL peaks for the interaction

```{r}
# trans-eQTL
# to do: extract BCIs programmatically from iis 
step_2L.int <- DEunderPeak(merge_int,  arm="X", 
                   upper = 15470000, lower =15390000)
Ilp5_3L.int <- DEunderPeak(merge_int,  arm="X", 
                   upper = 22980000, lower =22600000)

# Lifespan QTL
LinDiscrDR_2R.int <- DEunderPeak(merge_int,  arm="2R", 
                   upper = 9630000, lower =6120000)
LinDiscrDRC_2R.int <- DEunderPeak(merge_int,  arm="2R", 
                   upper = 9470000, lower =7710000)
MedLifeDR_3L.int <-   DEunderPeak(merge_int,  arm="3L", 
                   upper = 3840000, lower =3230000)
```

# Table of interaction DEGs under peaks

```{r}
# add identifier colum to each list
step_2L.int$QTL <- "step"
Ilp5_3L.int$QTL <- "Ilp5"
LinDiscrDR_2R.int$QTL <- "LinDiscrA"
LinDiscrDRC_2R.int$QTL <- "LinDiscrB"
MedLifeDR_3L.int$QTL <- "MedLife" 


qtl_degs_int <- rbind(step_2L.int,
                    LinDiscrDR_2R.int,LinDiscrDRC_2R.int,
                    MedLifeDR_3L.int)

ppi <- list(step_2L.int,
            LinDiscrDR_2R.int,LinDiscrDRC_2R.int,
            MedLifeDR_3L.int)
str(ppi)

qtl_degs_int <- qtl_degs_int[order(qtl_degs_int$QTL), ]
subset(qtl_degs_int,qtl_degs_int$QTL=="slif")

```

# Interaction and treatment tables together

```{r}
colnames(qtl_degs)
colnames(qtl_degs_int)

qtl_degs$effect <- "treat"
qtl_degs_int$effect <- "int"
all_qtl_degs <- rbind(qtl_degs,qtl_degs_int)
```

# Gene association table with GO IDs

```{r}
GO_fly <- read_lines(file = "../../processed/gene_association.fb",
                                       skip = 5) %>% 
  str_split(pattern = "\t", simplify = TRUE) %>% 
  as_tibble()

colnames(GO_fly) <- c("DB","gname","g_symbol",
                      "Qualifier","GOid","reference",
                      "Evidence","sourceE","Aspect","full_gname",
                      "altname","type","taxon","date","sourceA")
GO_fly[,c(16:17)] <- NULL
GO_fly <- select(GO_fly, gname, GOid)
colnames(GO_fly)[1] <- "FBgn"

allDEGs.GO <- inner_join(all_qtl_degs,GO_fly, by="FBgn")

```


```{r}
load("../../processed/DESEQ/DEG_QTL/all_iisPeak.Rda")
#R6pos = peak position in v6

transQTL <- subset(allPeak, allPeak$id==c('Ilp5','step'))
transQTL <- select(transQTL, id, as.numeric(R6pos), as.numeric(R6st))
colnames(transQTL)[1] <- "QTL"
transQTL$R6pos <- as.numeric(as.character(transQTL$R6pos)) 
tQTL.targets <- rbind(step_2L,step_2L.int,Ilp5_3L)

# Find distance of DEGs to  qtl position
distt <- inner_join(tQTL.targets,transQTL, by="QTL")
distt$qdist <- (distt$startp-distt$R6pos)

```

# Look for slif in DEG lists

```{r}
k <- gmtable[which(gmtable$gname=="slif"),]
kk <- which(DEGs_treat$FBgn %in% k$FBgn)
ki <- which(DEGs_int$FBgn %in% k$FBgn)
# slif not among DEGs!
```

# Gene summaries

```{r}
# view few lines in the terminal: head -n 3 automated_gene_summaries.tsv | tail -n 3

gsum <- read_lines(file = "../../processed/automated_gene_summaries.tsv",
                   skip = 1)%>%
  str_split(pattern = "\t", simplify = TRUE) %>% 
  as_tibble() 

ahcy <- grep("Ahcy", gsum$V2)
gsum[ahcy[1],'V2']
pull(gsum[ahcy[1],'V2'])

CG40486 <- grep("CG40486", gsum$V2)
gsum[CG40486[1],'V2']
pull(gsum[CG40486[1],'V2'])

Cda4 <- grep("Cda4", gsum$V2)
gsum[Cda4[1],'V2']
pull(gsum[Cda4[1],'V2'])
```

GO terms

```{r}
CVTerms <- read.csv("../../processed/DESEQ/CV_terms.csv",stringsAsFactors = FALSE)

#GO:0007568 - aging
age <- subset(CVTerms, CVTerms$GO_cat=="aging")
age.treat <- merge_treat %>%
      filter(FBgn %in% age$FBgn)
age.int <- merge_int %>%
  filter(FBgn %in% age$FBgn)
age.treat$effect <- "treat"
age.int$effect <- "int"
AGE <- rbind(age.treat,age.int)

#GO:0008340 - determination of adult lifespan
doal <- subset(CVTerms, CVTerms$GO_cat=="det_adul_life")
doal.treat <- merge_treat %>%
      filter(FBgn %in% doal$FBgn)
doal.int <- merge_int %>%
  filter(FBgn %in% doal$FBgn)
doal.treat$effect <- "treat"
doal.int$effect <- "int"
DOAL <- rbind(doal.treat,doal.int)

#GO:0008286 - insulin receptor signaling pathway
irsp <- subset(CVTerms, CVTerms$GO_cat=="IR_sig_path")
irsp <- merge_treat %>%
      filter(FBgn %in% irsp$FBgn)
irsp.int <- merge_int %>%
  filter(FBgn %in% irsp$FBgn)
irsp$effect <- "treat"
irsp.int$effect <- "int"
IRSP <- rbind(irsp,irsp.int)

#GO:0042593 - glucose homeostasis
gluc <- subset(CVTerms, CVTerms$GO_cat=="gluc_homeo")
gluc.treat <- merge_treat %>%
      filter(FBgn %in% gluc$FBgn)
gluc.int <- merge_int %>%
  filter(FBgn %in% gluc$FBgn)
gluc.treat$effect <- "treat"
gluc.int$effect <- "int"
GLUC <- rbind(gluc.treat,gluc.int)

#GO:0006629 - lipid metabolic process
lipmetp <- subset(CVTerms, CVTerms$GO_cat=="lip_met_proc")
lipmetp.treat <- merge_treat %>%
      filter(FBgn %in% lipmetp$FBgn)
lipmetp.int <- merge_int %>%
  filter(FBgn %in% lipmetp$FBgn)
lipmetp.treat$effect <- "treat"
lipmetp.int$effect <- "int"
LIPMETP <- rbind(lipmetp.treat,lipmetp.int)

#GO:0010897 - negative regulation of triglyceride catabolic process
nrtcp <- subset(CVTerms, CVTerms$GO_cat=="negRegl_trigly_catProc")
nrtcp.treat <- merge_treat %>%
      filter(FBgn %in% nrtcp$FBgn)
nrtcp.int <- merge_int %>%
  filter(FBgn %in% nrtcp$FBgn)
# zero

#GO:0042594 - response to starvation
rstarv <- subset(CVTerms, CVTerms$GO_cat=="resp_to_starv")
rstarv.treat <- merge_treat %>%
      filter(FBgn %in% rstarv$FBgn)
rstarv.int <- merge_int %>%
  filter(FBgn %in% rstarv$FBgn)
rstarv.treat$effect <- "treat"
rstarv.int$effect <- "int"
RSTARV <- rbind(rstarv.treat,rstarv.int)

#GO:0009267 - cellular response to starvation
crts <- subset(CVTerms, CVTerms$GO_cat=="cell_resp_starv")
crts.treat <- merge_treat %>%
      filter(FBgn %in% crts$FBgn)
crts.int <- merge_int %>%
  filter(FBgn %in% crts$FBgn)
crts.treat$effect <- "treat"
#crts.int$effect <- "int"
#CRTS <- rbind(crts.treat,crts.int)

```

# Nutrient-sensing pathways

```{r}
# IIS/TOR
iis_stan <- read.csv("../../processed/DESEQ/iisprobe_gene_CG_decoder.csv",stringsAsFactors = FALSE)

colnames(iis_stan)[2] <- "gname"
trt <- inner_join(iis_stan,merge_treat, by="gname")
trt$effect <- "treat"

xk <- merge_treat
colnames(xk)[10] <- "CG"
trt1 <- inner_join(iis_stan,xk, by="CG")

int <- inner_join(iis_stan,merge_int, by="gname")
int$effect <- "intr"

iis_tor <- rbind(trt,int)
```

