---
title: "Prep DESeq-worthy counts from StringTie output"
author: "Enoch Ng'oma"
date: "7/19/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("DESeq2")
library(dplyr)
library(stringr)
```

# Generate a list of gtf commands - ref p.92-93 Dunn book

##grep -li assemb *.gtf > ~/MyGithub/BasePop_RNAseq/processed/gtf_list.txt
# searches and list all .gtf which includes assemb*

#Open gtf_list.txt in TextWrangler [command+F]
#In the search window type ^
#In the replace window, type full path i.e: ~/MyGithub/BasePop_RNAseq/scripts/DESeq_scripts/
#Do Replace All
#This gives full path to each of the .gtf

#Alternatively:
#While in processed/ (i.e. where directory "ballgown" is located), run:
#python ../scripts/DESeq_scripts/prepDE.py
#Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`

# Load gene(/transcript) count matrix and labels

```{r}
countdata <- read.csv("../../processed/gene_count_matrix.csv", row.names="gene_id")

# Reformat column names
names(countdata) = sub("\\.","",names(countdata)) #regex escape the escape
countdata <- as.matrix(countdata)

phenDat <- read.csv("../../processed/describe_samplesDE.csv", stringsAsFactors = FALSE)
names(phenDat)[1]<-"ids"
phenDat <- phenDat %>%
  select(ids,treatment,tissue)

rownames(phenDat) <- phenDat[,1]
phenDat[,1] <- NULL

#colData <- pData(phenDat)[,c("treatment","tissue")]
phenDat <- as.matrix(phenDat)

# Check all sample IDs in colData are also in CountData and match their orders
all(rownames(phenDat) %in% colnames(countdata))
countdata <- countdata[, rownames(phenDat)]
all(rownames(phenDat) == colnames(countdata))

# Create a DESeqDataSet from count matrix and labels
dds <- DESeqDataSetFromMatrix(countData = countdata,
colData = phenDat, design = ~ treatment + tissue)

# Run the default analysis for DESeq2 and generate results table

ddsDE <- DESeq(dds)
res <- results(ddsDE)

#Sort by adjusted p-value and display
(resOrdered <- res[order(res$padj), ]) 
```






