---
title: "Prep DESeq-worthy counts from StringTie output"
author: "Enoch Ng'oma"
date: "7/19/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(dplyr)
library(stringr)
library(sva)
library(ggplot2)

set.seed(38142903)
```

# Ref: http://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#de

# Requirements:
#1. List of .gtf files produced by StringTie with the -B option (see Pertea et al 2016) and stored in a folder called `ballgown`
#2. A Python script `prepDE.py`
#3. A .txt file containing sample descriptions (see example at: http://ccb.jhu.edu/software/stringtie/dl/sample_lst.txt)

#1. Generate count matrices using `prepDE.py`

#a.    Assuming default ballgown directory structure produced by StringTie
#    $ python prepDE.py
#    List of GTFs sample IDs and paths (.txt) 
#    $ python prepDE.py -i sample_lst.txt # I used TextWrangler + Regex instead:
    
# Ref p.92-93 Haddok & Dunn book

# Locate the folder containing .gtf for each sample
#grep -li assemb *.gtf > ~/MyGithub/BasePop_RNAseq/processed/gtf_list.txt
# searches and list all .gtf which includes assemb*

#Open gtf_list.txt in TextWrangler [command+F]
#In the search window type ^
#In the replace window, type full path i.e: ~/MyGithub/BasePop_RNAseq/scripts/DESeq_scripts/
#Do Replace All
#This gives full path to each of the .gtf

#Alternatively:
#While in processed/ (i.e. where directory "ballgown" is located), run:
#python ../scripts/DESeq_scripts/prepDE.py
#Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`

# Load gene(/transcript) count matrix and labels

```{r}
countdata <- read.csv("../../processed/gene_count_matrix.csv", row.names="gene_id")

# Reformat column names
names(countdata) = sub("\\.","",names(countdata)) #regex escape the escape
countdata <- as.matrix(countdata)

phenDat <- read.csv("../../processed/describe_samplesDE.csv", stringsAsFactors = FALSE)
names(phenDat)[1]<-"ids"
phenDat <- phenDat %>%
  select(ids,treatment,tissue)

rownames(phenDat) <- phenDat[,1]
phenDat[,1] <- NULL

#colData <- pData(phenDat)[,c("treatment","tissue")]
phenDat <- as.matrix(phenDat)

# Check all sample IDs in colData are also in CountData and match their orders
all(rownames(phenDat) %in% colnames(countdata))
countdata <- countdata[, rownames(phenDat)]
all(rownames(phenDat) == colnames(countdata))
```

# Model with treatment and tissue

```{r}
# Create a DESeqDataSet from count matrix and labels
dds <- DESeqDataSetFromMatrix(countData = countdata,
colData = phenDat, design = ~ treatment + tissue)

# Run the default analysis for DESeq2 and generate results table

#ddsDE <- DESeq(dds)
#res <- results(ddsDE)

#Sort by adjusted p-value and display
#(resOrdered <- res[order(res$padj), ]) 

#save(ddsDE, file = "../../processed/DESEQ/ddsDE.Rda")
#save(resOrdered, file = "../../processed/DESEQ/resOrdered.Rda")
```

# Model with treatment, tissue and interaction

```{r}
dds2 <- DESeqDataSetFromMatrix(countData = countdata,
colData = phenDat, design = ~ treatment + tissue + treatment*tissue)

# Run the default analysis for DESeq2 and generate results table

#ddsDE2 <- DESeq(dds2)
#res2 <- results(ddsDE2)

#Sort by adjusted p-value and display
#(resOrdered2 <- res2[order(res2$padj), ]) 

#save(ddsDE2, file = "../../processed/DESEQ/ddsDE2.Rda")
#save(resOrdered2, file = "../../processed/DESEQ/resOrdered2.Rda")
```

# Control for batch effects with sva

```{r}
dds.sva <- estimateSizeFactors(dds)
dat.sva <- counts(dds.sva, normalized=TRUE)
cc.c <- rowSums(dat.sva)

#remove all genes with 0s for all samples
dat.sva <- dat.sva[which(cc.c>0),]

phenDat.sva <- read.csv("../../processed/describe_samples_batch.csv", stringsAsFactors = FALSE)
names(phenDat.sva)[5]<-"batch"
#cbind(phenDat.sva, colnames(dat.sva))

mod <- model.matrix(~as.factor(treatment)* 
                              as.factor(tissue) + 
                              as.factor(batch), 
                              data=phenDat.sva)

mod0 <- model.matrix(~as.factor(batch), data=phenDat.sva) 

# calculate number of surrogate variables
n.sv <- num.sv(dat.sva, mod, method = "be")
print(c("Calculated number of significant SVs = ", n.sv))

svobj <- svaseq(dat.sva, mod, mod0, n.sv=n.sv) 

phenDat.sva$SV1 <- svobj$sv[,1]
phenDat.sva$SV2 <- svobj$sv[,2]

ggplot(phenDat.sva, aes(x=SV1, y=SV2)) +
  geom_point(alpha=1/5, size=3)
```

# DESeq on batch-controlled data

```{r}
countdata <- read.csv("../../processed/gene_count_matrix.csv", row.names="gene_id")

# Reformat column names
names(countdata) = sub("\\.","",names(countdata)) #regex escape the escape
countdata <- as.matrix(countdata)

phenDat.sva <- phenDat.sva %>%
  select(id,treatment,tissue,batch,SV1,SV2)

rownames(phenDat.sva) <- phenDat.sva[,1]
phenDat.sva[,1] <- NULL

# Reformat column names
rownames(phenDat.sva) = sub("\\-","",rownames(phenDat.sva)) #regex escape the escape
#phenDat.sva <- as.matrix(phenDat.sva)

# countData colnames() must be identical to colData rownames()
all(rownames(phenDat.sva) %in% colnames(countdata))
countdata <- countdata[, rownames(phenDat.sva)]
all(rownames(phenDat.sva) == colnames(countdata))

phenDat.sva$batch <- as.factor(phenDat.sva$batch)
phenDat.sva$treatment <- as.factor(phenDat.sva$treatment)
phenDat.sva$tissue <- as.factor(phenDat.sva$tissue)

# DESeqDataSet for all variables
dds.ttSS <- DESeqDataSetFromMatrix(countData = countdata,
colData = phenDat.sva, design = ~ treatment + tissue + batch + SV1 + SV2)

# Run the default analysis for DESeq2 and generate results table
#ddsDE.ttSS <- DESeq(dds.ttSS)
#res.ttSS <- results(ddsDE.ttSS)

# Sort by adjusted p-value and display
(resOrdered <- res.ttSS[order(res.ttSS$padj), ]) 

## code burns out: "22 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest"
```

# Try suggestions in: https://support.bioconductor.org/p/65091/

```{r}
# Filter out a normalized count of at least 10 in two or more samples
ddsDE_filt10 <- estimateSizeFactors(dds.ttSS)
nc <- counts(ddsDE_filt10, normalized=TRUE)
filter <- rowSums(nc >= 10) >= 2
ddsDE_filt10 <- ddsDE_filt10[filter,]

# Run DESeq
#de_filt10 <- DESeq(ddsDE_filt10)
#res.filt10 <- results(de_filt10)

# Still: "8 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest"
```

# Try increasing the 2 sample requirement to 3 or 4

```{r}
set.seed(4364317)
ddsDE_filt5 <- estimateSizeFactors(dds.ttSS)
nc <- counts(ddsDE_filt5, normalized=TRUE)
filter <- rowSums(nc >= 5) >= 2
ddsDE_filt5 <- ddsDE_filt5[filter,]

#de_filt10 <- DESeq(ddsDE_filt10)
#res.filt10 <- results(de_filt10)
# still 6 rows not converged

#either:  omit non-converging rows from the results step
#ddsDE_Clean <- ddsDE_filt10[which(mcols(ddsDE_filt10)$betaConv),]

# or: increase the maximum iterations (maxit) instead of running DESeq()
dds <- estimateSizeFactors(ddsDE_filt5)
dds <- estimateDispersions(dds)
dds <- nbinomWaldTest(dds, maxit=500)

save(dds, file = "../../processed/DESEQ/DEseq_sva_filt5_2.Rda")
# 9 rows not converged

# At least 10 reads in two or more samples
ddsDE_filt10 <- estimateSizeFactors(dds.ttSS)
nc <- counts(ddsDE_filt10, normalized=TRUE)
filter <- rowSums(nc >= 10) >= 3
ddsDE_filt10 <- ddsDE_filt10[filter,]

# Increase the maximum iterations (maxit) instead of running DESeq()
dds <- estimateSizeFactors(ddsDE_filt10)
dds <- estimateDispersions(dds)
dds <- nbinomWaldTest(dds, maxit=500)

save(dds, file = "../../processed/DESEQ/DEseq_sva_filt10_3.Rda")
```

# DESeq2 3-factor conditions: https://www.biostars.org/p/115685/ (LRT is akin 1-WAY ANOVA)

```{r}
load("../../processed/DESEQ/DEseq_sva_filt10_3.Rda")

#res.filt10 <- results(de_filt10)

design(dds) = ~ group
dds = DESeq(dds, test = "LRT", reduced = ~ 1)
results(dds)
```


