---
title: "Visualize RNASeq on BasePop using DESeq2"
author: "Enoch Ng'oma"
date: "October 28, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(DESeq2)
library(genefilter)
library(tidyverse)
library(cowplot)
library(reshape)
library(ggthemes)
library(ggrepel)
library(data.table)
library(limma)
library(gridExtra)
library(RColorBrewer)

source("modified_plotPCA_function.R")
source("../color_map.R")
source("../ggplot_theme.R")

set.seed(354902)
```

# Global trends with ballgown

```{r}
phenDat <- read.csv("../../processed/describe_samples.csv",
                    stringsAsFactors = FALSE)

bg_ballG <- ballgown(dataDir = "../../processed/S03_short_ballG/ballgown", samplePattern = "", pData = phenDat)

# Save .Rda
save(bg_ballG, file = "../../processed/S03_short_ballG/bgOut_shortP.Rda")

```

# Gene abundances across libraries, colored by treatment

```{r}
load("../../processed/S03_short_ballG/bgOut_shortP.Rda")
bg_ballG

# Filter out low-abundance transcripts 
short_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)
short_filt
#whole_tx_table <- texpr(short_filt, meas = "all")
#write.csv(whole_tx_table, file = "../../processed/S03_short_ballG/whole_tx_table.csv")

# Get fpkm values
tfpkm_short<-texpr(short_filt, meas="FPKM")
tfpkm_short <-log2(tfpkm_short+1)
colnames(tfpkm_short)<-phenDat$id
#save(tfpkm_short, file = "../../processed/S03_short_ballG/tfpkm_short.Rda")

tbg<- as.data.frame(tfpkm_short) %>% gather(id, fpkm)
tbg$treatment <- rep(c("C","DR","HS"), each=18*nrow(tfpkm_short))
tbg$tissue <- rep(c("B","H","O"),each=nrow(tfpkm_short),6)
tbg$replicate <- rep(c(1,2,3,4,5,6),each=nrow(tfpkm_short)*3)
tbg$prep <- paste(tbg$treatment,tbg$replicate,sep="_")
tbg$trep <- paste(tbg$tissue,tbg$replicate,sep="_")

# Remove underscore
tbg$prep <- sub("_", "", tbg$prep)
tbg$trep <- sub("_", "", tbg$trep)

b1 <- ggplot(tbg, aes(x=as.factor(prep), y=fpkm,color=treatment)) +
  geom_boxplot(alpha=0.8) +
  theme(axis.title.x=element_blank(),
        axis.text.x  = element_text(angle=90, vjust=0.5))+
  #scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +       
  #scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Samples")+
  facet_grid(tissue~.) +
  scale_colour_brewer(palette = "Dark2") +
  my_theme
  #tx_color_map() 
b1 + theme(legend.position="none")
ggsave(filename = "../../plots/Expression_by_sample1.pdf",width = 5,height=5)


b2 <- ggplot(tbg, aes(x=as.factor(trep), y=fpkm,color=tissue)) +
  geom_boxplot(alpha=0.8) +
  theme(axis.title.x=element_blank(),
        axis.text.x  = element_text(angle=90, vjust=0.5))+
  scale_color_manual(values=c("indianred4", "olivedrab", "turquoise4")) +
  #scale_colour_brewer(palette = "Dark2") +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Tissue") +
  #scale_fill_discrete(name="Samples")+
  #scale_colour_tableau() +
  facet_grid(treatment~.) +
  my_theme
b2 + theme(legend.position="none")

ggsave(filename = "../../plots/Expression_by_sample2.pdf",width = 5,height=5)
```

# Number of transcripts per gene

```{r}
# Load transcript gene indexes from ballgown objects
t.genes_table_short <- indexes(short_filt)$t2g

# Plot # of transcripts per gene
counts.bg <- table(t.genes_table_short[,"g_id"])
bg_one <- length(which(counts.bg==1))
bg_more <- length(which(counts.bg>1))
bg_max <- max(counts.bg)
hist(counts.bg, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", bg_one), paste("Genes with more than one transcript =", bg_more), paste("Max transcripts for single gene = ", bg_max))
legend("topright", legend_text, lty = NULL)
```

# Distribution of transcript sizes

```{r}
full_table.bg<-texpr(short_filt, 'all')
hist(full_table.bg$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")

ggplot(full_table.bg, aes(x=length, col=length)) +
  geom_histogram(binwidth = 100) +
  labs(x = "Transcript length (bp)", y = "Count") +
  theme(axis.title.y = element_text(size = 12),
        text = element_text(size = 12))
```

# Tranform data to visualize global trends

```{r}
# Models created in DESeq_DExpr.Rmd
load("../../processed/DESEQ/dds_deseq01.Rda")
load("../../processed/DESEQ/dds_deseq.02.Rda")

# log transform null DESeq model: design ~ SV1 + batch + treatment + tissue
rlog.trt <- rlog(dds_deseq.01, blind = FALSE, fitType = "parametric")
#rlog.int <- rlog(dds_deseq.02, blind = FALSE, fitType = "parametric")

# rlog.trt <- read.table(
#             file="../../processed/DESEQ/rlog.trt.txt", sep="\t")
```

# Sample info

```{r}
phenDat.sva <- read.csv("../../processed/DESEQ/sampleDat_with_SV.csv")
rownames(phenDat.sva) <- phenDat.sva[,1]
phenDat.sva[,1] <- NULL
phenDat.sva$batch <- as.factor(phenDat.sva$batch)
phenDat.sva$treat_tissue <- paste(phenDat.sva$treatment,"_",phenDat.sva$tissue, sep="")

```

# PCA plot of uncorrected data

```{r}

uncor.pca <- plotPCA(rlog.trt, intgroup=c("treatment", "tissue"), returnData=TRUE)
percentVar <- round(100 * attr(uncor.pca, "percentVar"))

ll <- ggplot(uncor.pca, aes(PC1, PC2, color=treatment, shape=tissue)) +
  geom_point(size=5, alpha = 0.6) +
  theme(axis.title.y = element_text(size = 12),
                text = element_text(size = 12))+
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
    coord_cartesian(xlim = c(-105, 130)) +
  coord_cartesian(ylim = c(-105, 100)) +
  my_theme +
  labs(color="Treatment", shape="Tissue")
ll 

ggsave(last_plot(),file = "../../plots/PCAplot_rlog_uncorrected_trt.pdf",width = 5,height=5)

```

# Remove batches

```{r}
# check variation in size factors
hist(dds_deseq.01$sizeFactor)

d.mat <- model.matrix(~ treatment * tissue, phenDat.sva)

rlog.trt.batch <- rlog.trt

mat <- assay(rlog.trt.batch)
mat <- limma::removeBatchEffect(mat, batch = phenDat.sva$batch, 
                                batch2 = phenDat.sva$SV1, design = d.mat) 

assay(rlog.trt.batch) <- mat
bcor.pca <- pcaPlot(rlog.trt.batch, intgroup=c("treatment", "tissue"), returnData=TRUE)
percentVar <- round(100 * attr(bcor.pca, "percentVar"))

gg <- ggplot(bcor.pca, aes(PC1, PC2, color=treatment, shape=tissue)) + 
  geom_point(size=3, alpha = 0.6) + 
  theme(axis.title = element_text(size = 10),
                text = element_text(size = 10))+
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  xlab(paste0("PC1: ", percentVar[1],"% variance")) + 
  ylab(paste0("PC2: ", percentVar[2],"% variance")) + 
  #coord_cartesian(xlim = c(-130, 130)) +
  coord_cartesian(ylim = c(-100, 100)) +
  my_theme +
  labs(color="Treatment", shape="Tissue")
gg    
ggsave(gg,file = "../../plots/PCAplot_rlog_corrected_trt.pdf",width = 3,height=2.5)

#pcaplot <- plot_grid(ll, gg, nrow = 1)
pcaplot <- plot_grid(ll,gg, align = 'h',rel_widths = c(1.2,1.5), label_size = 10)
ggsave(pcaplot,file = "../../plots/PCAplot_rlog_trt.pdf",width = 3,height=2.5)
```

