---
title: "Visualize RNASeq on BasePop using DESeq2"
author: "Enoch Ng'oma"
date: "October 28, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(DESeq2)
library(genefilter)
library(tidyverse)
library(cowplot)
library(reshape)
library(ggthemes)
library(ggrepel)
library(data.table)
library(pheatmap)
library(limma)
library(RColorBrewer)

source("modified_plotPCA_function.R")
source("../color_map.R")
source("../ggplot_theme.R")

set.seed(354902)
```

# Global trends with ballgown

```{r}
phenDat <- read.csv("../../processed/describe_samples.csv",
                    stringsAsFactors = FALSE)

bg_ballG <- ballgown(dataDir = "../../processed/S03_short_ballG/ballgown", samplePattern = "", pData = phenDat)

# Save .Rda
save(bg_ballG, file = "../../processed/S03_short_ballG/bgOut_shortP.Rda")

```

# Gene abundances across libraries, colored by treatment

```{r}
load("../../processed/S03_short_ballG/bgOut_shortP.Rda")
bg_ballG

# Filter out low-abundance transcripts 
short_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)
short_filt

# Get fpkm values
tfpkm_short<-texpr(short_filt, meas="FPKM")
tfpkm_short <-log2(tfpkm_short+1)
colnames(tfpkm_short)<-phenDat$id
#save(tfpkm_short, file = "../../processed/S03_short_ballG/tfpkm_short.Rda")

tbg<- as.data.frame(tfpkm_short) %>% gather(id, fpkm)
tbg$treatment <- rep(c("C","DR","HS"), each=18*nrow(tfpkm_short))
tbg$tissue <- rep(c("B","H","O"),each=nrow(tfpkm_short),6)
tbg$replicate <- rep(c(1,2,3,4,5,6),each=nrow(tfpkm_short)*3)
tbg$prep <- paste(tbg$treatment,tbg$replicate,sep="_")
tbg$trep <- paste(tbg$tissue,tbg$replicate,sep="_")

# Remove underscore
tbg$prep <- sub("_", "", tbg$prep)
tbg$trep <- sub("_", "", tbg$trep)

b1 <- ggplot(tbg, aes(x=as.factor(prep), y=fpkm,color=treatment)) +
  geom_boxplot(alpha=0.8) +
  theme(axis.title.x=element_blank(),
        axis.text.x  = element_text(angle=90, vjust=0.5))+
  #scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +       
  #scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Samples")+
  facet_grid(tissue~.) +
  scale_colour_brewer(palette = "Dark2") +
  my_theme
  #tx_color_map() 
b1 + theme(legend.position="none")
ggsave(filename = "../../plots/Expression_by_sample1.pdf",width = 5,height=5)


b2 <- ggplot(tbg, aes(x=as.factor(trep), y=fpkm,color=tissue)) +
  geom_boxplot(alpha=0.8) +
  theme(axis.title.x=element_blank(),
        axis.text.x  = element_text(angle=90, vjust=0.5))+
  scale_color_manual(values=c("indianred4", "olivedrab", "turquoise4")) +
  #scale_colour_brewer(palette = "Dark2") +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Tissue") +
  #scale_fill_discrete(name="Samples")+
  #scale_colour_tableau() +
  facet_grid(treatment~.) +
  my_theme
b2 + theme(legend.position="none")

ggsave(filename = "../../plots/Expression_by_sample2.pdf",width = 5,height=5)
```

# Number of transcripts per gene

```{r}
# Load transcript gene indexes from ballgown objects
t.genes_table_short <- indexes(short_filt)$t2g

# Plot # of transcripts per gene
counts.bg <- table(t.genes_table_short[,"g_id"])
bg_one <- length(which(counts.bg==1))
bg_more <- length(which(counts.bg>1))
bg_max <- max(counts.bg)
hist(counts.bg, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", bg_one), paste("Genes with more than one transcript =", bg_more), paste("Max transcripts for single gene = ", bg_max))
legend("topright", legend_text, lty = NULL)
```

# Distribution of transcript sizes

```{r}
full_table.bg<-texpr(short_filt, 'all')
hist(full_table.bg$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")

ggplot(full_table.bg, aes(x=length, col=length)) +
  geom_histogram(binwidth = 100) +
  labs(x = "Transcript length (bp)", y = "Count") +
  theme(axis.title.y = element_text(size = 12),
        text = element_text(size = 12))
```

# Tranform data to visualize global trends

```{r}
# Models created in DESeq_DExpr.Rmd
load("../../processed/DESEQ/dds_deseq01.Rda")
load("../../processed/DESEQ/dds_deseq.02.Rda")

# log transform null DESeq model: design ~ SV1 + batch + treatment + tissue
rlog.trt <- rlog(dds_deseq.01, blind = FALSE, fitType = "parametric")
rlog.int <- rlog(dds_deseq.02, blind = FALSE, fitType = "parametric")

```

# Sample info

```{r}
phenDat.sva <- read.csv("../../processed/DESEQ/sampleDat_with_SV.csv")
rownames(phenDat.sva) <- phenDat.sva[,1]
phenDat.sva[,1] <- NULL
phenDat.sva$batch <- as.factor(phenDat.sva$batch)
phenDat.sva$treat_tissue <- paste(phenDat.sva$treatment,"_",phenDat.sva$tissue, sep="")

```

# PCA plots

```{r}
# PC1/PC2: Choose one variance stabilization (vst) plot 

rlog_pca <- plotPCA(rlog.trt, intgroup=c("treatment", "tissue"), returnData=TRUE)
percentVar <- round(100 * attr(rlog_pca, "percentVar"))
ggplot(rlog_pca, aes(PC1, PC2, color=treatment, shape=tissue)) +
  geom_point(size=5, alpha = 0.6) +
  theme(axis.title.y = element_text(size = 12),
                text = element_text(size = 12))+
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  my_theme
ggsave(last_plot(),file = "../../plots/PCAplot_rlog_trt.pdf",width = 5,height=5)


# PC1/PC2: Choose one regularized log transformation (rlog) plot
pcaData <- plotPCA(rlog.trt, intgroup=c("treatment", "tissue"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=treatment, shape=tissue)) +
  geom_point(size=5, alpha = 0.6) +
  theme(axis.title.y = element_text(size = 12),
                text = element_text(size = 12))+
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  my_theme
#(pcaData <- pcaData[order(pcaData$tissue), ])
ggsave(last_plot(),file = "../../plots/PCAplot_rlog_02.pdf",width = 5,height=5)


# PC3/PC4: source code modified
# ref: http://seqanswers.com/forums/showthread.php?t=66769
pc34 <- pcaPlot(rlog.trt, intgroup=c("treatment", "tissue"), returnData=TRUE)
percentVar <- round(100 * attr(pc34, "percentVar"))

ggplot(pc34, aes(PC3, PC4, color=treatment, shape=tissue)) + 
  geom_point(size=5, alpha = 0.6) + 
  theme(axis.title.y = element_text(size = 12),
                text = element_text(size = 12))+
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  xlab(paste0("PC3: ", percentVar[3],"% variance")) + 
  ylab(paste0("PC4: ", percentVar[4],"% variance")) + 
  coord_fixed() +
  my_theme

#ggsave(last_plot(),file = "../../plots/PCAplot_rlog_02_PC3PC4.pdf",width = 5,height=5)

```

# Remove batches

```{r}
# check variation in size factors
hist(dds_deseq.01$sizeFactor)

d.mat <- model.matrix(~ treatment * tissue, phenDat.sva)

rlog.trt.batch <- rlog.trt
# Treatment effect data
mat <- assay(rlog.trt.batch)
mat <- limma::removeBatchEffect(mat, batch = phenDat.sva$batch, batch2 = phenDat.sva$SV1, design = d.mat) 

assay(rlog.trt.batch) <- mat
print(plotPCA(rlog.trt.batch, intgroup=c("treatment", "tissue")))   
pc34 <- pcaPlot(rlog.trt.batch, intgroup=c("treatment", "tissue"), returnData=TRUE)
percentVar <- round(100 * attr(pc34, "percentVar"))

ggplot(pc34, aes(PC3, PC4, color=treatment, shape=tissue)) + 
  geom_point(size=5, alpha = 0.6) + 
  theme(axis.title.y = element_text(size = 12),
                text = element_text(size = 12))+
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  xlab(paste0("PC3: ", percentVar[3],"% variance")) + 
  ylab(paste0("PC4: ", percentVar[4],"% variance")) + 
  coord_fixed() +
  my_theme
                    
```

# Significantly DE genes
# data prep

```{r}
# DEGs
trtDEGs <- read.csv("../../processed/DESEQ/DEGs_lrt.treatment_0.05.csv")
intDEGs <- read.csv("../../processed/DESEQ/DEGs_lrt.int_0.05.csv")

# Normalized count data (accounted for sizeFactors)
load("../../processed/DESEQ/dds_deseq01.Rda")
trtNormCounts <- dds_deseq.01
trtNormCounts <- as.data.frame(counts(trtNormCounts, normalized=TRUE))
setDT(trtNormCounts, keep.rownames = TRUE)[]
names(trtNormCounts)[1] <- "Gene"

load("../../processed/DESEQ/dds_deseq.02.Rda")
intNormCounts <- dds_deseq.02
intNormCounts <- as.data.frame(counts(intNormCounts, normalized=TRUE))
setDT(intNormCounts, keep.rownames = TRUE)[]
names(intNormCounts)[1] <- "Gene"

# Metadata
phenDat.sva <- read.csv("../../processed/DESEQ/sampleDat_with_SV.csv",
                        stringsAsFactors = FALSE)
names(phenDat.sva)[1] <- "id"
```

# Plotting subsets of genes
#   Top 20 significant DEGs

```{r}
# Treatment

top20_sigTrt_genes <- trtDEGs %>%
  arrange(padj) %>%
  pull(Gene) %>%
  .[1:20]

# Extract normalized counts for these genes
top20_sigTrt_norm <- trtNormCounts %>%
  filter(Gene %in% top20_sigTrt_genes)

# Gather counts for all samples into one column for ggplot
gathered_top20_sigTrt <- top20_sigTrt_norm %>%
  gather(colnames(top20_sigTrt_norm)[2:55], 
         key="id", value = "normalized_counts")
#View(gathered_top20_sigTrt)

# color counts by treatment
gathered_top20_sigTrt <- inner_join(phenDat.sva, gathered_top20_sigTrt)

# Plot top 20 trt DEGs
ggplot(gathered_top20_sigTrt) +
        geom_point(aes(x = Gene, y = normalized_counts, color = tissue,treatment)) +
        scale_y_log10() +
        xlab("Genes") +
        ylab("log10 Normalized Counts") +
        ggtitle("Top 20 Treatment DE Genes") +
        theme_bw() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    theme(plot.title = element_text(hjust = 0.5))

# Interaction

top20_sigInt_genes <- intDEGs %>%
  arrange(padj) %>%
  pull(Gene) %>%
  .[1:20]

# Extract normalized counts for these genes
top20_sigInt_norm <- intNormCounts %>%
  filter(Gene %in% top20_sigInt_genes)

# Gather counts for all samples into one column for ggplot
gathered_top20_sigInt <- top20_sigInt_norm %>%
  gather(colnames(top20_sigInt_norm)[2:55], 
         key="id", value = "normalized_counts")
#View(gathered_top20_sigInt)

# color counts by treatment
gathered_top20_sigInt <- inner_join(phenDat.sva, gathered_top20_sigInt)

# Plot top 20 trt DEGs
ggplot(gathered_top20_sigInt) +
        geom_point(aes(x = Gene, y = normalized_counts, color = tissue)) +
        scale_y_log10() +
        xlab("Genes") +
        ylab("log10 Normalized Counts") +
        ggtitle("Top 20 Interaction DE Genes") +
        theme_bw() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    theme(plot.title = element_text(hjust = 0.5))
```

# Single gene plots

```{r}
# Save plotcounts to a data frame object
d <- plotCounts(dds_deseq.02, gene="FBgn0037801", 
                intgroup=c("treatment", "tissue"), returnData=TRUE)

# Plotting normalized counts
ggplot(d, aes(x = treatment, y = count, color = treatment)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  #geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("FBgn0037801") +
  theme(plot.title = element_text(hjust = 0.5))
```

# Heatmaps

```{r}
# Treatment
# Get normalized expression for DEGs 
# set the gene column to row names
norm_sigTrt <- trtNormCounts %>% 
              filter(Gene %in% as.character(trtDEGs$Gene)) %>% 
          data.frame() %>%
          column_to_rownames(var = "Gene") 

# Annotate heatmap (optional)
annotation <- phenDat.sva %>% 
    select(id, treatment) %>% 
    data.frame(row.names = "id")

# Set a color palette
heat_colors <- brewer.pal(6, "YlOrRd")

# Run pheatmap
pheatmap(norm_sigTrt, 
         color = heat_colors, 
         cluster_rows = T, 
         show_rownames = F,
         annotation = annotation, 
         border_color = NA, 
         fontsize = 10, 
         scale = "row", 
         fontsize_row = 10, 
         height = 20)


# Interaction
norm_sigInt <- intNormCounts %>% 
              filter(Gene %in% as.character(intDEGs$Gene)) %>% 
          data.frame() %>%
          column_to_rownames(var = "Gene") 

# Annotate heatmap (optional)
annotation <- phenDat.sva %>% 
    select(id, treatment) %>% 
    data.frame(row.names = "id")

# Set a color palette
heat_colors <- brewer.pal(6, "YlOrRd")

# Run pheatmap
pheatmap(norm_sigInt, 
         color = heat_colors, 
         cluster_rows = T, 
         show_rownames = F,
         annotation = annotation, 
         border_color = NA, 
         fontsize = 10, 
         scale = "row", 
         fontsize_row = 10, 
         height = 20)
#order samples
```

# Volcano plots

```{r}
# Load LRT data saved in DESeq_DExpr.Rmd
load("../../processed/DESEQ/lrtTreat_allPadj.Rda")
load("../../processed/DESEQ/lrtInt_allPadj.Rda")

# Treatment effect
lrt.treatAll <- subset(lrt.treatAll)
summary(lrt.treatAll)
sum(lrt.treatAll$padj < 0.1, na.rm = TRUE)
lrt.treatAll$Gene <- rownames(lrt.treatAll)
lrt.treatAll <- lrt.treatAll[, c(7,1:6)]
lrt.treatAll <- lrt.treatAll[order(lrt.treatAll$padj), ]

# Logical vector where TRUE means padj < 0.05
lrt.trtAll <- as_tibble(lrt.treatAll) %>% 
  mutate(thresh = padj < 0.05)
lrt.trtAll <- subset(lrt.trtAll, lrt.trtAll$padj != 'NA')

# Make column for genes to label
lrt.trtAll <- lrt.trtAll %>% 
  arrange(padj) %>% 
  mutate(genelabels = "")
lrt.trtAll$genelabels[1:10] <- lrt.trtAll$Gene[1:10]
#View(lrt.trtAll)

ggplot(lrt.trtAll, aes(x = log2FoldChange, y = -log10(padj))) +
        geom_point(aes(colour = thresh)) +
        #geom_text_repel(aes(label = genelabels)) +
        #ggtitle("Treatment effect") +
        xlab("log2 fold change") + 
        ylab("-log10 adj p-value") +
        #scale_y_continuous(limits = c(0,25)) +
        scale_x_continuous(limits = c(-5,5)) +
        theme(legend.position = "none",
              plot.title = element_text(size = rel(1.5), hjust = 0.5),
              axis.title = element_text(size = rel(1.25))) 

# Interacation effect

```

# Custom visuals

```{r}

# Define the genes of interest
load("../../processed/DESEQ/lrtTreat_allPadj.Rda")
goi <- read.csv("../../processed/DESEQ/DEGs_lrt.treatment_0.05.csv") %>%
  as.tbl()

lrt.treatment <- subset(lrt.treatAll, lrt.treatAll$padj < 0.05)

tcounts <- t(log2((counts(lrt.treatment[goi, ], normalized=TRUE, replaced=FALSE)+.5))) %>%
  merge(colData(phenDat.sva), ., by="row.names") %>%
  gather(gene, expression, (ncol(.)-length(goi)+1):ncol(.))
```

```{r}

# DEGs at padj < 0.05 cutoff
load(file="../../processed/DESEQ/lrt.treatment.Rda")

# Treatment effect

# Merge with normalized count data
res_trt <- subset(lrt.treatment, lrt.treatment$padj < 0.05)
res_trt_tab <- merge(as.data.frame(res_trt), 
                       as.data.frame(counts(dds_deseq.01, normalized=TRUE)), 
                       by="row.names", sort=FALSE)
#setDT(res_trt_tab, keep.rownames = TRUE)[] #in data.table
names(res_trt_tab)[1] <- "Gene"

# Bodies 
ff <- as.data.frame(res_trt_tab[1:7])
kk <- grep("_B",colnames(res_trt_tab))
grep.res_body <- res_trt_tab[,kk]
colnames(grep.res_body) <- gsub("[[:digit:]]","",colnames(grep.res_body))
#colnames(grep.res_body) <- grep.res_body[1,]
#colnames(grep.res_body) <- str_split_fixed(colnames(grep.res_body),pattern = "_", 1)
res_body <- cbind(ff, grep.res_body)
#names(res.dds_body)[1] <- "Gene"

ggplot(res_body, aes(x=log2FoldChange, y))


# Heads
hh <- grep("_H",colnames(res.dds_table))
grep.dds_head <- res.dds_table[,hh]
res.dds_head <- cbind(ff, grep.dds_head)

# Ovaries
oo <- grep("_O",colnames(res.dds_table))
grep.dds_ovary <- res.dds_table[,oo]
res.dds_ovary <- cbind(ff, grep.dds_ovary)

# Examine plot of p-values
hist(res.dds_body$padj, breaks=50, col="grey")


# Interaction effect
load(file="../../processed/DESEQ/lrt.int.Rda")

```


