---
title: "WGCNA_ANOVA and GO analysis"
author: "Enoch Ng'oma"
date: "4/8/2019"
output:
  html_document:
    theme: flatly
    toc: true
    toc_float:
      collapsed: false
      smooth_scroll: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(WGCNA)
library(tidyverse)
library(cowplot)
library(AnnotationDbi)
library(org.Dm.eg.db)

```

# Gene Ontology on WGCNA modules

```{r}
# Modules & module names
load("../../processed/DESEQ/Coexpression/modules.RData")


# Output gene lists
# see Index of ftp://ftp.flybase.net/releases/FB2018_05/precomputed_files/genes/

annot = read.csv("../../processed/DESEQ/Coexpression/FlyAnnotation.csv")
dim(annot)
names(annot)
colnames(annot)[colnames(annot)=="X"] <- "FBgn"

rlogtrt <- read.csv("../../processed/DESEQ/rlogtrt_batchCor.csv")
rownames(rlogtrt) <- rlogtrt[,1]
FBgenes = rownames(rlogtrt)
FBgenes2annot = match(FBgenes, annot$FBgn)
allIDs = annot$entrez[FBgenes2annot]

# allIDs = cbind(as.data.frame(annot$FBgn[FBgenes2annot]), as.data.frame(annot$symbol[FBgenes2annot]))
# colnames(allIDs) <- NULL
# colnames(allIDs) <- c("FBgn", "Symbol")

# Number genes without annotation:
sum(is.na(FBgenes2annot))
# Should return 0

# Get module genes
modNames
table(moduleColors)

getMods <- modNames[1:22]

for (module in getMods) {
  # select module genes
  modGenes = (moduleColors==module)
  # Get their entrez ids
  modIDs = allIDs[modGenes]
  # Write file
  fileName = paste("../../processed/DESEQ/Coexpression/minMod30/", 
                   module,".txt", sep="")
  write.table(as.data.frame(modIDs), file = fileName,
              row.names = FALSE, col.names = FALSE)
}

# Save background list
fileName = paste("../../processed/DESEQ/Coexpression/minMod30/backList.txt",
                 sep="");
write.table(as.data.frame(allIDs), file = fileName,
           row.names = FALSE, col.names = FALSE) 


# Module GO term enrichment

# Enrichment table containing the 10 best terms for each module
GOenr <- GOenrichmentAnalysis(moduleColors, allIDs, 
                              organism = "fly", nBestP = 10)
tab <- GOenr$bestPTerms[[4]]$enrichment
names(tab)
write.table(tab, file = "../../processed/DESEQ/Coexpression/minMod30/GOEnrichmentTable.csv", 
              sep = ",", quote = TRUE, row.names = FALSE)
### Note message: GOenrichmentAnalysis: loading annotation data...
###  ..of the 12614  Entrez identifiers submitted, 10334 are mapped ###in current GO categories.
###  ..will use 10334 background genes for enrichment calculations.

# View table on screen
keepCols = c(1, 2, 5, 6, 7, 12, 13);
screenTab = tab[, keepCols];

# Round numeric columns to 2 decimal places:
numCols = c(3, 4);
screenTab[, numCols] = signif(apply(screenTab[, numCols], 2, as.numeric), 2)

# Truncate the the term name to at most 40 characters
screenTab[, 7] = substring(screenTab[, 7], 1, 40)

# Shorten the column names:
colnames(screenTab) = c("module", "size", "p-val", "Bonf", "nInTerm", "ont", "term name");
rownames(screenTab) = NULL;

# Set the width of Râ€™s output
options(width=95)

# Enrichment table
screenTab
write.table(screenTab, file = "../../processed/DESEQ/Coexpression/minMod30/GOEnrichmentTable_simple.txt",
              sep = ",", quote = TRUE, row.names = FALSE)
```

# ANOVA on module eigengenes to test for interaction effect

```{r}
# Read and prep sample data
phenDat.sva <- read.csv("../../processed/DESEQ/sampleDat_with_SV.csv")
rownames(phenDat.sva) <- phenDat.sva[,1]
phenDat.sva[,1] <- NULL
phenDat.sva$batch <- as.factor(phenDat.sva$batch)
phenDat.sva$treat_tissue <- paste(phenDat.sva$treatment,"_",phenDat.sva$tissue, sep="")


eg.genes <- read.table(file="../../processed/DESEQ/Coexpression/WCGNA_eigengenes.txt", sep="\t",
                       header=TRUE, stringsAsFactors = FALSE) 
# write.csv(eg.genes, file = "../../processed/DESEQ/Coexpression/WCGNA_eigengenes.csv", 
#           row.names = FALSE)

# ANOVA for treatment
aov.mods1 <- aov(as.matrix(eg.genes) ~ phenDat.sva$treat_tissue) 
summary(aov.mods1)

# ANOVA for interaction
aov.mods2 <- aov(as.matrix(eg.genes) ~ phenDat.sva$treatment* phenDat.sva$tissue)
sum.aov <- summary(aov.mods2)

# Extract P values
kk <- lapply(sum.aov, function(x) x$`Pr(>F)`)
kk <- as.data.frame(kk)
rownames(kk) <- c("Diet", "Tissue", "Interaction", "Residuals")
colnames(kk) <- paste(letters[1:22])
kk <- as.data.frame(t(kk))
kk <- dplyr::select(kk, Diet, Tissue, Interaction,)
kk <- kk %>% 
  rownames_to_column("key") 

# Get FDR coreected P values
dietFDR <- p.adjust(kk$Diet, "BH", n = length(kk$Diet))
tissueFDR <- p.adjust(kk$Tissue, "BH", length(kk$Tissue))
interFDR <- p.adjust(kk$Interaction, "BH", length(kk$Interaction))
anova_adj <- as.data.frame(cbind(kk,dietFDR,tissueFDR,interFDR))

anova_adj <- format(anova_adj[,1:7], format = "e", digits = 2)

msKeys <- read.csv("../../processed/DESEQ/Coexpression/ms_module_keys.csv") %>%
  dplyr::select(-X)

msKeys$Percent <- msKeys$no_rows/12614*100

out_fig <- left_join(msKeys, anova_adj)
out_fig <- dplyr::select(out_fig, ModID,key,no_rows, everything()) %>%
  dplyr::rename(Genes = no_rows, MS_ID = key)

out_fig$Percent <- round(out_fig$Percent, digits = 1)
out_fig
write.csv(out_fig, file = "../../processed/DESEQ/Coexpression/Module_ANOVA.csv")
```

# Visualize modules across diets

```{r}
all.dat <- cbind(phenDat.sva, eg.genes)

ccs <- colnames(all.dat[,6:27])
all.plots <- vector(mode="list", length=length(ccs))
names(all.plots) <- ccs

for(cc in ccs)
{

pp<- ggplot(all.dat, aes_string(x="treatment", y=cc, color="tissue")) +
  geom_point(position = position_jitter(width = 0.15)) +
  stat_summary(fun.y = mean, geom = "point", size = 3, pch=2)

all.plots[[cc]] <- pp

}

```

# Notes for each module

## Black module

```{r}
all.plots[[1]]
aov.mod_black <- aov(eg.genes[,"MEblack"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_black)
black_ph <- TukeyHSD(aov.mod_black)
black_ph
subset(screenTab,screenTab$module=="black")
```

2956 genes
Effect of tissue*** (ovaries differ from rest); treatment**; no interaction. 
Top 10 GO terms: gene expression (RNA metabolism, DNA binding and organellar activiity


## Blue module

```{r}
all.plots[[2]]
aov.mod_blue <- aov(eg.genes[,"MEblue"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_blue)
blue_ph <- TukeyHSD(aov.mod_blue)
blue_ph
subset(screenTab,screenTab$module=="blue")
```

1747 genes
Effect of tissue*** (all tissues differ); treatment*** some interaction* effect
Treatment highly sig for HS-C and HS-DR, insig for DR-C
Tissue highly sig for all pairwise comparisons
Interaction highly sig for 30 diet-tissue comparisons
(The three tissues all different in expression, with slight increase of B&O in HS)
GO: nervous system process, signalling and neuronal activity


## Brown module

```{r}
all.plots[[3]]
aov.mod_brown <- aov(eg.genes[,"MEbrown"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_brown)
brown_ph <- TukeyHSD(aov.mod_brown)
brown_ph
subset(screenTab,screenTab$module=="brown")
```

1084 genes
Effect of tissue*** (heads and overies are together); some interaction effect* 
No treatment effect
Tissue highly sig for H-B and OB, insig for O-H
Interaction sig for 18 diet-tissue contrasts
GO: proteolysis and chitin metabolic process


## Cyan module

```{r}
all.plots[[4]]
aov.mod_cyan <- aov(eg.genes[,"MEcyan"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_cyan)
cyan_ph <- TukeyHSD(aov.mod_cyan)
cyan_ph
subset(screenTab,screenTab$module=="cyan")
```

507 genes
Tissue*** treatment*** interaction***
30 out of 36 interaction contrasts highly significant padj
Treatment highly significant in DR-C and HS-DR comparisons, mildly in HS-C
Tissue highly significant in all 3 comparisons
Interaction highly significant in 30 of 36 comparisons
GO: Redox process, transmembrane ion transport


## Darkgreen module

```{r}
all.plots[[5]]
aov.mod_darkgreen <- aov(eg.genes[,"MEdarkgreen"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_darkgreen)
darkgreen_ph <- TukeyHSD(aov.mod_darkgreen)
subset(screenTab,screenTab$module=="darkgreen")
```

199 genes
Tissue*** treatment*** interaction*
O is high in C and HS but low in DR
Strong treatment effects in HS-DR, mild effect for DR-C, no effect for HS-C 
Strong tissue effect in H-B and O-H, no tissue effect for O-B
Interaction is significant in 17 of 36 diet contrasts
GO: several BPs including signaling

## Darkgrey

```{r}
all.plots[[6]]
aov.mod_darkgrey <- aov(eg.genes[,"MEdarkgrey"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_darkgrey)
darkgrey_ph <- TukeyHSD(aov.mod_darkgrey)
darkgrey_ph
subset(screenTab,screenTab$module=="darkgrey")
```

309 genes
Tissue*** treatment*** interaction***
Treatment highly sig for HS-C and HS-DR, insig for DR-C
Tissue higly sig for 3 comparisons: H-B, O-B, O-H
Interaction sig for 31 of 36 diet-tissue comparisons
(Low in DR high in HS in general)
GO: Ras signaling and binding


## Darkred 

```{r}
all.plots[[7]]
aov.mod_darkred <- aov(eg.genes[,"MEdarkred"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_darkred)
darkred_ph <- TukeyHSD(aov.mod_darkred)
darkred_ph
subset(screenTab,screenTab$module=="darkred")
```

84 genes
Tissue*** treatment*** interaction**
Low in C, high in DR and low in HS
Treatment highly sig for HS-DR and DR-C, insig for HS-C
Tissue higly sig for 2 comparisons: O-B, O-H; mildly for H-B
Interaction sig for 21 of 36 diet-tissue comparisons
(Low in C, high in DR and low in HS)
GO: peptide metabolism, cytosolic ribosome activity

## Darkturquoise 

```{r}
all.plots[[8]]
aov.mod_darkturquoise <- aov(eg.genes[,"MEdarkturquoise"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_darkturquoise)
darkturquoise_ph <- TukeyHSD(aov.mod_darkturquoise)
darkturquoise_ph
subset(screenTab,screenTab$module=="darkturquoise")

```

79 genes
Tissue*** treatment** interaction**
Treatment mildly sig in DR-C and HS-C, insig in HS-DR
Tissue higly sig in all pairwise comparisons
Interaction highly sig in 30 diet-tissue combinations
(Body expression low in C, high in DR & HS)
GO: amino acid transport, proteasome binding


## Greenyellow

```{r}
all.plots[[9]]
aov.mod_greenyellow <- aov(eg.genes[,"MEgreenyellow"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_greenyellow)
greenyellow_ph <- TukeyHSD(aov.mod_greenyellow)
greenyellow_ph
subset(screenTab,screenTab$module=="greenyellow")

```

165 genes
Tissue*** treatment*** interaction***
Treatment highly sig in HS-DR, DR-C, mildly in HS-C
Tissue higly sig in all comparisons
Interaction highly sig in 29 diet-tissue contrasts
(B&O low in C, high in DR & low in HS; H higher in C than other diets)
GO: proteasome catabolic, ubiquitin processes


## Grey module - unassigned genes 

```{r}
all.plots[[10]]
aov.mod_grey <- aov(eg.genes[,"MEgrey"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_grey)
subset(screenTab,screenTab$module=="grey")
```

45 genes
Tissue* treatment*** interaction***
B&H high in C, high intermediate in DR & low in HS; O low in C high in DR & lower in HS
See p-values!!


## Lightcyan module

```{r}
all.plots[[11]]
aov.mod_lightcyan <- aov(eg.genes[,"MElightcyan"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_lightcyan)
lightcyan_ph <- TukeyHSD(aov.mod_lightcyan)
lightcyan_ph
subset(screenTab,screenTab$module=="lightcyan")

```

117 genes
Tissue*** treatment*** interaction**
Treatment highly sig for DR-C and HS-DR, insig for HS-C
Tissue insig for H-B, highly sig for O-B and O-H
(O high in C and HS, low in DR)


## Lightgreen module

```{r}
all.plots[[12]]
aov.mod_lightgreen <- aov(eg.genes[,"MElightgreen"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_lightgreen)
lightgreen_ph <- TukeyHSD(aov.mod_lightgreen)
lightgreen_ph
subset(screenTab,screenTab$module=="lightgreen")
```

107 genes 
Tissue*** treatment*** interaction***
Treatment highly sig DR-C and HS-DR, insig in HS-C
Tissue highly sig in all comparisons
Interaction sig in 26 diet-tissue comparisons
(B&O low in C, high in DR, low in HS; H higher in C)
GO: peptide metabolism, mitochondrial translation


## Lightyellow module

```{r}
all.plots[[13]]
aov.mod_lightyellow <- aov(eg.genes[,"MElightyellow"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_lightyellow)
lightyellow_ph <- TukeyHSD(aov.mod_lightyellow)
lightyellow_ph
subset(screenTab,screenTab$module=="lightyellow")
```

107 genes 
Tissue*** treatment*** interaction***
Treatment highly sig for DR-C and HS-DR, insig for HS-C
Tissue highly sig for H-B and O-H, insig for O-B
Interaction sig for 17 diet-tissue contrasts
(B&O low in C, high in DR, low in HS; H higher in C)
(trend too similar to Lightgreen)
GO: Golgi vesicle transport


## Pink module  

```{r}
all.plots[[14]]
aov.mod_pink <- aov(eg.genes[,"MEpink"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_pink)
pink_ph <- TukeyHSD(aov.mod_pink)
pink_ph
subset(screenTab,screenTab$module=="pink")
```

382 genes
Tissue*** treatment*** interaction**
Treatment highly sig for DR-C and HS-DR, insig for HS-C
Tissue highly highly sig for all comparisons
Interaction highly sig for 24 diet-tissue comparisons
(B&O high in C, low in DR, high in HS; H no pattern)
GO: regulation metabolic process, gene expression, transcription


## Red module  

```{r}
all.plots[[15]]
aov.mod_red <- aov(eg.genes[,"MEred"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_red)
red_ph <- TukeyHSD(aov.mod_red)
red_ph
subset(screenTab,screenTab$module=="red")
```

787 genes
Tissue*** treatment*** interaction***
Treatment highly sig for HS-C and DR-C, mildly for HS-DR
Tissue highly sig for all 3 comparisons
(subtlely increasing in all tissue across C, DR, HS)
GO: olfactory and sensory perception of smell or chemical stimulus


## Royalblue  

```{r}
all.plots[[16]]
aov.mod_royalblue <- aov(eg.genes[,"MEroyalblue"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_royalblue)
royalblue_ph <- TukeyHSD(aov.mod_royalblue)
royalblue_ph
subset(screenTab,screenTab$module=="royalblue")
```

132 genes
Tissue*** treatment*** interaction**
Treatment highly sig for DR-C and HS-DR, insig for HS-C
Tissue highly sig for H-B and O-B, insig for O-H
(B&O high in C, low in DR, high in HS; H low, higher, high)
GO: regulation of metabolic process, regulation of gene expression


## Salmon module

```{r}
all.plots[[17]]
aov.mod_salmon <- aov(eg.genes[,"MEsalmon"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_salmon)
salmon_ph <- TukeyHSD(aov.mod_salmon)
salmon_ph
subset(screenTab,screenTab$module=="salmon")
```

157 genes
Tissue*** treatment*** interaction***
(B high, low, high; H&O low, higher, high)
Treatment highly sig in HS-C and HS-DR, insig in DR-C
Tissue highly sig in all 3 comparisons
Interaction highly sig in 27 diet-tissue comparisons


## Skyblue module

```{r}
all.plots[[18]]
aov.mod_skyblue <- aov(eg.genes[,"MEskyblue"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_skyblue)
skyblue_ph <- TukeyHSD(aov.mod_skyblue)
skyblue_ph
subset(screenTab,screenTab$module=="skyblue")
```

52 genes
Tissue* treatment*** interaction***
Treatment highly sig in DR-C and HS-DR, insig in HS-C
Tissue marginally sig for O-B
Interaction highly sig in 13 diet-tissue comparisons
(B&O low, high, low; H high, low, low)
GO: peptide metabolic process, ribosome


## Steelblue module 

```{r}
all.plots[[19]]
aov.mod_steelblue <- aov(eg.genes[,"MEsteelblue"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_steelblue)
steelblue_ph <- TukeyHSD(aov.mod_steelblue)
steelblue_ph
subset(screenTab,screenTab$module=="steelblue")
```
44 genes
Tissue*** treatment*** interaction*
high, low, high; larger effect in ovaries
Treatment highly sig in DR-C and HS-DR, insig in HS-C
Tissue sig for O-B and O-H only
Interaction highly sig in 8 diet-tissue comparisons
GO: protein localization and binding


## Tan module 

```{r}
all.plots[[20]]
aov.mod_tan <- aov(eg.genes[,"MEtan"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_tan)
tan_ph <- TukeyHSD(aov.mod_tan)
tan_ph
subset(screenTab,screenTab$module=="tan")
```

163 genes
Tissue*** treatment*** interaction***
Treatment highly sig in DR-C and HS-DR, insig in HS-C
Tissue highly sig for all comparisons
Interaction highly sig in 29 diet-tissue comparisons
(O low, high, low; larger effect in ovaries)
GO: pigment biosynthesis


## Turquoise module 

```{r}
all.plots[[21]]
aov.mod_turquoise <- aov(eg.genes[,"MEturquoise"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_turquoise)
turquoise_ph <- TukeyHSD(aov.mod_turquoise)
turquoise_ph
subset(screenTab,screenTab$module=="turquoise")
```

3337 genes
Tissue*** treatment*** interaction*
All tissues low in C and similar for DR and HS
Treatment highly sig in DR-C and HS-C, insig in HS-DR
Tissue highly sig for O-B and O-H comparisons, insig for H-B
Interaction highly sig in 22 diet-tissue comparisons
GO: Redox process, bacterial/defence respnse, iron ion binding


## White module

```{r}
all.plots[[22]]
aov.mod_white <- aov(eg.genes[,"MEwhite"] ~ phenDat.sva$treatment* phenDat.sva$tissue)
summary(aov.mod_white)
white_ph <- TukeyHSD(aov.mod_white)
white_ph
subset(screenTab,screenTab$module=="white")
```
54 genes
Tissue*** treatment*** interaction***
Treatment highly sig in HS-C, mildly in DR-C, insig in HS-DR
Tissue highly sig for H-B and O-B comparisons, insig for O-H
Interaction highly sig in 15 diet-tissue comparisons
(B low in C, high in DR, higher in HS; other tissues no clear trend)
GO: chitin, cuticle development, body morphogenesis



