---
title: "GO analysis on WGCNA after resampling"
author: "Enoch Ng'oma"
date: "4/8/2019"
output:
  html_document:
    css: style.css
    theme: flatly
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
  pdf_document:
    toc: yes
  word_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(WGCNA)
library(tidyverse)
library(cowplot)
library(AnnotationDbi)
library(org.Dm.eg.db)

```


```{r}
lld <- read.csv("../../processed/DESEQ/Coexpression/modueleMembership.csv")
```

# Compare module sizes

```{r}
# ll %>% 
#   group_by(mod) %>%
#   summarise(no_rows = length(mod)) %>%
#   as.data.frame()
# ll$analysis <- "full"

lls <- lld %>%
  group_by(ModID) %>%
  summarise(no_rows = length(ModID)) %>%
  as.data.frame()
lls$analysis <- "resamp"

# Check one module
#stlbl <- rbind(subset(ll, ll$mod=="steelblue"), subset(lld, lld$mod=="steelblue"))

```

# Gene Ontology on WGCNA modules

```{r}
# Fbgn annotation file "Genes:fbgn_annotation_ID_fb_2018_04.tsv"

annot = read.csv("../../processed/DESEQ/Coexpression/FlyAnnotation.csv")
dim(annot)
names(annot)
colnames(annot)[colnames(annot)=="X"] <- "FBgn"

FBgenes <- lld$FBgn
#FBgenes <- dd$FBgn

FBgenes2annot = match(FBgenes, annot$FBgn)
allIDs = annot$entrez[FBgenes2annot]


# Module GO term enrichment

# Enrichment table
GOenr <- GOenrichmentAnalysis(lld$ModID, allIDs, 
                              organism = "fly", 
                              #nBestP = 100,
                              pCut = 0.01)
# GOenr1 <- GOenrichmentAnalysis(lld$ModID, allIDs, 
#                               organism = "fly", 
#                               #nBestP = 100,
#                               pCut = 0.01)

tab <- GOenr$bestPTerms[[4]]$enrichment
names(tab)
write.csv(tab, file = "../../processed/DESEQ/Coexpression/resampMods/GOEnrichTab_Resamp.csv", 
              row.names = FALSE)
### Note message: GOenrichmentAnalysis: loading annotation data...
###  ..of the 12614  Entrez identifiers submitted, 10334 are mapped ###in current GO categories.
###  ..will use 10334 background genes for enrichment calculations.

# View table on screen
keepCols = c(1, 2, 5, 6, 7, 12, 13);
screenTab = tab[, keepCols];

# Round numeric columns to 2 decimal places:
numCols = c(3, 4);
screenTab[, numCols] = signif(apply(screenTab[, numCols], 2, as.numeric), 2)

# Truncate the the term name to at most 40 characters
screenTab[, 7] = substring(screenTab[, 7], 1, 40)

# Shorten the column names:
colnames(screenTab) = c("module", "size", "p-val", "Bonf", "nInTerm", "ont", "termName");
rownames(screenTab) = NULL;

# Set the width of Râ€™s output
options(width=95)

# Enrichment table
screenTab
subset(screenTab, module == "black" & termName == "nucleus")
write.csv(screenTab, file = "../../processed/DESEQ/Coexpression/resampMods/GOEnrichTab_Resamp_simple.csv",
              row.names = FALSE)

bb <- subset(screenTab, screenTab$Bonf<0.01 & screenTab$module !="grey")
bc <- subset(screenTab, screenTab$Bonf<0.05 & screenTab$module !="grey")

bd <- bb %>%
  group_by(module) %>%
  summarise(lowest = min(Bonf), highest = max(Bonf)) %>%
  as.data.frame()

blk <- subset(bb,bb$module=="black" & bb$Bonf<0.01)
bbk <- bb %>%
  group_by(module) %>%
  summarise(no_rows = length(module)) %>%
  as.data.frame()
```

# Notes for each module

```{r}
ff <- unique(screenTab$module)

ff.list <- vector(mode='list', length=length(ff))
names(ff.list)  <- ff

for (ii in 1:length(ff)) 
  {
  
  ffc <- subset(screenTab,screenTab==ff[ii] )

  ff.list[[ii]] <- ffc
  
}

# Note: # substitute 'screenTab' for 'tab' to see full details

#lapply(ff.list[[ii]], head)
```


## Black (M1)
- 2956 genes in module; 771 GO terms across all GO categories
- BP: nucleobase metabolic process, gene expression, RNA processing, organelle organization, cellular metabolic process, cellular process, female gamete generation, germ cell development, reproduction, reproductive process, cellular protein modification process, cell communication, cell cycle, etc

## Blue (M2)
- 1747 genes, 311 GO terms
- BP: signalling, cell communication, nervous system process, ion transport, signal transduction, locomotion, neuron differentiation, neuron development, nervous system development, neurogenesis, cell morphogenesis, etc

## Brown (M3)
- 1084 genes, 116 terms
- BP: proteolysis, chitin metabolic process, fatty acid metabolic process, transmembrane transpoort, lipid metabolic process, lipid biosynthetic process, cellular lipid metabolic process, sulfur compound metabolic process, etc

## Cyan (M4)
- 507 genes, 98 terms
- BP: transmembrane transport, chitin metabolic process, ion transport, homeostatic process, heart development, fatty acid metabolic process, digestive tract development, response to toxic substance, etc 
## Darkgreen (M5)
- 199 genes, 273 terms
- BP: multicellular organism development, anatomical structure development, signaling, cell communication, developmental process, signal transduction, neurogenesis, cell differentiation, organ development, etc

## Darkgrey (M6)
- 309 genes, 85 terms
- BP: signaling, regulation of signal transduction, regulation of Ras protein signal transduction, regulation of small GTPase mediated signal transduction, cell communication, etc

## Darkred (M7)
- 84 genes, 71 terms
- BP: cytoplasmic translation, translation, peptide metabolic process, cytosolic ribosome activity, cellular process, nitrogen compound metabolic process, cellular metabolic process, biosynthetic process, gene expression, protein metabolic process, ribosome biogenesis, etc

## Darkturquoise (M8)
- 79 genes, 22 terms
- BP: L-amino acid transport, amino acid transport, proteasome assembly, etc

## Greenyellow (M9)
- 165 genes, 109 terms
- BP: proteasome-mediated ubiquitin-dependent, proteasome catabolic process, ubiquitin processes, ubiquitin-dependent protein catabolic process, modification-dependent protein catabolic process, proteolysis involved in cellular protein, cellular protein catabolic process, proteasomal ubiquitin-independent protein catabolic process, protein catabolic process, organonitrogen compound catabolic proces, catabolic process, proteolysis, organic substance catabolic process, protein-containing complex assembly, etc

## Grey (M10)
- 45 + 1011 genes, 373 terms

## Lightcyan (M11)
- 117 genes, 209 terms
- BP: cellular component organization, multicellular organism development, organelle organization, cell development, anatomical structure development, regulation of metabolic process, developmental process, cellular process, cellular metabolic process, etc

## Lightgreen (M12)
- 107 genes, 47 terms 
- BP: mitochondrial translation, translation, peptide metabolic process, protein metabolic process, biosynthetic process, gene expression, cellular metabolic process, nitrogen compound metabolic process, etc

## Lightyellow (M13)
- 107 genes, 99 terms 
- BP: Golgi vesicle transport, ER to Golgi vesicle-mediated transport, translation, peptide metabolic process,intracellular transport, intracellular protein transport, protein transport, establishment of protein localization, vesicle-mediated transport, biosythetic process, cellular macromolecule localization, organic substance transport, etc

## Pink (M14)
- 382 genes, 163 terms
- BP: regulation of gene expression, regulation of metabolic process, regulation of RNA metabolic process, gene expression, RNA metabolic process, nucleobase-containing compound metabolic process, regulation of cellular metabolic process, transcription, nitrogen compound metabolic process, cellular process, cellular metabolic process, cellular protein modification process, cell differentiation, nervous system development, organelle organization, signaling, signal transduction, anatomical structure development, anatomical structure morphogenesis, gamete generation, etc, etc.

## Red (M15)
- 787 genes, 204 terms
- BP: nervous system process, sensory perception of chemical stimulus, detection of stimulus involved in sensory perception, detection of chemical stimulus, olfactory and sensory perception of smell, sensory perception of smell, phototransduction, response to light stimulus, cell communication, signaling, signal transduction, etc, etc.

## Royalblue (M16) 
- 132 genes, 154 terms
- BP: regulation of metabolic process, regulation of gene expression, transcription, regulation of cellular metabolic process, regulation of transcription, regulation of RNA metabolic process, gene expression, germ cell development, RNA metabolic process, cellular metabolic process, nitrogen compound metabolic process, oogenesis, female gamete generation, ovarian follicle, etc, etc.

## Salmon (M17)
- 157 genes, 30 terms 
- BP: nervous system development, neuron development, synapse assembly, synaptic growth at neuromuscular junction, anterograde axonal transport, multicellular organism development, animal organ morphogenesis, morphogenesis of an epithelium, etc.

## Skyblue (M18)
- 52 genes, 54 tems
- BP: cytoplasmic translation, peptide metabolic process, translation, biosynthetic process, gene expression, cellular metabolic process, protein metabolic process, ribosome biogenesis, cellular process, etc.

## Steelblue (M19)
- 44 genes, 68 terms
- BP: protein localization Golgi apparatus, protein retention in Golgi apparatus, Wnt signaling pathway, regulation of signal transduction, endocytosis, cell surface receptor signaling pathway, etc.

## Tan (M20)
- 163 genes, 24 terms
- BP: vesicle-mediated transport, ommochrome biosynthetic process, ocellus pigment biosynthetic process, eye pigment biosynthetic process, endosomal transport, eye pigment metabolic process, compound eye pigmentation, developmental pigmentation, intracellular transport, secretion by cell, regulation of lipid storage, etc. 

## Turquoise (M21)
- 3337 genes, 289 terms
- BP: oxidation-reduction process, response to bacterium, defence response, carboxylic acid metabolic - process, organic acid metabolic process, carbohydrate metabolic process, defense response to bacterium, defense response to other organism, cellular respiration, cellular carbohydrate metabolic process, ribose phosphate metabolic process, immune response, nucleoside phosphate metabolic process, nucleoside phosphate metabolic process, response to organic substance, nucleotide metabolic process, transmembrane transport, chitin-based cuticle development, lipid metabolic process, cellular lipid metabolic process, response to oxidative stress, fatty acid metabolic process, developmental pigmentation, ion transport, etc, etc, etc.

## White (M22)
- 54 genes, 13 terms
- BP: chitin-based cuticle development, cuticle development, body morphogenesis, multicellular organism development, anatomical structure development, developmental process.


