\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={GO analysis on WGCNA after resampling},
            pdfauthor={Enoch Ng'oma},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{GO analysis on WGCNA after resampling}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Enoch Ng'oma}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{4/8/2019}


\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\hypertarget{modules-for-the-full-dataset}{%
\section{Modules for the full
dataset}\label{modules-for-the-full-dataset}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Full dataset}

\CommentTok{# Load module names}
\KeywordTok{load}\NormalTok{(}\StringTok{"../../processed/DESEQ/Coexpression/modules.RData"}\NormalTok{)}

\CommentTok{# Read in multiple files by wildcard expansion on file paths}
\NormalTok{getMods <-}\StringTok{ }\KeywordTok{lapply}\NormalTok{(}\KeywordTok{Sys.glob}\NormalTok{(}
  \StringTok{"../../processed/DESEQ/Coexpression/Mods/fbgn_names/*.txt"}\NormalTok{,),read.table)}

\CommentTok{# Rename column}
\NormalTok{ss <-}\StringTok{ "FBgn"}
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\OperatorTok{:}\KeywordTok{length}\NormalTok{(getMods)) \{}
 \KeywordTok{colnames}\NormalTok{(getMods[[i]]) <-}\StringTok{ }\NormalTok{ss }
\NormalTok{\}}

\CommentTok{# Add column for module name to each module}
\NormalTok{getMods <-}\StringTok{ }\KeywordTok{mapply}\NormalTok{(cbind, getMods, }\StringTok{"mod"}\NormalTok{=}\KeywordTok{sort}\NormalTok{(modNames), }
                  \DataTypeTok{SIMPLIFY=}\OtherTok{FALSE}\NormalTok{)}
 
\CommentTok{# Merge modules in list into one table (12569 + 45 grey)}
\NormalTok{ll <-}\StringTok{ }\KeywordTok{bind_rows}\NormalTok{(getMods)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character
\end{verbatim}

\begin{verbatim}
## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector
\end{verbatim}

\begin{verbatim}
## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character
\end{verbatim}

\begin{verbatim}
## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector

## Warning in bind_rows_(x, .id): binding character and factor vector,
## coercing into character vector
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Resampled data}

\CommentTok{# Genes uniquely assigned to module in less than half 50 of 100 datasets}
\NormalTok{droppedGenes <-}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"../../processed/DESEQ/Coexpression/Resamp_droppedGenes.csv"}\NormalTok{)}
\KeywordTok{colnames}\NormalTok{(droppedGenes)[}\DecValTok{2}\NormalTok{] <-}\StringTok{ "mod"}

\CommentTok{# Delete droppedGenes in ll}
\NormalTok{dd <-}\StringTok{ }\NormalTok{ll[}\OperatorTok{!}\NormalTok{(ll}\OperatorTok{$}\NormalTok{FBgn }\OperatorTok{%in%}\StringTok{ }\NormalTok{droppedGenes}\OperatorTok{$}\NormalTok{FBgn),]}
\CommentTok{#str(dd)}
\CommentTok{#str(ll)}

\CommentTok{# Move deleted genes to the grey module}
\NormalTok{lld <-}\StringTok{ }\NormalTok{ll}
\NormalTok{lld[ll}\OperatorTok{$}\NormalTok{FBgn }\OperatorTok{%in%}\StringTok{ }\NormalTok{droppedGenes}\OperatorTok{$}\NormalTok{FBgn,}\StringTok{"mod"}\NormalTok{] <-}\StringTok{ "grey"}
\end{Highlighting}
\end{Shaded}

\hypertarget{compare-module-sizes}{%
\section{Compare module sizes}\label{compare-module-sizes}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ll }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{group_by}\NormalTok{(mod) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{summarise}\NormalTok{(}\DataTypeTok{no_rows =} \KeywordTok{length}\NormalTok{(mod)) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{as.data.frame}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##              mod no_rows
## 1          black    2956
## 2           blue    1747
## 3          brown    1084
## 4           cyan     507
## 5      darkgreen     199
## 6       darkgrey     309
## 7        darkred      84
## 8  darkturquoise      79
## 9    greenyellow     165
## 10          grey      45
## 11     lightcyan     117
## 12    lightgreen     107
## 13   lightyellow     107
## 14          pink     382
## 15           red     787
## 16     royalblue     132
## 17        salmon     157
## 18       skyblue      52
## 19     steelblue      44
## 20           tan     163
## 21     turquoise    3337
## 22         white      54
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ll}\OperatorTok{$}\NormalTok{analysis <-}\StringTok{ "full"}

\NormalTok{lld }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{group_by}\NormalTok{(mod) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{summarise}\NormalTok{(}\DataTypeTok{no_rows =} \KeywordTok{length}\NormalTok{(mod)) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{as.data.frame}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##              mod no_rows
## 1          black    2886
## 2           blue    1679
## 3          brown    1042
## 4           cyan     391
## 5      darkgreen      68
## 6       darkgrey     249
## 7        darkred      60
## 8  darkturquoise      65
## 9    greenyellow     135
## 10          grey    1049
## 11     lightcyan      82
## 12    lightgreen     102
## 13   lightyellow      79
## 14          pink     269
## 15           red     759
## 16     royalblue      85
## 17        salmon     114
## 18       skyblue      44
## 19     steelblue      39
## 20           tan     123
## 21     turquoise    3240
## 22         white      54
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{lld}\OperatorTok{$}\NormalTok{analysis <-}\StringTok{ "resamp"}

\CommentTok{# Check one module}
\NormalTok{stlbl <-}\StringTok{ }\KeywordTok{rbind}\NormalTok{(}\KeywordTok{subset}\NormalTok{(ll, ll}\OperatorTok{$}\NormalTok{mod}\OperatorTok{==}\StringTok{"steelblue"}\NormalTok{), }\KeywordTok{subset}\NormalTok{(lld, lld}\OperatorTok{$}\NormalTok{mod}\OperatorTok{==}\StringTok{"steelblue"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\hypertarget{gene-ontology-on-wgcna-modules}{%
\section{Gene Ontology on WGCNA
modules}\label{gene-ontology-on-wgcna-modules}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Fbgn annotation file "Genes:fbgn_annotation_ID_fb_2018_04.tsv"}

\NormalTok{annot =}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"../../processed/DESEQ/Coexpression/FlyAnnotation.csv"}\NormalTok{)}
\KeywordTok{dim}\NormalTok{(annot)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 12614    55
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{names}\NormalTok{(annot)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "X"              "baseMean"       "log2FoldChange" "lfcSE"         
##  [5] "stat"           "pvalue"         "padj"           "FC_DR_B"       
##  [9] "FCse_DR_B"      "stat_DR_B"      "p_DR_B"         "padj_DR_B"     
## [13] "FC_HS_B"        "FCse_HS_B"      "stat_HS_B"      "p_HS_B"        
## [17] "padj_HS_B"      "FC_DRHS_B"      "FCse_DRHS_B"    "stat_DRHS_B"   
## [21] "p_DRHS_B"       "padj_DRHS_B"    "FC_DR_H"        "FCse_DR_H"     
## [25] "stat_DR_H"      "p_DR_H"         "padj_DR_H"      "FC_HS_H"       
## [29] "FCse_HS_H"      "stat_HS_H"      "p_HS_H"         "padj_HS_H"     
## [33] "FC_DRHS_H"      "FCse_DRHS_H"    "stat_DRHS_H"    "p_DRHS_H"      
## [37] "padj_DRHS_H"    "FC_DR_O"        "FCse_DR_O"      "stat_DR_O"     
## [41] "p_DR_O"         "padj_DR_O"      "FC_HS_O"        "FCse_HS_O"     
## [45] "stat_HS_O"      "p_HS_O"         "padj_HS_O"      "FC_DRHS_O"     
## [49] "FCse_DRHS_O"    "stat_DRHS_O"    "p_DRHS_O"       "padj_DRHS_O"   
## [53] "symbol"         "entrez"         "name"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{colnames}\NormalTok{(annot)[}\KeywordTok{colnames}\NormalTok{(annot)}\OperatorTok{==}\StringTok{"X"}\NormalTok{] <-}\StringTok{ "FBgn"}

\NormalTok{FBgenes <-}\StringTok{ }\NormalTok{lld}\OperatorTok{$}\NormalTok{FBgn}
\CommentTok{#FBgenes <- dd$FBgn}

\NormalTok{FBgenes2annot =}\StringTok{ }\KeywordTok{match}\NormalTok{(FBgenes, annot}\OperatorTok{$}\NormalTok{FBgn)}
\NormalTok{allIDs =}\StringTok{ }\NormalTok{annot}\OperatorTok{$}\NormalTok{entrez[FBgenes2annot]}


\CommentTok{# Module GO term enrichment}

\CommentTok{# Enrichment table}
\NormalTok{GOenr <-}\StringTok{ }\KeywordTok{GOenrichmentAnalysis}\NormalTok{(lld}\OperatorTok{$}\NormalTok{mod, allIDs, }
                              \DataTypeTok{organism =} \StringTok{"fly"}\NormalTok{, }
                              \CommentTok{#nBestP = 100,}
                              \DataTypeTok{pCut =} \FloatTok{0.01}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in GOenrichmentAnalysis(lld$mod, allIDs, organism = "fly", pCut = 0.01): This function is deprecated and will be removed in the near future. 
## We suggest using the replacement function enrichmentAnalysis 
## in R package anRichment, available from the following URL:
## https://labs.genetics.ucla.edu/horvath/htdocs/CoexpressionNetwork/GeneAnnotation/
\end{verbatim}

\begin{verbatim}
## Loading required package: GO.db
\end{verbatim}

\begin{verbatim}
##  GOenrichmentAnalysis: loading annotation data...
##   ..of the 12614  Entrez identifiers submitted, 10334 are mapped in current GO categories.
##   ..will use 10334 background genes for enrichment calculations.
##   ..preparing term lists (this may take a while).. 
##   ..working on label set 1 ..
##     ..calculating enrichments (this may also take a while)..
##     ..putting together terms with highest enrichment significance..
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# GOenr1 <- GOenrichmentAnalysis(lld$mod, allIDs, }
\CommentTok{#                               organism = "fly", }
\CommentTok{#                               #nBestP = 100,}
\CommentTok{#                               pCut = 0.01)}

\NormalTok{tab <-}\StringTok{ }\NormalTok{GOenr}\OperatorTok{$}\NormalTok{bestPTerms[[}\DecValTok{4}\NormalTok{]]}\OperatorTok{$}\NormalTok{enrichment}
\KeywordTok{names}\NormalTok{(tab)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "module"             "modSize"            "bkgrModSize"       
##  [4] "rank"               "enrichmentP"        "BonferoniP"        
##  [7] "nModGenesInTerm"    "fracOfBkgrModSize"  "fracOfBkgrTermSize"
## [10] "bkgrTermSize"       "termID"             "termOntology"      
## [13] "termName"           "termDefinition"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{write.csv}\NormalTok{(tab, }\DataTypeTok{file =} \StringTok{"../../processed/DESEQ/Coexpression/resampMods/GOEnrichTab_Resamp.csv"}\NormalTok{, }
              \DataTypeTok{row.names =} \OtherTok{FALSE}\NormalTok{)}
\CommentTok{### Note message: GOenrichmentAnalysis: loading annotation data...}
\CommentTok{###  ..of the 12614  Entrez identifiers submitted, 10334 are mapped ###in current GO categories.}
\CommentTok{###  ..will use 10334 background genes for enrichment calculations.}

\CommentTok{# View table on screen}
\NormalTok{keepCols =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{2}\NormalTok{, }\DecValTok{5}\NormalTok{, }\DecValTok{6}\NormalTok{, }\DecValTok{7}\NormalTok{, }\DecValTok{12}\NormalTok{, }\DecValTok{13}\NormalTok{);}
\NormalTok{screenTab =}\StringTok{ }\NormalTok{tab[, keepCols];}

\CommentTok{# Round numeric columns to 2 decimal places:}
\NormalTok{numCols =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\DecValTok{3}\NormalTok{, }\DecValTok{4}\NormalTok{);}
\NormalTok{screenTab[, numCols] =}\StringTok{ }\KeywordTok{signif}\NormalTok{(}\KeywordTok{apply}\NormalTok{(screenTab[, numCols], }\DecValTok{2}\NormalTok{, as.numeric), }\DecValTok{2}\NormalTok{)}

\CommentTok{# Truncate the the term name to at most 40 characters}
\NormalTok{screenTab[, }\DecValTok{7}\NormalTok{] =}\StringTok{ }\KeywordTok{substring}\NormalTok{(screenTab[, }\DecValTok{7}\NormalTok{], }\DecValTok{1}\NormalTok{, }\DecValTok{40}\NormalTok{)}

\CommentTok{# Shorten the column names:}
\KeywordTok{colnames}\NormalTok{(screenTab) =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"module"}\NormalTok{, }\StringTok{"size"}\NormalTok{, }\StringTok{"p-val"}\NormalTok{, }\StringTok{"Bonf"}\NormalTok{, }\StringTok{"nInTerm"}\NormalTok{, }\StringTok{"ont"}\NormalTok{, }\StringTok{"termName"}\NormalTok{);}
\KeywordTok{rownames}\NormalTok{(screenTab) =}\StringTok{ }\OtherTok{NULL}\NormalTok{;}

\CommentTok{# Set the width of Râ€™s output}
\KeywordTok{options}\NormalTok{(}\DataTypeTok{width=}\DecValTok{95}\NormalTok{)}

\CommentTok{# Enrichment table}
\NormalTok{screenTab}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##             module size    p-val     Bonf nInTerm ont                                 termName
## 1            black 2886 3.9e-160 3.2e-156    1095  CC                                  nucleus
## 2            black 2886 6.9e-112 5.8e-108    1857  CC                            intracellular
## 3            black 2886 2.7e-103  2.3e-99    1399  CC intracellular membrane-bounded organelle
## 4            black 2886  9.6e-98  8.1e-94    1574  CC                  intracellular organelle
## 5            black 2886  1.3e-94  1.1e-90    1009  CC               protein-containing complex
## 6            black 2886  2.1e-89  1.8e-85     936  BP nucleobase-containing compound metabolic
## 7            black 2886  3.2e-82  2.6e-78     710  MF                     nucleic acid binding
## 8            black 2886  6.6e-80  5.5e-76     882  BP                          gene expression
## 9            black 2886  2.2e-79  1.9e-75     340  BP                           RNA processing
## 10           black 2886  1.3e-77  1.1e-73     748  BP                    RNA metabolic process
## 11           black 2886  1.6e-75  1.3e-71     348  BP                  chromosome organization
## 12           black 2886  6.2e-59  5.2e-55     209  BP                    DNA metabolic process
## 13           black 2886  2.3e-58  2.0e-54     308  CC                               chromosome
## 14           black 2886  1.3e-54  1.1e-50     261  BP                       mitotic cell cycle
## 15           black 2886  4.6e-52  3.8e-48     680  BP                   organelle organization
## 16           black 2886  6.4e-52  5.3e-48     302  MF                              RNA binding
## 17           black 2886  1.7e-47  1.4e-43     192  BP                         nuclear division
## 18           black 2886  2.0e-46  1.7e-42     143  CC                                nucleolus
## 19           black 2886  3.7e-46  3.1e-42    1401  BP      nitrogen compound metabolic process
## 20           black 2886  9.1e-43  7.6e-39    1429  BP               cellular metabolic process
## 21           black 2886  3.4e-42  2.9e-38     136  BP                   chromosome segregation
## 22           black 2886  2.1e-40  1.7e-36     126  BP           nuclear chromosome segregation
## 23           black 2886  5.1e-40  4.3e-36     270  CC                ribonucleoprotein complex
## 24           black 2886  1.3e-39  1.1e-35     310  BP                 female gamete generation
## 25           black 2886  2.0e-37  1.7e-33    1934  BP                         cellular process
## 26           black 2886  9.6e-37  8.1e-33     228  CC                              nucleoplasm
## 27           black 2886  7.0e-36  5.9e-32     175  BP                          mRNA processing
## 28           black 2886  7.6e-35  6.3e-31     173  BP                 regulation of cell cycle
## 29           black 2886  1.4e-34  1.2e-30     373  BP                        gamete generation
## 30           black 2886  1.7e-33  1.4e-29     196  BP                   chromatin organization
## 31           black 2886  2.0e-32  1.7e-28     890  BP          cellular component organization
## 32           black 2886  2.6e-32  2.2e-28     134  BP                       meiotic cell cycle
## 33           black 2886  2.7e-32  2.2e-28     151  CC                       nuclear chromosome
## 34           black 2886  3.9e-32  3.3e-28      86  BP                          DNA replication
## 35           black 2886  2.3e-31  1.9e-27     131  BP cellular response to DNA damage stimulus
## 36           black 2886  2.5e-30  2.1e-26     272  BP                                oogenesis
## 37           black 2886  3.3e-30  2.7e-26     151  BP                             RNA splicing
## 38           black 2886  7.4e-30  6.2e-26     154  CC                                chromatin
## 39           black 2886  8.7e-30  7.3e-26     109  BP                               DNA repair
## 40           black 2886  1.2e-29  1.0e-25     121  BP                      ribosome biogenesis
## 41           black 2886  1.4e-29  1.2e-25     302  BP                    germ cell development
## 42           black 2886  1.9e-29  1.6e-25     143  BP RNA splicing, via transesterification re
## 43           black 2886  1.9e-29  1.6e-25     143  BP           mRNA splicing, via spliceosome
## 44           black 2886  1.2e-28  1.0e-24      94  BP             sister chromatid segregation
## 45           black 2886  2.9e-27  2.4e-23     116  BP         regulation of mitotic cell cycle
## 46           black 2886  3.6e-27  3.0e-23      71  BP            DNA-dependent DNA replication
## 47           black 2886  7.3e-27  6.1e-23     433  BP                     reproductive process
## 48           black 2886  1.3e-25  1.1e-21      84  BP                          rRNA processing
## 49           black 2886  6.7e-24  5.7e-20     438  BP                             reproduction
## 50           black 2886  6.0e-23  5.0e-19      79  BP     mitotic sister chromatid segregation
## 51           black 2886  7.6e-23  6.3e-19      86  BP                   tRNA metabolic process
## 52           black 2886  2.5e-22  2.1e-18      69  BP                        DNA recombination
## 53           black 2886  1.1e-21  9.0e-18      64  MF                        helicase activity
## 54           black 2886  2.7e-21  2.3e-17     324  MF                              DNA binding
## 55           black 2886  3.9e-21  3.2e-17      82  BP              cell cycle phase transition
## 56           black 2886  8.0e-21  6.7e-17     638  BP          regulation of metabolic process
## 57           black 2886  1.5e-20  1.2e-16      76  BP              nucleocytoplasmic transport
## 58           black 2886  2.0e-20  1.7e-16      78  BP      mitotic cell cycle phase transition
## 59           black 2886  8.7e-20  7.3e-16      60  CC                     spliceosomal complex
## 60           black 2886  2.3e-19  1.9e-15     384  BP      multicellular organism reproduction
## 61           black 2886  2.4e-19  2.0e-15     488  BP            regulation of gene expression
## 62           black 2886  3.6e-19  3.0e-15      72  CC                         nuclear envelope
## 63           black 2886  5.5e-19  4.6e-15      70  BP regulation of cell cycle phase transitio
## 64           black 2886  6.0e-19  5.0e-15      60  CC           chromosome, centromeric region
## 65           black 2886  6.2e-19  5.2e-15      98  BP                         RNA localization
## 66           black 2886  7.6e-19  6.4e-15      45  BP          female meiotic nuclear division
## 67           black 2886  8.4e-19  7.1e-15      68  BP regulation of mitotic cell cycle phase t
## 68           black 2886  3.7e-18  3.1e-14      48  BP           meiotic chromosome segregation
## 69           black 2886  1.1e-17  9.5e-14      32  CC                 small-subunit processome
## 70           black 2886  1.9e-17  1.6e-13      55  BP                    RNA 3'-end processing
## 71           black 2886  3.4e-17  2.9e-13      90  CC                        nuclear chromatin
## 72           black 2886  3.4e-17  2.9e-13      71  BP                       eggshell formation
## 73           black 2886  3.8e-17  3.2e-13     210  BP      protein-containing complex assembly
## 74           black 2886  4.4e-17  3.7e-13      31  CC                         replication fork
## 75           black 2886  4.4e-17  3.7e-13      70  BP    chorion-containing eggshell formation
## 76           black 2886  8.7e-17  7.3e-13      35  CC                                  chorion
## 77           black 2886  2.3e-16  1.9e-12     400  BP    cellular protein modification process
## 78           black 2886  3.4e-16  2.8e-12      46  BP        chromatin assembly or disassembly
## 79           black 2886  1.0e-15  8.6e-12      45  CC            RNA polymerase II, holoenzyme
## 80           black 2886  2.0e-15  1.7e-11      48  BP                  nucleosome organization
## 81           black 2886  2.3e-15  1.9e-11      94  BP                     histone modification
## 82           black 2886  2.3e-15  1.9e-11      47  CC                     condensed chromosome
## 83           black 2886  2.8e-15  2.4e-11      99  MF                        chromatin binding
## 84           black 2886  3.0e-15  2.5e-11     719  BP                     biosynthetic process
## 85           black 2886  3.3e-15  2.8e-11      43  MF          ATP-dependent helicase activity
## 86           black 2886  3.5e-15  2.9e-11      66  BP        negative regulation of cell cycle
## 87           black 2886  3.9e-15  3.3e-11      70  BP                     spindle organization
## 88           black 2886  5.6e-15  4.7e-11      59  BP                    RNA catabolic process
## 89           black 2886  8.6e-15  7.2e-11      54  BP             protein-DNA complex assembly
## 90           black 2886  1.6e-14  1.3e-10     683  MF                          protein binding
## 91           black 2886  1.7e-14  1.4e-10      46  BP          protein localization to nucleus
## 92           black 2886  2.0e-14  1.7e-10      84  BP    regulation of chromosome organization
## 93           black 2886  2.2e-14  1.8e-10      59  BP                         RNA modification
## 94           black 2886  2.8e-14  2.3e-10      42  BP                    chromosome separation
## 95           black 2886  3.4e-14  2.8e-10      36  CC                             nuclear pore
## 96           black 2886  4.0e-14  3.3e-10      66  MF               methyltransferase activity
## 97           black 2886  8.9e-14  7.5e-10     221  BP      cellular macromolecule localization
## 98           black 2886  9.1e-14  7.6e-10      40  BP                  RNA export from nucleus
## 99           black 2886  9.8e-14  8.2e-10     139  BP    microtubule cytoskeleton organization
## 100          black 2886  9.8e-14  8.3e-10      39  BP            mitotic cell cycle checkpoint
## 101          black 2886  1.2e-13  9.6e-10      45  BP                           nuclear export
## 102          black 2886  1.4e-13  1.2e-09      57  BP microtubule cytoskeleton organization in
## 103          black 2886  2.0e-13  1.7e-09      50  BP   regulation of mitotic nuclear division
## 104          black 2886  2.4e-13  2.0e-09      40  BP                  chromosome condensation
## 105          black 2886  2.5e-13  2.1e-09      49  BP negative regulation of mitotic cell cycl
## 106          black 2886  3.0e-13  2.5e-09      84  CC             transcription factor complex
## 107          black 2886  3.2e-13  2.7e-09      48  BP             mitotic spindle organization
## 108          black 2886  3.2e-13  2.7e-09      37  CC                              kinetochore
## 109          black 2886  3.4e-13  2.9e-09      34  BP                           mRNA transport
## 110          black 2886  3.5e-13  2.9e-09      35  MF            DNA-dependent ATPase activity
## 111          black 2886  4.1e-13  3.4e-09     227  MF                              ATP binding
## 112          black 2886  5.3e-13  4.5e-09      61  MF                        nuclease activity
## 113          black 2886  5.3e-13  4.5e-09      41  BP              protein export from nucleus
## 114          black 2886  5.6e-13  4.7e-09     123  BP                            cell division
## 115          black 2886  7.1e-13  5.9e-09      39  BP chromosome organization involved in meio
## 116          black 2886  7.3e-13  6.1e-09      93  BP                           gene silencing
## 117          black 2886  9.0e-13  7.5e-09      30  BP                      nucleosome assembly
## 118          black 2886  9.0e-13  7.5e-09      30  CC             catalytic step 2 spliceosome
## 119          black 2886  9.4e-13  7.9e-09     548  BP regulation of cellular metabolic process
## 120          black 2886  1.0e-12  8.5e-09      98  BP        protein localization to organelle
## 121          black 2886  1.3e-12  1.1e-08      41  BP                      import into nucleus
## 122          black 2886  1.3e-12  1.1e-08      67  BP                              methylation
## 123          black 2886  1.5e-12  1.3e-08      40  BP              protein import into nucleus
## 124          black 2886  2.1e-12  1.7e-08      38  BP                       chromatin assembly
## 125          black 2886  2.2e-12  1.8e-08      49  MF          nucleotidyltransferase activity
## 126          black 2886  2.5e-12  2.1e-08      28  BP                 mRNA export from nucleus
## 127          black 2886  2.8e-12  2.3e-08      45  BP        positive regulation of cell cycle
## 128          black 2886  2.9e-12  2.4e-08      35  BP ribonucleoprotein complex export from nu
## 129          black 2886  3.4e-12  2.9e-08      24  BP               nucleotide-excision repair
## 130          black 2886  3.4e-12  2.9e-08      24  CC                          90S preribosome
## 131          black 2886  3.8e-12  3.1e-08      80  BP                   oocyte differentiation
## 132          black 2886  4.6e-12  3.8e-08      46  BP transcription initiation from RNA polyme
## 133          black 2886  4.7e-12  4.0e-08     228  MF            adenyl ribonucleotide binding
## 134          black 2886  4.8e-12  4.0e-08      51  BP               double-strand break repair
## 135          black 2886  5.2e-12  4.4e-08     220  BP   negative regulation of gene expression
## 136          black 2886  6.3e-12  5.3e-08      50  BP  DNA-templated transcription, initiation
## 137          black 2886  6.9e-12  5.8e-08     274  MF                purine nucleotide binding
## 138          black 2886  7.0e-12  5.9e-08      26  BP         reciprocal meiotic recombination
## 139          black 2886  8.6e-12  7.2e-08      34  CC                               pole plasm
## 140          black 2886  9.0e-12  7.5e-08      51  BP                      protein acetylation
## 141          black 2886  1.0e-11  8.5e-08      32  MF           RNA methyltransferase activity
## 142          black 2886  1.2e-11  9.7e-08      20  MF                           snoRNA binding
## 143          black 2886  1.9e-11  1.6e-07      92  BP posttranscriptional regulation of gene e
## 144          black 2886  1.9e-11  1.6e-07     158  BP                              translation
## 145          black 2886  2.0e-11  1.7e-07      58  BP                     chromatin remodeling
## 146          black 2886  2.3e-11  2.0e-07      45  CC    cytoplasmic ribonucleoprotein granule
## 147          black 2886  2.5e-11  2.1e-07    2262  CC                       cellular_component
## 148          black 2886  2.6e-11  2.2e-07      21  BP               DNA replication initiation
## 149          black 2886  2.8e-11  2.3e-07      68  BP       ribonucleoprotein complex assembly
## 150          black 2886  2.9e-11  2.4e-07      26  BP                 homologous recombination
## 151          black 2886  2.9e-11  2.4e-07      49  BP                          tRNA processing
## 152          black 2886  3.1e-11  2.6e-07      31  BP                      tRNA aminoacylation
## 153          black 2886  3.1e-11  2.6e-07      31  BP                 DNA biosynthetic process
## 154          black 2886  3.3e-11  2.7e-07      30  BP tRNA aminoacylation for protein translat
## 155          black 2886  4.5e-11  3.7e-07      19  MF   structural constituent of nuclear pore
## 156          black 2886  4.6e-11  3.8e-07      45  CC                ribonucleoprotein granule
## 157          black 2886  5.0e-11  4.2e-07      42  BP                     nucleus organization
## 158          black 2886  5.0e-11  4.2e-07      47  BP                      histone acetylation
## 159          black 2886  5.6e-11  4.7e-07      58  CC     nuclear transcription factor complex
## 160          black 2886  6.1e-11  5.1e-07      38  BP       ribosomal small subunit biogenesis
## 161          black 2886  7.1e-11  6.0e-07      50  CC RNA polymerase II transcription factor c
## 162          black 2886  8.7e-11  7.3e-07      45  CC  small nuclear ribonucleoprotein complex
## 163          black 2886  9.3e-11  7.8e-07      47  BP              peptidyl-lysine acetylation
## 164          black 2886  9.7e-11  8.1e-07      20  BP                  snRNA 3'-end processing
## 165          black 2886  1.0e-10  8.4e-07      29  MF                    RNA helicase activity
## 166          black 2886  1.1e-10  8.9e-07      27  MF      ATP-dependent RNA helicase activity
## 167          black 2886  1.1e-10  8.9e-07      27  MF            RNA-dependent ATPase activity
## 168          black 2886  1.2e-10  9.9e-07     162  BP                microtubule-based process
## 169          black 2886  1.2e-10  1.0e-06      46  BP                   mRNA catabolic process
## 170          black 2886  1.5e-10  1.3e-06      53  BP        establishment of RNA localization
## 171          black 2886  1.6e-10  1.3e-06      21  BP                         snRNA processing
## 172          black 2886  1.9e-10  1.6e-06     127  BP        ovarian follicle cell development
## 173          black 2886  1.9e-10  1.6e-06      32  BP                          RNA methylation
## 174          black 2886  3.0e-10  2.6e-06      28  BP                    DNA damage checkpoint
## 175          black 2886  3.1e-10  2.6e-06      54  CC                                  spindle
## 176          black 2886  3.3e-10  2.7e-06      40  BP nucleic acid phosphodiester bond hydroly
## 177          black 2886  3.3e-10  2.8e-06      25  BP               oocyte karyosome formation
## 178          black 2886  3.6e-10  3.0e-06      19  CC                 nuclear replication fork
## 179          black 2886  3.7e-10  3.1e-06      33  BP     regulation of chromosome segregation
## 180          black 2886  4.1e-10  3.4e-06     360  BP             transcription, DNA-templated
## 181          black 2886  6.2e-10  5.2e-06      86  BP                       axis specification
## 182          black 2886  6.3e-10  5.3e-06      30  BP      mitotic sister chromatid separation
## 183          black 2886  7.0e-10  5.9e-06      37  BP nuclear-transcribed mRNA catabolic proce
## 184          black 2886  7.1e-10  5.9e-06      46  BP                eggshell chorion assembly
## 185          black 2886  7.8e-10  6.5e-06      51  BP                            RNA transport
## 186          black 2886  8.2e-10  6.9e-06      28  BP      regulation of chromosome separation
## 187          black 2886  8.4e-10  7.0e-06      33  BP double-strand break repair via homologou
## 188          black 2886  8.4e-10  7.0e-06      33  BP                   recombinational repair
## 189          black 2886  9.1e-10  7.6e-06      41  MF                    endonuclease activity
## 190          black 2886  9.1e-10  7.6e-06      47  BP          intracellular mRNA localization
## 191          black 2886  9.2e-10  7.7e-06      27  MF           aminoacyl-tRNA ligase activity
## 192          black 2886  9.2e-10  7.7e-06      22  BP establishment of chromosome localization
## 193          black 2886  9.9e-10  8.3e-06      26  CC             condensed nuclear chromosome
## 194          black 2886  1.1e-09  8.9e-06      67  BP                       oocyte development
## 195          black 2886  1.1e-09  9.3e-06      83  BP regulation of gene expression, epigeneti
## 196          black 2886  1.2e-09  1.0e-05      40  BP                         spindle assembly
## 197          black 2886  1.8e-09  1.5e-05      29  BP    G2/M transition of mitotic cell cycle
## 198          black 2886  2.0e-09  1.7e-05      19  BP                        DNA amplification
## 199          black 2886  2.0e-09  1.7e-05      19  BP              regulation of transposition
## 200          black 2886  2.0e-09  1.7e-05     391  MF                     transferase activity
## 201          black 2886  2.2e-09  1.9e-05     297  MF                       nucleotide binding
## 202          black 2886  2.4e-09  2.0e-05      27  BP metaphase/anaphase transition of mitotic
## 203          black 2886  2.4e-09  2.0e-05      27  BP regulation of mitotic sister chromatid s
## 204          black 2886  2.4e-09  2.0e-05      27  CC                                P granule
## 205          black 2886  2.7e-09  2.3e-05      26  BP regulation of G2/M transition of mitotic
## 206          black 2886  3.3e-09  2.8e-05     366  BP      regulation of RNA metabolic process
## 207          black 2886  4.8e-09  4.0e-05      17  BP              spliceosomal snRNP assembly
## 208          black 2886  5.2e-09  4.4e-05      81  MF                             mRNA binding
## 209          black 2886  5.9e-09  5.0e-05      34  BP                        tRNA modification
## 210          black 2886  6.0e-09  5.0e-05      27  BP                   maturation of SSU-rRNA
## 211          black 2886  6.0e-09  5.0e-05      27  MF                    DNA helicase activity
## 212          black 2886  8.0e-09  6.7e-05      25  BP                     telomere maintenance
## 213          black 2886  8.9e-09  7.4e-05      19  CC condensed chromosome, centromeric region
## 214          black 2886  9.0e-09  7.6e-05      24  BP regulation of mitotic metaphase/anaphase
## 215          black 2886  9.0e-09  7.6e-05      24  MF DNA-directed 5'-3' RNA polymerase activi
## 216          black 2886  1.2e-08  1.0e-04     183  BP            cellular protein localization
## 217          black 2886  1.3e-08  1.1e-04      59  BP                oocyte axis specification
## 218          black 2886  1.4e-08  1.2e-04      27  CC                               germ plasm
## 219          black 2886  1.6e-08  1.4e-04      53  BP                mitochondrial translation
## 220          black 2886  2.3e-08  1.9e-04      36  MF                          histone binding
## 221          black 2886  2.4e-08  2.0e-04      43  CC        protein acetyltransferase complex
## 222          black 2886  2.5e-08  2.1e-04      17  BP      eggshell chorion gene amplification
## 223          black 2886  2.5e-08  2.1e-04      17  BP     negative regulation of transposition
## 224          black 2886  3.0e-08  2.5e-04      22  BP                      RNA polyadenylation
## 225          black 2886  3.2e-08  2.6e-04      27  MF              single-stranded DNA binding
## 226          black 2886  3.3e-08  2.7e-04      19  BP              metaphase plate congression
## 227          black 2886  3.4e-08  2.9e-04     131  BP              epithelial cell development
## 228          black 2886  3.7e-08  3.1e-04      14  BP    female meiosis chromosome segregation
## 229          black 2886  3.7e-08  3.1e-04      14  CC                                 U5 snRNP
## 230          black 2886  3.8e-08  3.2e-04      59  BP                      oocyte construction
## 231          black 2886  4.9e-08  4.1e-04      31  MF                    chromatin DNA binding
## 232          black 2886  5.2e-08  4.4e-04      37  BP                      pole plasm assembly
## 233          black 2886  5.3e-08  4.5e-04      28  BP                   mRNA 3'-end processing
## 234          black 2886  6.3e-08  5.3e-04      15  BP               DNA replication checkpoint
## 235          black 2886  6.9e-08  5.8e-04     640  BP                protein metabolic process
## 236          black 2886  7.2e-08  6.1e-04      44  BP oocyte anterior/posterior axis specifica
## 237          black 2886  7.9e-08  6.6e-04      22  BP                sister chromatid cohesion
## 238          black 2886  8.3e-08  6.9e-04      45  BP maternal determination of anterior/poste
## 239          black 2886  8.5e-08  7.1e-04      16  BP       mitotic G2/M transition checkpoint
## 240          black 2886  8.5e-08  7.1e-04      16  CC small nucleolar ribonucleoprotein comple
## 241          black 2886  8.5e-08  7.1e-04      16  CC     preribosome, large subunit precursor
## 242          black 2886  8.5e-08  7.1e-04      16  CC             U4/U6 x U5 tri-snRNP complex
## 243          black 2886  8.5e-08  7.1e-04      16  MF cyclin-dependent protein serine/threonin
## 244          black 2886  9.8e-08  8.2e-04      20  CC             chromosome, telomeric region
## 245          black 2886  1.0e-07  8.5e-04      51  BP                 asymmetric cell division
## 246          black 2886  1.1e-07  8.8e-04      18  BP negative regulation of G2/M transition o
## 247          black 2886  1.1e-07  8.8e-04      18  CC                               nucleosome
## 248          black 2886  1.1e-07  8.8e-04      18  MF endoribonuclease activity, producing 5'-
## 249          black 2886  1.3e-07  1.1e-03      24  BP                            fertilization
## 250          black 2886  1.5e-07  1.2e-03     177  MF                 pyrophosphatase activity
## 251          black 2886  1.5e-07  1.3e-03      43  MF S-adenosylmethionine-dependent methyltra
## 252          black 2886  1.6e-07  1.3e-03     112  CC                 microtubule cytoskeleton
## 253          black 2886  1.7e-07  1.4e-03      53  BP                      chromatin silencing
## 254          black 2886  1.7e-07  1.5e-03     179  MF hydrolase activity, acting on acid anhyd
## 255          black 2886  1.9e-07  1.6e-03      22  CC                      spindle microtubule
## 256          black 2886  2.1e-07  1.7e-03     332  BP regulation of transcription, DNA-templat
## 257          black 2886  2.1e-07  1.8e-03      11  BP       mitotic DNA replication checkpoint
## 258          black 2886  2.1e-07  1.8e-03      11  MF           DNA replication origin binding
## 259          black 2886  2.6e-07  2.2e-03     170  MF       nucleoside-triphosphatase activity
## 260          black 2886  2.6e-07  2.2e-03      20  BP                     mRNA polyadenylation
## 261          black 2886  2.7e-07  2.2e-03     177  MF hydrolase activity, acting on acid anhyd
## 262          black 2886  3.1e-07  2.6e-03      31  BP              pole plasm RNA localization
## 263          black 2886  3.1e-07  2.6e-03      62  MF       transcription coregulator activity
## 264          black 2886  3.5e-07  2.9e-03      47  BP  germarium-derived egg chamber formation
## 265          black 2886  3.6e-07  3.0e-03      23  BP            regulation of DNA replication
## 266          black 2886  3.9e-07  3.3e-03      40  BP                         centrosome cycle
## 267          black 2886  4.4e-07  3.7e-03     220  BP                     protein localization
## 268          black 2886  4.4e-07  3.7e-03      22  BP                     single fertilization
## 269          black 2886  4.5e-07  3.8e-03      80  CC                      polytene chromosome
## 270          black 2886  4.7e-07  3.9e-03     128  BP          intracellular protein transport
## 271          black 2886  5.2e-07  4.4e-03      27  BP positive regulation of cell cycle proces
## 272          black 2886  5.3e-07  4.4e-03      12  BP vitelline membrane formation involved in
## 273          black 2886  5.5e-07  4.6e-03      58  BP    anterior/posterior axis specification
## 274          black 2886  5.6e-07  4.7e-03      81  BP protein modification by small protein co
## 275          black 2886  5.9e-07  5.0e-03      24  CC                        nuclear periphery
## 276          black 2886  7.2e-07  6.0e-03      30  BP             pole plasm mRNA localization
## 277          black 2886  7.3e-07  6.1e-03      28  CC                   protein kinase complex
## 278          black 2886  7.3e-07  6.1e-03      28  MF       histone acetyltransferase activity
## 279          black 2886  7.7e-07  6.5e-03      19  CC                                 U2 snRNP
## 280          black 2886  7.7e-07  6.5e-03      19  MF                      damaged DNA binding
## 281          black 2886  8.0e-07  6.7e-03     178  BP                peptide metabolic process
## 282          black 2886  8.1e-07  6.8e-03      13  BP                         rRNA methylation
## 283          black 2886  8.2e-07  6.9e-03      39  BP                    gene silencing by RNA
## 284          black 2886  8.9e-07  7.4e-03      18  CC                          spindle midzone
## 285          black 2886  9.0e-07  7.6e-03      25  BP            mitotic cell cycle, embryonic
## 286          black 2886  9.0e-07  7.6e-03      25  BP positive regulation of mitotic cell cycl
## 287          black 2886  9.2e-07  7.7e-03      31  BP       ribosomal large subunit biogenesis
## 288          black 2886  9.9e-07  8.3e-03      17  CC                             chromocenter
## 289          black 2886  9.9e-07  8.3e-03      14  BP         regulation of meiotic cell cycle
## 290          black 2886  1.1e-06  9.0e-03      15  BP negative regulation of mitotic metaphase
## 291          black 2886  1.2e-06  9.9e-03      24  CC  serine/threonine protein kinase complex
## 292          black 2886  1.3e-06  1.1e-02      44  BP microtubule organizing center organizati
## 293          black 2886  1.3e-06  1.1e-02     197  BP                cytoskeleton organization
## 294          black 2886  1.4e-06  1.1e-02      36  MF             N-acetyltransferase activity
## 295          black 2886  1.4e-06  1.1e-02      33  BP intracellular mRNA localization involved
## 296          black 2886  1.5e-06  1.3e-02      20  BP maturation of SSU-rRNA from tricistronic
## 297          black 2886  1.6e-06  1.3e-02      34  BP intracellular mRNA localization involved
## 298          black 2886  1.6e-06  1.3e-02      34  MF                    ribonuclease activity
## 299          black 2886  1.8e-06  1.5e-02      55  BP             embryonic axis specification
## 300          black 2886  2.0e-06  1.7e-02      11  CC mRNA cleavage and polyadenylation specif
## 301          black 2886  2.0e-06  1.7e-02     127  BP                       cell proliferation
## 302          black 2886  2.1e-06  1.7e-02      40  MF               acetyltransferase activity
## 303          black 2886  2.3e-06  1.9e-02      28  CC                          heterochromatin
## 304          black 2886  2.3e-06  2.0e-02      37  MF               N-acyltransferase activity
## 305          black 2886  2.5e-06  2.1e-02      53  BP anterior/posterior axis specification, e
## 306          black 2886  2.5e-06  2.1e-02      34  BP            asymmetric stem cell division
## 307          black 2886  2.8e-06  2.4e-02      12  BP spindle assembly involved in female meio
## 308          black 2886  2.8e-06  2.4e-02      12  BP regulation of mitotic cell cycle, embryo
## 309          black 2886  2.8e-06  2.4e-02      12  BP N-terminal protein amino acid modificati
## 310          black 2886  2.8e-06  2.4e-02      12  BP                         RNA surveillance
## 311          black 2886  2.8e-06  2.4e-02      12  CC         condensed chromosome kinetochore
## 312          black 2886  2.8e-06  2.4e-02      12  CC                       integrator complex
## 313          black 2886  3.0e-06  2.5e-02      23  BP       ATP-dependent chromatin remodeling
## 314          black 2886  3.0e-06  2.5e-02      16  BP          mitotic chromosome condensation
## 315          black 2886  3.0e-06  2.5e-02      16  BP negative regulation of sister chromatid 
## 316          black 2886  3.1e-06  2.6e-02      48  BP                       stem cell division
## 317          black 2886  3.2e-06  2.7e-02      20  CC                          mitotic spindle
## 318          black 2886  3.3e-06  2.8e-02      13  BP DNA strand elongation involved in DNA re
## 319          black 2886  3.3e-06  2.8e-02      13  BP anaphase-promoting complex-dependent cat
## 320          black 2886  3.3e-06  2.8e-02      13  MF     DNA-directed DNA polymerase activity
## 321          black 2886  3.3e-06  2.8e-02      56  BP                regulation of translation
## 322          black 2886  3.4e-06  2.9e-02      14  BP        transcription by RNA polymerase I
## 323          black 2886  3.4e-06  2.9e-02      14  BP      mitotic metaphase plate congression
## 324          black 2886  3.4e-06  2.9e-02      14  BP      mitotic spindle assembly checkpoint
## 325          black 2886  3.4e-06  2.9e-02      33  BP       posttranscriptional gene silencing
## 326          black 2886  3.5e-06  2.9e-02       9  CC                        chaperone complex
## 327          black 2886  4.3e-06  3.6e-02      48  BP     regulation of chromatin organization
## 328          black 2886  5.2e-06  4.4e-02      27  BP       pole plasm oskar mRNA localization
## 329          black 2886  5.2e-06  4.4e-02      27  MF                endoribonuclease activity
## 330          black 2886  5.2e-06  4.4e-02      37  BP        dorsal/ventral axis specification
## 331          black 2886  5.6e-06  4.7e-02      23  BP             meiotic spindle organization
## 332          black 2886  6.3e-06  5.3e-02      17  CC             U2-type spliceosomal complex
## 333          black 2886  6.7e-06  5.6e-02     314  MF                            anion binding
## 334          black 2886  6.8e-06  5.7e-02     112  MF                          ATPase activity
## 335          black 2886  6.9e-06  5.8e-02     240  MF                             drug binding
## 336          black 2886  7.3e-06  6.1e-02      10  CC          nuclear exosome (RNase complex)
## 337          black 2886  7.3e-06  6.1e-02      10  CC  transcription factor TFIIH holo complex
## 338          black 2886  8.9e-06  7.4e-02      15  BP            mitotic DNA damage checkpoint
## 339          black 2886  8.9e-06  7.4e-02      15  MF              3'-5' DNA helicase activity
## 340          black 2886  9.8e-06  8.2e-02      35  CC        histone acetyltransferase complex
## 341          black 2886  9.9e-06  8.3e-02      11  BP                 nuclear RNA surveillance
## 342          black 2886  9.9e-06  8.3e-02      11  CC                  kinetochore microtubule
## 343          black 2886  1.0e-05  8.4e-02      31  BP posttranscriptional gene silencing by RN
## 344          black 2886  1.0e-05  8.4e-02      14  BP                         tRNA methylation
## 345          black 2886  1.1e-05  8.9e-02     287  MF          carbohydrate derivative binding
## 346          black 2886  1.1e-05  9.1e-02      13  BP                  maturation of 5.8S rRNA
## 347          black 2886  1.1e-05  9.1e-02      13  BP negative regulation of DNA recombination
## 348          black 2886  1.1e-05  9.1e-02      13  CC             mRNA cleavage factor complex
## 349          black 2886  1.1e-05  9.1e-02      13  MF                            snRNA binding
## 350          black 2886  1.1e-05  9.2e-02      12  BP                  piRNA metabolic process
## 351          black 2886  1.1e-05  9.2e-02      12  CC               anaphase-promoting complex
## 352          black 2886  1.1e-05  9.2e-02      12  CC carboxy-terminal domain protein kinase c
## 353          black 2886  1.3e-05  1.1e-01      22  MF               deoxyribonuclease activity
## 354          black 2886  1.4e-05  1.1e-01      30  MF                kinase regulator activity
## 355          black 2886  1.4e-05  1.1e-01      42  CC     cullin-RING ubiquitin ligase complex
## 356          black 2886  1.4e-05  1.2e-01       8  BP promoter clearance from RNA polymerase I
## 357          black 2886  1.4e-05  1.2e-01       8  BP transcriptional open complex formation a
## 358          black 2886  1.4e-05  1.2e-01       8  BP                         female meiosis I
## 359          black 2886  1.4e-05  1.2e-01       8  CC          chaperonin-containing T-complex
## 360          black 2886  1.5e-05  1.3e-01     165  BP   positive regulation of gene expression
## 361          black 2886  1.6e-05  1.4e-01      28  MF        protein kinase regulator activity
## 362          black 2886  1.7e-05  1.4e-01      19  BP                 mitotic spindle assembly
## 363          black 2886  1.7e-05  1.5e-01     145  MF transferase activity, transferring phosp
## 364          black 2886  1.7e-05  1.5e-01      16  BP                    DNA endoreduplication
## 365          black 2886  1.7e-05  1.5e-01      16  CC       transcription factor TFIID complex
## 366          black 2886  1.7e-05  1.5e-01      16  MF      ATP-dependent DNA helicase activity
## 367          black 2886  2.1e-05  1.8e-01      15  BP negative regulation of mitotic nuclear d
## 368          black 2886  2.1e-05  1.8e-01      15  BP     spindle assembly involved in meiosis
## 369          black 2886  2.2e-05  1.9e-01      18  BP nuclear-transcribed mRNA catabolic proce
## 370          black 2886  2.4e-05  2.0e-01      20  MF               3'-5' exonuclease activity
## 371          black 2886  2.6e-05  2.2e-01     153  MF                         zinc ion binding
## 372          black 2886  2.6e-05  2.2e-01      25  BP             heterochromatin organization
## 373          black 2886  2.7e-05  2.2e-01      33  BP                      mitotic cytokinesis
## 374          black 2886  2.7e-05  2.3e-01     113  MF              double-stranded DNA binding
## 375          black 2886  2.7e-05  2.3e-01       9  BP                nuclear pore organization
## 376          black 2886  2.7e-05  2.3e-01       9  BP           histone mRNA metabolic process
## 377          black 2886  2.7e-05  2.3e-01       9  BP                   rRNA catabolic process
## 378          black 2886  2.7e-05  2.3e-01       9  BP                 snoRNA 3'-end processing
## 379          black 2886  2.7e-05  2.3e-01       9  CC   condensed chromosome outer kinetochore
## 380          black 2886  2.8e-05  2.4e-01      56  CC            microtubule organizing center
## 381          black 2886  2.9e-05  2.4e-01      17  BP                         histone exchange
## 382          black 2886  3.0e-05  2.5e-01      35  BP            male meiotic nuclear division
## 383          black 2886  3.0e-05  2.5e-01      54  CC                 ubiquitin ligase complex
## 384          black 2886  3.2e-05  2.7e-01      21  MF RNA polymerase II transcription factor b
## 385          black 2886  3.4e-05  2.8e-01      12  BP            extracellular matrix assembly
## 386          black 2886  3.4e-05  2.9e-01      10  BP            meiotic chromosome separation
## 387          black 2886  3.4e-05  2.9e-01      10  CC condensed nuclear chromosome, centromeri
## 388          black 2886  3.4e-05  2.9e-01      10  CC                            Prp19 complex
## 389          black 2886  3.4e-05  2.9e-01      10  CC            U12-type spliceosomal complex
## 390          black 2886  3.4e-05  2.9e-01      10  MF          rRNA methyltransferase activity
## 391          black 2886  3.6e-05  3.0e-01      11  BP                          mismatch repair
## 392          black 2886  3.6e-05  3.0e-01      11  BP                  nuclear DNA replication
## 393          black 2886  3.6e-05  3.0e-01      11  CC                  exosome (RNase complex)
## 394          black 2886  3.7e-05  3.1e-01     401  BP                         cell development
## 395          black 2886  3.9e-05  3.3e-01      28  BP                       protein lipidation
## 396          black 2886  4.0e-05  3.4e-01      22  BP                   centrosome duplication
## 397          black 2886  4.3e-05  3.6e-01      18  BP          GPI anchor biosynthetic process
## 398          black 2886  4.3e-05  3.6e-01      18  BP                           response to UV
## 399          black 2886  4.6e-05  3.8e-01      46  BP                          protein folding
## 400          black 2886  4.8e-05  4.0e-01      15  BP production of small RNA involved in gene
## 401          black 2886  5.2e-05  4.3e-01     148  BP    establishment of protein localization
## 402          black 2886  5.5e-05  4.6e-01      88  BP                   male gamete generation
## 403          black 2886  5.7e-05  4.8e-01       7  BP            peptidyl-arginine methylation
## 404          black 2886  5.7e-05  4.8e-01       7  BP             histone arginine methylation
## 405          black 2886  5.7e-05  4.8e-01       7  BP          peptidyl-arginine N-methylation
## 406          black 2886  5.7e-05  4.8e-01       7  BP                    rRNA base methylation
## 407          black 2886  5.7e-05  4.8e-01       7  CC  transcription factor TFIIH core complex
## 408          black 2886  5.7e-05  4.8e-01       7  CC                              MCM complex
## 409          black 2886  5.7e-05  4.8e-01       7  CC                                  Yb body
## 410          black 2886  5.7e-05  4.8e-01       7  CC                 cytoplasmic U snRNP body
## 411          black 2886  5.7e-05  4.8e-01       7  MF                DNA clamp loader activity
## 412          black 2886  5.7e-05  4.8e-01       7  MF histone-arginine N-methyltransferase act
## 413          black 2886  5.7e-05  4.8e-01      24  BP             germ-line stem cell division
## 414          black 2886  5.8e-05  4.9e-01      19  BP    G1/S transition of mitotic cell cycle
## 415          black 2886  5.9e-05  4.9e-01      70  BP                   protein ubiquitination
## 416          black 2886  6.0e-05  5.0e-01      14  BP regulation of cyclin-dependent protein s
## 417          black 2886  6.0e-05  5.0e-01      14  MF peptidyl-prolyl cis-trans isomerase acti
## 418          black 2886  6.2e-05  5.2e-01      44  CC                               centrosome
## 419          black 2886  7.4e-05  6.2e-01      13  MF transcription factor activity, RNA polym
## 420          black 2886  7.4e-05  6.2e-01      13  MF                       nucleosome binding
## 421          black 2886  8.3e-05  7.0e-01      36  CC                   mitochondrial ribosome
## 422          black 2886  8.9e-05  7.5e-01      12  BP    protein peptidyl-prolyl isomerization
## 423          black 2886  8.9e-05  7.5e-01      12  BP production of miRNAs involved in gene si
## 424          black 2886  9.1e-05  7.7e-01      21  MF                             rRNA binding
## 425          black 2886  9.9e-05  8.3e-01       8  BP                            mRNA cleavage
## 426          black 2886  9.9e-05  8.3e-01       8  BP N-terminal protein amino acid acetylatio
## 427          black 2886  9.9e-05  8.3e-01       8  BP            intra-S DNA damage checkpoint
## 428          black 2886  9.9e-05  8.3e-01       8  BP                nuclear mRNA surveillance
## 429          black 2886  9.9e-05  8.3e-01       8  CC                  nuclear pore outer ring
## 430          black 2886  9.9e-05  8.3e-01       8  CC              site of double-strand break
## 431          black 2886  9.9e-05  8.3e-01       8  MF peptide alpha-N-acetyltransferase activi
## 432          black 2886  1.0e-04  8.4e-01      25  BP        regulation of chromatin silencing
## 433          black 2886  1.0e-04  8.7e-01      11  BP        homologous chromosome segregation
## 434          black 2886  1.1e-04  8.8e-01      17  BP                       centriole assembly
## 435          black 2886  1.1e-04  9.1e-01      33  BP                      protein methylation
## 436          black 2886  1.1e-04  9.2e-01     142  BP                        protein transport
## 437          black 2886  1.1e-04  9.5e-01      36  BP             regulation of gene silencing
## 438          black 2886  1.1e-04  9.5e-01      38  CC                              microtubule
## 439          black 2886  1.1e-04  9.6e-01      10  BP                     base-excision repair
## 440          black 2886  1.1e-04  9.6e-01      10  BP        poly(A)+ mRNA export from nucleus
## 441          black 2886  1.1e-04  9.6e-01      10  CC                 RNA polymerase I complex
## 442          black 2886  1.2e-04  9.7e-01       9  BP                   endomitotic cell cycle
## 443          black 2886  1.2e-04  9.7e-01       9  MF      3'-5'-exodeoxyribonuclease activity
## 444          black 2886  1.2e-04  9.7e-01       9  MF RNA polymerase II CTD heptapeptide repea
## 445          black 2886  1.2e-04  1.0e+00      27  BP                 translational initiation
## 446          black 2886  1.2e-04  1.0e+00      20  BP       protein targeting to mitochondrion
## 447          black 2886  1.2e-04  1.0e+00      20  BP           response to ionizing radiation
## 448          black 2886  1.2e-04  1.0e+00      20  BP       positive regulation of translation
## 449          black 2886  1.3e-04  1.0e+00      14  CC                             SAGA complex
## 450          black 2886  1.3e-04  1.0e+00      14  CC                        SAGA-type complex
## 451          black 2886  1.3e-04  1.0e+00      29  BP                      histone methylation
## 452          black 2886  1.3e-04  1.0e+00      29  CC                             nuclear body
## 453          black 2886  1.4e-04  1.0e+00      18  BP             GPI anchor metabolic process
## 454          black 2886  1.4e-04  1.0e+00      18  CC                                   P-body
## 455          black 2886  1.4e-04  1.0e+00      94  MF                 ATPase activity, coupled
## 456          black 2886  1.4e-04  1.0e+00      16  BP regulation of response to DNA damage sti
## 457          black 2886  1.4e-04  1.0e+00      16  CC                              euchromatin
## 458          black 2886  1.7e-04  1.0e+00     123  BP negative regulation of transcription, DN
## 459          black 2886  1.9e-04  1.0e+00      17  CC                         mediator complex
## 460          black 2886  2.1e-04  1.0e+00      24  MF        ribonucleoprotein complex binding
## 461          black 2886  2.1e-04  1.0e+00      12  BP female germ-line stem cell asymmetric di
## 462          black 2886  2.3e-04  1.0e+00       6  BP regulation of syncytial blastoderm mitot
## 463          black 2886  2.3e-04  1.0e+00       6  BP                      protein sumoylation
## 464          black 2886  2.3e-04  1.0e+00       6  BP peptidyl-arginine methylation, to asymme
## 465          black 2886  2.3e-04  1.0e+00       6  BP              replication fork processing
## 466          black 2886  2.3e-04  1.0e+00       6  BP                 distributive segregation
## 467          black 2886  2.3e-04  1.0e+00       6  BP nucleotide-excision repair, DNA incision
## 468          black 2886  2.3e-04  1.0e+00       6  CC               origin recognition complex
## 469          black 2886  2.3e-04  1.0e+00       6  CC nuclear origin of replication recognitio
## 470          black 2886  2.3e-04  1.0e+00       6  CC Pwp2p-containing subcomplex of 90S preri
## 471          black 2886  2.3e-04  1.0e+00       6  CC                                 egg coat
## 472          black 2886  2.3e-04  1.0e+00       6  MF protein-arginine omega-N asymmetric meth
## 473          black 2886  2.3e-04  1.0e+00      21  CC            polytene chromosome interband
## 474          black 2886  2.3e-04  1.0e+00      18  BP transcription elongation from RNA polyme
## 475          black 2886  2.5e-04  1.0e+00      16  BP                    centriole replication
## 476          black 2886  2.5e-04  1.0e+00      16  CC                                centriole
## 477          black 2886  2.5e-04  1.0e+00      16  CC        transcriptional repressor complex
## 478          black 2886  2.5e-04  1.0e+00      16  MF basal RNA polymerase II transcription ma
## 479          black 2886  2.6e-04  1.0e+00      28  BP            lipoprotein metabolic process
## 480          black 2886  2.6e-04  1.0e+00      11  BP     cleavage involved in rRNA processing
## 481          black 2886  2.7e-04  1.0e+00      29  MF       transcription coactivator activity
## 482          black 2886  2.8e-04  1.0e+00      19  CC  transcription elongation factor complex
## 483          black 2886  2.9e-04  1.0e+00      74  MF       protein-containing complex binding
## 484          black 2886  3.1e-04  1.0e+00      10  BP maturation of 5.8S rRNA from tricistroni
## 485          black 2886  3.1e-04  1.0e+00      10  BP                    mitotic recombination
## 486          black 2886  3.1e-04  1.0e+00      10  BP  NLS-bearing protein import into nucleus
## 487          black 2886  3.1e-04  1.0e+00      10  MF ATP-dependent 3'-5' RNA helicase activit
## 488          black 2886  3.3e-04  1.0e+00      13  MF transcription factor activity, RNA polym
## 489          black 2886  3.6e-04  1.0e+00       7  BP mRNA 3'-end processing by stem-loop bind
## 490          black 2886  3.6e-04  1.0e+00       7  BP         septin cytoskeleton organization
## 491          black 2886  3.6e-04  1.0e+00       7  BP nuclear-transcribed mRNA catabolic proce
## 492          black 2886  3.6e-04  1.0e+00       7  CC                        prefoldin complex
## 493          black 2886  3.6e-04  1.0e+00       7  CC                            Ndc80 complex
## 494          black 2886  3.6e-04  1.0e+00       7  CC   nuclear pore central transport channel
## 495          black 2886  3.6e-04  1.0e+00       7  CC outer mitochondrial membrane protein com
## 496          black 2886  3.6e-04  1.0e+00       7  MF ubiquitin-like modifier activating enzym
## 497          black 2886  3.6e-04  1.0e+00       7  MF protein-arginine N-methyltransferase act
## 498          black 2886  3.6e-04  1.0e+00       7  MF                   DNA polymerase binding
## 499          black 2886  3.6e-04  1.0e+00       9  BP            interstrand cross-link repair
## 500          black 2886  3.6e-04  1.0e+00       9  CC                              Myb complex
## 501          black 2886  3.6e-04  1.0e+00       9  CC                           brahma complex
## 502          black 2886  3.6e-04  1.0e+00       9  CC    Cul4-RING E3 ubiquitin ligase complex
## 503          black 2886  3.6e-04  1.0e+00       9  CC                       DNA repair complex
## 504          black 2886  3.6e-04  1.0e+00       9  MF                RNA polymerase I activity
## 505          black 2886  3.7e-04  1.0e+00      29  MF             protein transporter activity
## 506          black 2886  3.9e-04  1.0e+00       8  BP double-strand break repair via nonhomolo
## 507          black 2886  3.9e-04  1.0e+00       8  BP                           egg activation
## 508          black 2886  3.9e-04  1.0e+00       8  BP double-strand break repair via synthesis
## 509          black 2886  3.9e-04  1.0e+00       8  BP positive regulation of transcription by 
## 510          black 2886  3.9e-04  1.0e+00       8  CC     preribosome, small subunit precursor
## 511          black 2886  3.9e-04  1.0e+00       8  MF          aminoacyl-tRNA editing activity
## 512          black 2886  4.4e-04  1.0e+00      12  BP                   maturation of LSU-rRNA
## 513          black 2886  4.4e-04  1.0e+00      12  CC               RNA polymerase III complex
## 514          black 2886  5.2e-04  1.0e+00      17  BP                  gene silencing by miRNA
## 515          black 2886  5.2e-04  1.0e+00      17  BP                 germ-line cyst formation
## 516          black 2886  5.2e-04  1.0e+00      17  CC    mitochondrial small ribosomal subunit
## 517          black 2886  5.7e-04  1.0e+00      11  BP regulation of chromatin assembly or disa
## 518          black 2886  5.7e-04  1.0e+00      11  BP pyrimidine nucleotide biosynthetic proce
## 519          black 2886  5.7e-04  1.0e+00      11  BP   negative regulation of DNA replication
## 520          black 2886  5.7e-04  1.0e+00      11  CC   NuA4 histone acetyltransferase complex
## 521          black 2886  6.1e-04  1.0e+00      18  BP                oocyte fate determination
## 522          black 2886  6.5e-04  1.0e+00      33  MF        ubiquitin protein ligase activity
## 523          black 2886  7.2e-04  1.0e+00      16  BP                   histone H3 acetylation
## 524          black 2886  7.3e-04  1.0e+00      10  BP     regulation of oskar mRNA translation
## 525          black 2886  7.3e-04  1.0e+00      10  CC cyclin-dependent protein kinase holoenzy
## 526          black 2886  7.3e-04  1.0e+00      10  CC                      nuclear euchromatin
## 527          black 2886  7.3e-04  1.0e+00      10  MF cyclin-dependent protein serine/threonin
## 528          black 2886  7.5e-04  1.0e+00     460  BP                     cell differentiation
## 529          black 2886  7.7e-04  1.0e+00     957  CC                                cytoplasm
## 530          black 2886  7.8e-04  1.0e+00     201  BP       transcription by RNA polymerase II
## 531          black 2886  8.3e-04  1.0e+00      17  BP germarium-derived oocyte differentiation
## 532          black 2886  8.8e-04  1.0e+00      31  BP               regulation of RNA splicing
## 533          black 2886  9.1e-04  1.0e+00       9  CC     nuclear chromosome, telomeric region
## 534          black 2886  9.1e-04  1.0e+00       9  CC          RNA polymerase II, core complex
## 535          black 2886  9.1e-04  1.0e+00       9  CC                            Ino80 complex
## 536          black 2886  9.1e-04  1.0e+00       9  MF endodeoxyribonuclease activity, producin
## 537          black 2886  9.3e-04  1.0e+00       5  BP transcription-dependent tethering of RNA
## 538          black 2886  9.3e-04  1.0e+00       5  BP spindle assembly involved in female meio
## 539          black 2886  9.3e-04  1.0e+00       5  BP polarity specification of anterior/poste
## 540          black 2886  9.3e-04  1.0e+00       5  BP      attachment of GPI anchor to protein
## 541          black 2886  9.3e-04  1.0e+00       5  BP DNA replication, Okazaki fragment proces
## 542          black 2886  9.3e-04  1.0e+00       5  BP    CENP-A containing nucleosome assembly
## 543          black 2886  9.3e-04  1.0e+00       5  BP                     kinetochore assembly
## 544          black 2886  9.3e-04  1.0e+00       5  BP maintenance of protein location in nucle
## 545          black 2886  9.3e-04  1.0e+00       5  BP nuclear polyadenylation-dependent ncRNA 
## 546          black 2886  9.3e-04  1.0e+00       5  BP protein localization to nuclear envelope
## 547          black 2886  9.3e-04  1.0e+00       5  BP pre-mRNA cleavage required for polyadeny
## 548          black 2886  9.3e-04  1.0e+00       5  CC         DNA replication factor A complex
## 549          black 2886  9.3e-04  1.0e+00       5  CC         DNA replication factor C complex
## 550          black 2886  9.3e-04  1.0e+00       5  CC                    Elg1 RFC-like complex
## 551          black 2886  9.3e-04  1.0e+00       5  CC                   box C/D snoRNP complex
## 552          black 2886  9.3e-04  1.0e+00       5  CC                              SMN complex
## 553          black 2886  9.3e-04  1.0e+00       5  CC                   SMN-Sm protein complex
## 554          black 2886  9.3e-04  1.0e+00       5  CC                           Rb-E2F complex
## 555          black 2886  9.3e-04  1.0e+00       5  CC          GPI-anchor transamidase complex
## 556          black 2886  9.3e-04  1.0e+00       5  CC                       vitelline envelope
## 557          black 2886  9.3e-04  1.0e+00       5  CC                      Lsm1-7-Pat1 complex
## 558          black 2886  9.3e-04  1.0e+00       5  MF RNA polymerase II core promoter sequence
## 559          black 2886  9.3e-04  1.0e+00       5  MF               DNA topoisomerase activity
## 560          black 2886  9.3e-04  1.0e+00       5  MF                         U2 snRNA binding
## 561          black 2886  9.3e-04  1.0e+00       5  MF              5'-3' DNA helicase activity
## 562          black 2886  9.4e-04  1.0e+00     136  CC                             cytoskeleton
## 563          black 2886  9.8e-04  1.0e+00      62  BP anterior/posterior pattern specification
## 564          black 2886  9.9e-04  1.0e+00      15  BP                    pole cell development
## 565          black 2886  9.9e-04  1.0e+00      15  BP germarium-derived oocyte fate determinat
## 566          black 2886  9.9e-04  1.0e+00      15  MF           endodeoxyribonuclease activity
## 567          black 2886  1.0e-03  1.0e+00      19  BP phosphatidylinositol biosynthetic proces
## 568          black 2886  1.1e-03  1.0e+00     129  BP positive regulation of transcription, DN
## 569          black 2886  1.1e-03  1.0e+00      34  BP      regulation of embryonic development
## 570          black 2886  1.1e-03  1.0e+00      20  BP          glycolipid biosynthetic process
## 571          black 2886  1.1e-03  1.0e+00       8  BP nuclear-transcribed mRNA catabolic proce
## 572          black 2886  1.1e-03  1.0e+00       8  BP     cytoplasmic translational initiation
## 573          black 2886  1.1e-03  1.0e+00       8  BP attachment of spindle microtubules to ki
## 574          black 2886  1.1e-03  1.0e+00       8  CC                           nuclear lamina
## 575          black 2886  1.1e-03  1.0e+00       8  CC                   ribonuclease P complex
## 576          black 2886  1.1e-03  1.0e+00       8  CC                          meiotic spindle
## 577          black 2886  1.1e-03  1.0e+00       8  MF                  ribonuclease P activity
## 578          black 2886  1.1e-03  1.0e+00      13  BP  syncytial blastoderm mitotic cell cycle
## 579          black 2886  1.1e-03  1.0e+00      32  BP germ-line stem cell population maintenan
## 580          black 2886  1.1e-03  1.0e+00      11  BP                     DNA duplex unwinding
## 581          black 2886  1.2e-03  1.0e+00      64  MF             transcription factor binding
## 582          black 2886  1.2e-03  1.0e+00      28  MF translation factor activity, RNA binding
## 583          black 2886  1.2e-03  1.0e+00      58  CC                              cell cortex
## 584          black 2886  1.3e-03  1.0e+00       7  BP                    protein deneddylation
## 585          black 2886  1.3e-03  1.0e+00       7  BP resolution of meiotic recombination inte
## 586          black 2886  1.3e-03  1.0e+00       7  BP                   tRNA 5'-leader removal
## 587          black 2886  1.3e-03  1.0e+00       7  BP pyrimidine ribonucleotide biosynthetic p
## 588          black 2886  1.3e-03  1.0e+00       7  BP positive regulation of mitotic metaphase
## 589          black 2886  1.3e-03  1.0e+00       7  BP                 kinetochore organization
## 590          black 2886  1.3e-03  1.0e+00       7  CC      cytoplasmic exosome (RNase complex)
## 591          black 2886  1.3e-03  1.0e+00       7  CC             transcription export complex
## 592          black 2886  1.3e-03  1.0e+00       7  CC                     synaptonemal complex
## 593          black 2886  1.3e-03  1.0e+00       7  CC                         COP9 signalosome
## 594          black 2886  1.3e-03  1.0e+00       7  CC      replication fork protection complex
## 595          black 2886  1.3e-03  1.0e+00       7  CC    polytene chromosome, telomeric region
## 596          black 2886  1.3e-03  1.0e+00       6  BP                leading strand elongation
## 597          black 2886  1.3e-03  1.0e+00       6  BP                lagging strand elongation
## 598          black 2886  1.3e-03  1.0e+00       6  BP              transposition, DNA-mediated
## 599          black 2886  1.3e-03  1.0e+00       6  BP deoxyribonucleotide biosynthetic process
## 600          black 2886  1.3e-03  1.0e+00       6  BP                 primary miRNA processing
## 601          black 2886  1.3e-03  1.0e+00       6  BP                 septin ring organization
## 602          black 2886  1.3e-03  1.0e+00       6  BP               U4 snRNA 3'-end processing
## 603          black 2886  1.3e-03  1.0e+00       6  BP positive regulation of methylation-depen
## 604          black 2886  1.3e-03  1.0e+00       6  CC                                 U4 snRNP
## 605          black 2886  1.3e-03  1.0e+00       6  CC   post-mRNA release spliceosomal complex
## 606          black 2886  1.3e-03  1.0e+00       6  MF            four-way junction DNA binding
## 607          black 2886  1.3e-03  1.0e+00      17  CC                         nuclear membrane
## 608          black 2886  1.3e-03  1.0e+00      17  MF       histone methyltransferase activity
## 609          black 2886  1.4e-03  1.0e+00     136  MF                           enzyme binding
## 610          black 2886  1.5e-03  1.0e+00      10  CC         polytene chromosome chromocenter
## 611          black 2886  1.5e-03  1.0e+00      10  MF              RNA polymerase III activity
## 612          black 2886  1.5e-03  1.0e+00      12  BP positive regulation of chromatin silenci
## 613          black 2886  1.5e-03  1.0e+00      12  BP regulation of DNA-dependent DNA replicat
## 614          black 2886  1.5e-03  1.0e+00      12  MF                       Ran GTPase binding
## 615          black 2886  1.6e-03  1.0e+00      20  MF   translation initiation factor activity
## 616          black 2886  1.6e-03  1.0e+00      67  BP                  blastoderm segmentation
## 617          black 2886  1.6e-03  1.0e+00      15  BP regulation of DNA-templated transcriptio
## 618          black 2886  1.6e-03  1.0e+00     191  BP                  intracellular transport
## 619          black 2886  1.7e-03  1.0e+00      73  MF ubiquitin-like protein transferase activ
## 620          black 2886  1.7e-03  1.0e+00      25  BP regulation of protein serine/threonine k
## 621          black 2886  1.8e-03  1.0e+00      72  BP          embryonic pattern specification
## 622          black 2886  1.8e-03  1.0e+00      33  MF                 unfolded protein binding
## 623          black 2886  1.8e-03  1.0e+00      16  CC                             spindle pole
## 624          black 2886  1.8e-03  1.0e+00      38  BP         stem cell population maintenance
## 625          black 2886  1.8e-03  1.0e+00      69  BP                          spermatogenesis
## 626          black 2886  2.0e-03  1.0e+00      30  BP               protein polyubiquitination
## 627          black 2886  2.0e-03  1.0e+00       9  MF    nuclear localization sequence binding
## 628          black 2886  2.1e-03  1.0e+00      18  BP                     nucleus localization
## 629          black 2886  2.1e-03  1.0e+00      11  MF          tRNA methyltransferase activity
## 630          black 2886  2.1e-03  1.0e+00      27  BP regulation of mRNA splicing, via spliceo
## 631          black 2886  2.5e-03  1.0e+00      30  BP            regulation of mRNA processing
## 632          black 2886  2.6e-03  1.0e+00       8  BP                       rRNA transcription
## 633          black 2886  2.6e-03  1.0e+00       8  CC                   U2-type prespliceosome
## 634          black 2886  2.6e-03  1.0e+00      12  BP positive regulation of DNA-templated tra
## 635          black 2886  2.6e-03  1.0e+00      12  BP          female germ-line cyst formation
## 636          black 2886  2.6e-03  1.0e+00      12  MF            exodeoxyribonuclease activity
## 637          black 2886  2.7e-03  1.0e+00      61  BP               mitochondrion organization
## 638          black 2886  2.7e-03  1.0e+00      16  BP                         RNA interference
## 639          black 2886  2.9e-03  1.0e+00      10  CC        mitochondrial intermembrane space
## 640          black 2886  3.1e-03  1.0e+00      20  BP                      neuroblast division
## 641          black 2886  3.1e-03  1.0e+00      13  BP regulation of pole plasm oskar mRNA loca
## 642          black 2886  3.1e-03  1.0e+00      13  BP regulation of G1/S transition of mitotic
## 643          black 2886  3.3e-03  1.0e+00       7  BP DNA-templated transcription, termination
## 644          black 2886  3.3e-03  1.0e+00       7  BP DNA replication-independent nucleosome o
## 645          black 2886  3.3e-03  1.0e+00       7  CC                                 U6 snRNP
## 646          black 2886  3.3e-03  1.0e+00       7  CC                         RSC-type complex
## 647          black 2886  3.3e-03  1.0e+00       7  CC aminoacyl-tRNA synthetase multienzyme co
## 648          black 2886  3.3e-03  1.0e+00       7  CC                       histone locus body
## 649          black 2886  3.4e-03  1.0e+00      68  MF   ubiquitin-protein transferase activity
## 650          black 2886  3.6e-03  1.0e+00      11  BP   regulation of translational initiation
## 651          black 2886  3.6e-03  1.0e+00      11  MF                             tRNA binding
## 652          black 2886  3.7e-03  1.0e+00      15  BP                        nuclear migration
## 653          black 2886  3.8e-03  1.0e+00       4  BP exonucleolytic trimming to generate matu
## 654          black 2886  3.8e-03  1.0e+00       4  BP                        oocyte maturation
## 655          black 2886  3.8e-03  1.0e+00       4  BP          astral microtubule organization
## 656          black 2886  3.8e-03  1.0e+00       4  BP                    sperm aster formation
## 657          black 2886  3.8e-03  1.0e+00       4  BP meiotic DNA double-strand break formatio
## 658          black 2886  3.8e-03  1.0e+00       4  BP establishment of meiotic spindle localiz
## 659          black 2886  3.8e-03  1.0e+00       4  BP        protein K11-linked ubiquitination
## 660          black 2886  3.8e-03  1.0e+00       4  BP nuclear retention of pre-mRNA at the sit
## 661          black 2886  3.8e-03  1.0e+00       4  BP nuclear polyadenylation-dependent rRNA c
## 662          black 2886  3.8e-03  1.0e+00       4  BP nuclear polyadenylation-dependent tRNA c
## 663          black 2886  3.8e-03  1.0e+00       4  BP positive regulation of oxidative phospho
## 664          black 2886  3.8e-03  1.0e+00       4  CC     alpha DNA polymerase:primase complex
## 665          black 2886  3.8e-03  1.0e+00       4  CC           epsilon DNA polymerase complex
## 666          black 2886  3.8e-03  1.0e+00       4  CC        cortical microtubule cytoskeleton
## 667          black 2886  3.8e-03  1.0e+00       4  CC                  mismatch repair complex
## 668          black 2886  3.8e-03  1.0e+00       4  CC                 male germline ring canal
## 669          black 2886  3.8e-03  1.0e+00       4  CC                  nuclear pore inner ring
## 670          black 2886  3.8e-03  1.0e+00       4  MF         GPI-anchor transamidase activity
## 671          black 2886  3.8e-03  1.0e+00       4  MF     RNA-DNA hybrid ribonuclease activity
## 672          black 2886  3.8e-03  1.0e+00       4  MF        structural constituent of chorion
## 673          black 2886  3.8e-03  1.0e+00       4  MF                         U6 snRNA binding
## 674          black 2886  3.8e-03  1.0e+00       4  MF                        U3 snoRNA binding
## 675          black 2886  3.8e-03  1.0e+00       4  MF ATP-dependent 5'-3' DNA helicase activit
## 676          black 2886  3.9e-03  1.0e+00       9  BP                 regulation of DNA repair
## 677          black 2886  3.9e-03  1.0e+00       9  MF     chromatin insulator sequence binding
## 678          black 2886  4.0e-03  1.0e+00       6  BP meiotic DNA double-strand break processi
## 679          black 2886  4.0e-03  1.0e+00       6  BP formation of translation preinitiation c
## 680          black 2886  4.0e-03  1.0e+00       6  BP                   DNA topological change
## 681          black 2886  4.0e-03  1.0e+00       6  BP negative regulation of oskar mRNA transl
## 682          black 2886  4.0e-03  1.0e+00       6  BP        male germ-line cyst encapsulation
## 683          black 2886  4.0e-03  1.0e+00       6  BP     positive regulation of cell division
## 684          black 2886  4.0e-03  1.0e+00       6  BP phosphorylation of RNA polymerase II C-t
## 685          black 2886  4.0e-03  1.0e+00       6  BP positive regulation of protein localizat
## 686          black 2886  4.0e-03  1.0e+00       6  CC        multimeric ribonuclease P complex
## 687          black 2886  4.0e-03  1.0e+00       6  MF               flap endonuclease activity
## 688          black 2886  4.1e-03  1.0e+00      18  BP              regulation of cell division
## 689          black 2886  4.2e-03  1.0e+00      19  BP           asymmetric neuroblast division
## 690          black 2886  4.4e-03  1.0e+00       5  BP negative regulation of transposition, DN
## 691          black 2886  4.4e-03  1.0e+00       5  BP                        female meiosis II
## 692          black 2886  4.4e-03  1.0e+00       5  BP positive regulation of transcription of 
## 693          black 2886  4.4e-03  1.0e+00       5  BP polarity specification of dorsal/ventral
## 694          black 2886  4.4e-03  1.0e+00       5  BP regulation of transposition, RNA-mediate
## 695          black 2886  4.4e-03  1.0e+00       5  BP                   tRNA catabolic process
## 696          black 2886  4.4e-03  1.0e+00       5  BP                   nucleosome positioning
## 697          black 2886  4.4e-03  1.0e+00       5  BP snRNA transcription by RNA polymerase II
## 698          black 2886  4.4e-03  1.0e+00       5  BP exonucleolytic nuclear-transcribed mRNA 
## 699          black 2886  4.4e-03  1.0e+00       5  BP negative regulation of meiotic nuclear d
## 700          black 2886  4.4e-03  1.0e+00       5  BP positive regulation of translational ini
## 701          black 2886  4.4e-03  1.0e+00       5  BP              replication fork protection
## 702          black 2886  4.4e-03  1.0e+00       5  BP RNA phosphodiester bond hydrolysis, exon
## 703          black 2886  4.4e-03  1.0e+00       5  CC                 ribonuclease MRP complex
## 704          black 2886  4.4e-03  1.0e+00       5  CC             chromosome passenger complex
## 705          black 2886  4.4e-03  1.0e+00       5  CC                      septin cytoskeleton
## 706          black 2886  4.4e-03  1.0e+00       5  CC              nuclear pore nuclear basket
## 707          black 2886  4.4e-03  1.0e+00       5  MF                  nucleosomal DNA binding
## 708          black 2886  4.7e-03  1.0e+00      13  BP heterochromatin organization involved in
## 709          black 2886  4.7e-03  1.0e+00      13  BP regulation of embryonic pattern specific
## 710          black 2886  5.0e-03  1.0e+00      10  BP                      pole cell formation
## 711          black 2886  5.0e-03  1.0e+00      10  BP                 heterochromatin assembly
## 712          black 2886  5.0e-03  1.0e+00      10  BP                regulation of cytokinesis
## 713          black 2886  5.0e-03  1.0e+00      10  MF protein transmembrane transporter activi
## 714          black 2886  5.3e-03  1.0e+00      91  MF sequence-specific double-stranded DNA bi
## 715          black 2886  5.3e-03  1.0e+00       8  BP                        exit from mitosis
## 716          black 2886  5.3e-03  1.0e+00       8  BP             regulation of mRNA stability
## 717          black 2886  5.3e-03  1.0e+00       8  MF     aspartic-type endopeptidase activity
## 718          black 2886  5.4e-03  1.0e+00      36  BP                 neuroblast proliferation
## 719          black 2886  5.5e-03  1.0e+00      21  BP       negative regulation of translation
## 720          black 2886  5.5e-03  1.0e+00      21  CC    mitochondrial large ribosomal subunit
## 721          black 2886  5.9e-03  1.0e+00      11  BP germarium-derived female germ-line cyst 
## 722          black 2886  5.9e-03  1.0e+00      11  MF peptide transmembrane transporter activi
## 723          black 2886  6.0e-03  1.0e+00      55  CC                     mitochondrial matrix
## 724          black 2886  6.3e-03  1.0e+00      25  BP  regulation of protein catabolic process
## 725          black 2886  6.6e-03  1.0e+00      12  MF                  signal sequence binding
## 726          black 2886  7.0e-03  1.0e+00       9  BP nuclear-transcribed mRNA catabolic proce
## 727          black 2886  7.0e-03  1.0e+00       9  BP              regulation of RNA stability
## 728          black 2886  7.0e-03  1.0e+00       9  BP positive regulation of mitotic nuclear d
## 729          black 2886  7.0e-03  1.0e+00       9  MF    translation initiation factor binding
## 730          black 2886  7.0e-03  1.0e+00       9  MF                   RNA polymerase binding
## 731          black 2886  7.0e-03  1.0e+00      22  BP               somatic stem cell division
## 732          black 2886  7.1e-03  1.0e+00      13  BP transcription preinitiation complex asse
## 733          black 2886  7.1e-03  1.0e+00      13  BP regulation of intracellular mRNA localiz
## 734          black 2886  7.1e-03  1.0e+00       7  BP             mitochondrial RNA processing
## 735          black 2886  7.1e-03  1.0e+00       7  BP                   postreplication repair
## 736          black 2886  7.1e-03  1.0e+00       7  BP        mitotic sister chromatid cohesion
## 737          black 2886  7.1e-03  1.0e+00       7  BP          regulation of exit from mitosis
## 738          black 2886  7.1e-03  1.0e+00       7  BP           sperm chromatin decondensation
## 739          black 2886  7.1e-03  1.0e+00       7  BP                  nucleosome mobilization
## 740          black 2886  7.1e-03  1.0e+00       7  MF               5'-3' exonuclease activity
## 741          black 2886  7.2e-03  1.0e+00      21  BP   morphogenesis of follicular epithelium
## 742          black 2886  7.6e-03  1.0e+00      42  MF                          tubulin binding
## 743          black 2886  7.7e-03  1.0e+00      16  MF                      mRNA 3'-UTR binding
## 744          black 2886  7.8e-03  1.0e+00      33  BP    regulation of protein kinase activity
## 745          black 2886  8.3e-03  1.0e+00      10  BP oocyte microtubule cytoskeleton polariza
## 746          black 2886  8.3e-03  1.0e+00      10  BP              sterol biosynthetic process
## 747          black 2886  8.8e-03  1.0e+00      81  BP positive regulation of cellular protein 
## 748          black 2886  9.2e-03  1.0e+00      11  BP                    centrosome separation
## 749          black 2886  9.2e-03  1.0e+00      11  MF               methylated histone binding
## 750          black 2886  9.3e-03  1.0e+00      21  BP          regulation of protein stability
## 751          black 2886  9.3e-03  1.0e+00      21  BP glycerophospholipid biosynthetic process
## 752          black 2886  9.5e-03  1.0e+00       6  BP DNA replication-independent nucleosome a
## 753          black 2886  9.5e-03  1.0e+00       6  BP         mitotic G2 DNA damage checkpoint
## 754          black 2886  9.5e-03  1.0e+00       6  BP                                 synapsis
## 755          black 2886  9.5e-03  1.0e+00       6  BP   regulation of meiotic nuclear division
## 756          black 2886  9.5e-03  1.0e+00       6  BP nucleolar large rRNA transcription by RN
## 757          black 2886  9.5e-03  1.0e+00       6  BP positive regulation of nuclear-transcrib
## 758          black 2886  9.5e-03  1.0e+00       6  CC                                  exocyst
## 759          black 2886  9.5e-03  1.0e+00       6  MF core promoter sequence-specific DNA bind
## 760          black 2886  9.7e-03  1.0e+00       8  BP                        response to X-ray
## 761          black 2886  9.7e-03  1.0e+00       8  BP         regulation of chromatin assembly
## 762          black 2886  9.7e-03  1.0e+00       8  BP                 female sex determination
## 763          black 2886  9.7e-03  1.0e+00       8  BP production of siRNA involved in RNA inte
## 764          black 2886  9.7e-03  1.0e+00       8  BP oocyte localization involved in germariu
## 765          black 2886  9.7e-03  1.0e+00       8  BP            germline ring canal formation
## 766          black 2886  9.7e-03  1.0e+00       8  BP negative regulation of chromatin silenci
## 767          black 2886  9.7e-03  1.0e+00       8  BP        regulation of histone acetylation
## 768          black 2886  9.7e-03  1.0e+00       8  MF P-P-bond-hydrolysis-driven protein trans
## 769          black 2886  9.9e-03  1.0e+00      12  BP      transcription by RNA polymerase III
## 770          black 2886  9.9e-03  1.0e+00      12  BP               microtubule polymerization
## 771          black 2886  9.9e-03  1.0e+00      25  BP   negative regulation of phosphorylation
## 772           blue 1679  4.1e-32  3.4e-28     261  CC                          plasma membrane
## 773           blue 1679  6.2e-32  5.2e-28     161  CC                          cell projection
## 774           blue 1679  1.5e-28  1.3e-24     136  CC    integral component of plasma membrane
## 775           blue 1679  3.4e-28  2.9e-24     138  CC   intrinsic component of plasma membrane
## 776           blue 1679  1.9e-27  1.6e-23     122  CC                        neuron projection
## 777           blue 1679  4.9e-27  4.1e-23     271  CC                           cell periphery
## 778           blue 1679  7.9e-25  6.7e-21     149  BP                   nervous system process
## 779           blue 1679  4.1e-24  3.4e-20     265  CC          intrinsic component of membrane
## 780           blue 1679  9.4e-24  7.9e-20     261  CC           integral component of membrane
## 781           blue 1679  3.7e-23  3.1e-19      98  MF transmembrane signaling receptor activit
## 782           blue 1679  8.9e-23  7.5e-19     111  MF              signaling receptor activity
## 783           blue 1679  1.2e-19  9.8e-16     409  CC                                 membrane
## 784           blue 1679  7.8e-18  6.6e-14      38  MF       neurotransmitter receptor activity
## 785           blue 1679  1.4e-15  1.2e-11      83  BP           chemical synaptic transmission
## 786           blue 1679  2.2e-15  1.9e-11      83  BP                       synaptic signaling
## 787           blue 1679  2.3e-14  2.0e-10      45  MF      G protein-coupled receptor activity
## 788           blue 1679  3.8e-14  3.2e-10     296  BP                                signaling
## 789           blue 1679  4.8e-14  4.1e-10      61  MF                     ion channel activity
## 790           blue 1679  5.1e-14  4.3e-10      51  MF                  cation channel activity
## 791           blue 1679  5.4e-14  4.5e-10      63  MF                         channel activity
## 792           blue 1679  7.5e-14  6.3e-10      65  BP G protein-coupled receptor signaling pat
## 793           blue 1679  9.6e-14  8.0e-10      40  CC        transmembrane transporter complex
## 794           blue 1679  1.2e-13  1.0e-09      77  CC                                  synapse
## 795           blue 1679  2.2e-13  1.8e-09     298  BP                       cell communication
## 796           blue 1679  6.5e-13  5.4e-09      59  BP  sensory perception of chemical stimulus
## 797           blue 1679  6.9e-13  5.8e-09      33  BP         regulation of membrane potential
## 798           blue 1679  2.1e-12  1.8e-08      41  CC                                   cilium
## 799           blue 1679  3.0e-12  2.5e-08      61  CC                                     axon
## 800           blue 1679  3.4e-12  2.8e-08      40  CC                       neuronal cell body
## 801           blue 1679  5.4e-12  4.5e-08      42  CC                                cell body
## 802           blue 1679  7.1e-12  5.9e-08     103  BP                            ion transport
## 803           blue 1679  7.1e-11  6.0e-07      16  MF G protein-coupled amine receptor activit
## 804           blue 1679  7.3e-11  6.2e-07      73  BP                         cation transport
## 805           blue 1679  4.9e-10  4.1e-06      73  MF inorganic cation transmembrane transport
## 806           blue 1679  7.6e-10  6.3e-06      75  MF cation transmembrane transporter activit
## 807           blue 1679  9.2e-10  7.7e-06     103  BP                      cell-cell signaling
## 808           blue 1679  1.3e-09  1.1e-05      63  BP              ion transmembrane transport
## 809           blue 1679  2.7e-09  2.3e-05      28  CC               neuron projection membrane
## 810           blue 1679  4.2e-09  3.5e-05      44  CC                                 dendrite
## 811           blue 1679  5.9e-09  4.9e-05     241  BP                      signal transduction
## 812           blue 1679  1.0e-08  8.4e-05      69  BP                     synapse organization
## 813           blue 1679  1.4e-08  1.2e-04      41  BP                                   mating
## 814           blue 1679  1.9e-08  1.6e-04      92  MF   ion transmembrane transporter activity
## 815           blue 1679  2.7e-08  2.2e-04      22  MF extracellular ligand-gated ion channel a
## 816           blue 1679  2.8e-08  2.3e-04      49  MF metal ion transmembrane transporter acti
## 817           blue 1679  2.8e-08  2.4e-04      16  MF    voltage-gated cation channel activity
## 818           blue 1679  2.9e-08  2.4e-04      39  BP                          mating behavior
## 819           blue 1679  3.9e-08  3.3e-04      32  BP                       courtship behavior
## 820           blue 1679  4.0e-08  3.3e-04      23  CC                   cation channel complex
## 821           blue 1679  4.5e-08  3.8e-04      28  BP sensory perception of mechanical stimulu
## 822           blue 1679  4.9e-08  4.1e-04       9  BP                      monoamine transport
## 823           blue 1679  5.2e-08  4.4e-04      42  CC                               presynapse
## 824           blue 1679  6.4e-08  5.3e-04      21  BP                  potassium ion transport
## 825           blue 1679  7.8e-08  6.6e-04      30  BP                     male mating behavior
## 826           blue 1679  1.0e-07  8.6e-04      41  BP                    reproductive behavior
## 827           blue 1679  1.4e-07  1.2e-03      77  BP                                    taxis
## 828           blue 1679  1.5e-07  1.3e-03      28  BP                  male courtship behavior
## 829           blue 1679  1.6e-07  1.3e-03      24  BP              sensory perception of sound
## 830           blue 1679  2.3e-07  1.9e-03      18  BP    potassium ion transmembrane transport
## 831           blue 1679  2.5e-07  2.1e-03      27  CC                        secretory vesicle
## 832           blue 1679  3.4e-07  2.8e-03      27  MF        ligand-gated ion channel activity
## 833           blue 1679  3.6e-07  3.0e-03       8  BP                       dopamine transport
## 834           blue 1679  4.0e-07  3.3e-03      34  BP modulation of chemical synaptic transmis
## 835           blue 1679  4.7e-07  3.9e-03      21  BP G protein-coupled receptor signaling pat
## 836           blue 1679  5.5e-07  4.6e-03      17  MF               potassium channel activity
## 837           blue 1679  7.5e-07  6.3e-03      20  BP adenylate cyclase-modulating G protein-c
## 838           blue 1679  8.2e-07  6.9e-03      22  BP     cyclic-nucleotide-mediated signaling
## 839           blue 1679  1.1e-06  8.8e-03      41  BP                       learning or memory
## 840           blue 1679  1.1e-06  9.4e-03      35  BP                       cell-cell adhesion
## 841           blue 1679  1.4e-06  1.1e-02      28  BP                          cilium assembly
## 842           blue 1679  1.6e-06  1.4e-02       8  CC      acetylcholine-gated channel complex
## 843           blue 1679  1.6e-06  1.4e-02       8  MF acetylcholine-gated cation-selective cha
## 844           blue 1679  1.7e-06  1.4e-02      29  CC                        cell leading edge
## 845           blue 1679  1.9e-06  1.6e-02      42  BP           cation transmembrane transport
## 846           blue 1679  2.1e-06  1.8e-02      30  BP                    chemosensory behavior
## 847           blue 1679  2.1e-06  1.8e-02      22  CC                        synaptic membrane
## 848           blue 1679  2.6e-06  2.1e-02      24  BP               motor neuron axon guidance
## 849           blue 1679  2.6e-06  2.2e-02      10  BP       synaptic transmission, cholinergic
## 850           blue 1679  3.2e-06  2.7e-02      16  BP                           drug transport
## 851           blue 1679  3.3e-06  2.8e-02      43  BP                      metal ion transport
## 852           blue 1679  3.6e-06  3.0e-02      45  BP                      locomotory behavior
## 853           blue 1679  3.9e-06  3.3e-02      48  BP                         response to drug
## 854           blue 1679  4.1e-06  3.4e-02     103  MF       transmembrane transporter activity
## 855           blue 1679  4.4e-06  3.7e-02      13  CC                  presynaptic active zone
## 856           blue 1679  5.1e-06  4.2e-02      59  BP                            axon guidance
## 857           blue 1679  5.7e-06  4.8e-02      10  BP           inter-male aggressive behavior
## 858           blue 1679  5.8e-06  4.9e-02     114  BP                               locomotion
## 859           blue 1679  6.2e-06  5.2e-02       9  CC                synaptic vesicle membrane
## 860           blue 1679  6.4e-06  5.3e-02      35  BP               neurotransmitter transport
## 861           blue 1679  6.4e-06  5.4e-02      17  MF           neuropeptide receptor activity
## 862           blue 1679  6.7e-06  5.7e-02      32  BP                   synaptic vesicle cycle
## 863           blue 1679  7.1e-06  6.0e-02      39  BP                           adult behavior
## 864           blue 1679  8.4e-06  7.1e-02      21  CC                         exocytic vesicle
## 865           blue 1679  8.6e-06  7.2e-02      85  BP cell morphogenesis involved in neuron di
## 866           blue 1679  8.6e-06  7.2e-02      28  BP                      cilium organization
## 867           blue 1679  1.1e-05  9.1e-02       7  MF G protein-coupled serotonin receptor act
## 868           blue 1679  1.3e-05  1.1e-01      17  BP               circadian sleep/wake cycle
## 869           blue 1679  1.3e-05  1.1e-01      17  MF G protein-coupled peptide receptor activ
## 870           blue 1679  1.3e-05  1.1e-01     128  BP                   neuron differentiation
## 871           blue 1679  1.4e-05  1.2e-01       9  CC                exocytic vesicle membrane
## 872           blue 1679  1.4e-05  1.2e-01       8  CC  voltage-gated potassium channel complex
## 873           blue 1679  1.4e-05  1.2e-01       8  MF   calcium-dependent phospholipid binding
## 874           blue 1679  1.5e-05  1.3e-01      18  MF                peptide receptor activity
## 875           blue 1679  1.5e-05  1.3e-01      46  BP                            cell adhesion
## 876           blue 1679  1.6e-05  1.3e-01      85  BP          neuron projection morphogenesis
## 877           blue 1679  1.8e-05  1.5e-01      14  BP adenylate cyclase-activating G protein-c
## 878           blue 1679  1.8e-05  1.5e-01      35  BP    monovalent inorganic cation transport
## 879           blue 1679  2.0e-05  1.6e-01       6  BP         regulation of dopamine secretion
## 880           blue 1679  2.1e-05  1.8e-01     109  BP                       neuron development
## 881           blue 1679  2.2e-05  1.9e-01      36  BP inorganic cation transmembrane transport
## 882           blue 1679  2.6e-05  2.2e-01       5  MF monoamine transmembrane transporter acti
## 883           blue 1679  2.6e-05  2.2e-01       5  MF        sodium channel regulator activity
## 884           blue 1679  2.6e-05  2.2e-01      36  BP                         circadian rhythm
## 885           blue 1679  2.8e-05  2.4e-01      12  BP heterophilic cell-cell adhesion via plas
## 886           blue 1679  3.1e-05  2.6e-01      88  BP            neuron projection development
## 887           blue 1679  3.3e-05  2.8e-01       8  CC                potassium channel complex
## 888           blue 1679  3.8e-05  3.2e-01      10  BP                      aggressive behavior
## 889           blue 1679  3.9e-05  3.3e-01      19  CC                         synaptic vesicle
## 890           blue 1679  4.0e-05  3.4e-01      89  BP cell morphogenesis involved in different
## 891           blue 1679  4.1e-05  3.4e-01     172  BP               nervous system development
## 892           blue 1679  4.6e-05  3.8e-01      24  CC                        transport vesicle
## 893           blue 1679  5.1e-05  4.3e-01      20  BP         calcium ion regulated exocytosis
## 894           blue 1679  5.3e-05  4.5e-01       9  MF voltage-gated potassium channel activity
## 895           blue 1679  5.3e-05  4.5e-01      67  BP                         axon development
## 896           blue 1679  5.4e-05  4.5e-01      65  BP                             axonogenesis
## 897           blue 1679  5.5e-05  4.6e-01      29  BP                       neuron recognition
## 898           blue 1679  5.5e-05  4.6e-01      27  BP              sensory perception of smell
## 899           blue 1679  5.6e-05  4.7e-01      24  BP                adult locomotory behavior
## 900           blue 1679  5.6e-05  4.7e-01      24  MF sodium ion transmembrane transporter act
## 901           blue 1679  6.1e-05  5.1e-01      92  BP                  transmembrane transport
## 902           blue 1679  6.5e-05  5.5e-01      10  MF               channel regulator activity
## 903           blue 1679  7.3e-05  6.2e-01      23  BP                  regulation of secretion
## 904           blue 1679  8.3e-05  6.9e-01      20  BP                     regulated exocytosis
## 905           blue 1679  8.3e-05  7.0e-01     110  MF                     transporter activity
## 906           blue 1679  9.0e-05  7.5e-01      13  MF chloride transmembrane transporter activ
## 907           blue 1679  9.0e-05  7.5e-01      15  BP                  cAMP-mediated signaling
## 908           blue 1679  9.0e-05  7.6e-01      23  BP           regulation of circadian rhythm
## 909           blue 1679  9.5e-05  7.9e-01      27  BP                       circadian behavior
## 910           blue 1679  9.9e-05  8.3e-01      16  BP              synaptic target recognition
## 911           blue 1679  1.1e-04  8.9e-01      10  BP                          cilium movement
## 912           blue 1679  1.3e-04  1.0e+00      27  BP                        rhythmic behavior
## 913           blue 1679  1.4e-04  1.0e+00       5  BP         cellular response to calcium ion
## 914           blue 1679  1.4e-04  1.0e+00       5  MF             octopamine receptor activity
## 915           blue 1679  1.4e-04  1.0e+00       5  MF       sodium:chloride symporter activity
## 916           blue 1679  1.6e-04  1.0e+00      15  CC                    postsynaptic membrane
## 917           blue 1679  1.6e-04  1.0e+00      25  BP                               exocytosis
## 918           blue 1679  1.7e-04  1.0e+00      98  BP                       cell morphogenesis
## 919           blue 1679  1.7e-04  1.0e+00      16  BP                  calcium ion homeostasis
## 920           blue 1679  1.7e-04  1.0e+00       7  BP phospholipase C-activating G protein-cou
## 921           blue 1679  1.8e-04  1.0e+00      13  BP          response to mechanical stimulus
## 922           blue 1679  1.8e-04  1.0e+00      13  MF    neurotransmitter transporter activity
## 923           blue 1679  1.9e-04  1.0e+00      21  BP                                    sleep
## 924           blue 1679  1.9e-04  1.0e+00       6  MF               phosphatidylserine binding
## 925           blue 1679  2.0e-04  1.0e+00      23  BP                   regulation of behavior
## 926           blue 1679  2.0e-04  1.0e+00      14  BP regulation of circadian sleep/wake cycle
## 927           blue 1679  2.1e-04  1.0e+00      22  BP               synaptic vesicle transport
## 928           blue 1679  2.1e-04  1.0e+00       4  MF dopamine neurotransmitter receptor activ
## 929           blue 1679  2.1e-04  1.0e+00      15  BP homophilic cell adhesion via plasma memb
## 930           blue 1679  2.2e-04  1.0e+00     141  BP                             neurogenesis
## 931           blue 1679  2.3e-04  1.0e+00      21  BP      neuromuscular synaptic transmission
## 932           blue 1679  2.4e-04  1.0e+00      11  CC                                  axoneme
## 933           blue 1679  2.4e-04  1.0e+00      13  BP        circadian sleep/wake cycle, sleep
## 934           blue 1679  2.6e-04  1.0e+00      19  BP                                 learning
## 935           blue 1679  2.6e-04  1.0e+00       9  BP               response to carbon dioxide
## 936           blue 1679  2.7e-04  1.0e+00      18  BP              synaptic vesicle exocytosis
## 937           blue 1679  3.0e-04  1.0e+00      20  BP          cell-cell junction organization
## 938           blue 1679  3.1e-04  1.0e+00      22  BP            synaptic vesicle localization
## 939           blue 1679  3.1e-04  1.0e+00      37  MF                      calcium ion binding
## 940           blue 1679  3.7e-04  1.0e+00      34  BP                 cell projection assembly
## 941           blue 1679  3.7e-04  1.0e+00      32  BP               response to light stimulus
## 942           blue 1679  4.0e-04  1.0e+00      12  BP regulation of circadian sleep/wake cycle
## 943           blue 1679  4.1e-04  1.0e+00       9  CC                       ciliary basal body
## 944           blue 1679  4.4e-04  1.0e+00       5  BP neurotransmitter loading into synaptic v
## 945           blue 1679  4.4e-04  1.0e+00       5  MF          acetylcholine receptor activity
## 946           blue 1679  4.6e-04  1.0e+00      16  BP           neuropeptide signaling pathway
## 947           blue 1679  4.7e-04  1.0e+00      27  BP                                   memory
## 948           blue 1679  4.9e-04  1.0e+00      30  BP                          ion homeostasis
## 949           blue 1679  5.0e-04  1.0e+00      24  BP               neurotransmitter secretion
## 950           blue 1679  5.5e-04  1.0e+00     133  BP                    generation of neurons
## 951           blue 1679  5.9e-04  1.0e+00      14  BP                       olfactory learning
## 952           blue 1679  5.9e-04  1.0e+00      14  CC                         septate junction
## 953           blue 1679  5.9e-04  1.0e+00       7  MF ATP-dependent microtubule motor activity
## 954           blue 1679  6.4e-04  1.0e+00      20  BP           cellular metal ion homeostasis
## 955           blue 1679  6.6e-04  1.0e+00      11  BP      calcium ion transmembrane transport
## 956           blue 1679  7.1e-04  1.0e+00      16  BP                        phototransduction
## 957           blue 1679  7.1e-04  1.0e+00      16  BP                     associative learning
## 958           blue 1679  8.4e-04  1.0e+00       6  MF ammonium transmembrane transporter activ
## 959           blue 1679  9.1e-04  1.0e+00       9  BP male courtship behavior, veined wing vib
## 960           blue 1679  9.1e-04  1.0e+00       9  BP                  response to insecticide
## 961           blue 1679  9.2e-04  1.0e+00      24  BP detection of stimulus involved in sensor
## 962           blue 1679  9.2e-04  1.0e+00      18  CC                              postsynapse
## 963           blue 1679  9.5e-04  1.0e+00      28  BP                       cation homeostasis
## 964           blue 1679  9.7e-04  1.0e+00       4  BP          cellular sodium ion homeostasis
## 965           blue 1679  9.7e-04  1.0e+00       4  BP     serotonin receptor signaling pathway
## 966           blue 1679  9.7e-04  1.0e+00       4  BP       cellular potassium ion homeostasis
## 967           blue 1679  9.7e-04  1.0e+00       4  BP sodium ion export across plasma membrane
## 968           blue 1679  9.7e-04  1.0e+00       4  BP  cellular response to glucose starvation
## 969           blue 1679  9.7e-04  1.0e+00       4  BP regulation of presynaptic cytosolic calc
## 970           blue 1679  9.7e-04  1.0e+00       4  CC sodium:potassium-exchanging ATPase compl
## 971           blue 1679  9.7e-04  1.0e+00       4  MF delayed rectifier potassium channel acti
## 972           blue 1679  9.7e-04  1.0e+00       4  MF sodium:potassium-exchanging ATPase activ
## 973           blue 1679  9.7e-04  1.0e+00      13  BP                   response to anesthetic
## 974           blue 1679  9.9e-04  1.0e+00       8  BP negative regulation of synaptic transmis
## 975           blue 1679  9.9e-04  1.0e+00       8  CC                        secretory granule
## 976           blue 1679  1.0e-03  1.0e+00       7  BP       sodium ion transmembrane transport
## 977           blue 1679  1.1e-03  1.0e+00       5  CC  cytoskeleton of presynaptic active zone
## 978           blue 1679  1.1e-03  1.0e+00       5  MF             adrenergic receptor activity
## 979           blue 1679  1.1e-03  1.0e+00      24  BP                    metal ion homeostasis
## 980           blue 1679  1.1e-03  1.0e+00      18  BP              cell-cell junction assembly
## 981           blue 1679  1.2e-03  1.0e+00      15  BP                     sodium ion transport
## 982           blue 1679  1.2e-03  1.0e+00      14  BP         cellular calcium ion homeostasis
## 983           blue 1679  1.3e-03  1.0e+00      20  CC                         receptor complex
## 984           blue 1679  1.3e-03  1.0e+00       9  BP male courtship behavior, veined wing ext
## 985           blue 1679  1.5e-03  1.0e+00       8  BP       sensory perception of bitter taste
## 986           blue 1679  1.5e-03  1.0e+00      10  BP              neuron cellular homeostasis
## 987           blue 1679  1.5e-03  1.0e+00      10  CC                           dynein complex
## 988           blue 1679  1.6e-03  1.0e+00      45  BP                  regulation of transport
## 989           blue 1679  1.6e-03  1.0e+00      11  CC plasma membrane bounded cell projection 
## 990           blue 1679  1.6e-03  1.0e+00       7  BP     presynaptic active zone organization
## 991           blue 1679  1.6e-03  1.0e+00      12  BP              sensory perception of taste
## 992           blue 1679  1.8e-03  1.0e+00       3  BP                        peptide amidation
## 993           blue 1679  1.8e-03  1.0e+00       3  BP            cellular response to dopamine
## 994           blue 1679  1.8e-03  1.0e+00       3  MF dopamine neurotransmitter receptor activ
## 995           blue 1679  1.8e-03  1.0e+00       3  MF    peptidylamidoglycolate lyase activity
## 996           blue 1679  1.8e-03  1.0e+00       3  MF      serotonin:sodium symporter activity
## 997           blue 1679  1.8e-03  1.0e+00       9  BP      cyclic nucleotide metabolic process
## 998           blue 1679  1.8e-03  1.0e+00       9  BP regulation of cytosolic calcium ion conc
## 999           blue 1679  1.9e-03  1.0e+00      23  BP                 cellular ion homeostasis
## 1000          blue 1679  2.0e-03  1.0e+00      19  BP                       olfactory behavior
## 1001          blue 1679  2.0e-03  1.0e+00      12  BP       amino acid transmembrane transport
## 1002          blue 1679  2.1e-03  1.0e+00      11  BP               larval locomotory behavior
## 1003          blue 1679  2.1e-03  1.0e+00      17  BP            eye-antennal disc development
## 1004          blue 1679  2.1e-03  1.0e+00       5  BP       regulation of sodium ion transport
## 1005          blue 1679  2.1e-03  1.0e+00       5  BP                potassium ion homeostasis
## 1006          blue 1679  2.1e-03  1.0e+00       5  CC presynaptic active zone cytoplasmic comp
## 1007          blue 1679  2.1e-03  1.0e+00       5  MF   voltage-gated calcium channel activity
## 1008          blue 1679  2.2e-03  1.0e+00      28  BP               microtubule-based movement
## 1009          blue 1679  2.2e-03  1.0e+00      28  BP                           signal release
## 1010          blue 1679  2.3e-03  1.0e+00      14  BP                          larval behavior
## 1011          blue 1679  2.5e-03  1.0e+00       9  BP                     antennal development
## 1012          blue 1679  2.5e-03  1.0e+00       9  CC                              growth cone
## 1013          blue 1679  2.5e-03  1.0e+00       9  MF                  taste receptor activity
## 1014          blue 1679  2.5e-03  1.0e+00       6  BP              anesthesia-resistant memory
## 1015          blue 1679  2.5e-03  1.0e+00       6  BP                  mechanosensory behavior
## 1016          blue 1679  2.5e-03  1.0e+00       6  BP           regulation of response to drug
## 1017          blue 1679  2.5e-03  1.0e+00       6  CC                  axonemal dynein complex
## 1018          blue 1679  2.6e-03  1.0e+00       4  CC nucleotide-activated protein kinase comp
## 1019          blue 1679  2.6e-03  1.0e+00       4  CC                                   BBSome
## 1020          blue 1679  2.6e-03  1.0e+00       4  MF oxidoreductase activity, acting on paire
## 1021          blue 1679  2.6e-03  1.0e+00      10  MF              glutamate receptor activity
## 1022          blue 1679  2.8e-03  1.0e+00      14  MF               microtubule motor activity
## 1023          blue 1679  2.9e-03  1.0e+00      17  MF                           motor activity
## 1024          blue 1679  3.0e-03  1.0e+00       8  BP male courtship behavior, veined wing gen
## 1025          blue 1679  3.1e-03  1.0e+00      18  CC               neuron projection terminus
## 1026          blue 1679  3.2e-03  1.0e+00      40  BP                        secretion by cell
## 1027          blue 1679  3.2e-03  1.0e+00      12  MF                 calcium channel activity
## 1028          blue 1679  3.6e-03  1.0e+00      18  MF                          odorant binding
## 1029          blue 1679  3.7e-03  1.0e+00      13  BP                     amino acid transport
## 1030          blue 1679  3.8e-03  1.0e+00       5  BP                       calcium ion import
## 1031          blue 1679  3.8e-03  1.0e+00       5  BP potassium ion import across plasma membr
## 1032          blue 1679  3.8e-03  1.0e+00       5  MF     cell-cell adhesion mediator activity
## 1033          blue 1679  3.9e-03  1.0e+00       6  CC                  ciliary transition zone
## 1034          blue 1679  4.1e-03  1.0e+00       8  BP     synaptic transmission, glutamatergic
## 1035          blue 1679  4.4e-03  1.0e+00      15  BP                 apical junction assembly
## 1036          blue 1679  4.4e-03  1.0e+00      10  MF  drug transmembrane transporter activity
## 1037          blue 1679  4.4e-03  1.0e+00       9  MF   ionotropic glutamate receptor activity
## 1038          blue 1679  4.5e-03  1.0e+00      13  BP                septate junction assembly
## 1039          blue 1679  4.7e-03  1.0e+00      17  BP detection of chemical stimulus involved 
## 1040          blue 1679  4.9e-03  1.0e+00      12  MF                            SNARE binding
## 1041          blue 1679  5.0e-03  1.0e+00       7  BP                  dendrite self-avoidance
## 1042          blue 1679  5.1e-03  1.0e+00      15  CC             cytoplasmic vesicle membrane
## 1043          blue 1679  5.1e-03  1.0e+00      23  CC                       cell-cell junction
## 1044          blue 1679  5.3e-03  1.0e+00      11  BP               detection of visible light
## 1045          blue 1679  5.3e-03  1.0e+00      22  CC           microtubule associated complex
## 1046          blue 1679  5.3e-03  1.0e+00      16  CC                         vesicle membrane
## 1047          blue 1679  5.4e-03  1.0e+00      13  BP detection of chemical stimulus involved 
## 1048          blue 1679  5.5e-03  1.0e+00       4  BP              camera-type eye development
## 1049          blue 1679  5.5e-03  1.0e+00       4  BP adenylate cyclase-activating adrenergic 
## 1050          blue 1679  5.5e-03  1.0e+00       4  BP           postsynaptic membrane assembly
## 1051          blue 1679  5.5e-03  1.0e+00       4  CC    voltage-gated calcium channel complex
## 1052          blue 1679  5.5e-03  1.0e+00       4  CC integral component of synaptic vesicle m
## 1053          blue 1679  5.5e-03  1.0e+00       4  CC                         outer dynein arm
## 1054          blue 1679  5.6e-03  1.0e+00       8  BP                          visual behavior
## 1055          blue 1679  5.8e-03  1.0e+00      37  BP                     cellular homeostasis
## 1056          blue 1679  5.8e-03  1.0e+00      14  BP                    calcium ion transport
## 1057          blue 1679  5.9e-03  1.0e+00       6  BP                   adult walking behavior
## 1058          blue 1679  5.9e-03  1.0e+00       6  BP positive regulation of circadian sleep/w
## 1059          blue 1679  6.3e-03  1.0e+00      34  BP                    response to radiation
## 1060          blue 1679  6.3e-03  1.0e+00      17  CC                            axon terminus
## 1061          blue 1679  6.4e-03  1.0e+00       5  BP adenylate cyclase-inhibiting G protein-c
## 1062          blue 1679  6.4e-03  1.0e+00       5  BP                   adult feeding behavior
## 1063          blue 1679  6.4e-03  1.0e+00       5  BP        presynaptic membrane organization
## 1064          blue 1679  6.4e-03  1.0e+00       5  BP                      response to odorant
## 1065          blue 1679  6.4e-03  1.0e+00      32  CC                     extracellular matrix
## 1066          blue 1679  6.4e-03  1.0e+00       3  BP             glial cell apoptotic process
## 1067          blue 1679  6.4e-03  1.0e+00       3  CC intracellular cyclic nucleotide activate
## 1068          blue 1679  6.4e-03  1.0e+00       3  CC              neuronal cell body membrane
## 1069          blue 1679  6.4e-03  1.0e+00       3  MF intracellular cyclic nucleotide activate
## 1070          blue 1679  6.5e-03  1.0e+00      13  MF amino acid transmembrane transporter act
## 1071          blue 1679  7.0e-03  1.0e+00       7  BP   cyclic nucleotide biosynthetic process
## 1072          blue 1679  7.0e-03  1.0e+00       7  BP    neurotransmitter biosynthetic process
## 1073          blue 1679  7.0e-03  1.0e+00       7  BP  positive regulation of circadian rhythm
## 1074          blue 1679  7.3e-03  1.0e+00       9  BP         photoreceptor cell axon guidance
## 1075          blue 1679  7.3e-03  1.0e+00       8  MF neurotransmitter:sodium symporter activi
## 1076          blue 1679  7.7e-03  1.0e+00      32  BP regulation of synapse structure or activ
## 1077          blue 1679  8.3e-03  1.0e+00      16  CC                  apical junction complex
## 1078          blue 1679  8.5e-03  1.0e+00       6  MF               dynein light chain binding
## 1079          blue 1679  8.6e-03  1.0e+00      12  BP          protein complex oligomerization
## 1080          blue 1679  8.9e-03  1.0e+00      97  BP        intracellular signal transduction
## 1081          blue 1679  9.8e-03  1.0e+00       5  CC                  calcium channel complex
## 1082          blue 1679  9.8e-03  1.0e+00       5  MF               guanylate cyclase activity
## 1083         brown 1042  8.8e-39  7.4e-35      85  MF           serine-type peptidase activity
## 1084         brown 1042  1.3e-38  1.1e-34      85  MF                serine hydrolase activity
## 1085         brown 1042  1.5e-34  1.2e-30      76  MF       serine-type endopeptidase activity
## 1086         brown 1042  3.7e-32  3.1e-28     119  MF                       peptidase activity
## 1087         brown 1042  3.5e-24  3.0e-20      86  MF                   endopeptidase activity
## 1088         brown 1042  1.3e-21  1.1e-17     228  MF                       hydrolase activity
## 1089         brown 1042  9.4e-19  7.9e-15     133  BP                              proteolysis
## 1090         brown 1042  1.3e-14  1.1e-10      32  MF                    exopeptidase activity
## 1091         brown 1042  2.3e-14  2.0e-10      34  BP                 chitin metabolic process
## 1092         brown 1042  1.3e-13  1.1e-09      34  MF                           chitin binding
## 1093         brown 1042  1.1e-10  9.3e-07     370  MF                       catalytic activity
## 1094         brown 1042  1.8e-10  1.5e-06      15  MF                carboxypeptidase activity
## 1095         brown 1042  5.2e-09  4.4e-05      22  BP          fatty acid biosynthetic process
## 1096         brown 1042  6.2e-09  5.2e-05      18  MF             metalloexopeptidase activity
## 1097         brown 1042  6.8e-09  5.7e-05      10  MF            alkaline phosphatase activity
## 1098         brown 1042  1.9e-08  1.6e-04       7  MF                phospholipase A1 activity
## 1099         brown 1042  2.0e-08  1.7e-04     113  CC                     extracellular region
## 1100         brown 1042  1.2e-07  9.8e-04      75  BP                  transmembrane transport
## 1101         brown 1042  4.2e-07  3.5e-03      27  BP             fatty acid metabolic process
## 1102         brown 1042  4.5e-07  3.8e-03      22  MF proton transmembrane transporter activit
## 1103         brown 1042  1.3e-06  1.1e-02      15  MF                  aminopeptidase activity
## 1104         brown 1042  1.4e-06  1.2e-02      10  MF         metallocarboxypeptidase activity
## 1105         brown 1042  2.4e-06  2.0e-02      10  MF             fatty acid synthase activity
## 1106         brown 1042  2.4e-06  2.0e-02      10  MF         solute:proton symporter activity
## 1107         brown 1042  3.1e-06  2.6e-02       5  MF    serine-type carboxypeptidase activity
## 1108         brown 1042  5.7e-06  4.8e-02       8  CC plasma membrane proton-transporting V-ty
## 1109         brown 1042  6.4e-06  5.3e-02      10  CC proton-transporting V-type ATPase comple
## 1110         brown 1042  6.4e-06  5.3e-02      10  MF proton-transporting ATPase activity, rot
## 1111         brown 1042  8.7e-06  7.3e-02       9  BP very long-chain fatty acid biosynthetic 
## 1112         brown 1042  9.2e-06  7.7e-02      25  MF                metallopeptidase activity
## 1113         brown 1042  1.4e-05  1.2e-01       9  BP                    fatty acid elongation
## 1114         brown 1042  1.5e-05  1.2e-01      35  MF active transmembrane transporter activit
## 1115         brown 1042  1.5e-05  1.2e-01      10  BP  ATP hydrolysis coupled proton transport
## 1116         brown 1042  1.7e-05  1.4e-01       5  MF        serine-type exopeptidase activity
## 1117         brown 1042  1.9e-05  1.6e-01       8  BP fatty acid elongation, saturated fatty a
## 1118         brown 1042  1.9e-05  1.6e-01       8  BP fatty acid elongation, unsaturated fatty
## 1119         brown 1042  1.9e-05  1.6e-01       8  BP fatty acid elongation, monounsaturated f
## 1120         brown 1042  1.9e-05  1.6e-01       8  BP fatty acid elongation, polyunsaturated f
## 1121         brown 1042  1.9e-05  1.6e-01       8  MF   carbohydrate:proton symporter activity
## 1122         brown 1042  1.9e-05  1.6e-01      22  MF                     carbohydrate binding
## 1123         brown 1042  2.9e-05  2.4e-01      71  MF       transmembrane transporter activity
## 1124         brown 1042  2.9e-05  2.4e-01      12  BP        sphingolipid biosynthetic process
## 1125         brown 1042  3.2e-05  2.6e-01       8  MF             fatty acid elongase activity
## 1126         brown 1042  3.4e-05  2.9e-01       9  BP                peptide catabolic process
## 1127         brown 1042  3.4e-05  2.9e-01      14  CC                        vacuolar membrane
## 1128         brown 1042  3.8e-05  3.2e-01       6  BP                            somitogenesis
## 1129         brown 1042  3.8e-05  3.2e-01       6  MF       maltose alpha-glucosidase activity
## 1130         brown 1042  3.8e-05  3.2e-01      33  MF transferase activity, transferring acyl 
## 1131         brown 1042  4.8e-05  4.0e-01      61  BP                  lipid metabolic process
## 1132         brown 1042  4.9e-05  4.1e-01      34  BP               lipid biosynthetic process
## 1133         brown 1042  5.1e-05  4.3e-01       8  MF           metalloaminopeptidase activity
## 1134         brown 1042  5.1e-05  4.3e-01       9  MF               fatty acid ligase activity
## 1135         brown 1042  5.1e-05  4.3e-01       9  MF                      CoA-ligase activity
## 1136         brown 1042  5.4e-05  4.6e-01      14  BP           sphingolipid metabolic process
## 1137         brown 1042  6.9e-05  5.8e-01       7  CC vacuolar proton-transporting V-type ATPa
## 1138         brown 1042  1.1e-04  9.5e-01      59  CC                      extracellular space
## 1139         brown 1042  1.3e-04  1.0e+00      14  MF hydrolase activity, acting on carbon-nit
## 1140         brown 1042  1.5e-04  1.0e+00       9  MF                   phospholipase activity
## 1141         brown 1042  1.5e-04  1.0e+00      19  CC                               peroxisome
## 1142         brown 1042  1.8e-04  1.0e+00       7  MF            dipeptidyl-peptidase activity
## 1143         brown 1042  1.9e-04  1.0e+00      14  MF                          lipase activity
## 1144         brown 1042  2.0e-04  1.0e+00      23  CC                                  vacuole
## 1145         brown 1042  2.2e-04  1.0e+00      17  MF hydrolase activity, hydrolyzing O-glycos
## 1146         brown 1042  3.0e-04  1.0e+00      17  MF hydrolase activity, acting on carbon-nit
## 1147         brown 1042  3.0e-04  1.0e+00      29  MF transferase activity, transferring acyl 
## 1148         brown 1042  3.6e-04  1.0e+00      15  CC                             cell surface
## 1149         brown 1042  4.6e-04  1.0e+00      18  BP                  cell fate specification
## 1150         brown 1042  5.1e-04  1.0e+00       4  MF                    aminoacylase activity
## 1151         brown 1042  5.5e-04  1.0e+00       5  CC vacuolar proton-transporting V-type ATPa
## 1152         brown 1042  5.9e-04  1.0e+00      11  MF ATPase coupled ion transmembrane transpo
## 1153         brown 1042  5.9e-04  1.0e+00      74  MF                     transporter activity
## 1154         brown 1042  6.0e-04  1.0e+00      17  MF hydrolase activity, acting on glycosyl b
## 1155         brown 1042  7.4e-04  1.0e+00      53  MF   ion transmembrane transporter activity
## 1156         brown 1042  9.4e-04  1.0e+00       5  CC proton-transporting V-type ATPase, V0 do
## 1157         brown 1042  9.6e-04  1.0e+00       6  BP                 wax biosynthetic process
## 1158         brown 1042  9.6e-04  1.0e+00       6  MF fatty-acyl-CoA reductase (alcohol-formin
## 1159         brown 1042  1.2e-03  1.0e+00      18  BP                  lipid catabolic process
## 1160         brown 1042  1.4e-03  1.0e+00       6  BP long-chain fatty-acyl-CoA metabolic proc
## 1161         brown 1042  1.5e-03  1.0e+00       5  MF                        galactose binding
## 1162         brown 1042  1.7e-03  1.0e+00      14  BP      membrane lipid biosynthetic process
## 1163         brown 1042  1.9e-03  1.0e+00      10  BP               neuroblast differentiation
## 1164         brown 1042  1.9e-03  1.0e+00      12  MF                       symporter activity
## 1165         brown 1042  2.1e-03  1.0e+00       4  CC vacuolar proton-transporting V-type ATPa
## 1166         brown 1042  2.1e-03  1.0e+00      16  BP         membrane lipid metabolic process
## 1167         brown 1042  2.3e-03  1.0e+00       8  BP             stem cell fate determination
## 1168         brown 1042  2.8e-03  1.0e+00       9  BP               acyl-CoA metabolic process
## 1169         brown 1042  3.3e-03  1.0e+00       7  BP            neuroblast fate determination
## 1170         brown 1042  3.3e-03  1.0e+00       5  MF               alpha-mannosidase activity
## 1171         brown 1042  3.5e-03  1.0e+00       4  BP                mannose metabolic process
## 1172         brown 1042  3.5e-03  1.0e+00       4  BP Malpighian tubule tip cell differentiati
## 1173         brown 1042  3.5e-03  1.0e+00       4  MF secondary active sulfate transmembrane t
## 1174         brown 1042  3.6e-03  1.0e+00      44  BP         cellular lipid metabolic process
## 1175         brown 1042  4.2e-03  1.0e+00      14  MF      carboxylic ester hydrolase activity
## 1176         brown 1042  4.4e-03  1.0e+00       3  BP cuticle hydrocarbon biosynthetic process
## 1177         brown 1042  4.4e-03  1.0e+00       3  BP sex determination, establishment of X:A 
## 1178         brown 1042  4.7e-03  1.0e+00       5  MF                     mannosidase activity
## 1179         brown 1042  5.5e-03  1.0e+00       4  CC proton-transporting V-type ATPase, V1 do
## 1180         brown 1042  5.5e-03  1.0e+00       4  MF single-stranded DNA endodeoxyribonucleas
## 1181         brown 1042  5.8e-03  1.0e+00      37  MF cation transmembrane transporter activit
## 1182         brown 1042  6.0e-03  1.0e+00       6  BP           regulation of intracellular pH
## 1183         brown 1042  6.1e-03  1.0e+00       8  BP cell fate commitment involved in pattern
## 1184         brown 1042  6.2e-03  1.0e+00       2  BP          sphingomyelin catabolic process
## 1185         brown 1042  6.2e-03  1.0e+00       2  BP mitochondrial proton-transporting ATP sy
## 1186         brown 1042  6.2e-03  1.0e+00       2  BP     induction of bacterial agglutination
## 1187         brown 1042  6.2e-03  1.0e+00       2  BP         smooth septate junction assembly
## 1188         brown 1042  6.2e-03  1.0e+00       2  MF          strictosidine synthase activity
## 1189         brown 1042  6.2e-03  1.0e+00       2  MF acid sphingomyelin phosphodiesterase act
## 1190         brown 1042  6.8e-03  1.0e+00      10  BP           proton transmembrane transport
## 1191         brown 1042  7.5e-03  1.0e+00       6  BP sensory organ precursor cell fate determ
## 1192         brown 1042  7.5e-03  1.0e+00       6  BP                regulation of cellular pH
## 1193         brown 1042  7.6e-03  1.0e+00       7  BP               neuroblast fate commitment
## 1194         brown 1042  7.7e-03  1.0e+00      21  BP        sulfur compound metabolic process
## 1195         brown 1042  8.1e-03  1.0e+00      35  MF inorganic cation transmembrane transport
## 1196         brown 1042  8.2e-03  1.0e+00       4  BP                        sulfate transport
## 1197         brown 1042  8.2e-03  1.0e+00       4  BP           sphingolipid catabolic process
## 1198         brown 1042  8.2e-03  1.0e+00       4  MF sulfate transmembrane transporter activi
## 1199          cyan  391  1.0e-08  8.6e-05       9  MF proton-transporting ATPase activity, rot
## 1200          cyan  391  7.8e-08  6.5e-04       7  CC plasma membrane proton-transporting V-ty
## 1201          cyan  391  2.0e-07  1.7e-03       8  CC proton-transporting V-type ATPase comple
## 1202          cyan  391  4.1e-07  3.4e-03       8  BP  ATP hydrolysis coupled proton transport
## 1203          cyan  391  4.2e-07  3.5e-03      10  MF ATPase coupled ion transmembrane transpo
## 1204          cyan  391  5.8e-07  4.8e-03      36  BP                  transmembrane transport
## 1205          cyan  391  9.7e-07  8.1e-03      17  BP inorganic cation transmembrane transport
## 1206          cyan  391  1.1e-06  9.5e-03      14  MF                           chitin binding
## 1207          cyan  391  1.7e-06  1.5e-02      10  BP           proton transmembrane transport
## 1208          cyan  391  3.6e-06  3.0e-02      13  BP                 chitin metabolic process
## 1209          cyan  391  5.3e-06  4.4e-02      17  BP           cation transmembrane transport
## 1210          cyan  391  9.5e-06  7.9e-02       4  BP                   response to copper ion
## 1211          cyan  391  1.0e-05  8.5e-02      12  MF ATPase activity, coupled to transmembran
## 1212          cyan  391  1.7e-05  1.5e-01      22  BP                         cation transport
## 1213          cyan  391  1.9e-05  1.6e-01      20  BP              ion transmembrane transport
## 1214          cyan  391  3.7e-05  3.1e-01      37  MF                     transporter activity
## 1215          cyan  391  4.2e-05  3.5e-01       8  BP                    response to metal ion
## 1216          cyan  391  4.4e-05  3.7e-01       5  CC vacuolar proton-transporting V-type ATPa
## 1217          cyan  391  5.2e-05  4.3e-01      33  MF       transmembrane transporter activity
## 1218          cyan  391  5.5e-05  4.6e-01      28  BP                            ion transport
## 1219          cyan  391  8.6e-05  7.2e-01      14  BP              response to toxic substance
## 1220          cyan  391  9.8e-05  8.2e-01      17  MF active transmembrane transporter activit
## 1221          cyan  391  1.0e-04  8.5e-01       9  BP            Malpighian tubule development
## 1222          cyan  391  1.0e-04  8.7e-01       8  CC                        vacuolar membrane
## 1223          cyan  391  1.2e-04  1.0e+00       4  CC proton-transporting V-type ATPase, V1 do
## 1224          cyan  391  1.3e-04  1.0e+00       9  BP            digestive tract morphogenesis
## 1225          cyan  391  1.5e-04  1.0e+00      10  MF proton transmembrane transporter activit
## 1226          cyan  391  1.9e-04  1.0e+00      13  BP    monovalent inorganic cation transport
## 1227          cyan  391  1.9e-04  1.0e+00       4  CC proton-transporting V-type ATPase, V0 do
## 1228          cyan  391  2.2e-04  1.0e+00      11  BP              digestive tract development
## 1229          cyan  391  2.4e-04  1.0e+00      12  MF                             heme binding
## 1230          cyan  391  2.5e-04  1.0e+00      13  MF                         iron ion binding
## 1231          cyan  391  2.6e-04  1.0e+00      20  MF inorganic cation transmembrane transport
## 1232          cyan  391  2.7e-04  1.0e+00       4  BP   anterior Malpighian tubule development
## 1233          cyan  391  2.8e-04  1.0e+00      13  MF                            lipid binding
## 1234          cyan  391  4.6e-04  1.0e+00      22  MF                         cofactor binding
## 1235          cyan  391  4.8e-04  1.0e+00      20  MF cation transmembrane transporter activit
## 1236          cyan  391  5.8e-04  1.0e+00       7  BP            glutathione metabolic process
## 1237          cyan  391  6.0e-04  1.0e+00      11  BP             fatty acid metabolic process
## 1238          cyan  391  6.1e-04  1.0e+00      25  MF   ion transmembrane transporter activity
## 1239          cyan  391  6.1e-04  1.0e+00      10  BP                 cellular ion homeostasis
## 1240          cyan  391  7.0e-04  1.0e+00       4  BP                     response to zinc ion
## 1241          cyan  391  7.7e-04  1.0e+00      10  BP                        heart development
## 1242          cyan  391  8.2e-04  1.0e+00       2  BP   iron ion import across plasma membrane
## 1243          cyan  391  8.2e-04  1.0e+00       2  BP               detoxification of iron ion
## 1244          cyan  391  8.2e-04  1.0e+00       2  CC                  collagen type IV trimer
## 1245          cyan  391  8.2e-04  1.0e+00       2  CC           intracellular ferritin complex
## 1246          cyan  391  8.2e-04  1.0e+00       2  MF                   alpha-amylase activity
## 1247          cyan  391  8.2e-04  1.0e+00       2  MF phosphatidylinositol-5-phosphate binding
## 1248          cyan  391  8.2e-04  1.0e+00       2  MF          pristanoyl-CoA oxidase activity
## 1249          cyan  391  9.7e-04  1.0e+00      11  BP                       cation homeostasis
## 1250          cyan  391  1.1e-03  1.0e+00       5  BP       striated muscle tissue development
## 1251          cyan  391  1.1e-03  1.0e+00       5  BP               cardiocyte differentiation
## 1252          cyan  391  1.2e-03  1.0e+00       3  BP            cellular iron ion homeostasis
## 1253          cyan  391  1.2e-03  1.0e+00       3  CC vacuolar proton-transporting V-type ATPa
## 1254          cyan  391  1.2e-03  1.0e+00      11  CC                                  vacuole
## 1255          cyan  391  1.5e-03  1.0e+00      11  BP                          ion homeostasis
## 1256          cyan  391  1.6e-03  1.0e+00       8  BP           cellular metal ion homeostasis
## 1257          cyan  391  1.7e-03  1.0e+00       5  BP                muscle tissue development
## 1258          cyan  391  2.4e-03  1.0e+00       3  BP        cardiac muscle tissue development
## 1259          cyan  391  2.4e-03  1.0e+00       3  CC vacuolar proton-transporting V-type ATPa
## 1260          cyan  391  2.4e-03  1.0e+00       2  BP        ectoderm and mesoderm interaction
## 1261          cyan  391  2.4e-03  1.0e+00       2  BP                     response to iron ion
## 1262          cyan  391  2.4e-03  1.0e+00       2  MF                         amylase activity
## 1263          cyan  391  2.7e-03  1.0e+00       9  BP                    metal ion homeostasis
## 1264          cyan  391  3.0e-03  1.0e+00       5  BP                       midgut development
## 1265          cyan  391  3.2e-03  1.0e+00       3  BP                   vacuolar acidification
## 1266          cyan  391  3.4e-03  1.0e+00       5  MF                     antioxidant activity
## 1267          cyan  391  3.7e-03  1.0e+00      23  BP                      homeostatic process
## 1268          cyan  391  4.2e-03  1.0e+00       3  BP                     iron ion homeostasis
## 1269          cyan  391  4.3e-03  1.0e+00       5  BP         transition metal ion homeostasis
## 1270          cyan  391  4.5e-03  1.0e+00      12  BP                      metal ion transport
## 1271          cyan  391  4.7e-03  1.0e+00       2  BP positive regulation of axon extension in
## 1272          cyan  391  4.7e-03  1.0e+00       2  CC                          collagen trimer
## 1273          cyan  391  4.7e-03  1.0e+00       2  MF  gamma-glutamylcyclotransferase activity
## 1274          cyan  391  4.7e-03  1.0e+00       2  MF          glutathione peroxidase activity
## 1275          cyan  391  4.7e-03  1.0e+00       2  MF proton-exporting ATPase activity, phosph
## 1276          cyan  391  4.8e-03  1.0e+00       5  MF         glutathione transferase activity
## 1277          cyan  391  5.1e-03  1.0e+00       7  BP             divalent metal ion transport
## 1278          cyan  391  5.2e-03  1.0e+00       4  MF                      peroxidase activity
## 1279          cyan  391  5.4e-03  1.0e+00       3  MF ATPase activity, coupled to transmembran
## 1280          cyan  391  6.0e-03  1.0e+00       4  MF oxidoreductase activity, acting on perox
## 1281          cyan  391  7.7e-03  1.0e+00      37  CC                     extracellular region
## 1282          cyan  391  7.7e-03  1.0e+00       2  BP                     nucleoside transport
## 1283          cyan  391  7.7e-03  1.0e+00       2  BP      cardiac muscle cell differentiation
## 1284          cyan  391  7.7e-03  1.0e+00       2  BP       nucleoside transmembrane transport
## 1285          cyan  391  7.7e-03  1.0e+00       2  MF nucleoside transmembrane transporter act
## 1286          cyan  391  7.7e-03  1.0e+00       2  MF                     ferrous iron binding
## 1287          cyan  391  7.7e-03  1.0e+00       2  MF              acyl-CoA hydrolase activity
## 1288          cyan  391  7.7e-03  1.0e+00       2  MF phosphatidylinositol-3,5-bisphosphate bi
## 1289          cyan  391  7.8e-03  1.0e+00       6  MF transferase activity, transferring alkyl
## 1290          cyan  391  8.2e-03  1.0e+00       3  CC                       lysosomal membrane
## 1291          cyan  391  8.9e-03  1.0e+00       4  BP           transition metal ion transport
## 1292          cyan  391  8.9e-03  1.0e+00       4  BP                  cell-substrate adhesion
## 1293          cyan  391  8.9e-03  1.0e+00       5  BP          Malpighian tubule morphogenesis
## 1294          cyan  391  9.3e-03  1.0e+00      49  CC          intrinsic component of membrane
## 1295          cyan  391  9.6e-03  1.0e+00       7  MF hydrolase activity, hydrolyzing O-glycos
## 1296          cyan  391  9.9e-03  1.0e+00       3  BP               intracellular pH reduction
## 1297     darkgreen   68  8.4e-11  7.1e-07      37  BP       multicellular organism development
## 1298     darkgreen   68  4.0e-09  3.3e-05      37  BP         anatomical structure development
## 1299     darkgreen   68  4.1e-09  3.5e-05      28  BP       anatomical structure morphogenesis
## 1300     darkgreen   68  7.7e-09  6.4e-05      29  BP                                signaling
## 1301     darkgreen   68  1.3e-08  1.1e-04      29  BP                       cell communication
## 1302     darkgreen   68  1.4e-08  1.2e-04      37  BP                    developmental process
## 1303     darkgreen   68  1.7e-08  1.4e-04      19  BP        regulation of signal transduction
## 1304     darkgreen   68  2.7e-08  2.3e-04      17  BP           morphogenesis of an epithelium
## 1305     darkgreen   68  3.2e-08  2.7e-04      26  BP                      signal transduction
## 1306     darkgreen   68  3.3e-08  2.8e-04      20  BP                    generation of neurons
## 1307     darkgreen   68  7.2e-08  6.1e-04      20  BP                             neurogenesis
## 1308     darkgreen   68  7.4e-08  6.2e-04      32  MF                          protein binding
## 1309     darkgreen   68  8.1e-08  6.8e-04      22  BP               nervous system development
## 1310     darkgreen   68  1.4e-07  1.2e-03      18  BP               animal organ morphogenesis
## 1311     darkgreen   68  1.5e-07  1.2e-03      27  BP                     cell differentiation
## 1312     darkgreen   68  1.9e-07  1.6e-03      14  BP               regulation of localization
## 1313     darkgreen   68  2.1e-07  1.8e-03      16  BP                cytoskeleton organization
## 1314     darkgreen   68  3.2e-07  2.7e-03      20  BP                   epithelium development
## 1315     darkgreen   68  4.8e-07  4.0e-03      22  BP                 animal organ development
## 1316     darkgreen   68  5.4e-07  4.5e-03      11  BP          actin cytoskeleton organization
## 1317     darkgreen   68  5.4e-07  4.5e-03      16  BP        intracellular signal transduction
## 1318     darkgreen   68  5.6e-07  4.7e-03      15  BP                imaginal disc development
## 1319     darkgreen   68  8.0e-07  6.7e-03      17  BP                   neuron differentiation
## 1320     darkgreen   68  9.0e-07  7.6e-03      11  BP             actin filament-based process
## 1321     darkgreen   68  1.2e-06  1.0e-02      20  BP                       tissue development
## 1322     darkgreen   68  1.8e-06  1.5e-02      13  BP            epithelial tube morphogenesis
## 1323     darkgreen   68  1.9e-06  1.6e-02      16  BP                         tube development
## 1324     darkgreen   68  2.0e-06  1.7e-02      15  BP                       neuron development
## 1325     darkgreen   68  3.3e-06  2.8e-02      13  BP            neuron projection development
## 1326     darkgreen   68  3.7e-06  3.1e-02      16  BP      regulation of developmental process
## 1327     darkgreen   68  4.0e-06  3.4e-02       9  BP establishment or maintenance of cell pol
## 1328     darkgreen   68  4.1e-06  3.5e-02      13  BP                       tube morphogenesis
## 1329     darkgreen   68  4.3e-06  3.6e-02      14  BP                       cell morphogenesis
## 1330     darkgreen   68  4.8e-06  4.0e-02       6  BP                mushroom body development
## 1331     darkgreen   68  5.0e-06  4.1e-02       7  BP                        brain development
## 1332     darkgreen   68  5.6e-06  4.7e-02      22  BP                         cell development
## 1333     darkgreen   68  1.0e-05  8.4e-02       7  BP      regulation of cellular localization
## 1334     darkgreen   68  1.1e-05  9.6e-02      20  CC                           cell periphery
## 1335     darkgreen   68  1.1e-05  9.6e-02      13  BP       instar larval or pupal development
## 1336     darkgreen   68  1.6e-05  1.3e-01       7  BP                         head development
## 1337     darkgreen   68  1.8e-05  1.5e-01      10  BP negative regulation of signal transducti
## 1338     darkgreen   68  2.1e-05  1.8e-01      11  BP                  protein phosphorylation
## 1339     darkgreen   68  2.2e-05  1.9e-01      31  BP          cellular component organization
## 1340     darkgreen   68  3.0e-05  2.5e-01      18  CC                          plasma membrane
## 1341     darkgreen   68  3.6e-05  3.0e-01       6  BP                           dorsal closure
## 1342     darkgreen   68  3.8e-05  3.2e-01      14  BP  cell surface receptor signaling pathway
## 1343     darkgreen   68  4.1e-05  3.4e-01       5  BP                      pole plasm assembly
## 1344     darkgreen   68  4.3e-05  3.6e-01      13  BP               post-embryonic development
## 1345     darkgreen   68  5.3e-05  4.5e-01       8  BP       central nervous system development
## 1346     darkgreen   68  5.4e-05  4.5e-01      10  BP                    wing disc development
## 1347     darkgreen   68  6.3e-05  5.3e-01      11  BP                            metamorphosis
## 1348     darkgreen   68  6.5e-05  5.4e-01       6  BP                   synaptic vesicle cycle
## 1349     darkgreen   68  7.9e-05  6.6e-01      10  BP              imaginal disc morphogenesis
## 1350     darkgreen   68  8.2e-05  6.9e-01       6  BP    morphogenesis of embryonic epithelium
## 1351     darkgreen   68  1.3e-04  1.0e+00       4  BP                          hippo signaling
## 1352     darkgreen   68  1.3e-04  1.0e+00       3  BP     negative regulation of cell adhesion
## 1353     darkgreen   68  1.3e-04  1.0e+00       9  MF                          kinase activity
## 1354     darkgreen   68  1.3e-04  1.0e+00       5  BP      regulation of embryonic development
## 1355     darkgreen   68  1.4e-04  1.0e+00       5  BP oocyte anterior/posterior axis specifica
## 1356     darkgreen   68  1.6e-04  1.0e+00       5  BP maternal determination of anterior/poste
## 1357     darkgreen   68  1.7e-04  1.0e+00       7  CC                                     axon
## 1358     darkgreen   68  1.8e-04  1.0e+00       4  BP              regulation of cell adhesion
## 1359     darkgreen   68  1.8e-04  1.0e+00       3  BP               adherens junction assembly
## 1360     darkgreen   68  1.8e-04  1.0e+00       9  BP                         axon development
## 1361     darkgreen   68  1.9e-04  1.0e+00       4  BP       pole plasm oskar mRNA localization
## 1362     darkgreen   68  2.0e-04  1.0e+00       2  BP                heparin metabolic process
## 1363     darkgreen   68  2.0e-04  1.0e+00       2  BP        ephrin receptor signaling pathway
## 1364     darkgreen   68  2.0e-04  1.0e+00       2  MF                guanylate kinase activity
## 1365     darkgreen   68  2.1e-04  1.0e+00       3  BP establishment of mitotic spindle orienta
## 1366     darkgreen   68  2.1e-04  1.0e+00       9  BP                                    taxis
## 1367     darkgreen   68  2.2e-04  1.0e+00       8  BP                  wing disc morphogenesis
## 1368     darkgreen   68  2.2e-04  1.0e+00       5  BP microtubule cytoskeleton organization in
## 1369     darkgreen   68  2.4e-04  1.0e+00       7  BP    regulation of protein phosphorylation
## 1370     darkgreen   68  2.4e-04  1.0e+00       6  CC                      apical part of cell
## 1371     darkgreen   68  2.5e-04  1.0e+00       9  BP                     developmental growth
## 1372     darkgreen   68  2.6e-04  1.0e+00       4  BP             pole plasm mRNA localization
## 1373     darkgreen   68  2.8e-04  1.0e+00       9  CC                        neuron projection
## 1374     darkgreen   68  2.8e-04  1.0e+00      10  BP          neuron projection morphogenesis
## 1375     darkgreen   68  2.8e-04  1.0e+00      12  BP                               locomotion
## 1376     darkgreen   68  2.8e-04  1.0e+00       4  BP              pole plasm RNA localization
## 1377     darkgreen   68  3.0e-04  1.0e+00       4  BP    regulation of intracellular transport
## 1378     darkgreen   68  3.0e-04  1.0e+00       9  BP                 compound eye development
## 1379     darkgreen   68  3.5e-04  1.0e+00       8  BP               compound eye morphogenesis
## 1380     darkgreen   68  3.6e-04  1.0e+00       3  BP      negative regulation of organ growth
## 1381     darkgreen   68  3.8e-04  1.0e+00       6  CC                              cell cortex
## 1382     darkgreen   68  3.8e-04  1.0e+00      11  BP                          phosphorylation
## 1383     darkgreen   68  4.0e-04  1.0e+00      10  BP                      cell-cell signaling
## 1384     darkgreen   68  4.1e-04  1.0e+00       3  BP regulation of pole plasm oskar mRNA loca
## 1385     darkgreen   68  4.6e-04  1.0e+00       8  BP                        eye morphogenesis
## 1386     darkgreen   68  4.6e-04  1.0e+00       3  BP                      response to gravity
## 1387     darkgreen   68  4.6e-04  1.0e+00       3  BP                               gravitaxis
## 1388     darkgreen   68  4.6e-04  1.0e+00       3  BP     establishment of spindle orientation
## 1389     darkgreen   68  4.6e-04  1.0e+00       3  BP regulation of embryonic pattern specific
## 1390     darkgreen   68  4.6e-04  1.0e+00       7  BP            regulation of phosphorylation
## 1391     darkgreen   68  4.7e-04  1.0e+00       9  BP                          eye development
## 1392     darkgreen   68  4.7e-04  1.0e+00       4  BP           establishment of cell polarity
## 1393     darkgreen   68  4.7e-04  1.0e+00       4  BP intracellular mRNA localization involved
## 1394     darkgreen   68  4.8e-04  1.0e+00       6  BP embryonic development via the syncytial 
## 1395     darkgreen   68  4.8e-04  1.0e+00      10  BP                sensory organ development
## 1396     darkgreen   68  4.9e-04  1.0e+00      11  BP      cellular macromolecule localization
## 1397     darkgreen   68  4.9e-04  1.0e+00       7  BP                  regulation of transport
## 1398     darkgreen   68  5.0e-04  1.0e+00       8  BP   post-embryonic appendage morphogenesis
## 1399     darkgreen   68  5.0e-04  1.0e+00       4  BP               regulation of organ growth
## 1400     darkgreen   68  5.1e-04  1.0e+00       2  BP heparan sulfate proteoglycan biosyntheti
## 1401     darkgreen   68  5.1e-04  1.0e+00       2  BP        border follicle cell delamination
## 1402     darkgreen   68  5.1e-04  1.0e+00       2  MF                         spectrin binding
## 1403     darkgreen   68  5.1e-04  1.0e+00       6  MF protein serine/threonine kinase activity
## 1404     darkgreen   68  5.1e-04  1.0e+00       7  MF                  protein kinase activity
## 1405     darkgreen   68  5.2e-04  1.0e+00       3  BP regulation of intracellular mRNA localiz
## 1406     darkgreen   68  5.2e-04  1.0e+00       3  CC                   type I terminal bouton
## 1407     darkgreen   68  5.3e-04  1.0e+00      10  BP cell morphogenesis involved in different
## 1408     darkgreen   68  5.5e-04  1.0e+00       8  BP    microtubule cytoskeleton organization
## 1409     darkgreen   68  5.7e-04  1.0e+00       4  BP intracellular mRNA localization involved
## 1410     darkgreen   68  5.8e-04  1.0e+00       3  BP establishment of mitotic spindle localiz
## 1411     darkgreen   68  6.0e-04  1.0e+00       8  BP imaginal disc-derived appendage morphoge
## 1412     darkgreen   68  6.1e-04  1.0e+00       5  BP  morphogenesis of a polarized epithelium
## 1413     darkgreen   68  6.4e-04  1.0e+00       8  BP                  appendage morphogenesis
## 1414     darkgreen   68  6.4e-04  1.0e+00       6  BP transmembrane receptor protein tyrosine 
## 1415     darkgreen   68  6.9e-04  1.0e+00       9  BP       regulation of cell differentiation
## 1416     darkgreen   68  6.9e-04  1.0e+00       5  BP anterior/posterior axis specification, e
## 1417     darkgreen   68  7.0e-04  1.0e+00       8  BP imaginal disc-derived appendage developm
## 1418     darkgreen   68  7.1e-04  1.0e+00       2  BP negative regulation of response to wound
## 1419     darkgreen   68  7.1e-04  1.0e+00       3  BP melanotic encapsulation of foreign targe
## 1420     darkgreen   68  7.1e-04  1.0e+00       4  BP              cell-cell junction assembly
## 1421     darkgreen   68  7.2e-04  1.0e+00       8  BP                     cell fate commitment
## 1422     darkgreen   68  7.4e-04  1.0e+00       5  BP                oocyte axis specification
## 1423     darkgreen   68  7.5e-04  1.0e+00       4  BP                  regulation of oogenesis
## 1424     darkgreen   68  8.2e-04  1.0e+00      25  CC                                 membrane
## 1425     darkgreen   68  8.3e-04  1.0e+00       5  BP             embryonic axis specification
## 1426     darkgreen   68  8.3e-04  1.0e+00       5  BP                      oocyte construction
## 1427     darkgreen   68  8.6e-04  1.0e+00       5  BP                     response to wounding
## 1428     darkgreen   68  8.6e-04  1.0e+00      10  CC                          cell projection
## 1429     darkgreen   68  8.8e-04  1.0e+00       4  BP          cell-cell junction organization
## 1430     darkgreen   68  9.0e-04  1.0e+00       9  MF transferase activity, transferring phosp
## 1431     darkgreen   68  9.3e-04  1.0e+00       6  BP embryo development ending in birth or eg
## 1432     darkgreen   68  9.4e-04  1.0e+00       2  CC                  type Ib terminal bouton
## 1433     darkgreen   68  9.6e-04  1.0e+00       9  BP                                   growth
## 1434     darkgreen   68  9.8e-04  1.0e+00       6  BP                  blastoderm segmentation
## 1435     darkgreen   68  9.9e-04  1.0e+00       5  BP    anterior/posterior axis specification
## 1436     darkgreen   68  1.0e-03  1.0e+00       7  BP imaginal disc-derived wing morphogenesis
## 1437     darkgreen   68  1.0e-03  1.0e+00       3  BP                     spindle localization
## 1438     darkgreen   68  1.1e-03  1.0e+00       4  BP                             organ growth
## 1439     darkgreen   68  1.1e-03  1.0e+00       9  BP cell morphogenesis involved in neuron di
## 1440     darkgreen   68  1.2e-03  1.0e+00       7  BP                            cell division
## 1441     darkgreen   68  1.2e-03  1.0e+00       6  BP                            cell adhesion
## 1442     darkgreen   68  1.3e-03  1.0e+00      51  BP                         cellular process
## 1443     darkgreen   68  1.3e-03  1.0e+00       5  BP                       oocyte development
## 1444     darkgreen   68  1.3e-03  1.0e+00      20  BP                   organelle organization
## 1445     darkgreen   68  1.5e-03  1.0e+00       4  BP          intracellular mRNA localization
## 1446     darkgreen   68  1.5e-03  1.0e+00       4  BP negative regulation of developmental gro
## 1447     darkgreen   68  1.5e-03  1.0e+00       2  BP negative regulation of microtubule depol
## 1448     darkgreen   68  1.5e-03  1.0e+00       3  BP           adherens junction organization
## 1449     darkgreen   68  1.5e-03  1.0e+00       3  BP          encapsulation of foreign target
## 1450     darkgreen   68  1.5e-03  1.0e+00       6  BP          embryonic pattern specification
## 1451     darkgreen   68  1.6e-03  1.0e+00      11  BP                     protein localization
## 1452     darkgreen   68  1.7e-03  1.0e+00       3  BP establishment or maintenance of microtub
## 1453     darkgreen   68  1.8e-03  1.0e+00       3  BP regulation of smoothened signaling pathw
## 1454     darkgreen   68  1.9e-03  1.0e+00       3  BP       regulation of imaginal disc growth
## 1455     darkgreen   68  2.0e-03  1.0e+00       7  MF phosphotransferase activity, alcohol gro
## 1456     darkgreen   68  2.2e-03  1.0e+00       2  BP          pole plasm protein localization
## 1457     darkgreen   68  2.2e-03  1.0e+00       2  BP            regulation of ATPase activity
## 1458     darkgreen   68  2.2e-03  1.0e+00       2  BP               basal protein localization
## 1459     darkgreen   68  2.2e-03  1.0e+00       2  BP                             delamination
## 1460     darkgreen   68  2.2e-03  1.0e+00       2  MF nucleoside monophosphate kinase activity
## 1461     darkgreen   68  2.2e-03  1.0e+00       6  BP       regulation of developmental growth
## 1462     darkgreen   68  2.2e-03  1.0e+00      15  BP    cellular protein modification process
## 1463     darkgreen   68  2.3e-03  1.0e+00       4  BP       eye photoreceptor cell development
## 1464     darkgreen   68  2.5e-03  1.0e+00       5  BP                   oocyte differentiation
## 1465     darkgreen   68  2.6e-03  1.0e+00       4  BP               neurotransmitter secretion
## 1466     darkgreen   68  2.6e-03  1.0e+00       4  BP                 neuroblast proliferation
## 1467     darkgreen   68  2.6e-03  1.0e+00       3  BP            melanization defense response
## 1468     darkgreen   68  2.6e-03  1.0e+00       3  MF                       Rho GTPase binding
## 1469     darkgreen   68  2.7e-03  1.0e+00       9  BP            pattern specification process
## 1470     darkgreen   68  2.7e-03  1.0e+00       6  BP                  embryonic morphogenesis
## 1471     darkgreen   68  2.8e-03  1.0e+00       5  CC                            cell junction
## 1472     darkgreen   68  2.9e-03  1.0e+00       6  BP                             segmentation
## 1473     darkgreen   68  3.0e-03  1.0e+00       3  BP               synaptic vesicle recycling
## 1474     darkgreen   68  3.0e-03  1.0e+00       3  CC              basolateral plasma membrane
## 1475     darkgreen   68  3.0e-03  1.0e+00       2  BP regulation of microtubule depolymerizati
## 1476     darkgreen   68  3.0e-03  1.0e+00       2  BP     negative regulation of ion transport
## 1477     darkgreen   68  3.1e-03  1.0e+00       8  BP                microtubule-based process
## 1478     darkgreen   68  3.1e-03  1.0e+00       5  MF                       Ras GTPase binding
## 1479     darkgreen   68  3.3e-03  1.0e+00       7  BP                             axonogenesis
## 1480     darkgreen   68  3.4e-03  1.0e+00       4  BP                 asymmetric cell division
## 1481     darkgreen   68  3.4e-03  1.0e+00       5  MF                     small GTPase binding
## 1482     darkgreen   68  3.4e-03  1.0e+00       2  BP heparan sulfate proteoglycan biosyntheti
## 1483     darkgreen   68  3.4e-03  1.0e+00       2  BP                        axon regeneration
## 1484     darkgreen   68  3.5e-03  1.0e+00       4  BP         establishment of planar polarity
## 1485     darkgreen   68  3.5e-03  1.0e+00       4  BP         establishment of tissue polarity
## 1486     darkgreen   68  3.6e-03  1.0e+00       5  BP                         RNA localization
## 1487     darkgreen   68  3.6e-03  1.0e+00       5  BP anterior/posterior pattern specification
## 1488     darkgreen   68  3.9e-03  1.0e+00       4  BP            negative regulation of growth
## 1489     darkgreen   68  3.9e-03  1.0e+00       2  BP heparan sulfate proteoglycan metabolic p
## 1490     darkgreen   68  4.1e-03  1.0e+00      19  BP regulation of cellular metabolic process
## 1491     darkgreen   68  4.1e-03  1.0e+00       4  BP           photoreceptor cell development
## 1492     darkgreen   68  4.1e-03  1.0e+00       3  BP                     imaginal disc growth
## 1493     darkgreen   68  4.1e-03  1.0e+00       3  BP           behavioral response to ethanol
## 1494     darkgreen   68  4.2e-03  1.0e+00       7  BP              epithelial cell development
## 1495     darkgreen   68  4.2e-03  1.0e+00       4  BP                    Wnt signaling pathway
## 1496     darkgreen   68  4.3e-03  1.0e+00      61  BP                       biological_process
## 1497     darkgreen   68  4.4e-03  1.0e+00       8  BP                       mitotic cell cycle
## 1498     darkgreen   68  4.4e-03  1.0e+00       2  BP negative regulation of imaginal disc gro
## 1499     darkgreen   68  4.4e-03  1.0e+00       2  BP positive regulation of smoothened signal
## 1500     darkgreen   68  4.7e-03  1.0e+00       4  BP                  stem cell proliferation
## 1501     darkgreen   68  4.8e-03  1.0e+00       7  BP                       cell proliferation
## 1502     darkgreen   68  4.9e-03  1.0e+00       2  BP   positive regulation of hippo signaling
## 1503     darkgreen   68  4.9e-03  1.0e+00       2  BP              apical protein localization
## 1504     darkgreen   68  5.0e-03  1.0e+00       3  BP             smoothened signaling pathway
## 1505     darkgreen   68  5.0e-03  1.0e+00       5  MF                           GTPase binding
## 1506     darkgreen   68  5.3e-03  1.0e+00       5  BP                       axis specification
## 1507     darkgreen   68  5.3e-03  1.0e+00       3  BP                 apical junction assembly
## 1508     darkgreen   68  5.3e-03  1.0e+00       6  BP                            axon guidance
## 1509     darkgreen   68  5.5e-03  1.0e+00      22  BP                protein metabolic process
## 1510     darkgreen   68  5.5e-03  1.0e+00       2  BP             microtubule depolymerization
## 1511     darkgreen   68  5.5e-03  1.0e+00       2  BP asymmetric protein localization involved
## 1512     darkgreen   68  5.5e-03  1.0e+00       2  MF protein tyrosine/serine/threonine phosph
## 1513     darkgreen   68  5.5e-03  1.0e+00      20  BP          regulation of metabolic process
## 1514     darkgreen   68  5.5e-03  1.0e+00       3  BP cortical actin cytoskeleton organization
## 1515     darkgreen   68  5.6e-03  1.0e+00       4  BP positive regulation of protein phosphory
## 1516     darkgreen   68  5.8e-03  1.0e+00       3  BP                 regulation of cell shape
## 1517     darkgreen   68  5.8e-03  1.0e+00       3  BP              synaptic vesicle exocytosis
## 1518     darkgreen   68  5.8e-03  1.0e+00       3  CC                       cell cortex region
## 1519     darkgreen   68  5.9e-03  1.0e+00       1  BP                         ultradian rhythm
## 1520     darkgreen   68  5.9e-03  1.0e+00       1  BP positive regulation of peptidyl-threonin
## 1521     darkgreen   68  5.9e-03  1.0e+00       1  BP regulation of border follicle cell delam
## 1522     darkgreen   68  5.9e-03  1.0e+00       1  BP 5S class rRNA transcription by RNA polym
## 1523     darkgreen   68  5.9e-03  1.0e+00       1  BP              zonula adherens maintenance
## 1524     darkgreen   68  5.9e-03  1.0e+00       1  BP negative regulation of focal adhesion as
## 1525     darkgreen   68  5.9e-03  1.0e+00       1  BP cell morphogenesis involved in Malpighia
## 1526     darkgreen   68  5.9e-03  1.0e+00       1  BP negative regulation of ecdysone receptor
## 1527     darkgreen   68  5.9e-03  1.0e+00       1  BP regulation of axo-dendritic protein tran
## 1528     darkgreen   68  5.9e-03  1.0e+00       1  BP       protein localization to presynapse
## 1529     darkgreen   68  5.9e-03  1.0e+00       1  CC interphase microtubule organizing center
## 1530     darkgreen   68  5.9e-03  1.0e+00       1  CC                    microtubule minus-end
## 1531     darkgreen   68  5.9e-03  1.0e+00       1  MF phosphatidylinositol-3,4,5-trisphosphate
## 1532     darkgreen   68  5.9e-03  1.0e+00       1  MF        activin receptor activity, type I
## 1533     darkgreen   68  5.9e-03  1.0e+00       1  MF                  ephrin receptor binding
## 1534     darkgreen   68  5.9e-03  1.0e+00       1  MF               protein N-terminus binding
## 1535     darkgreen   68  5.9e-03  1.0e+00       1  MF                       PTB domain binding
## 1536     darkgreen   68  5.9e-03  1.0e+00       1  MF inositol-1,3,4,5-tetrakisphosphate 3-pho
## 1537     darkgreen   68  5.9e-03  1.0e+00       1  MF phosphatidylinositol-3,4-bisphosphate 3-
## 1538     darkgreen   68  6.0e-03  1.0e+00       6  BP           chemical synaptic transmission
## 1539     darkgreen   68  6.0e-03  1.0e+00       3  BP  regulation of protein catabolic process
## 1540     darkgreen   68  6.1e-03  1.0e+00       2  BP                     cell-matrix adhesion
## 1541     darkgreen   68  6.1e-03  1.0e+00       2  BP        proteoglycan biosynthetic process
## 1542     darkgreen   68  6.1e-03  1.0e+00       2  BP   negative regulation of hippo signaling
## 1543     darkgreen   68  6.1e-03  1.0e+00       2  BP                  response to axon injury
## 1544     darkgreen   68  6.2e-03  1.0e+00       6  BP                       synaptic signaling
## 1545     darkgreen   68  6.3e-03  1.0e+00       4  MF                            actin binding
## 1546     darkgreen   68  6.4e-03  1.0e+00       4  BP                pigment metabolic process
## 1547     darkgreen   68  6.6e-03  1.0e+00       3  BP         calcium ion regulated exocytosis
## 1548     darkgreen   68  6.6e-03  1.0e+00       4  BP                      response to alcohol
## 1549     darkgreen   68  6.7e-03  1.0e+00       2  CC             apicolateral plasma membrane
## 1550     darkgreen   68  6.8e-03  1.0e+00       4  BP   positive regulation of phosphorylation
## 1551     darkgreen   68  7.2e-03  1.0e+00       3  BP                instar larval development
## 1552     darkgreen   68  7.2e-03  1.0e+00       3  BP                     regulated exocytosis
## 1553     darkgreen   68  7.2e-03  1.0e+00       3  CC                          terminal bouton
## 1554     darkgreen   68  7.3e-03  1.0e+00       2  BP negative regulation of smoothened signal
## 1555     darkgreen   68  7.8e-03  1.0e+00       4  BP                           signal release
## 1556     darkgreen   68  7.8e-03  1.0e+00       6  MF             cytoskeletal protein binding
## 1557     darkgreen   68  7.8e-03  1.0e+00       3  BP       cortical cytoskeleton organization
## 1558     darkgreen   68  8.0e-03  1.0e+00       4  BP               neurotransmitter transport
## 1559     darkgreen   68  8.0e-03  1.0e+00       2  BP regulation of microtubule polymerization
## 1560     darkgreen   68  8.3e-03  1.0e+00       6  BP                 regulation of cell cycle
## 1561     darkgreen   68  8.5e-03  1.0e+00      12  BP phosphate-containing compound metabolic 
## 1562     darkgreen   68  8.7e-03  1.0e+00       2  BP                      apical constriction
## 1563     darkgreen   68  8.7e-03  1.0e+00       3  BP regulation of canonical Wnt signaling pa
## 1564     darkgreen   68  9.4e-03  1.0e+00       6  BP                     regulation of growth
## 1565     darkgreen   68  9.4e-03  1.0e+00       2  BP establishment of imaginal disc-derived w
## 1566     darkgreen   68  9.4e-03  1.0e+00       2  BP           clathrin-dependent endocytosis
## 1567     darkgreen   68  9.4e-03  1.0e+00       3  BP             mitotic spindle organization
## 1568     darkgreen   68  9.4e-03  1.0e+00       3  CC                            axon terminus
## 1569     darkgreen   68  9.8e-03  1.0e+00       3  CC               neuron projection terminus
## 1570      darkgrey  249  1.2e-08  9.9e-05      18  MF                       Ras GTPase binding
## 1571      darkgrey  249  1.2e-08  1.0e-04      19  MF                           GTPase binding
## 1572      darkgrey  249  1.6e-08  1.3e-04      18  MF                     small GTPase binding
## 1573      darkgrey  249  2.5e-07  2.1e-03      37  BP        regulation of signal transduction
## 1574      darkgrey  249  6.1e-07  5.1e-03      11  BP regulation of Ras protein signal transdu
## 1575      darkgrey  249  1.6e-06  1.4e-02      11  BP regulation of small GTPase mediated sign
## 1576      darkgrey  249  3.5e-06  2.9e-02       9  MF Ras guanyl-nucleotide exchange factor ac
## 1577      darkgrey  249  3.8e-06  3.2e-02      10  MF guanyl-nucleotide exchange factor activi
## 1578      darkgrey  249  8.6e-06  7.2e-02      13  BP          Ras protein signal transduction
## 1579      darkgrey  249  1.0e-05  8.7e-02      59  BP                                signaling
## 1580      darkgrey  249  1.1e-05  8.9e-02      53  BP                      signal transduction
## 1581      darkgrey  249  2.0e-05  1.7e-01      59  BP                       cell communication
## 1582      darkgrey  249  3.3e-05  2.7e-01      13  BP small GTPase mediated signal transductio
## 1583      darkgrey  249  3.5e-05  2.9e-01      14  BP                              endocytosis
## 1584      darkgrey  249  1.1e-04  9.1e-01      13  MF                             mRNA binding
## 1585      darkgrey  249  1.4e-04  1.0e+00       5  CC                  clathrin-coated vesicle
## 1586      darkgrey  249  1.7e-04  1.0e+00      17  BP                          mRNA processing
## 1587      darkgrey  249  2.9e-04  1.0e+00       5  BP positive regulation of Ras protein signa
## 1588      darkgrey  249  3.1e-04  1.0e+00      11  MF                     phosphatase activity
## 1589      darkgrey  249  3.9e-04  1.0e+00       6  MF                       Rho GTPase binding
## 1590      darkgrey  249  3.9e-04  1.0e+00      11  BP                        dephosphorylation
## 1591      darkgrey  249  4.4e-04  1.0e+00       6  BP            receptor-mediated endocytosis
## 1592      darkgrey  249  5.9e-04  1.0e+00      14  BP RNA splicing, via transesterification re
## 1593      darkgrey  249  5.9e-04  1.0e+00      14  BP           mRNA splicing, via spliceosome
## 1594      darkgrey  249  6.2e-04  1.0e+00       5  MF    protein tyrosine phosphatase activity
## 1595      darkgrey  249  7.2e-04  1.0e+00       8  BP                protein dephosphorylation
## 1596      darkgrey  249  7.5e-04  1.0e+00      20  MF                           enzyme binding
## 1597      darkgrey  249  8.5e-04  1.0e+00      16  BP         regulation of catalytic activity
## 1598      darkgrey  249  9.8e-04  1.0e+00       4  BP negative regulation of insulin receptor 
## 1599      darkgrey  249  1.1e-03  1.0e+00      34  CC                      endomembrane system
## 1600      darkgrey  249  1.1e-03  1.0e+00      14  BP                             RNA splicing
## 1601      darkgrey  249  1.2e-03  1.0e+00      21  BP               vesicle-mediated transport
## 1602      darkgrey  249  1.3e-03  1.0e+00       2  BP       mitotic G1/S transition checkpoint
## 1603      darkgrey  249  1.3e-03  1.0e+00       2  MF phosphatidylinositol-4-phosphate phospha
## 1604      darkgrey  249  1.6e-03  1.0e+00       4  BP           clathrin-dependent endocytosis
## 1605      darkgrey  249  1.7e-03  1.0e+00       5  BP positive regulation of Wnt signaling pat
## 1606      darkgrey  249  1.7e-03  1.0e+00       5  BP          Rab protein signal transduction
## 1607      darkgrey  249  1.7e-03  1.0e+00      11  MF      phosphoric ester hydrolase activity
## 1608      darkgrey  249  1.7e-03  1.0e+00       3  BP regulation of Rab protein signal transdu
## 1609      darkgrey  249  2.1e-03  1.0e+00       3  BP   phosphatidylinositol dephosphorylation
## 1610      darkgrey  249  2.2e-03  1.0e+00      26  BP  cell surface receptor signaling pathway
## 1611      darkgrey  249  2.3e-03  1.0e+00       7  MF                       Rab GTPase binding
## 1612      darkgrey  249  2.5e-03  1.0e+00       2  MF       insulin-like growth factor binding
## 1613      darkgrey  249  2.6e-03  1.0e+00       7  BP            regulation of GTPase activity
## 1614      darkgrey  249  2.7e-03  1.0e+00       3  BP negative regulation of G1/S transition o
## 1615      darkgrey  249  2.7e-03  1.0e+00       3  MF phosphatidylinositol phosphate phosphata
## 1616      darkgrey  249  2.8e-03  1.0e+00       4  MF Rho guanyl-nucleotide exchange factor ac
## 1617      darkgrey  249  3.2e-03  1.0e+00      15  BP negative regulation of signal transducti
## 1618      darkgrey  249  3.3e-03  1.0e+00       9  MF                            lipid binding
## 1619      darkgrey  249  3.6e-03  1.0e+00       4  BP regulation of Rho protein signal transdu
## 1620      darkgrey  249  3.9e-03  1.0e+00      37  BP    cellular protein modification process
## 1621      darkgrey  249  4.1e-03  1.0e+00       2  BP          protein localization to synapse
## 1622      darkgrey  249  4.1e-03  1.0e+00       2  MF                       Ral GTPase binding
## 1623      darkgrey  249  4.1e-03  1.0e+00      17  CC                                  vesicle
## 1624      darkgrey  249  4.5e-03  1.0e+00       6  BP            regulation of mRNA processing
## 1625      darkgrey  249  4.7e-03  1.0e+00       7  MF                     phospholipid binding
## 1626      darkgrey  249  4.8e-03  1.0e+00       3  BP positive regulation of smoothened signal
## 1627      darkgrey  249  4.8e-03  1.0e+00       6  MF      phosphoprotein phosphatase activity
## 1628      darkgrey  249  4.8e-03  1.0e+00      14  BP                     regulation of growth
## 1629      darkgrey  249  5.4e-03  1.0e+00      47  MF                     nucleic acid binding
## 1630      darkgrey  249  5.7e-03  1.0e+00       4  BP regulation of insulin receptor signaling
## 1631      darkgrey  249  6.1e-03  1.0e+00       2  BP clathrin-dependent endocytosis involved 
## 1632      darkgrey  249  6.3e-03  1.0e+00      21  MF                              RNA binding
## 1633      darkgrey  249  6.5e-03  1.0e+00      23  BP        intracellular signal transduction
## 1634      darkgrey  249  6.7e-03  1.0e+00      13  BP                     synapse organization
## 1635      darkgrey  249  6.8e-03  1.0e+00       5  BP alternative mRNA splicing, via spliceoso
## 1636      darkgrey  249  6.8e-03  1.0e+00       5  BP regulation of alternative mRNA splicing,
## 1637      darkgrey  249  6.9e-03  1.0e+00      14  CC                          Golgi apparatus
## 1638      darkgrey  249  6.9e-03  1.0e+00       4  BP positive regulation of canonical Wnt sig
## 1639      darkgrey  249  7.4e-03  1.0e+00       7  BP            negative regulation of growth
## 1640      darkgrey  249  7.7e-03  1.0e+00       3  BP        proteoglycan biosynthetic process
## 1641      darkgrey  249  7.7e-03  1.0e+00       3  BP  positive regulation of circadian rhythm
## 1642      darkgrey  249  7.9e-03  1.0e+00      15  CC                      cytoplasmic vesicle
## 1643      darkgrey  249  8.3e-03  1.0e+00      14  BP                  appendage morphogenesis
## 1644      darkgrey  249  8.4e-03  1.0e+00       2  BP               plasma membrane tubulation
## 1645      darkgrey  249  8.4e-03  1.0e+00       2  BP clathrin-dependent synaptic vesicle endo
## 1646      darkgrey  249  8.4e-03  1.0e+00       2  MF        GTPase activating protein binding
## 1647      darkgrey  249  8.8e-03  1.0e+00       3  BP     inositol phosphate metabolic process
## 1648      darkgrey  249  8.8e-03  1.0e+00       3  BP           phospholipid dephosphorylation
## 1649      darkgrey  249  9.0e-03  1.0e+00       7  BP               developmental pigmentation
## 1650      darkgrey  249  9.1e-03  1.0e+00      17  BP                                   growth
## 1651      darkgrey  249  9.7e-03  1.0e+00       5  BP negative regulation of protein phosphory
## 1652      darkgrey  249  9.7e-03  1.0e+00       5  CC                           coated vesicle
## 1653      darkgrey  249  9.9e-03  1.0e+00      25  BP      regulation of developmental process
## 1654      darkgrey  249  9.9e-03  1.0e+00       4  BP regulation of smoothened signaling pathw
## 1655       darkred   60  2.0e-43  1.7e-39      28  CC                       cytosolic ribosome
## 1656       darkred   60  4.4e-39  3.7e-35      27  BP                  cytoplasmic translation
## 1657       darkred   60  4.4e-35  3.7e-31      28  CC                                 ribosome
## 1658       darkred   60  1.3e-34  1.1e-30      27  MF       structural constituent of ribosome
## 1659       darkred   60  1.7e-28  1.5e-24      30  BP                              translation
## 1660       darkred   60  8.4e-27  7.1e-23      17  CC        cytosolic large ribosomal subunit
## 1661       darkred   60  6.0e-25  5.0e-21      30  BP                peptide metabolic process
## 1662       darkred   60  7.0e-25  5.9e-21      31  CC                                  cytosol
## 1663       darkred   60  2.5e-24  2.1e-20      28  MF             structural molecule activity
## 1664       darkred   60  2.2e-23  1.9e-19      29  CC                ribonucleoprotein complex
## 1665       darkred   60  1.4e-21  1.2e-17      17  CC                  large ribosomal subunit
## 1666       darkred   60  1.7e-16  1.4e-12      11  CC        cytosolic small ribosomal subunit
## 1667       darkred   60  2.5e-15  2.1e-11      41  CC               protein-containing complex
## 1668       darkred   60  1.0e-13  8.5e-10      46  CC                                cytoplasm
## 1669       darkred   60  1.3e-13  1.1e-09      11  CC                  small ribosomal subunit
## 1670       darkred   60  1.5e-11  1.3e-07      13  BP                      ribosome biogenesis
## 1671       darkred   60  4.4e-10  3.7e-06       8  BP       ribosomal large subunit biogenesis
## 1672       darkred   60  4.5e-10  3.8e-06      33  BP                protein metabolic process
## 1673       darkred   60  1.4e-09  1.2e-05      32  BP                          gene expression
## 1674       darkred   60  1.9e-09  1.6e-05      33  BP                     biosynthetic process
## 1675       darkred   60  5.4e-08  4.5e-04      44  BP               cellular metabolic process
## 1676       darkred   60  1.7e-07  1.4e-03      43  CC                  intracellular organelle
## 1677       darkred   60  4.9e-07  4.1e-03       9  CC             mitochondrial inner membrane
## 1678       darkred   60  5.9e-07  4.9e-03      47  CC                            intracellular
## 1679       darkred   60  1.5e-06  1.2e-02      41  BP      nitrogen compound metabolic process
## 1680       darkred   60  2.8e-06  2.3e-02       9  CC                   mitochondrial membrane
## 1681       darkred   60  2.9e-06  2.4e-02       7  BP                    ATP metabolic process
## 1682       darkred   60  4.5e-06  3.8e-02       9  CC                   mitochondrial envelope
## 1683       darkred   60  1.8e-05  1.5e-01       4  BP mitochondrial electron transport, NADH t
## 1684       darkred   60  2.1e-05  1.8e-01       5  BP mitochondrial ATP synthesis coupled elec
## 1685       darkred   60  2.3e-05  1.9e-01       5  BP ATP synthesis coupled electron transport
## 1686       darkred   60  2.9e-05  2.4e-01       5  CC          mitochondrial respiratory chain
## 1687       darkred   60  3.0e-05  2.5e-01      12  MF                              RNA binding
## 1688       darkred   60  3.1e-05  2.6e-01       7  BP generation of precursor metabolites and 
## 1689       darkred   60  3.5e-05  3.0e-01       5  BP     respiratory electron transport chain
## 1690       darkred   60  3.9e-05  3.3e-01       6  BP                          rRNA processing
## 1691       darkred   60  4.3e-05  3.6e-01       5  BP                 electron transport chain
## 1692       darkred   60  5.1e-05  4.3e-01       4  CC mitochondrial respiratory chain complex 
## 1693       darkred   60  5.1e-05  4.3e-01       4  CC               NADH dehydrogenase complex
## 1694       darkred   60  5.6e-05  4.7e-01       5  BP                oxidative phosphorylation
## 1695       darkred   60  8.6e-05  7.2e-01      11  CC                       organelle membrane
## 1696       darkred   60  1.3e-04  1.0e+00       4  BP                        ribosome assembly
## 1697       darkred   60  1.3e-04  1.0e+00       7  BP      purine nucleotide metabolic process
## 1698       darkred   60  1.3e-04  1.0e+00       7  BP         ribonucleotide metabolic process
## 1699       darkred   60  1.3e-04  1.0e+00       3  BP                   maturation of LSU-rRNA
## 1700       darkred   60  1.6e-04  1.0e+00       7  BP       ribose phosphate metabolic process
## 1701       darkred   60  1.9e-04  1.0e+00       5  BP                     cellular respiration
## 1702       darkred   60  2.0e-04  1.0e+00       3  BP         ribosomal large subunit assembly
## 1703       darkred   60  2.1e-04  1.0e+00      48  BP                         cellular process
## 1704       darkred   60  2.5e-04  1.0e+00       4  BP       ribosomal small subunit biogenesis
## 1705       darkred   60  5.5e-04  1.0e+00       7  BP             nucleotide metabolic process
## 1706       darkred   60  5.8e-04  1.0e+00       3  MF              NADH dehydrogenase activity
## 1707       darkred   60  5.8e-04  1.0e+00       7  BP   nucleoside phosphate metabolic process
## 1708       darkred   60  5.9e-04  1.0e+00      10  BP                           RNA processing
## 1709       darkred   60  2.2e-03  1.0e+00       3  MF oxidoreductase activity, acting on NAD(P
## 1710       darkred   60  2.4e-03  1.0e+00      10  CC                            mitochondrion
## 1711       darkred   60  2.8e-03  1.0e+00       2  MF      intramolecular transferase activity
## 1712       darkred   60  3.6e-03  1.0e+00       2  CC mitochondrial proton-transporting ATP sy
## 1713       darkred   60  5.0e-03  1.0e+00       2  MF                         ribosome binding
## 1714       darkred   60  5.2e-03  1.0e+00       4  BP       ribonucleoprotein complex assembly
## 1715       darkred   60  5.3e-03  1.0e+00       1  BP protein insertion into mitochondrial inn
## 1716       darkred   60  5.3e-03  1.0e+00       1  BP positive regulation by host of viral gen
## 1717       darkred   60  5.3e-03  1.0e+00       1  BP positive regulation of glycogen biosynth
## 1718       darkred   60  5.3e-03  1.0e+00       1  BP IRES-dependent viral translational initi
## 1719       darkred   60  5.3e-03  1.0e+00       1  CC mitochondrial processing peptidase compl
## 1720       darkred   60  5.3e-03  1.0e+00       1  CC           eRF1 methyltransferase complex
## 1721       darkred   60  5.3e-03  1.0e+00       1  MF         phosphoglycerate mutase activity
## 1722       darkred   60  5.3e-03  1.0e+00       1  MF                      ribokinase activity
## 1723       darkred   60  5.3e-03  1.0e+00       1  MF                UFM1 transferase activity
## 1724       darkred   60  6.7e-03  1.0e+00      54  MF                       molecular_function
## 1725       darkred   60  9.2e-03  1.0e+00       8  BP                          phosphorylation
## 1726 darkturquoise   65  1.0e-05  8.6e-02       3  MF                       proteasome binding
## 1727 darkturquoise   65  8.1e-05  6.8e-01       3  BP                   L-amino acid transport
## 1728 darkturquoise   65  2.3e-04  1.0e+00       3  MF L-amino acid transmembrane transporter a
## 1729 darkturquoise   65  2.1e-03  1.0e+00       2  MF polypeptide N-acetylgalactosaminyltransf
## 1730 darkturquoise   65  2.7e-03  1.0e+00       3  BP                     amino acid transport
## 1731 darkturquoise   65  3.2e-03  1.0e+00       3  MF amino acid transmembrane transporter act
## 1732 darkturquoise   65  4.3e-03  1.0e+00       2  BP                      proteasome assembly
## 1733 darkturquoise   65  5.8e-03  1.0e+00       1  BP   peptidyl-pyrromethane cofactor linkage
## 1734 darkturquoise   65  5.8e-03  1.0e+00       1  BP negative regulation of NF-kappaB transcr
## 1735 darkturquoise   65  5.8e-03  1.0e+00       1  BP                    tRNA methylthiolation
## 1736 darkturquoise   65  5.8e-03  1.0e+00       1  BP negative regulation of dopamine metaboli
## 1737 darkturquoise   65  5.8e-03  1.0e+00       1  BP                       L-glutamate import
## 1738 darkturquoise   65  5.8e-03  1.0e+00       1  BP         proteasome core complex assembly
## 1739 darkturquoise   65  5.8e-03  1.0e+00       1  MF   dimethylallyltranstransferase activity
## 1740 darkturquoise   65  5.8e-03  1.0e+00       1  MF         geranyltranstransferase activity
## 1741 darkturquoise   65  5.8e-03  1.0e+00       1  MF    hydroxymethylbilane synthase activity
## 1742 darkturquoise   65  5.8e-03  1.0e+00       1  MF          pyroglutamyl-peptidase activity
## 1743 darkturquoise   65  5.8e-03  1.0e+00       1  MF N6-threonylcarbomyladenosine methylthiot
## 1744 darkturquoise   65  5.8e-03  1.0e+00       1  MF sphingolipid delta-4 desaturase activity
## 1745 darkturquoise   65  5.8e-03  1.0e+00       1  MF betaine-homocysteine S-methyltransferase
## 1746 darkturquoise   65  5.8e-03  1.0e+00       1  MF                        NF-kappaB binding
## 1747 darkturquoise   65  8.4e-03  1.0e+00       2  MF acetylgalactosaminyltransferase activity
## 1748   greenyellow  135  4.3e-60  3.6e-56      36  CC                       proteasome complex
## 1749   greenyellow  135  1.7e-32  1.4e-28      35  BP proteasome-mediated ubiquitin-dependent 
## 1750   greenyellow  135  1.1e-31  9.5e-28      35  BP    proteasomal protein catabolic process
## 1751   greenyellow  135  1.3e-28  1.1e-24      15  CC                  proteasome core complex
## 1752   greenyellow  135  1.8e-28  1.5e-24      36  BP ubiquitin-dependent protein catabolic pr
## 1753   greenyellow  135  3.2e-28  2.7e-24      36  BP modification-dependent protein catabolic
## 1754   greenyellow  135  1.1e-27  9.3e-24      15  MF    threonine-type endopeptidase activity
## 1755   greenyellow  135  9.2e-27  7.7e-23      36  BP proteolysis involved in cellular protein
## 1756   greenyellow  135  1.5e-26  1.2e-22      36  BP       cellular protein catabolic process
## 1757   greenyellow  135  9.0e-26  7.5e-22      14  BP proteasomal ubiquitin-independent protei
## 1758   greenyellow  135  1.7e-25  1.4e-21      16  CC           proteasome regulatory particle
## 1759   greenyellow  135  2.4e-25  2.0e-21      36  BP                protein catabolic process
## 1760   greenyellow  135  7.2e-20  6.0e-16      36  BP organonitrogen compound catabolic proces
## 1761   greenyellow  135  6.5e-19  5.4e-15      47  BP                        catabolic process
## 1762   greenyellow  135  3.7e-18  3.1e-14      43  BP                              proteolysis
## 1763   greenyellow  135  8.5e-18  7.1e-14      42  BP      organic substance catabolic process
## 1764   greenyellow  135  3.9e-16  3.3e-12       8  CC proteasome core complex, alpha-subunit c
## 1765   greenyellow  135  1.4e-14  1.2e-10      70  CC               protein-containing complex
## 1766   greenyellow  135  1.7e-14  1.4e-10       8  CC proteasome regulatory particle, lid subc
## 1767   greenyellow  135  6.7e-13  5.6e-09      63  BP                protein metabolic process
## 1768   greenyellow  135  1.1e-12  9.2e-09       8  CC proteasome regulatory particle, base sub
## 1769   greenyellow  135  7.0e-12  5.9e-08     104  CC                            intracellular
## 1770   greenyellow  135  1.1e-11  9.2e-08      81  CC                                cytoplasm
## 1771   greenyellow  135  2.0e-11  1.7e-07       6  CC proteasome core complex, beta-subunit co
## 1772   greenyellow  135  2.4e-10  2.0e-06       6  MF    proteasome-activating ATPase activity
## 1773   greenyellow  135  8.8e-09  7.4e-05      24  BP      protein-containing complex assembly
## 1774   greenyellow  135  9.7e-09  8.2e-05      23  MF                       peptidase activity
## 1775   greenyellow  135  1.4e-08  1.1e-04       5  BP positive regulation of RNA polymerase II
## 1776   greenyellow  135  3.2e-08  2.7e-04       6  BP                      proteasome assembly
## 1777   greenyellow  135  3.4e-08  2.9e-04      86  BP               cellular metabolic process
## 1778   greenyellow  135  6.0e-08  5.0e-04       5  BP regulation of RNA polymerase II transcri
## 1779   greenyellow  135  9.6e-08  8.0e-04      18  MF                   endopeptidase activity
## 1780   greenyellow  135  2.0e-07  1.7e-03     108  BP                         cellular process
## 1781   greenyellow  135  4.6e-07  3.8e-03       5  BP positive regulation of transcription ini
## 1782   greenyellow  135  6.7e-07  5.6e-03      81  BP      nitrogen compound metabolic process
## 1783   greenyellow  135  1.4e-06  1.2e-02       4  CC             cytosolic proteasome complex
## 1784   greenyellow  135  4.3e-06  3.6e-02       5  BP RNA polymerase II preinitiation complex 
## 1785   greenyellow  135  5.4e-06  4.5e-02       7  BP positive regulation of protein complex a
## 1786   greenyellow  135  1.1e-05  9.5e-02      79  CC                  intracellular organelle
## 1787   greenyellow  135  1.6e-05  1.4e-01       5  BP transcription preinitiation complex asse
## 1788   greenyellow  135  2.2e-05  1.9e-01      69  CC intracellular membrane-bounded organelle
## 1789   greenyellow  135  3.4e-05  2.9e-01       3  BP              protein geranylgeranylation
## 1790   greenyellow  135  6.0e-05  5.0e-01       3  BP                      protein prenylation
## 1791   greenyellow  135  8.1e-05  6.8e-01      19  CC                                  cytosol
## 1792   greenyellow  135  1.2e-04  9.9e-01      36  MF                       hydrolase activity
## 1793   greenyellow  135  1.3e-04  1.0e+00       4  BP                   lamellipodium assembly
## 1794   greenyellow  135  1.5e-04  1.0e+00       2  CC CAAX-protein geranylgeranyltransferase c
## 1795   greenyellow  135  1.5e-04  1.0e+00       2  MF CAAX-protein geranylgeranyltransferase a
## 1796   greenyellow  135  2.0e-04  1.0e+00       3  BP               cellular response to virus
## 1797   greenyellow  135  2.0e-04  1.0e+00       3  CC              ER membrane protein complex
## 1798   greenyellow  135  2.7e-04  1.0e+00       3  BP protein folding in endoplasmic reticulum
## 1799   greenyellow  135  4.4e-04  1.0e+00       2  BP regulation of dopamine metabolic process
## 1800   greenyellow  135  5.1e-04  1.0e+00       6  BP             protein-DNA complex assembly
## 1801   greenyellow  135  5.8e-04  1.0e+00       3  MF                TBP-class protein binding
## 1802   greenyellow  135  8.7e-04  1.0e+00       2  BP                       peroxisome fission
## 1803   greenyellow  135  8.7e-04  1.0e+00       2  BP molybdopterin cofactor biosynthetic proc
## 1804   greenyellow  135  8.7e-04  1.0e+00       2  BP regulation of catecholamine metabolic pr
## 1805   greenyellow  135  9.9e-04  1.0e+00       5  BP  regulation of protein catabolic process
## 1806   greenyellow  135  1.1e-03  1.0e+00       3  BP     regulation of lamellipodium assembly
## 1807   greenyellow  135  1.1e-03  1.0e+00       7  BP       ribonucleoprotein complex assembly
## 1808   greenyellow  135  1.2e-03  1.0e+00       6  BP               neurotransmitter secretion
## 1809   greenyellow  135  1.8e-03  1.0e+00       5  CC integral component of endoplasmic reticu
## 1810   greenyellow  135  1.9e-03  1.0e+00       5  BP transcription initiation from RNA polyme
## 1811   greenyellow  135  2.1e-03  1.0e+00      15  CC                ribonucleoprotein complex
## 1812   greenyellow  135  2.1e-03  1.0e+00       8  CC           endoplasmic reticulum membrane
## 1813   greenyellow  135  2.1e-03  1.0e+00       2  BP Mo-molybdopterin cofactor biosynthetic p
## 1814   greenyellow  135  2.1e-03  1.0e+00       2  BP molybdopterin cofactor metabolic process
## 1815   greenyellow  135  2.2e-03  1.0e+00       5  CC intrinsic component of endoplasmic retic
## 1816   greenyellow  135  2.6e-03  1.0e+00       3  BP positive regulation of proteasomal ubiqu
## 1817   greenyellow  135  2.9e-03  1.0e+00       4  BP                        ribosome assembly
## 1818   greenyellow  135  3.0e-03  1.0e+00       2  MF vitamin transmembrane transporter activi
## 1819   greenyellow  135  3.0e-03  1.0e+00       5  BP                           macroautophagy
## 1820   greenyellow  135  3.0e-03  1.0e+00       5  CC                   mitochondrial ribosome
## 1821   greenyellow  135  3.3e-03  1.0e+00       3  BP       endoplasmic reticulum organization
## 1822   greenyellow  135  3.3e-03  1.0e+00       3  BP positive regulation of proteasomal prote
## 1823   greenyellow  135  3.4e-03  1.0e+00       5  BP  DNA-templated transcription, initiation
## 1824   greenyellow  135  3.6e-03  1.0e+00      59  MF                       catalytic activity
## 1825   greenyellow  135  3.9e-03  1.0e+00       2  CC                   Arp2/3 protein complex
## 1826   greenyellow  135  4.2e-03  1.0e+00       4  BP                     vacuole organization
## 1827   greenyellow  135  4.2e-03  1.0e+00       3  BP positive regulation of ubiquitin-depende
## 1828   greenyellow  135  4.3e-03  1.0e+00       7  MF       structural constituent of ribosome
## 1829   greenyellow  135  4.3e-03  1.0e+00      12  CC                    endoplasmic reticulum
## 1830   greenyellow  135  4.9e-03  1.0e+00      10  MF                          ATPase activity
## 1831   greenyellow  135  5.0e-03  1.0e+00       2  BP negative regulation of phosphatase activ
## 1832   greenyellow  135  5.0e-03  1.0e+00       2  BP          actin filament depolymerization
## 1833   greenyellow  135  5.0e-03  1.0e+00       2  BP                        vitamin transport
## 1834   greenyellow  135  5.0e-03  1.0e+00       2  CC                            lamellipodium
## 1835   greenyellow  135  5.0e-03  1.0e+00       2  MF polyubiquitin modification-dependent pro
## 1836   greenyellow  135  5.1e-03  1.0e+00       3  BP regulation of proteasomal ubiquitin-depe
## 1837   greenyellow  135  5.1e-03  1.0e+00       3  BP positive regulation of proteolysis invol
## 1838   greenyellow  135  5.7e-03  1.0e+00       6  BP                           signal release
## 1839   greenyellow  135  5.7e-03  1.0e+00       4  BP            actin filament polymerization
## 1840   greenyellow  135  5.9e-03  1.0e+00       6  BP               neurotransmitter transport
## 1841   greenyellow  135  6.1e-03  1.0e+00       4  BP       positive regulation of proteolysis
## 1842   greenyellow  135  6.2e-03  1.0e+00       2  CC integral component of peroxisomal membra
## 1843   greenyellow  135  6.2e-03  1.0e+00       2  MF               sulfurtransferase activity
## 1844   greenyellow  135  6.4e-03  1.0e+00       4  MF transferase activity, transferring alkyl
## 1845   greenyellow  135  6.8e-03  1.0e+00       3  BP                   autophagosome assembly
## 1846   greenyellow  135  6.8e-03  1.0e+00       3  BP positive regulation of cellular protein 
## 1847   greenyellow  135  6.8e-03  1.0e+00       3  CC    mitochondrial small ribosomal subunit
## 1848   greenyellow  135  6.8e-03  1.0e+00      16  CC                            mitochondrion
## 1849   greenyellow  135  7.2e-03  1.0e+00       9  BP           chemical synaptic transmission
## 1850   greenyellow  135  7.4e-03  1.0e+00       3  BP regulation of proteasomal protein catabo
## 1851   greenyellow  135  7.5e-03  1.0e+00      12  MF             structural molecule activity
## 1852   greenyellow  135  7.5e-03  1.0e+00       9  BP                       synaptic signaling
## 1853   greenyellow  135  7.5e-03  1.0e+00       2  MF               prenyltransferase activity
## 1854   greenyellow  135  7.6e-03  1.0e+00       7  CC                                 ribosome
## 1855   greenyellow  135  8.9e-03  1.0e+00       4  BP actin polymerization or depolymerization
## 1856   greenyellow  135  9.5e-03  1.0e+00       5  BP                mitochondrial translation
## 1857          grey 1049  7.2e-18  6.0e-14     508  CC                  intracellular organelle
## 1858          grey 1049  3.4e-17  2.8e-13     593  CC                            intracellular
## 1859          grey 1049  8.2e-13  6.9e-09     423  CC intracellular membrane-bounded organelle
## 1860          grey 1049  9.8e-12  8.2e-08     401  CC                                cytoplasm
## 1861          grey 1049  1.2e-09  1.0e-05      87  CC                       organelle membrane
## 1862          grey 1049  3.5e-09  2.9e-05      64  BP regulation of nervous system development
## 1863          grey 1049  7.4e-09  6.2e-05     212  BP regulation of cellular metabolic process
## 1864          grey 1049  8.0e-09  6.7e-05      38  BP small GTPase mediated signal transductio
## 1865          grey 1049  1.5e-08  1.3e-04      79  BP               vesicle-mediated transport
## 1866          grey 1049  1.6e-08  1.3e-04     299  BP          cellular component organization
## 1867          grey 1049  3.1e-08  2.6e-04      40  CC             mitochondrial inner membrane
## 1868          grey 1049  3.4e-08  2.9e-04      49  BP               regulation of neurogenesis
## 1869          grey 1049  4.4e-08  3.7e-04      28  BP                           dorsal closure
## 1870          grey 1049  5.6e-08  4.7e-04     650  BP                         cellular process
## 1871          grey 1049  9.6e-08  8.1e-04     204  BP                   organelle organization
## 1872          grey 1049  1.0e-07  8.6e-04      30  BP    morphogenesis of embryonic epithelium
## 1873          grey 1049  1.2e-07  9.7e-04     465  BP               cellular metabolic process
## 1874          grey 1049  2.3e-07  2.0e-03     139  BP               nervous system development
## 1875          grey 1049  2.6e-07  2.2e-03      67  BP       regulation of cell differentiation
## 1876          grey 1049  2.6e-07  2.2e-03      36  BP embryonic development via the syncytial 
## 1877          grey 1049  3.1e-07  2.6e-03      39  BP embryo development ending in birth or eg
## 1878          grey 1049  3.9e-07  3.3e-03      32  BP          Ras protein signal transduction
## 1879          grey 1049  4.1e-07  3.4e-03      83  BP                          phosphorylation
## 1880          grey 1049  4.8e-07  4.0e-03     221  BP          regulation of metabolic process
## 1881          grey 1049  5.7e-07  4.8e-03      43  CC                   mitochondrial membrane
## 1882          grey 1049  5.8e-07  4.8e-03      20  CC          mitochondrial respiratory chain
## 1883          grey 1049  6.0e-07  5.1e-03      28  BP                    ATP metabolic process
## 1884          grey 1049  7.2e-07  6.0e-03      91  BP                  intracellular transport
## 1885          grey 1049  7.6e-07  6.4e-03      52  BP                           cell migration
## 1886          grey 1049  8.0e-07  6.7e-03     124  CC                      endomembrane system
## 1887          grey 1049  8.0e-07  6.7e-03      39  CC                                 endosome
## 1888          grey 1049  9.1e-07  7.6e-03      35  BP generation of precursor metabolites and 
## 1889          grey 1049  1.0e-06  8.8e-03      35  BP     regulation of neuron differentiation
## 1890          grey 1049  1.6e-06  1.3e-02      63  CC                                  vesicle
## 1891          grey 1049  1.7e-06  1.4e-02      58  CC                      cytoplasmic vesicle
## 1892          grey 1049  1.8e-06  1.5e-02     122  BP phosphate-containing compound metabolic 
## 1893          grey 1049  2.0e-06  1.7e-02     142  BP    cellular protein modification process
## 1894          grey 1049  2.1e-06  1.7e-02      14  CC mitochondrial respiratory chain complex 
## 1895          grey 1049  2.1e-06  1.7e-02      14  CC               NADH dehydrogenase complex
## 1896          grey 1049  2.5e-06  2.1e-02      20  BP                 electron transport chain
## 1897          grey 1049  2.6e-06  2.2e-02     268  CC               protein-containing complex
## 1898          grey 1049  3.0e-06  2.5e-02      43  CC                   mitochondrial envelope
## 1899          grey 1049  3.2e-06  2.6e-02      20  BP      negative regulation of neurogenesis
## 1900          grey 1049  3.8e-06  3.2e-02      91  BP                     protein localization
## 1901          grey 1049  4.3e-06  3.6e-02      18  BP mitochondrial ATP synthesis coupled elec
## 1902          grey 1049  4.5e-06  3.8e-02      66  BP               regulation of localization
## 1903          grey 1049  4.9e-06  4.1e-02      67  BP    establishment of protein localization
## 1904          grey 1049  5.2e-06  4.4e-02      19  BP     respiratory electron transport chain
## 1905          grey 1049  5.5e-06  4.6e-02      18  BP ATP synthesis coupled electron transport
## 1906          grey 1049  6.1e-06  5.1e-02      20  BP                oxidative phosphorylation
## 1907          grey 1049  6.3e-06  5.3e-02      90  CC                            mitochondrion
## 1908          grey 1049  8.7e-06  7.3e-02     108  BP                    generation of neurons
## 1909          grey 1049  9.3e-06  7.8e-02     112  BP                             neurogenesis
## 1910          grey 1049  9.4e-06  7.9e-02     248  BP       multicellular organism development
## 1911          grey 1049  1.1e-05  8.9e-02      26  BP           border follicle cell migration
## 1912          grey 1049  1.1e-05  9.1e-02      23  BP                     cellular respiration
## 1913          grey 1049  1.2e-05  1.0e-01     138  BP      regulation of RNA metabolic process
## 1914          grey 1049  1.6e-05  1.3e-01     129  BP regulation of transcription, DNA-templat
## 1915          grey 1049  1.6e-05  1.3e-01      31  BP                                autophagy
## 1916          grey 1049  1.6e-05  1.3e-01      27  BP          ovarian follicle cell migration
## 1917          grey 1049  1.7e-05  1.5e-01      21  BP            regulation of GTPase activity
## 1918          grey 1049  1.8e-05  1.5e-01     238  MF                          protein binding
## 1919          grey 1049  1.8e-05  1.5e-01       7  BP       late endosome to vacuole transport
## 1920          grey 1049  2.0e-05  1.7e-01      52  BP                            cell motility
## 1921          grey 1049  2.3e-05  1.9e-01      63  BP                        protein transport
## 1922          grey 1049  2.4e-05  2.0e-01      20  BP                       Golgi organization
## 1923          grey 1049  2.6e-05  2.2e-01       6  BP dorsal closure, spreading of leading edg
## 1924          grey 1049  2.8e-05  2.4e-01      49  BP         regulation of catalytic activity
## 1925          grey 1049  3.1e-05  2.6e-01      53  BP          intracellular protein transport
## 1926          grey 1049  3.2e-05  2.7e-01      71  BP            cellular protein localization
## 1927          grey 1049  3.5e-05  2.9e-01      20  BP regulation of small GTPase mediated sign
## 1928          grey 1049  4.0e-05  3.3e-01      13  BP regulation of epidermal growth factor re
## 1929          grey 1049  4.4e-05  3.7e-01      94  BP      regulation of developmental process
## 1930          grey 1049  4.4e-05  3.7e-01      28  BP  regulation of cytoskeleton organization
## 1931          grey 1049  4.8e-05  4.0e-01      27  BP         regulation of cell morphogenesis
## 1932          grey 1049  4.9e-05  4.1e-01      20  CC                        adherens junction
## 1933          grey 1049  5.3e-05  4.5e-01      34  CC                      polytene chromosome
## 1934          grey 1049  5.4e-05  4.5e-01       4  BP protein transport to vacuole involved in
## 1935          grey 1049  5.5e-05  4.6e-01      27  BP regulation of cell projection organizati
## 1936          grey 1049  5.6e-05  4.7e-01      81  BP                       neuron development
## 1937          grey 1049  5.8e-05  4.8e-01      39  BP                  embryonic morphogenesis
## 1938          grey 1049  5.8e-05  4.9e-01      16  CC                           coated vesicle
## 1939          grey 1049  5.9e-05  5.0e-01      26  BP negative regulation of cell differentiat
## 1940          grey 1049  6.0e-05  5.0e-01      33  BP            ameboidal-type cell migration
## 1941          grey 1049  6.3e-05  5.3e-01      36  BP establishment or maintenance of cell pol
## 1942          grey 1049  6.4e-05  5.4e-01      92  BP        regulation of signal transduction
## 1943          grey 1049  7.6e-05  6.4e-01     130  BP             transcription, DNA-templated
## 1944          grey 1049  7.8e-05  6.6e-01      11  BP        actin cytoskeleton reorganization
## 1945          grey 1049  8.1e-05  6.8e-01      20  BP                  regulation of autophagy
## 1946          grey 1049  8.4e-05  7.1e-01     156  BP                         cell development
## 1947          grey 1049  8.6e-05  7.2e-01      82  BP                               locomotion
## 1948          grey 1049  8.9e-05  7.4e-01     270  BP         anatomical structure development
## 1949          grey 1049  9.0e-05  7.6e-01      10  BP negative regulation of epidermal growth 
## 1950          grey 1049  9.2e-05  7.7e-01      21  BP                      endosomal transport
## 1951          grey 1049  9.3e-05  7.8e-01      93  BP                   neuron differentiation
## 1952          grey 1049  1.1e-04  8.8e-01      75  BP                       cell morphogenesis
## 1953          grey 1049  1.2e-04  9.7e-01      81  BP regulation of transcription by RNA polym
## 1954          grey 1049  1.2e-04  1.0e+00       6  BP ubiquitin-dependent protein catabolic pr
## 1955          grey 1049  1.2e-04  1.0e+00      10  BP mitochondrial electron transport, NADH t
## 1956          grey 1049  1.3e-04  1.0e+00      32  BP                              endocytosis
## 1957          grey 1049  1.3e-04  1.0e+00     137  BP                 animal organ development
## 1958          grey 1049  1.3e-04  1.0e+00     279  BP                    developmental process
## 1959          grey 1049  1.3e-04  1.0e+00      31  BP transmembrane receptor protein tyrosine 
## 1960          grey 1049  1.3e-04  1.0e+00      14  CC                 Golgi-associated vesicle
## 1961          grey 1049  1.3e-04  1.0e+00      77  BP                cytoskeleton organization
## 1962          grey 1049  1.4e-04  1.0e+00      48  BP negative regulation of signal transducti
## 1963          grey 1049  1.4e-04  1.0e+00     158  BP            regulation of gene expression
## 1964          grey 1049  1.5e-04  1.0e+00      15  BP epidermal growth factor receptor signali
## 1965          grey 1049  1.5e-04  1.0e+00      60  BP                                   growth
## 1966          grey 1049  1.5e-04  1.0e+00      75  BP      cellular macromolecule localization
## 1967          grey 1049  1.6e-04  1.0e+00      72  CC                               chromosome
## 1968          grey 1049  1.7e-04  1.0e+00      10  MF              NADH dehydrogenase activity
## 1969          grey 1049  1.8e-04  1.0e+00      13  MF oxidoreductase activity, acting on NAD(P
## 1970          grey 1049  1.8e-04  1.0e+00      67  BP            pattern specification process
## 1971          grey 1049  2.0e-04  1.0e+00      14  BP  regulation of microtubule-based process
## 1972          grey 1049  2.0e-04  1.0e+00      55  BP                          eye development
## 1973          grey 1049  2.3e-04  1.0e+00      48  BP                             axonogenesis
## 1974          grey 1049  2.3e-04  1.0e+00      10  BP negative regulation of neuron differenti
## 1975          grey 1049  2.3e-04  1.0e+00     122  BP      multicellular organism reproduction
## 1976          grey 1049  2.4e-04  1.0e+00      33  BP                    membrane organization
## 1977          grey 1049  2.5e-04  1.0e+00     179  BP                     cell differentiation
## 1978          grey 1049  2.5e-04  1.0e+00      41  CC                                chromatin
## 1979          grey 1049  2.6e-04  1.0e+00      45  BP                     regulation of growth
## 1980          grey 1049  2.6e-04  1.0e+00      22  BP                  stem cell proliferation
## 1981          grey 1049  2.7e-04  1.0e+00       9  BP protein localization to endoplasmic reti
## 1982          grey 1049  2.7e-04  1.0e+00      52  BP                 compound eye development
## 1983          grey 1049  2.9e-04  1.0e+00      37  BP                  regulation of transport
## 1984          grey 1049  3.0e-04  1.0e+00       7  MF oxidoreductase activity, acting on NAD(P
## 1985          grey 1049  3.0e-04  1.0e+00      49  BP                         axon development
## 1986          grey 1049  3.1e-04  1.0e+00      17  BP regulation of Ras protein signal transdu
## 1987          grey 1049  3.1e-04  1.0e+00      83  BP       transcription by RNA polymerase II
## 1988          grey 1049  3.2e-04  1.0e+00      56  CC                    endoplasmic reticulum
## 1989          grey 1049  3.5e-04  1.0e+00      28  CC                            cell junction
## 1990          grey 1049  3.5e-04  1.0e+00      90  BP               animal organ morphogenesis
## 1991          grey 1049  3.5e-04  1.0e+00      46  CC                          Golgi apparatus
## 1992          grey 1049  3.6e-04  1.0e+00      63  BP            neuron projection development
## 1993          grey 1049  3.6e-04  1.0e+00      40  BP                 regulation of cell death
## 1994          grey 1049  3.7e-04  1.0e+00      16  BP             regulation of cell migration
## 1995          grey 1049  3.9e-04  1.0e+00      11  BP establishment or maintenance of apical/b
## 1996          grey 1049  4.0e-04  1.0e+00      26  BP positive regulation of cell differentiat
## 1997          grey 1049  4.0e-04  1.0e+00      84  BP              organic substance transport
## 1998          grey 1049  4.2e-04  1.0e+00      28  BP                epithelial cell migration
## 1999          grey 1049  4.3e-04  1.0e+00      59  BP cell morphogenesis involved in neuron di
## 2000          grey 1049  4.3e-04  1.0e+00      14  BP               regulation of axonogenesis
## 2001          grey 1049  4.4e-04  1.0e+00      58  CC                              nucleoplasm
## 2002          grey 1049  4.7e-04  1.0e+00      28  BP                     epithelium migration
## 2003          grey 1049  4.8e-04  1.0e+00      63  BP                sensory organ development
## 2004          grey 1049  5.1e-04  1.0e+00     238  CC                                 membrane
## 2005          grey 1049  5.6e-04  1.0e+00     424  BP      nitrogen compound metabolic process
## 2006          grey 1049  5.8e-04  1.0e+00      42  BP             actin filament-based process
## 2007          grey 1049  6.0e-04  1.0e+00       8  BP endosome transport via multivesicular bo
## 2008          grey 1049  6.2e-04  1.0e+00      14  BP                 regulation of cell shape
## 2009          grey 1049  6.2e-04  1.0e+00      11  CC                       cell division site
## 2010          grey 1049  6.3e-04  1.0e+00       3  CC                         ESCRT II complex
## 2011          grey 1049  6.3e-04  1.0e+00      63  BP cell morphogenesis involved in different
## 2012          grey 1049  6.3e-04  1.0e+00      30  CC           endoplasmic reticulum membrane
## 2013          grey 1049  6.5e-04  1.0e+00      33  BP      purine nucleotide metabolic process
## 2014          grey 1049  6.6e-04  1.0e+00       9  BP establishment or maintenance of epitheli
## 2015          grey 1049  6.6e-04  1.0e+00       9  CC       transcriptionally active chromatin
## 2016          grey 1049  6.9e-04  1.0e+00      12  BP                       vacuolar transport
## 2017          grey 1049  6.9e-04  1.0e+00      12  BP regulation of microtubule cytoskeleton o
## 2018          grey 1049  6.9e-04  1.0e+00     780  CC                       cellular_component
## 2019          grey 1049  7.0e-04  1.0e+00       4  BP late endosome to vacuole transport via m
## 2020          grey 1049  7.1e-04  1.0e+00      40  BP          actin cytoskeleton organization
## 2021          grey 1049  7.2e-04  1.0e+00      64  BP   positive regulation of gene expression
## 2022          grey 1049  7.6e-04  1.0e+00     162  BP       anatomical structure morphogenesis
## 2023          grey 1049  7.7e-04  1.0e+00      11  BP             fatty acid catabolic process
## 2024          grey 1049  8.1e-04  1.0e+00      18  BP                       lipid modification
## 2025          grey 1049  8.1e-04  1.0e+00      26  BP                       learning or memory
## 2026          grey 1049  8.2e-04  1.0e+00       8  BP       endoplasmic reticulum organization
## 2027          grey 1049  8.3e-04  1.0e+00      25  BP                      response to hormone
## 2028          grey 1049  8.4e-04  1.0e+00     131  BP                             reproduction
## 2029          grey 1049  8.9e-04  1.0e+00      57  MF                           enzyme binding
## 2030          grey 1049  9.1e-04  1.0e+00      54  BP positive regulation of transcription, DN
## 2031          grey 1049  9.3e-04  1.0e+00      16  BP                      histone acetylation
## 2032          grey 1049  9.8e-04  1.0e+00      72  CC                                  cytosol
## 2033          grey 1049  9.9e-04  1.0e+00       6  BP negative regulation of photoreceptor cel
## 2034          grey 1049  9.9e-04  1.0e+00       6  MF NADH dehydrogenase (ubiquinone) activity
## 2035          grey 1049  1.0e-03  1.0e+00      13  CC                        endosome membrane
## 2036          grey 1049  1.1e-03  1.0e+00      16  BP              peptidyl-lysine acetylation
## 2037          grey 1049  1.1e-03  1.0e+00      58  BP          neuron projection morphogenesis
## 2038          grey 1049  1.1e-03  1.0e+00      72  BP           morphogenesis of an epithelium
## 2039          grey 1049  1.1e-03  1.0e+00       9  BP                fatty acid beta-oxidation
## 2040          grey 1049  1.1e-03  1.0e+00      77  BP        intracellular signal transduction
## 2041          grey 1049  1.1e-03  1.0e+00      27  MF                       Ras GTPase binding
## 2042          grey 1049  1.2e-03  1.0e+00     100  BP                        gamete generation
## 2043          grey 1049  1.2e-03  1.0e+00      19  CC                               centrosome
## 2044          grey 1049  1.2e-03  1.0e+00      33  BP       ribose phosphate metabolic process
## 2045          grey 1049  1.2e-03  1.0e+00      53  BP                  protein phosphorylation
## 2046          grey 1049  1.3e-03  1.0e+00      37  BP positive regulation of cellular protein 
## 2047          grey 1049  1.3e-03  1.0e+00      10  BP           adherens junction organization
## 2048          grey 1049  1.3e-03  1.0e+00     109  BP                   epithelium development
## 2049          grey 1049  1.3e-03  1.0e+00      32  BP         ribonucleotide metabolic process
## 2050          grey 1049  1.3e-03  1.0e+00     125  BP                     reproductive process
## 2051          grey 1049  1.4e-03  1.0e+00      35  MF                  protein kinase activity
## 2052          grey 1049  1.4e-03  1.0e+00      13  MF Ras guanyl-nucleotide exchange factor ac
## 2053          grey 1049  1.4e-03  1.0e+00       8  CC COPII-coated ER to Golgi transport vesic
## 2054          grey 1049  1.5e-03  1.0e+00      27  MF                     small GTPase binding
## 2055          grey 1049  1.5e-03  1.0e+00      28  BP                     histone modification
## 2056          grey 1049  1.6e-03  1.0e+00      34  BP      regulation of programmed cell death
## 2057          grey 1049  1.6e-03  1.0e+00      48  BP                     developmental growth
## 2058          grey 1049  1.7e-03  1.0e+00     236  CC                                  nucleus
## 2059          grey 1049  1.7e-03  1.0e+00     177  BP                       cell communication
## 2060          grey 1049  1.7e-03  1.0e+00      20  BP                                   memory
## 2061          grey 1049  1.8e-03  1.0e+00      27  BP       neuromuscular junction development
## 2062          grey 1049  1.8e-03  1.0e+00      47  BP                       cell proliferation
## 2063          grey 1049  1.9e-03  1.0e+00      14  BP   ER to Golgi vesicle-mediated transport
## 2064          grey 1049  1.9e-03  1.0e+00      38  BP                            axon guidance
## 2065          grey 1049  2.0e-03  1.0e+00      10  BP                  axo-dendritic transport
## 2066          grey 1049  2.0e-03  1.0e+00      13  BP           establishment of cell polarity
## 2067          grey 1049  2.0e-03  1.0e+00      12  BP                regulation of endocytosis
## 2068          grey 1049  2.1e-03  1.0e+00      81  BP  cell surface receptor signaling pathway
## 2069          grey 1049  2.1e-03  1.0e+00      20  BP  morphogenesis of a polarized epithelium
## 2070          grey 1049  2.1e-03  1.0e+00       6  BP cell competition in a multicellular orga
## 2071          grey 1049  2.1e-03  1.0e+00       6  BP negative regulation of small GTPase medi
## 2072          grey 1049  2.1e-03  1.0e+00       6  CC mitochondrial proton-transporting ATP sy
## 2073          grey 1049  2.2e-03  1.0e+00       5  BP               lipid droplet organization
## 2074          grey 1049  2.2e-03  1.0e+00       5  CC                      multivesicular body
## 2075          grey 1049  2.2e-03  1.0e+00       9  CC                       recycling endosome
## 2076          grey 1049  2.4e-03  1.0e+00       3  BP       septum digestion after cytokinesis
## 2077          grey 1049  2.4e-03  1.0e+00       3  BP                  stress granule assembly
## 2078          grey 1049  2.4e-03  1.0e+00       3  CC                    CCR4-NOT core complex
## 2079          grey 1049  2.4e-03  1.0e+00       3  MF    acetyl-CoA C-acyltransferase activity
## 2080          grey 1049  2.4e-03  1.0e+00      16  BP                      protein acetylation
## 2081          grey 1049  2.4e-03  1.0e+00      10  BP          Rab protein signal transduction
## 2082          grey 1049  2.4e-03  1.0e+00      12  MF               electron transfer activity
## 2083          grey 1049  2.4e-03  1.0e+00      17  BP      positive regulation of neurogenesis
## 2084          grey 1049  2.4e-03  1.0e+00     222  BP                protein metabolic process
## 2085          grey 1049  2.5e-03  1.0e+00      71  BP                                oogenesis
## 2086          grey 1049  2.7e-03  1.0e+00      19  BP                 regulation of locomotion
## 2087          grey 1049  2.7e-03  1.0e+00      16  BP negative regulation of cell proliferatio
## 2088          grey 1049  2.7e-03  1.0e+00       9  BP                     fatty acid oxidation
## 2089          grey 1049  2.7e-03  1.0e+00      60  BP                       tube morphogenesis
## 2090          grey 1049  2.8e-03  1.0e+00      38  BP                  wing disc morphogenesis
## 2091          grey 1049  2.8e-03  1.0e+00      14  BP                         long-term memory
## 2092          grey 1049  2.8e-03  1.0e+00      14  BP                      response to insulin
## 2093          grey 1049  2.8e-03  1.0e+00       4  BP      semaphorin-plexin signaling pathway
## 2094          grey 1049  2.8e-03  1.0e+00       4  CC mitochondrial proton-transporting ATP sy
## 2095          grey 1049  2.8e-03  1.0e+00       4  MF          acyl-CoA dehydrogenase activity
## 2096          grey 1049  2.8e-03  1.0e+00      37  BP imaginal disc-derived wing morphogenesis
## 2097          grey 1049  2.9e-03  1.0e+00      28  MF                           GTPase binding
## 2098          grey 1049  3.0e-03  1.0e+00      15  MF                GTPase activator activity
## 2099          grey 1049  3.0e-03  1.0e+00       8  BP                    segment specification
## 2100          grey 1049  3.1e-03  1.0e+00      13  MF             protein transporter activity
## 2101          grey 1049  3.1e-03  1.0e+00      26  BP positive regulation of catalytic activit
## 2102          grey 1049  3.1e-03  1.0e+00       7  BP                 autophagosome maturation
## 2103          grey 1049  3.2e-03  1.0e+00      45  BP              epithelial cell development
## 2104          grey 1049  3.3e-03  1.0e+00       5  BP           peptidyl-lysine trimethylation
## 2105          grey 1049  3.3e-03  1.0e+00       9  BP                          lipid oxidation
## 2106          grey 1049  3.3e-03  1.0e+00      15  BP                           macroautophagy
## 2107          grey 1049  3.4e-03  1.0e+00      16  MF                           kinase binding
## 2108          grey 1049  3.4e-03  1.0e+00      10  BP regulation of photoreceptor cell differe
## 2109          grey 1049  3.5e-03  1.0e+00      49  BP              imaginal disc morphogenesis
## 2110          grey 1049  3.6e-03  1.0e+00      41  BP        ovarian follicle cell development
## 2111          grey 1049  3.6e-03  1.0e+00     171  BP                                signaling
## 2112          grey 1049  3.7e-03  1.0e+00      18  BP         establishment of planar polarity
## 2113          grey 1049  3.7e-03  1.0e+00      18  BP         establishment of tissue polarity
## 2114          grey 1049  3.8e-03  1.0e+00      17  BP                 neuroblast proliferation
## 2115          grey 1049  3.8e-03  1.0e+00      56  BP                            metamorphosis
## 2116          grey 1049  4.0e-03  1.0e+00      43  BP imaginal disc-derived appendage developm
## 2117          grey 1049  4.0e-03  1.0e+00       6  CC              cell-cell adherens junction
## 2118          grey 1049  4.0e-03  1.0e+00       9  BP negative regulation of stem cell prolife
## 2119          grey 1049  4.1e-03  1.0e+00      10  BP positive regulation of cell cycle proces
## 2120          grey 1049  4.1e-03  1.0e+00      47  BP                               cell death
## 2121          grey 1049  4.3e-03  1.0e+00      61  BP      protein-containing complex assembly
## 2122          grey 1049  4.4e-03  1.0e+00     149  BP                      signal transduction
## 2123          grey 1049  4.5e-03  1.0e+00      25  BP                         synapse assembly
## 2124          grey 1049  4.6e-03  1.0e+00      13  CC                            late endosome
## 2125          grey 1049  4.6e-03  1.0e+00      28  MF                        chromatin binding
## 2126          grey 1049  4.6e-03  1.0e+00       8  CC                                   P-body
## 2127          grey 1049  4.7e-03  1.0e+00      20  BP                      oocyte construction
## 2128          grey 1049  4.7e-03  1.0e+00      45  BP                                    taxis
## 2129          grey 1049  4.7e-03  1.0e+00      41  BP   post-embryonic appendage morphogenesis
## 2130          grey 1049  4.7e-03  1.0e+00      14  CC                   neuromuscular junction
## 2131          grey 1049  4.8e-03  1.0e+00       5  BP        regulation of histone acetylation
## 2132          grey 1049  4.8e-03  1.0e+00       5  BP                regulation of oviposition
## 2133          grey 1049  4.8e-03  1.0e+00       5  BP                  nonassociative learning
## 2134          grey 1049  4.8e-03  1.0e+00       5  MF                TBP-class protein binding
## 2135          grey 1049  4.8e-03  1.0e+00       4  BP    regulation of post-mating oviposition
## 2136          grey 1049  4.8e-03  1.0e+00       4  CC                           BLOC-1 complex
## 2137          grey 1049  4.8e-03  1.0e+00      24  BP                   oocyte differentiation
## 2138          grey 1049  5.0e-03  1.0e+00      12  BP cortical actin cytoskeleton organization
## 2139          grey 1049  5.0e-03  1.0e+00      18  CC                       cell-cell junction
## 2140          grey 1049  5.1e-03  1.0e+00      99  BP                        catabolic process
## 2141          grey 1049  5.1e-03  1.0e+00       7  BP         negative regulation of autophagy
## 2142          grey 1049  5.1e-03  1.0e+00       7  BP    regulation of cell fate specification
## 2143          grey 1049  5.1e-03  1.0e+00       7  BP                histone H3-K4 methylation
## 2144          grey 1049  5.1e-03  1.0e+00       7  CC                       cytochrome complex
## 2145          grey 1049  5.3e-03  1.0e+00     113  BP                       tissue development
## 2146          grey 1049  5.3e-03  1.0e+00      39  BP               compound eye morphogenesis
## 2147          grey 1049  5.3e-03  1.0e+00      29  BP          regulation of apoptotic process
## 2148          grey 1049  5.3e-03  1.0e+00      21  BP                  Golgi vesicle transport
## 2149          grey 1049  5.3e-03  1.0e+00       6  BP establishment of mitotic spindle orienta
## 2150          grey 1049  5.3e-03  1.0e+00      14  MF guanyl-nucleotide exchange factor activi
## 2151          grey 1049  5.4e-03  1.0e+00      15  CC                        transport vesicle
## 2152          grey 1049  5.4e-03  1.0e+00      36  BP             nucleotide metabolic process
## 2153          grey 1049  5.5e-03  1.0e+00       3  BP                    lysosome localization
## 2154          grey 1049  5.5e-03  1.0e+00       3  BP dorsal closure, leading edge cell differ
## 2155          grey 1049  5.5e-03  1.0e+00       3  BP             sensory neuron axon guidance
## 2156          grey 1049  5.6e-03  1.0e+00      24  CC                        nuclear chromatin
## 2157          grey 1049  5.7e-03  1.0e+00      29  BP        protein localization to organelle
## 2158          grey 1049  5.7e-03  1.0e+00       8  BP negative regulation of neuroblast prolif
## 2159          grey 1049  5.7e-03  1.0e+00       8  CC                          cleavage furrow
## 2160          grey 1049  5.7e-03  1.0e+00      25  BP regulation of synapse structure or activ
## 2161          grey 1049  5.8e-03  1.0e+00       9  BP                 ATP biosynthetic process
## 2162          grey 1049  5.9e-03  1.0e+00      39  MF                enzyme regulator activity
## 2163          grey 1049  5.9e-03  1.0e+00      13  BP    cellular response to insulin stimulus
## 2164          grey 1049  5.9e-03  1.0e+00      13  CC                  apical junction complex
## 2165          grey 1049  6.0e-03  1.0e+00      79  BP                         tube development
## 2166          grey 1049  6.1e-03  1.0e+00      24  CC                              cell cortex
## 2167          grey 1049  6.1e-03  1.0e+00     229  BP                     biosynthetic process
## 2168          grey 1049  6.1e-03  1.0e+00      25  MF protein serine/threonine kinase activity
## 2169          grey 1049  6.3e-03  1.0e+00      34  CC                       nuclear chromosome
## 2170          grey 1049  6.4e-03  1.0e+00      36  BP   nucleoside phosphate metabolic process
## 2171          grey 1049  6.5e-03  1.0e+00      78  BP                    germ cell development
## 2172          grey 1049  6.6e-03  1.0e+00      27  BP                       axis specification
## 2173          grey 1049  6.6e-03  1.0e+00      10  BP positive regulation of protein catabolic
## 2174          grey 1049  6.6e-03  1.0e+00       5  BP ecdysone-mediated induction of salivary 
## 2175          grey 1049  6.6e-03  1.0e+00       5  MF      intramolecular transferase activity
## 2176          grey 1049  6.7e-03  1.0e+00      14  MF                   protein kinase binding
## 2177          grey 1049  6.8e-03  1.0e+00      21  BP                       oocyte development
## 2178          grey 1049  6.9e-03  1.0e+00       9  BP positive regulation of mitotic cell cycl
## 2179          grey 1049  7.2e-03  1.0e+00      61  BP              oxidation-reduction process
## 2180          grey 1049  7.2e-03  1.0e+00      16  BP                  cell fate specification
## 2181          grey 1049  7.3e-03  1.0e+00       2  BP         regulation of glycolytic process
## 2182          grey 1049  7.3e-03  1.0e+00       2  BP monoubiquitinated histone H2A deubiquiti
## 2183          grey 1049  7.3e-03  1.0e+00       2  BP negative regulation of epithelial cell p
## 2184          grey 1049  7.3e-03  1.0e+00       2  BP positive regulation of histone H3-K9 tri
## 2185          grey 1049  7.3e-03  1.0e+00       2  BP semaphorin-plexin signaling pathway invo
## 2186          grey 1049  7.3e-03  1.0e+00       2  BP replication-born double-strand break rep
## 2187          grey 1049  7.3e-03  1.0e+00       2  BP semaphorin-plexin signaling pathway invo
## 2188          grey 1049  7.3e-03  1.0e+00       2  BP negative regulation of sodium ion transm
## 2189          grey 1049  7.3e-03  1.0e+00       2  CC              semaphorin receptor complex
## 2190          grey 1049  7.3e-03  1.0e+00       2  CC                           PR-DUB complex
## 2191          grey 1049  7.3e-03  1.0e+00       2  CC                             BORC complex
## 2192          grey 1049  7.3e-03  1.0e+00       2  MF glyceraldehyde-3-phosphate dehydrogenase
## 2193          grey 1049  7.3e-03  1.0e+00       2  MF MAP kinase kinase kinase kinase activity
## 2194          grey 1049  7.3e-03  1.0e+00       2  MF             semaphorin receptor activity
## 2195          grey 1049  7.3e-03  1.0e+00       2  MF                       SAM domain binding
## 2196          grey 1049  7.3e-03  1.0e+00       2  MF phosphatidylinositol 3-kinase catalytic 
## 2197          grey 1049  7.3e-03  1.0e+00       2  MF RNA polymerase II sequence-specific DNA-
## 2198          grey 1049  7.4e-03  1.0e+00       4  BP                      germ-band extension
## 2199          grey 1049  7.4e-03  1.0e+00       4  BP                     post-mating behavior
## 2200          grey 1049  7.4e-03  1.0e+00       4  CC                         CCR4-NOT complex
## 2201          grey 1049  7.4e-03  1.0e+00       4  MF histone methyltransferase activity (H3-K
## 2202          grey 1049  7.4e-03  1.0e+00      46  BP negative regulation of transcription, DN
## 2203          grey 1049  7.6e-03  1.0e+00      14  BP                regulation of cell growth
## 2204          grey 1049  7.6e-03  1.0e+00      41  BP imaginal disc-derived appendage morphoge
## 2205          grey 1049  7.7e-03  1.0e+00      19  BP                oocyte axis specification
## 2206          grey 1049  7.7e-03  1.0e+00      10  BP    positive regulation of cell migration
## 2207          grey 1049  7.8e-03  1.0e+00      37  BP                     synapse organization
## 2208          grey 1049  7.8e-03  1.0e+00      54  BP            epithelial tube morphogenesis
## 2209          grey 1049  7.9e-03  1.0e+00       7  BP establishment of mitotic spindle localiz
## 2210          grey 1049  7.9e-03  1.0e+00       7  CC        Golgi-associated vesicle membrane
## 2211          grey 1049  8.1e-03  1.0e+00       9  MF                           myosin binding
## 2212          grey 1049  8.2e-03  1.0e+00      22  BP       regulation of synapse organization
## 2213          grey 1049  8.2e-03  1.0e+00      39  BP positive regulation of transcription by 
## 2214          grey 1049  8.5e-03  1.0e+00      17  BP                  regulation of cell size
## 2215          grey 1049  8.7e-03  1.0e+00      73  BP                 female gamete generation
## 2216          grey 1049  8.8e-03  1.0e+00      22  CC                      apical part of cell
## 2217          grey 1049  8.9e-03  1.0e+00      10  BP          Rho protein signal transduction
## 2218          grey 1049  8.9e-03  1.0e+00      41  BP                  appendage morphogenesis
## 2219          grey 1049  9.0e-03  1.0e+00      29  BP                        secretion by cell
## 2220          grey 1049  9.0e-03  1.0e+00       5  BP positive regulation of mitotic nuclear d
## 2221          grey 1049  9.0e-03  1.0e+00       5  BP negative regulation of Ras protein signa
## 2222          grey 1049  9.1e-03  1.0e+00      62  BP       instar larval or pupal development
## 2223          grey 1049  9.1e-03  1.0e+00      28  BP       photoreceptor cell differentiation
## 2224          grey 1049  9.3e-03  1.0e+00      20  MF                enzyme activator activity
## 2225          grey 1049  9.3e-03  1.0e+00      64  BP                imaginal disc development
## 2226          grey 1049  9.5e-03  1.0e+00      13  BP positive regulation of neuron differenti
## 2227          grey 1049  9.7e-03  1.0e+00       7  BP         intestinal stem cell homeostasis
## 2228          grey 1049  9.7e-03  1.0e+00       7  BP positive regulation of Ras protein signa
## 2229          grey 1049  9.7e-03  1.0e+00       7  BP positive regulation of proteolysis invol
## 2230     lightcyan   82  1.4e-10  1.2e-06      46  BP          cellular component organization
## 2231     lightcyan   82  5.2e-09  4.4e-05      40  BP       multicellular organism development
## 2232     lightcyan   82  6.6e-09  5.6e-05      16  BP                   chromatin organization
## 2233     lightcyan   82  2.7e-08  2.3e-04      33  BP                   organelle organization
## 2234     lightcyan   82  4.5e-08  3.8e-04      29  BP                         cell development
## 2235     lightcyan   82  4.8e-08  4.0e-04      21  BP                   neuron differentiation
## 2236     lightcyan   82  6.5e-08  5.4e-04      19  BP                       neuron development
## 2237     lightcyan   82  6.6e-08  5.5e-04      41  BP         anatomical structure development
## 2238     lightcyan   82  1.3e-07  1.1e-03      34  BP          regulation of metabolic process
## 2239     lightcyan   82  1.6e-07  1.4e-03      18  BP                  chromosome organization
## 2240     lightcyan   82  1.7e-07  1.5e-03      31  BP                     cell differentiation
## 2241     lightcyan   82  2.3e-07  1.9e-03      41  BP                    developmental process
## 2242     lightcyan   82  3.3e-07  2.8e-03      36  MF                          protein binding
## 2243     lightcyan   82  4.1e-07  3.4e-03      21  BP                    generation of neurons
## 2244     lightcyan   82  7.4e-07  6.2e-03      10  BP                     histone modification
## 2245     lightcyan   82  8.8e-07  7.4e-03      21  BP                             neurogenesis
## 2246     lightcyan   82  1.1e-06  9.6e-03      27  BP            regulation of gene expression
## 2247     lightcyan   82  1.3e-06  1.1e-02      68  BP                         cellular process
## 2248     lightcyan   82  1.5e-06  1.3e-02      10  CC                      polytene chromosome
## 2249     lightcyan   82  1.6e-06  1.3e-02      36  CC                                  nucleus
## 2250     lightcyan   82  2.0e-06  1.6e-02      30  BP regulation of cellular metabolic process
## 2251     lightcyan   82  2.1e-06  1.8e-02      15  BP cell morphogenesis involved in different
## 2252     lightcyan   82  3.0e-06  2.5e-02      16  BP                       cell morphogenesis
## 2253     lightcyan   82  3.5e-06  3.0e-02      14  BP cell morphogenesis involved in neuron di
## 2254     lightcyan   82  4.1e-06  3.4e-02      27  BP       anatomical structure morphogenesis
## 2255     lightcyan   82  4.3e-06  3.6e-02      14  BP          neuron projection morphogenesis
## 2256     lightcyan   82  5.7e-06  4.8e-02       6  MF                          histone binding
## 2257     lightcyan   82  5.8e-06  4.8e-02      22  BP               nervous system development
## 2258     lightcyan   82  6.7e-06  5.6e-02      16  BP               post-embryonic development
## 2259     lightcyan   82  8.5e-06  7.1e-02      23  BP                 animal organ development
## 2260     lightcyan   82  8.5e-06  7.1e-02      14  BP            neuron projection development
## 2261     lightcyan   82  1.5e-05  1.2e-01      13  CC                              nucleoplasm
## 2262     lightcyan   82  1.6e-05  1.3e-01      52  BP               cellular metabolic process
## 2263     lightcyan   82  1.6e-05  1.3e-01      22  BP    cellular protein modification process
## 2264     lightcyan   82  1.8e-05  1.5e-01       4  MF               methylated histone binding
## 2265     lightcyan   82  2.0e-05  1.6e-01      32  BP                          gene expression
## 2266     lightcyan   82  2.3e-05  1.9e-01      15  BP                                oogenesis
## 2267     lightcyan   82  2.8e-05  2.3e-01      10  CC                                chromatin
## 2268     lightcyan   82  3.3e-05  2.8e-01      16  BP                    germ cell development
## 2269     lightcyan   82  3.7e-05  3.1e-01      50  BP      nitrogen compound metabolic process
## 2270     lightcyan   82  4.0e-05  3.4e-01      14  CC                               chromosome
## 2271     lightcyan   82  4.8e-05  4.0e-01      15  BP                               locomotion
## 2272     lightcyan   82  5.8e-05  4.9e-01       8  BP regulation of gene expression, epigeneti
## 2273     lightcyan   82  6.1e-05  5.1e-01      15  BP                 female gamete generation
## 2274     lightcyan   82  7.9e-05  6.6e-01       8  BP                           gene silencing
## 2275     lightcyan   82  8.4e-05  7.1e-01       6  BP         stem cell population maintenance
## 2276     lightcyan   82  9.2e-05  7.7e-01      14  BP   negative regulation of gene expression
## 2277     lightcyan   82  9.7e-05  8.2e-01       8  BP posttranscriptional regulation of gene e
## 2278     lightcyan   82  1.2e-04  9.8e-01       6  BP          imaginal disc pattern formation
## 2279     lightcyan   82  1.2e-04  1.0e+00      33  CC               protein-containing complex
## 2280     lightcyan   82  1.3e-04  1.0e+00       6  BP                              hemopoiesis
## 2281     lightcyan   82  1.3e-04  1.0e+00       6  BP     regulation of chromatin organization
## 2282     lightcyan   82  1.3e-04  1.0e+00      13  BP       instar larval or pupal development
## 2283     lightcyan   82  1.3e-04  1.0e+00       3  CC                 polytene chromosome band
## 2284     lightcyan   82  1.5e-04  1.0e+00       8  BP cellular response to DNA damage stimulus
## 2285     lightcyan   82  1.5e-04  1.0e+00      17  BP                        gamete generation
## 2286     lightcyan   82  1.5e-04  1.0e+00      10  BP                             axonogenesis
## 2287     lightcyan   82  1.6e-04  1.0e+00       8  BP          embryonic pattern specification
## 2288     lightcyan   82  1.7e-04  1.0e+00       9  BP                            axon guidance
## 2289     lightcyan   82  1.7e-04  1.0e+00       6  BP                      chromatin silencing
## 2290     lightcyan   82  1.8e-04  1.0e+00       5  BP              wing disc pattern formation
## 2291     lightcyan   82  1.9e-04  1.0e+00      49  CC                  intracellular organelle
## 2292     lightcyan   82  2.0e-04  1.0e+00      12  BP                sensory organ development
## 2293     lightcyan   82  2.1e-04  1.0e+00      10  BP                         axon development
## 2294     lightcyan   82  2.2e-04  1.0e+00       4  CC            polytene chromosome interband
## 2295     lightcyan   82  2.4e-04  1.0e+00      31  BP                     biosynthetic process
## 2296     lightcyan   82  2.5e-04  1.0e+00      10  BP                                    taxis
## 2297     lightcyan   82  3.4e-04  1.0e+00       3  BP            cellular response to ecdysone
## 2298     lightcyan   82  3.8e-04  1.0e+00       8  BP                             segmentation
## 2299     lightcyan   82  4.6e-04  1.0e+00       3  BP                     germ-band shortening
## 2300     lightcyan   82  4.7e-04  1.0e+00      18  BP                       tissue development
## 2301     lightcyan   82  4.8e-04  1.0e+00       4  BP               histone lysine methylation
## 2302     lightcyan   82  5.2e-04  1.0e+00      17  BP                   epithelium development
## 2303     lightcyan   82  5.6e-04  1.0e+00       7  BP                  blastoderm segmentation
## 2304     lightcyan   82  5.6e-04  1.0e+00       7  MF                        chromatin binding
## 2305     lightcyan   82  5.9e-04  1.0e+00      10  BP                          eye development
## 2306     lightcyan   82  6.7e-04  1.0e+00       8  BP                          immune response
## 2307     lightcyan   82  6.9e-04  1.0e+00       3  BP        regulation of histone methylation
## 2308     lightcyan   82  7.8e-04  1.0e+00       9  BP                            cell motility
## 2309     lightcyan   82  7.9e-04  1.0e+00       2  BP regulation of histone H3-K27 methylation
## 2310     lightcyan   82  8.5e-04  1.0e+00       4  CC        histone methyltransferase complex
## 2311     lightcyan   82  1.0e-03  1.0e+00       6  BP                       larval development
## 2312     lightcyan   82  1.0e-03  1.0e+00       5  BP                     chromatin remodeling
## 2313     lightcyan   82  1.1e-03  1.0e+00       6  BP    regulation of chromosome organization
## 2314     lightcyan   82  1.1e-03  1.0e+00       4  BP                      histone methylation
## 2315     lightcyan   82  1.1e-03  1.0e+00       2  BP central nervous system neuron axonogenes
## 2316     lightcyan   82  1.1e-03  1.0e+00       3  BP                    centriole replication
## 2317     lightcyan   82  1.1e-03  1.0e+00      18  BP                     reproductive process
## 2318     lightcyan   82  1.2e-03  1.0e+00       3  BP                       centriole assembly
## 2319     lightcyan   82  1.3e-03  1.0e+00      28  BP                protein metabolic process
## 2320     lightcyan   82  1.4e-03  1.0e+00      17  BP      multicellular organism reproduction
## 2321     lightcyan   82  1.4e-03  1.0e+00      54  CC                            intracellular
## 2322     lightcyan   82  1.5e-03  1.0e+00       2  BP                    histone demethylation
## 2323     lightcyan   82  1.5e-03  1.0e+00      13  BP  cell surface receptor signaling pathway
## 2324     lightcyan   82  1.5e-03  1.0e+00      17  BP regulation of transcription, DNA-templat
## 2325     lightcyan   82  1.6e-03  1.0e+00       9  BP negative regulation of transcription, DN
## 2326     lightcyan   82  1.6e-03  1.0e+00       3  BP             cellular response to hypoxia
## 2327     lightcyan   82  1.6e-03  1.0e+00       3  CC                                   P-body
## 2328     lightcyan   82  1.6e-03  1.0e+00       8  BP                           cell migration
## 2329     lightcyan   82  1.7e-03  1.0e+00      22  MF                     nucleic acid binding
## 2330     lightcyan   82  1.8e-03  1.0e+00       6  BP anterior/posterior pattern specification
## 2331     lightcyan   82  1.8e-03  1.0e+00       5  BP    cellular response to hormone stimulus
## 2332     lightcyan   82  1.9e-03  1.0e+00       2  BP                    protein demethylation
## 2333     lightcyan   82  1.9e-03  1.0e+00       2  BP central nervous system neuron developmen
## 2334     lightcyan   82  1.9e-03  1.0e+00       2  BP               histone H3-K27 methylation
## 2335     lightcyan   82  1.9e-03  1.0e+00       2  MF             histone demethylase activity
## 2336     lightcyan   82  1.9e-03  1.0e+00       2  MF         nuclear hormone receptor binding
## 2337     lightcyan   82  1.9e-03  1.0e+00      18  BP                             reproduction
## 2338     lightcyan   82  1.9e-03  1.0e+00       4  BP                         long-term memory
## 2339     lightcyan   82  1.9e-03  1.0e+00       3  CC                         nuclear membrane
## 2340     lightcyan   82  2.0e-03  1.0e+00       8  BP        ovarian follicle cell development
## 2341     lightcyan   82  2.0e-03  1.0e+00      28  BP nucleobase-containing compound metabolic
## 2342     lightcyan   82  2.0e-03  1.0e+00       8  BP                        eye morphogenesis
## 2343     lightcyan   82  2.0e-03  1.0e+00       4  BP                      protein methylation
## 2344     lightcyan   82  2.2e-03  1.0e+00       4  MF DNA-binding transcription repressor acti
## 2345     lightcyan   82  2.2e-03  1.0e+00      13  BP        regulation of signal transduction
## 2346     lightcyan   82  2.2e-03  1.0e+00       5  BP                      oocyte construction
## 2347     lightcyan   82  2.3e-03  1.0e+00       2  BP                     sex comb development
## 2348     lightcyan   82  2.3e-03  1.0e+00       2  CC                       histone locus body
## 2349     lightcyan   82  2.4e-03  1.0e+00      17  BP             transcription, DNA-templated
## 2350     lightcyan   82  2.4e-03  1.0e+00       5  BP                      response to alcohol
## 2351     lightcyan   82  2.4e-03  1.0e+00      23  BP                    RNA metabolic process
## 2352     lightcyan   82  2.6e-03  1.0e+00       4  BP                         centrosome cycle
## 2353     lightcyan   82  2.6e-03  1.0e+00       5  BP                regulation of translation
## 2354     lightcyan   82  2.7e-03  1.0e+00      13  BP               animal organ morphogenesis
## 2355     lightcyan   82  2.7e-03  1.0e+00      14  MF                              DNA binding
## 2356     lightcyan   82  2.7e-03  1.0e+00      20  BP                      signal transduction
## 2357     lightcyan   82  2.8e-03  1.0e+00       2  BP ecdysone receptor-mediated signaling pat
## 2358     lightcyan   82  2.8e-03  1.0e+00       2  BP                     convergent extension
## 2359     lightcyan   82  2.8e-03  1.0e+00       2  BP                            demethylation
## 2360     lightcyan   82  2.8e-03  1.0e+00       2  CC    phosphatidylinositol 3-kinase complex
## 2361     lightcyan   82  2.8e-03  1.0e+00       2  CC                           MLL3/4 complex
## 2362     lightcyan   82  2.8e-03  1.0e+00       2  MF                     demethylase activity
## 2363     lightcyan   82  3.1e-03  1.0e+00       3  BP                   centrosome duplication
## 2364     lightcyan   82  3.1e-03  1.0e+00       3  BP            regulation of TORC1 signaling
## 2365     lightcyan   82  3.4e-03  1.0e+00       2  BP        synaptonemal complex organization
## 2366     lightcyan   82  3.4e-03  1.0e+00      17  BP      regulation of RNA metabolic process
## 2367     lightcyan   82  3.4e-03  1.0e+00       5  BP                       oocyte development
## 2368     lightcyan   82  3.5e-03  1.0e+00       4  BP               double-strand break repair
## 2369     lightcyan   82  3.6e-03  1.0e+00      10  BP            response to organic substance
## 2370     lightcyan   82  3.7e-03  1.0e+00      10  BP            pattern specification process
## 2371     lightcyan   82  3.7e-03  1.0e+00       5  BP regulation of cell projection organizati
## 2372     lightcyan   82  3.8e-03  1.0e+00       8  BP              epithelial cell development
## 2373     lightcyan   82  3.9e-03  1.0e+00       8  BP                         defense response
## 2374     lightcyan   82  4.0e-03  1.0e+00       2  BP negative regulation of innate immune res
## 2375     lightcyan   82  4.1e-03  1.0e+00       3  BP                          TORC1 signaling
## 2376     lightcyan   82  4.1e-03  1.0e+00      22  BP                       cell communication
## 2377     lightcyan   82  4.6e-03  1.0e+00       2  BP                  mucosal immune response
## 2378     lightcyan   82  4.6e-03  1.0e+00       2  BP intracellular steroid hormone receptor s
## 2379     lightcyan   82  4.6e-03  1.0e+00       2  BP cytoplasmic mRNA processing body assembl
## 2380     lightcyan   82  4.6e-03  1.0e+00       3  BP                     response to ecdysone
## 2381     lightcyan   82  4.6e-03  1.0e+00       4  BP microtubule organizing center organizati
## 2382     lightcyan   82  4.7e-03  1.0e+00       5  BP                      response to hormone
## 2383     lightcyan   82  5.2e-03  1.0e+00       3  MF       histone acetyltransferase activity
## 2384     lightcyan   82  5.3e-03  1.0e+00       2  BP male germ-line stem cell population main
## 2385     lightcyan   82  5.3e-03  1.0e+00       2  BP steroid hormone mediated signaling pathw
## 2386     lightcyan   82  5.3e-03  1.0e+00       2  CC              pericentric heterochromatin
## 2387     lightcyan   82  5.3e-03  1.0e+00       2  MF           3'-5'-exoribonuclease activity
## 2388     lightcyan   82  5.6e-03  1.0e+00       8  BP                    wing disc development
## 2389     lightcyan   82  5.8e-03  1.0e+00       3  BP wing disc dorsal/ventral pattern formati
## 2390     lightcyan   82  5.8e-03  1.0e+00       3  CC                          heterochromatin
## 2391     lightcyan   82  6.0e-03  1.0e+00      39  CC intracellular membrane-bounded organelle
## 2392     lightcyan   82  6.0e-03  1.0e+00       2  BP axon extension involved in axon guidance
## 2393     lightcyan   82  6.0e-03  1.0e+00       2  MF                 exoribonuclease activity
## 2394     lightcyan   82  6.1e-03  1.0e+00       9  BP                            metamorphosis
## 2395     lightcyan   82  6.2e-03  1.0e+00       3  BP positive regulation of protein kinase ac
## 2396     lightcyan   82  6.2e-03  1.0e+00       3  BP regulation of cellular protein localizat
## 2397     lightcyan   82  6.2e-03  1.0e+00       3  MF                    chromatin DNA binding
## 2398     lightcyan   82  6.4e-03  1.0e+00       5  BP                   oocyte differentiation
## 2399     lightcyan   82  6.4e-03  1.0e+00       7  BP                 regulation of cell cycle
## 2400     lightcyan   82  6.5e-03  1.0e+00       5  BP                  Notch signaling pathway
## 2401     lightcyan   82  6.7e-03  1.0e+00       5  CC                        nuclear chromatin
## 2402     lightcyan   82  6.8e-03  1.0e+00       2  BP intrinsic apoptotic signaling pathway in
## 2403     lightcyan   82  6.8e-03  1.0e+00       2  BP positive regulation of smoothened signal
## 2404     lightcyan   82  6.9e-03  1.0e+00       3  BP   positive regulation of kinase activity
## 2405     lightcyan   82  7.1e-03  1.0e+00       5  BP                   dendrite morphogenesis
## 2406     lightcyan   82  7.2e-03  1.0e+00       3  BP           larval lymph gland hemopoiesis
## 2407     lightcyan   82  7.4e-03  1.0e+00       1  BP meiotic DNA repair synthesis involved in
## 2408     lightcyan   82  7.4e-03  1.0e+00       1  BP                      platelet activation
## 2409     lightcyan   82  7.4e-03  1.0e+00       1  BP negative regulation of glial cell apopto
## 2410     lightcyan   82  7.4e-03  1.0e+00       1  BP regulation of long-term neuronal synapti
## 2411     lightcyan   82  7.4e-03  1.0e+00       1  BP             histone H3-K27 demethylation
## 2412     lightcyan   82  7.4e-03  1.0e+00       1  BP         commissural neuron axon guidance
## 2413     lightcyan   82  7.4e-03  1.0e+00       1  BP     fasciculation of sensory neuron axon
## 2414     lightcyan   82  7.4e-03  1.0e+00       1  BP positive regulation of intrinsic apoptot
## 2415     lightcyan   82  7.4e-03  1.0e+00       1  BP positive regulation of intrinsic apoptot
## 2416     lightcyan   82  7.4e-03  1.0e+00       1  BP negative regulation of histone H3-K27 tr
## 2417     lightcyan   82  7.4e-03  1.0e+00       1  BP              histone H2A phosphorylation
## 2418     lightcyan   82  7.4e-03  1.0e+00       1  CC                   lamellipodium membrane
## 2419     lightcyan   82  7.4e-03  1.0e+00       1  CC       transcriptionally silent chromatin
## 2420     lightcyan   82  7.4e-03  1.0e+00       1  CC                              PAS complex
## 2421     lightcyan   82  7.4e-03  1.0e+00       1  MF glycerophosphodiester phosphodiesterase 
## 2422     lightcyan   82  7.4e-03  1.0e+00       1  MF histone demethylase activity (H4-R3 spec
## 2423     lightcyan   82  7.4e-03  1.0e+00       1  MF histone acetyltransferase activity (H3-K
## 2424     lightcyan   82  7.4e-03  1.0e+00       1  MF phosphatidylinositol-4,5-bisphosphate 3-
## 2425     lightcyan   82  7.4e-03  1.0e+00       1  MF        lysine-acetylated histone binding
## 2426     lightcyan   82  7.4e-03  1.0e+00       1  MF histone demethylase activity (H3-K27 spe
## 2427     lightcyan   82  7.4e-03  1.0e+00      10  BP                imaginal disc development
## 2428     lightcyan   82  7.6e-03  1.0e+00       2  BP              lamellocyte differentiation
## 2429     lightcyan   82  7.6e-03  1.0e+00       3  BP dorsal/ventral pattern formation, imagin
## 2430     lightcyan   82  8.2e-03  1.0e+00       5  BP embryonic development via the syncytial 
## 2431     lightcyan   82  8.4e-03  1.0e+00       2  BP anterior/posterior pattern specification
## 2432     lightcyan   82  8.4e-03  1.0e+00       2  BP       negative regulation of cell growth
## 2433     lightcyan   82  8.6e-03  1.0e+00       5  BP     regulation of neuron differentiation
## 2434     lightcyan   82  8.7e-03  1.0e+00       4  BP        negative regulation of cell cycle
## 2435     lightcyan   82  9.0e-03  1.0e+00       5  BP                     dendrite development
## 2436     lightcyan   82  9.3e-03  1.0e+00       2  BP somatic stem cell population maintenance
## 2437     lightcyan   82  9.3e-03  1.0e+00       2  BP  positive regulation of circadian rhythm
## 2438     lightcyan   82  9.9e-03  1.0e+00       7  BP negative regulation of signal transducti
## 2439    lightgreen  102  6.7e-31  5.6e-27      24  CC                   mitochondrial ribosome
## 2440    lightgreen  102  2.2e-29  1.8e-25      25  BP                mitochondrial translation
## 2441    lightgreen  102  2.7e-27  2.3e-23      27  CC                     mitochondrial matrix
## 2442    lightgreen  102  6.6e-23  5.5e-19      25  CC                                 ribosome
## 2443    lightgreen  102  2.5e-21  2.1e-17      23  MF       structural constituent of ribosome
## 2444    lightgreen  102  1.9e-20  1.6e-16      37  CC                            mitochondrion
## 2445    lightgreen  102  1.3e-19  1.1e-15      15  CC    mitochondrial large ribosomal subunit
## 2446    lightgreen  102  6.7e-18  5.6e-14      28  BP                              translation
## 2447    lightgreen  102  1.0e-17  8.5e-14      31  BP                peptide metabolic process
## 2448    lightgreen  102  1.9e-14  1.6e-10      15  CC                  large ribosomal subunit
## 2449    lightgreen  102  7.8e-14  6.5e-10      25  MF             structural molecule activity
## 2450    lightgreen  102  5.4e-12  4.5e-08      25  CC                ribonucleoprotein complex
## 2451    lightgreen  102  6.8e-12  5.7e-08       9  CC    mitochondrial small ribosomal subunit
## 2452    lightgreen  102  7.8e-09  6.5e-05      60  CC                                cytoplasm
## 2453    lightgreen  102  2.2e-08  1.9e-04       9  CC                  small ribosomal subunit
## 2454    lightgreen  102  2.7e-07  2.3e-03      42  BP                protein metabolic process
## 2455    lightgreen  102  9.5e-06  8.0e-02      40  BP                     biosynthetic process
## 2456    lightgreen  102  2.9e-05  2.4e-01      54  CC intracellular membrane-bounded organelle
## 2457    lightgreen  102  4.4e-05  3.7e-01      40  CC               protein-containing complex
## 2458    lightgreen  102  1.7e-04  1.0e+00      35  BP                          gene expression
## 2459    lightgreen  102  3.2e-04  1.0e+00       5  MF translation factor activity, RNA binding
## 2460    lightgreen  102  3.4e-04  1.0e+00       4  BP mitochondrial respiratory chain complex 
## 2461    lightgreen  102  3.4e-04  1.0e+00      67  CC                            intracellular
## 2462    lightgreen  102  3.7e-04  1.0e+00       3  BP mitochondrial respiratory chain complex 
## 2463    lightgreen  102  6.3e-04  1.0e+00      57  CC                  intracellular organelle
## 2464    lightgreen  102  9.1e-04  1.0e+00      57  BP               cellular metabolic process
## 2465    lightgreen  102  9.8e-04  1.0e+00       5  MF S-adenosylmethionine-dependent methyltra
## 2466    lightgreen  102  1.2e-03  1.0e+00       7  BP               mitochondrion organization
## 2467    lightgreen  102  2.2e-03  1.0e+00       2  MF                    nucleotidase activity
## 2468    lightgreen  102  2.6e-03  1.0e+00      54  BP      nitrogen compound metabolic process
## 2469    lightgreen  102  3.5e-03  1.0e+00       2  MF               sulfurtransferase activity
## 2470    lightgreen  102  3.8e-03  1.0e+00       5  MF               methyltransferase activity
## 2471    lightgreen  102  5.3e-03  1.0e+00       3  MF                             rRNA binding
## 2472    lightgreen  102  6.0e-03  1.0e+00       2  CC eukaryotic translation initiation factor
## 2473    lightgreen  102  6.9e-03  1.0e+00       3  MF   translation initiation factor activity
## 2474    lightgreen  102  6.9e-03  1.0e+00       3  MF           RNA methyltransferase activity
## 2475    lightgreen  102  9.1e-03  1.0e+00       1  BP                             angiogenesis
## 2476    lightgreen  102  9.1e-03  1.0e+00       1  BP ecdysone-mediated polytene chromosome pu
## 2477    lightgreen  102  9.1e-03  1.0e+00       1  BP          dihydrofolate metabolic process
## 2478    lightgreen  102  9.1e-03  1.0e+00       1  CC                      nuclear microtubule
## 2479    lightgreen  102  9.1e-03  1.0e+00       1  CC  methylcrotonoyl-CoA carboxylase complex
## 2480    lightgreen  102  9.1e-03  1.0e+00       1  MF         dihydrofolate reductase activity
## 2481    lightgreen  102  9.1e-03  1.0e+00       1  MF  Rab GDP-dissociation inhibitor activity
## 2482    lightgreen  102  9.1e-03  1.0e+00       1  MF 3'(2'),5'-bisphosphate nucleotidase acti
## 2483    lightgreen  102  9.1e-03  1.0e+00       1  MF translation release factor activity, cod
## 2484    lightgreen  102  9.1e-03  1.0e+00       1  MF           acireductone synthase activity
## 2485    lightgreen  102  9.1e-03  1.0e+00       1  MF tRNA (cytosine-3-)-methyltransferase act
## 2486   lightyellow   79  8.4e-11  7.0e-07       9  CC                 Golgi-associated vesicle
## 2487   lightyellow   79  1.9e-10  1.6e-06      51  CC                                cytoplasm
## 2488   lightyellow   79  3.6e-10  3.0e-06       9  CC                           coated vesicle
## 2489   lightyellow   79  5.6e-10  4.7e-06       7  CC        Golgi-associated vesicle membrane
## 2490   lightyellow   79  1.6e-09  1.4e-05       5  CC                        COPI vesicle coat
## 2491   lightyellow   79  1.6e-09  1.4e-05       5  CC                      COPI-coated vesicle
## 2492   lightyellow   79  2.0e-08  1.7e-04       8  BP   ER to Golgi vesicle-mediated transport
## 2493   lightyellow   79  2.4e-08  2.0e-04      10  BP                  Golgi vesicle transport
## 2494   lightyellow   79  1.7e-07  1.5e-03       7  CC             cytoplasmic vesicle membrane
## 2495   lightyellow   79  3.3e-07  2.8e-03       7  CC                         vesicle membrane
## 2496   lightyellow   79  1.9e-06  1.6e-02       9  MF       structural constituent of ribosome
## 2497   lightyellow   79  2.6e-06  2.2e-02      13  BP                              translation
## 2498   lightyellow   79  4.7e-06  3.9e-02       9  CC                                 ribosome
## 2499   lightyellow   79  7.9e-06  6.6e-02      13  MF             structural molecule activity
## 2500   lightyellow   79  8.3e-06  6.9e-02      16  CC                            mitochondrion
## 2501   lightyellow   79  1.4e-05  1.1e-01       4  BP retrograde vesicle-mediated transport, G
## 2502   lightyellow   79  2.0e-05  1.6e-01       4  BP   intra-Golgi vesicle-mediated transport
## 2503   lightyellow   79  2.0e-05  1.7e-01      12  CC                    endoplasmic reticulum
## 2504   lightyellow   79  2.7e-05  2.3e-01      55  CC                            intracellular
## 2505   lightyellow   79  3.2e-05  2.7e-01       4  CC COPII-coated ER to Golgi transport vesic
## 2506   lightyellow   79  4.8e-05  4.1e-01       6  BP                        protein targeting
## 2507   lightyellow   79  4.9e-05  4.1e-01      13  BP                peptide metabolic process
## 2508   lightyellow   79  6.2e-05  5.2e-01      10  CC                          Golgi apparatus
## 2509   lightyellow   79  8.8e-05  7.4e-01      42  CC intracellular membrane-bounded organelle
## 2510   lightyellow   79  9.6e-05  8.1e-01      14  BP                  intracellular transport
## 2511   lightyellow   79  1.0e-04  8.4e-01      47  CC                  intracellular organelle
## 2512   lightyellow   79  1.2e-04  1.0e+00       7  CC                     mitochondrial matrix
## 2513   lightyellow   79  1.3e-04  1.0e+00      10  BP          intracellular protein transport
## 2514   lightyellow   79  1.7e-04  1.0e+00       4  BP              regulation of lipid storage
## 2515   lightyellow   79  1.7e-04  1.0e+00      11  BP                        protein transport
## 2516   lightyellow   79  1.8e-04  1.0e+00      31  CC               protein-containing complex
## 2517   lightyellow   79  1.9e-04  1.0e+00      17  CC                      endomembrane system
## 2518   lightyellow   79  2.3e-04  1.0e+00      12  CC                       organelle membrane
## 2519   lightyellow   79  2.3e-04  1.0e+00       5  CC                   mitochondrial ribosome
## 2520   lightyellow   79  2.3e-04  1.0e+00      11  BP    establishment of protein localization
## 2521   lightyellow   79  2.9e-04  1.0e+00       7  CC           endoplasmic reticulum membrane
## 2522   lightyellow   79  3.4e-04  1.0e+00       4  MF        ribonucleoprotein complex binding
## 2523   lightyellow   79  4.5e-04  1.0e+00      11  BP               vesicle-mediated transport
## 2524   lightyellow   79  4.6e-04  1.0e+00       2  BP      vesicle fusion with Golgi apparatus
## 2525   lightyellow   79  4.6e-04  1.0e+00       4  BP                            lipid storage
## 2526   lightyellow   79  7.2e-04  1.0e+00       5  CC                  large ribosomal subunit
## 2527   lightyellow   79  7.6e-04  1.0e+00       9  CC                      cytoplasmic vesicle
## 2528   lightyellow   79  8.8e-04  1.0e+00      28  BP                     biosynthetic process
## 2529   lightyellow   79  1.3e-03  1.0e+00       5  BP                  cytoplasmic translation
## 2530   lightyellow   79  1.3e-03  1.0e+00       3  BP                fatty acid beta-oxidation
## 2531   lightyellow   79  1.6e-03  1.0e+00       2  MF             enoyl-CoA hydratase activity
## 2532   lightyellow   79  1.6e-03  1.0e+00       9  CC                                  vesicle
## 2533   lightyellow   79  1.9e-03  1.0e+00       3  BP                     fatty acid oxidation
## 2534   lightyellow   79  1.9e-03  1.0e+00       4  BP                alcohol metabolic process
## 2535   lightyellow   79  2.0e-03  1.0e+00       4  CC                           Golgi membrane
## 2536   lightyellow   79  2.0e-03  1.0e+00       3  BP                          lipid oxidation
## 2537   lightyellow   79  2.5e-03  1.0e+00       2  BP regulation of tube diameter, open trache
## 2538   lightyellow   79  2.5e-03  1.0e+00      10  BP            cellular protein localization
## 2539   lightyellow   79  2.9e-03  1.0e+00       2  BP      larval salivary gland morphogenesis
## 2540   lightyellow   79  2.9e-03  1.0e+00       2  CC   ER to Golgi transport vesicle membrane
## 2541   lightyellow   79  3.2e-03  1.0e+00       3  BP             fatty acid catabolic process
## 2542   lightyellow   79  3.2e-03  1.0e+00       3  MF   translation initiation factor activity
## 2543   lightyellow   79  3.3e-03  1.0e+00      10  CC                ribonucleoprotein complex
## 2544   lightyellow   79  3.5e-03  1.0e+00       2  BP regulation of reactive oxygen species me
## 2545   lightyellow   79  3.5e-03  1.0e+00       2  CC eukaryotic translation initiation factor
## 2546   lightyellow   79  4.0e-03  1.0e+00       2  BP SRP-dependent cotranslational protein ta
## 2547   lightyellow   79  4.0e-03  1.0e+00       2  MF P-P-bond-hydrolysis-driven protein trans
## 2548   lightyellow   79  4.6e-03  1.0e+00       3  BP                        ribosome assembly
## 2549   lightyellow   79  4.6e-03  1.0e+00       3  MF oxidoreductase activity, acting on NAD(P
## 2550   lightyellow   79  4.6e-03  1.0e+00       2  BP              regulation of tube diameter
## 2551   lightyellow   79  4.8e-03  1.0e+00       3  CC    mitochondrial large ribosomal subunit
## 2552   lightyellow   79  4.9e-03  1.0e+00       4  CC                       cytosolic ribosome
## 2553   lightyellow   79  5.4e-03  1.0e+00      11  BP                     protein localization
## 2554   lightyellow   79  5.8e-03  1.0e+00      10  BP      cellular macromolecule localization
## 2555   lightyellow   79  6.0e-03  1.0e+00       4  BP                mitochondrial translation
## 2556   lightyellow   79  6.4e-03  1.0e+00      10  CC                                  cytosol
## 2557   lightyellow   79  6.6e-03  1.0e+00      11  BP              organic substance transport
## 2558   lightyellow   79  6.6e-03  1.0e+00       2  MF protein transmembrane transporter activi
## 2559   lightyellow   79  6.6e-03  1.0e+00       3  BP                 translational initiation
## 2560   lightyellow   79  6.9e-03  1.0e+00       1  BP                        ethanol oxidation
## 2561   lightyellow   79  6.9e-03  1.0e+00       1  BP               thiamine metabolic process
## 2562   lightyellow   79  6.9e-03  1.0e+00       1  BP              heme a biosynthetic process
## 2563   lightyellow   79  6.9e-03  1.0e+00       1  BP thiamine diphosphate biosynthetic proces
## 2564   lightyellow   79  6.9e-03  1.0e+00       1  BP regulation of hydrogen peroxide metaboli
## 2565   lightyellow   79  6.9e-03  1.0e+00       1  BP               glycerol catabolic process
## 2566   lightyellow   79  6.9e-03  1.0e+00       1  BP positive regulation of histone phosphory
## 2567   lightyellow   79  6.9e-03  1.0e+00       1  BP glyceraldehyde-3-phosphate biosynthetic 
## 2568   lightyellow   79  6.9e-03  1.0e+00       1  BP  proteasome regulatory particle assembly
## 2569   lightyellow   79  6.9e-03  1.0e+00       1  BP regulation of nitric oxide metabolic pro
## 2570   lightyellow   79  6.9e-03  1.0e+00       1  BP regulation of peptidyl-cysteine S-nitros
## 2571   lightyellow   79  6.9e-03  1.0e+00       1  MF        lactoylglutathione lyase activity
## 2572   lightyellow   79  6.9e-03  1.0e+00       1  MF           octanol dehydrogenase activity
## 2573   lightyellow   79  6.9e-03  1.0e+00       1  MF        thiamine diphosphokinase activity
## 2574   lightyellow   79  6.9e-03  1.0e+00       1  MF      triose-phosphate isomerase activity
## 2575   lightyellow   79  6.9e-03  1.0e+00       1  MF                         thiamine binding
## 2576   lightyellow   79  6.9e-03  1.0e+00       1  MF dehydrodolichyl diphosphate synthase act
## 2577   lightyellow   79  6.9e-03  1.0e+00       1  MF S-(hydroxymethyl)glutathione dehydrogena
## 2578   lightyellow   79  6.9e-03  1.0e+00       1  MF ADP-dependent NAD(P)H-hydrate dehydratas
## 2579   lightyellow   79  6.9e-03  1.0e+00       1  MF  S-nitrosoglutathione reductase activity
## 2580   lightyellow   79  6.9e-03  1.0e+00       1  MF                   COPII adaptor activity
## 2581   lightyellow   79  8.2e-03  1.0e+00       2  MF                         ribosome binding
## 2582   lightyellow   79  9.0e-03  1.0e+00       4  BP             fatty acid metabolic process
## 2583   lightyellow   79  9.0e-03  1.0e+00       2  BP                  protein targeting to ER
## 2584   lightyellow   79  9.0e-03  1.0e+00       2  MF peptide transmembrane transporter activi
## 2585          pink  269  7.5e-13  6.3e-09     104  CC                                  nucleus
## 2586          pink  269  4.0e-12  3.4e-08      74  BP            regulation of gene expression
## 2587          pink  269  4.4e-12  3.6e-08      65  BP      regulation of RNA metabolic process
## 2588          pink  269  3.0e-11  2.5e-07      88  BP          regulation of metabolic process
## 2589          pink  269  6.1e-11  5.1e-07      94  BP                          gene expression
## 2590          pink  269  9.6e-11  8.1e-07      81  BP                    RNA metabolic process
## 2591          pink  269  1.3e-10  1.1e-06      59  BP regulation of transcription, DNA-templat
## 2592          pink  269  4.1e-10  3.5e-06      95  BP nucleobase-containing compound metabolic
## 2593          pink  269  5.8e-10  4.8e-06     176  CC                            intracellular
## 2594          pink  269  7.5e-10  6.3e-06      79  BP regulation of cellular metabolic process
## 2595          pink  269  1.9e-09  1.6e-05      58  BP             transcription, DNA-templated
## 2596          pink  269  2.5e-09  2.1e-05      34  MF                         zinc ion binding
## 2597          pink  269  5.1e-09  4.3e-05      37  CC                               chromosome
## 2598          pink  269  1.4e-08  1.2e-04      71  MF                     nucleic acid binding
## 2599          pink  269  2.2e-08  1.8e-04     148  CC                  intracellular organelle
## 2600          pink  269  3.0e-08  2.5e-04      20  CC                      polytene chromosome
## 2601          pink  269  5.9e-08  4.9e-04      38  BP regulation of transcription by RNA polym
## 2602          pink  269  9.5e-08  8.0e-04      39  BP       transcription by RNA polymerase II
## 2603          pink  269  1.3e-07  1.1e-03     128  CC intracellular membrane-bounded organelle
## 2604          pink  269  3.7e-07  3.1e-03      35  BP   negative regulation of gene expression
## 2605          pink  269  1.0e-06  8.3e-03     140  BP      nitrogen compound metabolic process
## 2606          pink  269  1.1e-06  8.9e-03     192  BP                         cellular process
## 2607          pink  269  2.1e-06  1.7e-02      25  BP negative regulation of transcription, DN
## 2608          pink  269  3.1e-06  2.6e-02      45  MF                           cation binding
## 2609          pink  269  3.8e-06  3.2e-02      44  MF                        metal ion binding
## 2610          pink  269  5.6e-06  4.7e-02     141  BP               cellular metabolic process
## 2611          pink  269  5.8e-06  4.9e-02      29  BP   positive regulation of gene expression
## 2612          pink  269  5.8e-06  4.9e-02      34  MF             transition metal ion binding
## 2613          pink  269  1.1e-05  9.3e-02      25  BP positive regulation of transcription, DN
## 2614          pink  269  4.2e-05  3.6e-01      30  BP                  chromosome organization
## 2615          pink  269  4.5e-05  3.8e-01      90  BP          cellular component organization
## 2616          pink  269  4.5e-05  3.8e-01      22  BP                   chromatin organization
## 2617          pink  269  4.8e-05  4.0e-01      38  MF                              DNA binding
## 2618          pink  269  5.3e-05  4.4e-01      15  MF                        chromatin binding
## 2619          pink  269  5.9e-05  5.0e-01      50  MF                     transferase activity
## 2620          pink  269  7.0e-05  5.9e-01      18  CC                                chromatin
## 2621          pink  269  8.2e-05  6.9e-01      47  BP    cellular protein modification process
## 2622          pink  269  8.3e-05  7.0e-01      24  CC                              nucleoplasm
## 2623          pink  269  1.1e-04  9.0e-01      19  BP positive regulation of transcription by 
## 2624          pink  269  1.1e-04  9.1e-01      13  CC                        nuclear chromatin
## 2625          pink  269  1.2e-04  9.8e-01      61  BP                     cell differentiation
## 2626          pink  269  1.2e-04  9.9e-01      14  BP                   protein ubiquitination
## 2627          pink  269  1.2e-04  1.0e+00       3  MF histone methyltransferase activity (H3-K
## 2628          pink  269  1.3e-04  1.0e+00      79  BP                     biosynthetic process
## 2629          pink  269  1.7e-04  1.0e+00      15  MF ubiquitin-like protein transferase activ
## 2630          pink  269  2.4e-04  1.0e+00       5  MF histone-lysine N-methyltransferase activ
## 2631          pink  269  2.5e-04  1.0e+00      14  BP protein modification by small protein co
## 2632          pink  269  2.6e-04  1.0e+00      52  BP                         cell development
## 2633          pink  269  2.6e-04  1.0e+00      33  BP                   neuron differentiation
## 2634          pink  269  2.9e-04  1.0e+00      36  BP                    generation of neurons
## 2635          pink  269  3.1e-04  1.0e+00      43  BP               nervous system development
## 2636          pink  269  3.1e-04  1.0e+00      61  BP                   organelle organization
## 2637          pink  269  3.4e-04  1.0e+00      37  BP                             neurogenesis
## 2638          pink  269  4.3e-04  1.0e+00       5  CC                      PcG protein complex
## 2639          pink  269  4.5e-04  1.0e+00      23  BP                            metamorphosis
## 2640          pink  269  5.1e-04  1.0e+00       5  MF protein-lysine N-methyltransferase activ
## 2641          pink  269  5.5e-04  1.0e+00       2  BP            isoleucyl-tRNA aminoacylation
## 2642          pink  269  5.5e-04  1.0e+00       2  BP                mitotic cohesin unloading
## 2643          pink  269  5.5e-04  1.0e+00       2  MF          isoleucine-tRNA ligase activity
## 2644          pink  269  5.8e-04  1.0e+00      13  BP                           gene silencing
## 2645          pink  269  8.0e-04  1.0e+00      18  BP imaginal disc-derived appendage developm
## 2646          pink  269  8.6e-04  1.0e+00      26  BP                       cell morphogenesis
## 2647          pink  269  8.8e-04  1.0e+00      11  BP         dorsal/ventral pattern formation
## 2648          pink  269  1.0e-03  1.0e+00      13  MF   ubiquitin-protein transferase activity
## 2649          pink  269  1.0e-03  1.0e+00      31  BP      regulation of developmental process
## 2650          pink  269  1.1e-03  1.0e+00       5  MF       histone methyltransferase activity
## 2651          pink  269  1.1e-03  1.0e+00      12  BP                     histone modification
## 2652          pink  269  1.1e-03  1.0e+00      19  BP                               cell death
## 2653          pink  269  1.2e-03  1.0e+00      17  BP   post-embryonic appendage morphogenesis
## 2654          pink  269  1.3e-03  1.0e+00      57  BP                                signaling
## 2655          pink  269  1.3e-03  1.0e+00       3  BP               protein monoubiquitination
## 2656          pink  269  1.3e-03  1.0e+00       3  CC                         ESC/E(Z) complex
## 2657          pink  269  1.6e-03  1.0e+00       2  BP negative regulation of mRNA polyadenylat
## 2658          pink  269  1.6e-03  1.0e+00       2  CC               Hedgehog signaling complex
## 2659          pink  269  1.6e-03  1.0e+00      17  BP imaginal disc-derived appendage morphoge
## 2660          pink  269  1.8e-03  1.0e+00      17  BP                  appendage morphogenesis
## 2661          pink  269  1.8e-03  1.0e+00       3  BP apical constriction involved in gastrula
## 2662          pink  269  1.8e-03  1.0e+00       3  BP                   histone ubiquitination
## 2663          pink  269  1.9e-03  1.0e+00      16  BP                          mRNA processing
## 2664          pink  269  1.9e-03  1.0e+00      50  BP                      signal transduction
## 2665          pink  269  1.9e-03  1.0e+00      26  MF                              ATP binding
## 2666          pink  269  2.3e-03  1.0e+00       6  CC        histone methyltransferase complex
## 2667          pink  269  2.4e-03  1.0e+00      11  MF protein serine/threonine kinase activity
## 2668          pink  269  2.4e-03  1.0e+00       3  BP                     response to caffeine
## 2669          pink  269  2.5e-03  1.0e+00       5  CC            polytene chromosome interband
## 2670          pink  269  2.6e-03  1.0e+00      26  BP                       neuron development
## 2671          pink  269  2.7e-03  1.0e+00      18  BP                    wing disc development
## 2672          pink  269  2.7e-03  1.0e+00      14  CC                       nuclear chromosome
## 2673          pink  269  2.8e-03  1.0e+00      26  MF            adenyl ribonucleotide binding
## 2674          pink  269  2.9e-03  1.0e+00       4  BP     cytoplasmic microtubule organization
## 2675          pink  269  3.0e-03  1.0e+00      80  BP         anatomical structure development
## 2676          pink  269  3.0e-03  1.0e+00       3  BP regulation of development, heterochronic
## 2677          pink  269  3.2e-03  1.0e+00      29  BP        regulation of signal transduction
## 2678          pink  269  3.2e-03  1.0e+00       2  BP negative regulation of mRNA 3'-end proce
## 2679          pink  269  3.2e-03  1.0e+00       2  CC                          cohesin complex
## 2680          pink  269  3.2e-03  1.0e+00       2  CC                             nBAF complex
## 2681          pink  269  3.2e-03  1.0e+00       6  BP                      histone methylation
## 2682          pink  269  3.3e-03  1.0e+00      17  BP              epithelial cell development
## 2683          pink  269  3.3e-03  1.0e+00      25  BP                                oogenesis
## 2684          pink  269  3.4e-03  1.0e+00       4  MF                      damaged DNA binding
## 2685          pink  269  3.4e-03  1.0e+00      51  BP       anatomical structure morphogenesis
## 2686          pink  269  3.6e-03  1.0e+00      56  BP                       cell communication
## 2687          pink  269  3.7e-03  1.0e+00      23  BP       instar larval or pupal development
## 2688          pink  269  3.8e-03  1.0e+00       3  BP positive regulation of photoreceptor cel
## 2689          pink  269  3.8e-03  1.0e+00      33  BP                        gamete generation
## 2690          pink  269  4.0e-03  1.0e+00      26  BP        intracellular signal transduction
## 2691          pink  269  4.1e-03  1.0e+00      69  MF                          protein binding
## 2692          pink  269  4.3e-03  1.0e+00      25  BP               post-embryonic development
## 2693          pink  269  4.3e-03  1.0e+00      18  BP              imaginal disc morphogenesis
## 2694          pink  269  4.4e-03  1.0e+00      82  BP                    developmental process
## 2695          pink  269  4.5e-03  1.0e+00      14  BP imaginal disc-derived wing morphogenesis
## 2696          pink  269  4.5e-03  1.0e+00      30  MF                purine nucleotide binding
## 2697          pink  269  4.7e-03  1.0e+00      71  BP       multicellular organism development
## 2698          pink  269  5.0e-03  1.0e+00      26  BP                 female gamete generation
## 2699          pink  269  5.0e-03  1.0e+00       4  CC       transcriptionally active chromatin
## 2700          pink  269  5.1e-03  1.0e+00       5  BP SCF-dependent proteasomal ubiquitin-depe
## 2701          pink  269  5.1e-03  1.0e+00       5  MF       protein methyltransferase activity
## 2702          pink  269  5.2e-03  1.0e+00      16  BP                    programmed cell death
## 2703          pink  269  5.2e-03  1.0e+00       2  BP positive regulation of hh target transcr
## 2704          pink  269  5.2e-03  1.0e+00       2  BP               histone monoubiquitination
## 2705          pink  269  5.2e-03  1.0e+00       2  BP determination of dorsal/ventral asymmetr
## 2706          pink  269  5.2e-03  1.0e+00       2  BP                     microtubule severing
## 2707          pink  269  5.2e-03  1.0e+00       2  MF     microtubule-severing ATPase activity
## 2708          pink  269  5.2e-03  1.0e+00      13  BP RNA splicing, via transesterification re
## 2709          pink  269  5.2e-03  1.0e+00      13  BP           mRNA splicing, via spliceosome
## 2710          pink  269  5.3e-03  1.0e+00      11  BP                       axis specification
## 2711          pink  269  5.3e-03  1.0e+00      21  BP cell morphogenesis involved in different
## 2712          pink  269  5.4e-03  1.0e+00      10  CC             transcription factor complex
## 2713          pink  269  5.6e-03  1.0e+00       9  MF       transcription coregulator activity
## 2714          pink  269  5.6e-03  1.0e+00       4  BP                        nuclear migration
## 2715          pink  269  5.7e-03  1.0e+00       3  BP            terminal region determination
## 2716          pink  269  5.7e-03  1.0e+00       3  BP positive regulation of protein ubiquitin
## 2717          pink  269  5.7e-03  1.0e+00       3  MF        steroid hormone receptor activity
## 2718          pink  269  5.9e-03  1.0e+00      23  BP                imaginal disc development
## 2719          pink  269  6.0e-03  1.0e+00      75  CC               protein-containing complex
## 2720          pink  269  6.0e-03  1.0e+00      27  BP                    germ cell development
## 2721          pink  269  6.0e-03  1.0e+00      14  BP                  wing disc morphogenesis
## 2722          pink  269  6.1e-03  1.0e+00       5  BP double-strand break repair via homologou
## 2723          pink  269  6.1e-03  1.0e+00       5  BP                   recombinational repair
## 2724          pink  269  6.1e-03  1.0e+00       5  BP               histone lysine methylation
## 2725          pink  269  6.3e-03  1.0e+00       4  BP            epithelial cell morphogenesis
## 2726          pink  269  6.7e-03  1.0e+00       6  CC        histone acetyltransferase complex
## 2727          pink  269  6.8e-03  1.0e+00       3  MF                nuclear receptor activity
## 2728          pink  269  6.8e-03  1.0e+00       8  BP    cellular response to hormone stimulus
## 2729          pink  269  7.0e-03  1.0e+00       4  BP       hormone-mediated signaling pathway
## 2730          pink  269  7.0e-03  1.0e+00       4  BP positive regulation of establishment of 
## 2731          pink  269  7.1e-03  1.0e+00       6  BP                      protein methylation
## 2732          pink  269  7.2e-03  1.0e+00       7  BP                        DNA recombination
## 2733          pink  269  7.7e-03  1.0e+00       2  BP       negative regulation of DNA binding
## 2734          pink  269  7.7e-03  1.0e+00       2  BP                  mesectoderm development
## 2735          pink  269  7.8e-03  1.0e+00      10  BP                     dendrite development
## 2736          pink  269  8.0e-03  1.0e+00       3  BP positive regulation of protein secretion
## 2737          pink  269  8.2e-03  1.0e+00      16  BP regulation of nervous system development
## 2738          pink  269  8.3e-03  1.0e+00       9  BP                      response to hormone
## 2739          pink  269  8.6e-03  1.0e+00       6  BP        dorsal/ventral axis specification
## 2740          pink  269  8.7e-03  1.0e+00       4  BP         positive regulation of secretion
## 2741          pink  269  8.8e-03  1.0e+00      20  BP            neuron projection development
## 2742          pink  269  9.0e-03  1.0e+00      18  BP       regulation of cell differentiation
## 2743          pink  269  9.0e-03  1.0e+00      13  BP                             RNA splicing
## 2744          pink  269  9.1e-03  1.0e+00       5  BP                        neuron remodeling
## 2745          pink  269  9.6e-03  1.0e+00      67  BP                protein metabolic process
## 2746          pink  269  9.8e-03  1.0e+00      13  BP                        apoptotic process
## 2747          pink  269  9.9e-03  1.0e+00       8  BP                      response to alcohol
## 2748           red  759  4.4e-31  3.7e-27     106  BP                   nervous system process
## 2749           red  759  1.9e-30  1.6e-26      61  BP  sensory perception of chemical stimulus
## 2750           red  759  3.8e-30  3.2e-26      46  BP detection of stimulus involved in sensor
## 2751           red  759  1.1e-29  9.6e-26     103  CC                          cell projection
## 2752           red  759  1.3e-29  1.1e-25      39  BP detection of chemical stimulus involved 
## 2753           red  759  2.5e-28  2.1e-24      43  BP           detection of chemical stimulus
## 2754           red  759  1.7e-27  1.4e-23      31  MF              olfactory receptor activity
## 2755           red  759  9.3e-27  7.8e-23      43  BP              sensory perception of smell
## 2756           red  759  2.8e-26  2.4e-22      31  BP detection of chemical stimulus involved 
## 2757           red  759  1.4e-23  1.2e-19      35  MF                          odorant binding
## 2758           red  759  1.7e-22  1.5e-18      27  CC                        dendrite membrane
## 2759           red  759  2.9e-19  2.4e-15      30  CC               neuron projection membrane
## 2760           red  759  4.1e-19  3.4e-15      27  BP                        phototransduction
## 2761           red  759  8.6e-19  7.3e-15      22  CC                               rhabdomere
## 2762           red  759  1.0e-18  8.5e-15      42  CC                                 dendrite
## 2763           red  759  2.3e-18  1.9e-14      67  MF              signaling receptor activity
## 2764           red  759  5.5e-18  4.6e-14      59  MF transmembrane signaling receptor activit
## 2765           red  759  1.2e-16  1.0e-12      23  BP      cellular response to light stimulus
## 2766           red  759  2.0e-16  1.7e-12      19  BP         phototransduction, visible light
## 2767           red  759  2.9e-16  2.4e-12      65  CC                        neuron projection
## 2768           red  759  1.1e-15  9.0e-12      31  CC                        cell leading edge
## 2769           red  759  3.7e-15  3.1e-11      20  BP               detection of visible light
## 2770           red  759  3.0e-14  2.6e-10      15  BP     rhodopsin mediated signaling pathway
## 2771           red  759  1.8e-13  1.5e-09      35  BP               response to light stimulus
## 2772           red  759  6.0e-13  5.0e-09     120  CC                          plasma membrane
## 2773           red  759  1.8e-12  1.5e-08      23  BP           cellular response to radiation
## 2774           red  759  6.6e-12  5.6e-08      12  BP deactivation of rhodopsin mediated signa
## 2775           red  759  9.4e-12  7.9e-08      15  BP     sensory perception of light stimulus
## 2776           red  759  1.9e-11  1.6e-07      12  BP regulation of rhodopsin mediated signali
## 2777           red  759  6.6e-11  5.5e-07      36  BP                    response to radiation
## 2778           red  759  7.0e-11  5.9e-07       9  CC                   inaD signaling complex
## 2779           red  759  1.1e-10  9.2e-07      13  BP                        visual perception
## 2780           red  759  2.1e-10  1.8e-06     123  CC                           cell periphery
## 2781           red  759  2.4e-09  2.0e-05      36  BP G protein-coupled receptor signaling pat
## 2782           red  759  3.5e-09  2.9e-05     195  CC                                 membrane
## 2783           red  759  6.4e-09  5.3e-05      12  BP regulation of G protein-coupled receptor
## 2784           red  759  9.4e-09  7.9e-05     117  CC           integral component of membrane
## 2785           red  759  1.2e-08  1.0e-04     118  CC          intrinsic component of membrane
## 2786           red  759  5.2e-08  4.3e-04       8  MF                   photoreceptor activity
## 2787           red  759  8.5e-08  7.2e-04      31  MF                     ion channel activity
## 2788           red  759  2.1e-07  1.8e-03     146  BP                       cell communication
## 2789           red  759  2.3e-07  1.9e-03       7  MF         AMPA glutamate receptor activity
## 2790           red  759  2.5e-07  2.1e-03      25  BP                                   mating
## 2791           red  759  2.8e-07  2.4e-03       8  CC                         sensory dendrite
## 2792           red  759  2.9e-07  2.4e-03      31  MF                         channel activity
## 2793           red  759  3.5e-07  2.9e-03      24  BP                          mating behavior
## 2794           red  759  4.8e-07  4.0e-03     142  BP                                signaling
## 2795           red  759  5.2e-07  4.4e-03      16  BP              sensory perception of sound
## 2796           red  759  6.0e-07  5.0e-03       7  CC          AMPA glutamate receptor complex
## 2797           red  759  6.4e-07  5.3e-03      21  CC                                   cilium
## 2798           red  759  7.1e-07  5.9e-03      11  MF              glutamate receptor activity
## 2799           red  759  9.8e-07  8.2e-03       6  CC                      rhabdomere membrane
## 2800           red  759  9.8e-07  8.2e-03       6  MF G protein-coupled photoreceptor activity
## 2801           red  759  2.8e-06  2.3e-02       6  BP              response to light intensity
## 2802           red  759  3.1e-06  2.6e-02      24  BP                    reproductive behavior
## 2803           red  759  3.8e-06  3.2e-02       5  BP                          detection of UV
## 2804           red  759  4.6e-06  3.8e-02     122  BP                      signal transduction
## 2805           red  759  5.5e-06  4.6e-02       8  BP                       retina homeostasis
## 2806           red  759  6.9e-06  5.8e-02      11  CC   extrinsic component of plasma membrane
## 2807           red  759  8.6e-06  7.2e-02      16  BP sensory perception of mechanical stimulu
## 2808           red  759  1.1e-05  9.6e-02       4  BP adaptation of rhodopsin mediated signali
## 2809           red  759  1.1e-05  9.6e-02       4  BP       protein localization to rhabdomere
## 2810           red  759  1.1e-05  9.6e-02       4  CC                  subrhabdomeral cisterna
## 2811           red  759  1.2e-05  1.0e-01      16  MF        ligand-gated ion channel activity
## 2812           red  759  1.3e-05  1.1e-01       5  CC                   rhabdomere microvillus
## 2813           red  759  1.5e-05  1.2e-01      13  CC                   cation channel complex
## 2814           red  759  1.9e-05  1.6e-01       9  MF   ionotropic glutamate receptor activity
## 2815           red  759  2.2e-05  1.8e-01      18  BP                    chemosensory behavior
## 2816           red  759  2.7e-05  2.2e-01      13  CC                         synaptic vesicle
## 2817           red  759  3.5e-05  2.9e-01      17  BP                       courtship behavior
## 2818           red  759  4.0e-05  3.4e-01       7  BP                              thermotaxis
## 2819           red  759  4.8e-05  4.1e-01      16  BP                     male mating behavior
## 2820           red  759  5.2e-05  4.4e-01      21  MF                  cation channel activity
## 2821           red  759  5.5e-05  4.6e-01       4  BP                    phototransduction, UV
## 2822           red  759  5.5e-05  4.6e-01       4  BP               metarhodopsin inactivation
## 2823           red  759  5.6e-05  4.7e-01      13  CC                         exocytic vesicle
## 2824           red  759  5.7e-05  4.8e-01       8  BP           clathrin-dependent endocytosis
## 2825           red  759  6.7e-05  5.7e-01      15  BP                  male courtship behavior
## 2826           red  759  7.3e-05  6.1e-01      45  CC    integral component of plasma membrane
## 2827           red  759  7.4e-05  6.2e-01      46  CC   intrinsic component of plasma membrane
## 2828           red  759  7.6e-05  6.4e-01      11  BP            receptor-mediated endocytosis
## 2829           red  759  8.0e-05  6.7e-01       9  MF                       calmodulin binding
## 2830           red  759  9.3e-05  7.8e-01      32  CC                                  synapse
## 2831           red  759  1.2e-04  1.0e+00      21  CC                               presynapse
## 2832           red  759  1.4e-04  1.0e+00      11  MF extracellular ligand-gated ion channel a
## 2833           red  759  1.4e-04  1.0e+00      14  BP                       olfactory behavior
## 2834           red  759  1.6e-04  1.0e+00       4  BP        bulk synaptic vesicle endocytosis
## 2835           red  759  1.7e-04  1.0e+00       9  BP               calcium-mediated signaling
## 2836           red  759  1.9e-04  1.0e+00       6  BP           photoreceptor cell maintenance
## 2837           red  759  2.0e-04  1.0e+00       3  BP               retinoid metabolic process
## 2838           red  759  2.0e-04  1.0e+00      15  BP   compound eye photoreceptor development
## 2839           red  759  2.2e-04  1.0e+00      14  CC                         receptor complex
## 2840           red  759  2.2e-04  1.0e+00      17  BP           photoreceptor cell development
## 2841           red  759  2.3e-04  1.0e+00       5  BP entrainment of circadian clock by photop
## 2842           red  759  3.0e-04  1.0e+00      21  BP                           adult behavior
## 2843           red  759  3.2e-04  1.0e+00       7  CC    ionotropic glutamate receptor complex
## 2844           red  759  3.2e-04  1.0e+00       9  BP              sensory perception of taste
## 2845           red  759  3.3e-04  1.0e+00      14  BP            synaptic vesicle localization
## 2846           red  759  3.3e-04  1.0e+00      15  BP       eye photoreceptor cell development
## 2847           red  759  3.5e-04  1.0e+00       4  BP              sequestering of calcium ion
## 2848           red  759  3.5e-04  1.0e+00       4  BP clathrin-dependent synaptic vesicle endo
## 2849           red  759  3.5e-04  1.0e+00       4  CC             smooth endoplasmic reticulum
## 2850           red  759  3.7e-04  1.0e+00       5  BP                           photoperiodism
## 2851           red  759  3.7e-04  1.0e+00       5  BP detection of light stimulus involved in 
## 2852           red  759  3.7e-04  1.0e+00       5  CC                              microvillus
## 2853           red  759  5.2e-04  1.0e+00      15  BP                      cilium organization
## 2854           red  759  5.5e-04  1.0e+00       7  MF                  taste receptor activity
## 2855           red  759  5.8e-04  1.0e+00      12  MF       neurotransmitter receptor activity
## 2856           red  759  6.0e-04  1.0e+00      13  CC                        secretory vesicle
## 2857           red  759  6.7e-04  1.0e+00       4  BP regulation of clathrin-dependent endocyt
## 2858           red  759  7.4e-04  1.0e+00       6  BP   cyclic nucleotide biosynthetic process
## 2859           red  759  7.4e-04  1.0e+00       6  BP           entrainment of circadian clock
## 2860           red  759  7.4e-04  1.0e+00       6  BP                    response to pheromone
## 2861           red  759  8.6e-04  1.0e+00       5  BP            transmission of nerve impulse
## 2862           red  759  8.6e-04  1.0e+00      13  BP               synaptic vesicle transport
## 2863           red  759  9.0e-04  1.0e+00       7  BP          regulation of protein secretion
## 2864           red  759  1.0e-03  1.0e+00      30  BP                       synaptic signaling
## 2865           red  759  1.0e-03  1.0e+00      14  CC        transmembrane transporter complex
## 2866           red  759  1.2e-03  1.0e+00      13  CC                        transport vesicle
## 2867           red  759  1.2e-03  1.0e+00       5  BP detection of chemical stimulus involved 
## 2868           red  759  1.2e-03  1.0e+00      56  BP        intracellular signal transduction
## 2869           red  759  1.3e-03  1.0e+00      16  BP                   synaptic vesicle cycle
## 2870           red  759  1.4e-03  1.0e+00      13  BP                adult locomotory behavior
## 2871           red  759  1.4e-03  1.0e+00      22  BP                              endocytosis
## 2872           red  759  1.6e-03  1.0e+00      11  BP                     regulated exocytosis
## 2873           red  759  1.7e-03  1.0e+00       6  MF ATP-dependent microtubule motor activity
## 2874           red  759  1.7e-03  1.0e+00       5  BP                 synaptic vesicle priming
## 2875           red  759  1.7e-03  1.0e+00       5  BP                      R7 cell development
## 2876           red  759  1.8e-03  1.0e+00       3  BP nitric oxide mediated signal transductio
## 2877           red  759  1.8e-03  1.0e+00       3  BP negative regulation of protein secretion
## 2878           red  759  1.8e-03  1.0e+00       3  CC          rhabdomere microvillus membrane
## 2879           red  759  1.8e-03  1.0e+00       3  MF                        pheromone binding
## 2880           red  759  1.8e-03  1.0e+00       4  BP                cGMP biosynthetic process
## 2881           red  759  1.8e-03  1.0e+00      17  BP                           signal release
## 2882           red  759  1.9e-03  1.0e+00      29  BP           chemical synaptic transmission
## 2883           red  759  1.9e-03  1.0e+00      15  MF      G protein-coupled receptor activity
## 2884           red  759  2.1e-03  1.0e+00      13  BP                          cilium assembly
## 2885           red  759  2.2e-03  1.0e+00       8  MF                         hormone activity
## 2886           red  759  2.2e-03  1.0e+00      40  MF   ion transmembrane transporter activity
## 2887           red  759  2.3e-03  1.0e+00      10  BP          regulation of protein transport
## 2888           red  759  2.3e-03  1.0e+00       5  BP              rhodopsin metabolic process
## 2889           red  759  2.3e-03  1.0e+00       5  BP positive regulation of receptor-mediated
## 2890           red  759  2.6e-03  1.0e+00       8  BP                   rhabdomere development
## 2891           red  759  2.6e-03  1.0e+00       6  BP      cyclic nucleotide metabolic process
## 2892           red  759  2.9e-03  1.0e+00      11  CC          extrinsic component of membrane
## 2893           red  759  2.9e-03  1.0e+00      10  BP              synaptic vesicle exocytosis
## 2894           red  759  3.2e-03  1.0e+00      22  CC                                     axon
## 2895           red  759  3.3e-03  1.0e+00      10  BP regulation of establishment of protein l
## 2896           red  759  3.4e-03  1.0e+00       2  BP desensitization of G protein-coupled rec
## 2897           red  759  3.4e-03  1.0e+00       2  BP   activation of phospholipase C activity
## 2898           red  759  3.4e-03  1.0e+00       2  BP light-induced release of internally sequ
## 2899           red  759  3.4e-03  1.0e+00       2  BP                      response to acetate
## 2900           red  759  3.4e-03  1.0e+00       2  BP                     response to chlorate
## 2901           red  759  3.4e-03  1.0e+00       2  BP                retinal metabolic process
## 2902           red  759  3.4e-03  1.0e+00       2  BP             cellular response to acetate
## 2903           red  759  3.4e-03  1.0e+00       2  BP              cellular response to anoxia
## 2904           red  759  3.4e-03  1.0e+00       2  BP              dendritic spine maintenance
## 2905           red  759  3.4e-03  1.0e+00       2  BP positive regulation of protein polyubiqu
## 2906           red  759  3.4e-03  1.0e+00       2  BP            dense core granule exocytosis
## 2907           red  759  3.4e-03  1.0e+00       2  BP positive regulation of DNA topoisomerase
## 2908           red  759  3.4e-03  1.0e+00       2  MF                            opsin binding
## 2909           red  759  3.4e-03  1.0e+00       2  MF     light-activated ion channel activity
## 2910           red  759  3.4e-03  1.0e+00       2  MF extracellular phenylacetaldehyde-gated i
## 2911           red  759  3.5e-03  1.0e+00       3  BP positive regulation of clathrin-dependen
## 2912           red  759  3.5e-03  1.0e+00       3  CC                     microvillus membrane
## 2913           red  759  3.9e-03  1.0e+00       4  BP            regulation of ATPase activity
## 2914           red  759  3.9e-03  1.0e+00       4  MF               guanylate cyclase activity
## 2915           red  759  4.0e-03  1.0e+00       6  BP                         axoneme assembly
## 2916           red  759  4.0e-03  1.0e+00      13  BP       regulation of protein localization
## 2917           red  759  4.1e-03  1.0e+00      10  BP         calcium ion regulated exocytosis
## 2918           red  759  4.4e-03  1.0e+00      13  BP               neurotransmitter secretion
## 2919           red  759  4.8e-03  1.0e+00       6  CC                                  axoneme
## 2920           red  759  5.4e-03  1.0e+00       4  BP                  cellular response to UV
## 2921           red  759  5.4e-03  1.0e+00       4  BP                   cGMP metabolic process
## 2922           red  759  5.8e-03  1.0e+00       6  BP                   female mating behavior
## 2923           red  759  5.8e-03  1.0e+00       3  BP negative regulation of compound eye reti
## 2924           red  759  5.8e-03  1.0e+00       3  BP regulation of short-term neuronal synapt
## 2925           red  759  6.3e-03  1.0e+00      20  BP                      locomotory behavior
## 2926           red  759  6.8e-03  1.0e+00       9  BP                       tissue homeostasis
## 2927           red  759  6.8e-03  1.0e+00      42  BP               vesicle-mediated transport
## 2928           red  759  7.6e-03  1.0e+00       5  BP                          cilium movement
## 2929           red  759  8.5e-03  1.0e+00      21  BP       photoreceptor cell differentiation
## 2930           red  759  8.9e-03  1.0e+00       3  BP     imaginal disc-derived wing expansion
## 2931           red  759  8.9e-03  1.0e+00       3  MF            sweet taste receptor activity
## 2932           red  759  9.3e-03  1.0e+00       5  BP                      dosage compensation
## 2933           red  759  9.3e-03  1.0e+00       5  BP     synaptic transmission, glutamatergic
## 2934           red  759  9.3e-03  1.0e+00       4  BP dosage compensation by hyperactivation o
## 2935           red  759  9.3e-03  1.0e+00       4  BP                  cGMP-mediated signaling
## 2936           red  759  9.8e-03  1.0e+00       2  BP               conditioned taste aversion
## 2937           red  759  9.8e-03  1.0e+00       2  BP            regulation of cilium movement
## 2938           red  759  9.8e-03  1.0e+00       2  BP                  manganese ion transport
## 2939           red  759  9.8e-03  1.0e+00       2  BP gamma-aminobutyric acid signaling pathwa
## 2940           red  759  9.8e-03  1.0e+00       2  BP              synaptic vesicle maturation
## 2941           red  759  9.8e-03  1.0e+00       2  BP   negative regulation of protein binding
## 2942           red  759  9.8e-03  1.0e+00       2  BP          Rap protein signal transduction
## 2943           red  759  9.8e-03  1.0e+00       2  BP                SNARE complex disassembly
## 2944           red  759  9.8e-03  1.0e+00       2  CC G protein-coupled receptor heterodimeric
## 2945           red  759  9.8e-03  1.0e+00       2  CC                        dendrite terminus
## 2946           red  759  9.8e-03  1.0e+00       2  CC neuromuscular junction of somatic muscle
## 2947           red  759  9.8e-03  1.0e+00       2  CC      messenger ribonucleoprotein complex
## 2948           red  759  9.8e-03  1.0e+00       2  MF phosphatidylinositol phospholipase C act
## 2949           red  759  9.8e-03  1.0e+00       2  MF                 phospholipase C activity
## 2950           red  759  9.8e-03  1.0e+00       2  MF  soluble NSF attachment protein activity
## 2951           red  759  9.8e-03  1.0e+00       2  MF      ecdysis-triggering hormone activity
## 2952     royalblue   85  1.0e-10  8.4e-07      41  BP          regulation of metabolic process
## 2953     royalblue   85  2.0e-09  1.7e-05      33  BP            regulation of gene expression
## 2954     royalblue   85  1.1e-08  9.0e-05      28  BP             transcription, DNA-templated
## 2955     royalblue   85  1.5e-08  1.3e-04       8  CC        histone methyltransferase complex
## 2956     royalblue   85  2.7e-08  2.3e-04      35  BP regulation of cellular metabolic process
## 2957     royalblue   85  8.3e-08  7.0e-04      26  BP regulation of transcription, DNA-templat
## 2958     royalblue   85  9.0e-08  7.6e-04      14  BP                           cell migration
## 2959     royalblue   85  9.7e-08  8.1e-04      27  BP      regulation of RNA metabolic process
## 2960     royalblue   85  1.8e-07  1.5e-03      38  BP                          gene expression
## 2961     royalblue   85  2.9e-07  2.5e-03      20  BP                    germ cell development
## 2962     royalblue   85  3.3e-07  2.8e-03      33  BP                    RNA metabolic process
## 2963     royalblue   85  3.5e-07  2.9e-03      14  BP                            cell motility
## 2964     royalblue   85  6.7e-07  5.6e-03      22  BP                        gamete generation
## 2965     royalblue   85  1.2e-06  9.7e-03      65  CC                            intracellular
## 2966     royalblue   85  3.3e-06  2.8e-02      17  BP                                oogenesis
## 2967     royalblue   85  4.2e-06  3.5e-02      23  BP                       tissue development
## 2968     royalblue   85  5.4e-06  4.5e-02      24  BP    cellular protein modification process
## 2969     royalblue   85  1.0e-05  8.5e-02      17  BP                 female gamete generation
## 2970     royalblue   85  1.4e-05  1.2e-01      21  BP                   epithelium development
## 2971     royalblue   85  1.5e-05  1.2e-01      55  CC                  intracellular organelle
## 2972     royalblue   85  1.5e-05  1.3e-01      22  BP      multicellular organism reproduction
## 2973     royalblue   85  2.2e-05  1.9e-01      55  BP               cellular metabolic process
## 2974     royalblue   85  2.3e-05  2.0e-01       9  CC                      polytene chromosome
## 2975     royalblue   85  3.2e-05  2.7e-01      11  BP        ovarian follicle cell development
## 2976     royalblue   85  3.7e-05  3.1e-01      35  CC                                  nucleus
## 2977     royalblue   85  3.8e-05  3.2e-01      25  BP                         cell development
## 2978     royalblue   85  3.9e-05  3.3e-01      70  BP                         cellular process
## 2979     royalblue   85  4.5e-05  3.7e-01       5  BP               histone lysine methylation
## 2980     royalblue   85  4.7e-05  4.0e-01      34  BP nucleobase-containing compound metabolic
## 2981     royalblue   85  4.7e-05  4.0e-01      22  BP                     reproductive process
## 2982     royalblue   85  4.8e-05  4.0e-01       4  BP                histone H3-K4 methylation
## 2983     royalblue   85  5.1e-05  4.3e-01       8  BP                epithelial cell migration
## 2984     royalblue   85  5.3e-05  4.4e-01       8  BP                     epithelium migration
## 2985     royalblue   85  5.7e-05  4.8e-01      15  BP   negative regulation of gene expression
## 2986     royalblue   85  8.1e-05  6.8e-01       7  BP          ovarian follicle cell migration
## 2987     royalblue   85  8.6e-05  7.2e-01      11  BP              epithelial cell development
## 2988     royalblue   85  9.0e-05  7.5e-01      22  BP                             reproduction
## 2989     royalblue   85  1.0e-04  8.6e-01      22  BP                 animal organ development
## 2990     royalblue   85  1.2e-04  1.0e+00      35  CC               protein-containing complex
## 2991     royalblue   85  1.2e-04  1.0e+00       8  BP            ameboidal-type cell migration
## 2992     royalblue   85  1.3e-04  1.0e+00       5  BP                      histone methylation
## 2993     royalblue   85  1.4e-04  1.0e+00      11  BP                   chromatin organization
## 2994     royalblue   85  1.5e-04  1.0e+00      12  CC                              nucleoplasm
## 2995     royalblue   85  1.5e-04  1.0e+00      16  BP        regulation of signal transduction
## 2996     royalblue   85  1.8e-04  1.0e+00       6  BP                     chromatin remodeling
## 2997     royalblue   85  2.5e-04  1.0e+00      36  BP          cellular component organization
## 2998     royalblue   85  2.6e-04  1.0e+00      51  BP      nitrogen compound metabolic process
## 2999     royalblue   85  2.7e-04  1.0e+00       5  BP                      protein methylation
## 3000     royalblue   85  2.8e-04  1.0e+00      26  BP                     cell differentiation
## 3001     royalblue   85  4.0e-04  1.0e+00       6  BP           border follicle cell migration
## 3002     royalblue   85  4.2e-04  1.0e+00      14  BP                               locomotion
## 3003     royalblue   85  4.3e-04  1.0e+00      31  MF                          protein binding
## 3004     royalblue   85  4.9e-04  1.0e+00      26  BP                   organelle organization
## 3005     royalblue   85  5.5e-04  1.0e+00       4  MF       histone acetyltransferase activity
## 3006     royalblue   85  5.5e-04  1.0e+00      12  BP   positive regulation of gene expression
## 3007     royalblue   85  5.7e-04  1.0e+00       7  BP                     histone modification
## 3008     royalblue   85  6.2e-04  1.0e+00      11  BP                                   growth
## 3009     royalblue   85  6.2e-04  1.0e+00       7  BP regulation of gene expression, epigeneti
## 3010     royalblue   85  6.6e-04  1.0e+00      13  BP                  chromosome organization
## 3011     royalblue   85  8.0e-04  1.0e+00       7  BP                           gene silencing
## 3012     royalblue   85  8.2e-04  1.0e+00      34  BP         anatomical structure development
## 3013     royalblue   85  8.7e-04  1.0e+00      31  BP       multicellular organism development
## 3014     royalblue   85  1.1e-03  1.0e+00      31  BP                     biosynthetic process
## 3015     royalblue   85  1.2e-03  1.0e+00      12  CC                               chromosome
## 3016     royalblue   85  1.3e-03  1.0e+00       2  CC                             WASH complex
## 3017     royalblue   85  1.3e-03  1.0e+00      13  BP regulation of transcription by RNA polym
## 3018     royalblue   85  1.4e-03  1.0e+00      10  BP positive regulation of transcription, DN
## 3019     royalblue   85  1.4e-03  1.0e+00       4  MF proximal promoter DNA-binding transcript
## 3020     royalblue   85  1.5e-03  1.0e+00       3  BP                     glial cell migration
## 3021     royalblue   85  1.7e-03  1.0e+00       3  BP                   histone H3 acetylation
## 3022     royalblue   85  1.8e-03  1.0e+00      34  BP                    developmental process
## 3023     royalblue   85  1.8e-03  1.0e+00       3  BP                        nuclear migration
## 3024     royalblue   85  2.2e-03  1.0e+00       2  BP chromatin-mediated maintenance of transc
## 3025     royalblue   85  2.2e-03  1.0e+00      13  BP       transcription by RNA polymerase II
## 3026     royalblue   85  2.2e-03  1.0e+00       3  BP                      endocytic recycling
## 3027     royalblue   85  2.3e-03  1.0e+00       6  BP embryonic development via the syncytial 
## 3028     royalblue   85  2.3e-03  1.0e+00       4  MF             N-acetyltransferase activity
## 3029     royalblue   85  2.3e-03  1.0e+00      10  MF                           enzyme binding
## 3030     royalblue   85  2.5e-03  1.0e+00       8  BP               compound eye morphogenesis
## 3031     royalblue   85  2.5e-03  1.0e+00       4  CC        histone acetyltransferase complex
## 3032     royalblue   85  2.5e-03  1.0e+00       8  MF transcription regulatory region DNA bind
## 3033     royalblue   85  2.6e-03  1.0e+00       9  BP                    wing disc development
## 3034     royalblue   85  2.6e-03  1.0e+00       3  MF thiol-dependent ubiquitin-specific prote
## 3035     royalblue   85  2.6e-03  1.0e+00       3  MF thiol-dependent ubiquitinyl hydrolase ac
## 3036     royalblue   85  2.6e-03  1.0e+00       4  BP             regulation of cell migration
## 3037     royalblue   85  2.7e-03  1.0e+00       9  BP                 compound eye development
## 3038     royalblue   85  2.7e-03  1.0e+00       9  BP negative regulation of transcription, DN
## 3039     royalblue   85  2.8e-03  1.0e+00       4  MF               N-acyltransferase activity
## 3040     royalblue   85  2.9e-03  1.0e+00       5  BP                              methylation
## 3041     royalblue   85  2.9e-03  1.0e+00       4  MF DNA-binding transcription repressor acti
## 3042     royalblue   85  3.0e-03  1.0e+00      13  BP  cell surface receptor signaling pathway
## 3043     royalblue   85  3.0e-03  1.0e+00       8  BP                     regulation of growth
## 3044     royalblue   85  3.0e-03  1.0e+00      10  MF DNA-binding transcription factor activit
## 3045     royalblue   85  3.1e-03  1.0e+00      21  BP                      signal transduction
## 3046     royalblue   85  3.2e-03  1.0e+00      13  BP                         tube development
## 3047     royalblue   85  3.2e-03  1.0e+00       8  BP                        eye morphogenesis
## 3048     royalblue   85  3.3e-03  1.0e+00       2  BP growth of a germarium-derived egg chambe
## 3049     royalblue   85  3.3e-03  1.0e+00       2  CC                           MLL3/4 complex
## 3050     royalblue   85  3.3e-03  1.0e+00       3  BP                     nucleus localization
## 3051     royalblue   85  3.7e-03  1.0e+00       4  CC        protein acetyltransferase complex
## 3052     royalblue   85  3.9e-03  1.0e+00       3  BP cellular response to topologically incor
## 3053     royalblue   85  4.0e-03  1.0e+00       9  BP                          eye development
## 3054     royalblue   85  4.0e-03  1.0e+00       4  MF               acetyltransferase activity
## 3055     royalblue   85  4.1e-03  1.0e+00       3  BP                 protein deubiquitination
## 3056     royalblue   85  4.1e-03  1.0e+00       3  BP       negative regulation of proteolysis
## 3057     royalblue   85  4.2e-03  1.0e+00       6  BP embryo development ending in birth or eg
## 3058     royalblue   85  4.4e-03  1.0e+00       4  BP                              gliogenesis
## 3059     royalblue   85  4.4e-03  1.0e+00       4  BP          canonical Wnt signaling pathway
## 3060     royalblue   85  4.6e-03  1.0e+00       2  BP protein quality control for misfolded or
## 3061     royalblue   85  4.6e-03  1.0e+00      42  CC intracellular membrane-bounded organelle
## 3062     royalblue   85  4.7e-03  1.0e+00       3  CC                 polytene chromosome puff
## 3063     royalblue   85  4.8e-03  1.0e+00      10  BP                sensory organ development
## 3064     royalblue   85  5.0e-03  1.0e+00       3  BP regulation of photoreceptor cell differe
## 3065     royalblue   85  5.7e-03  1.0e+00       3  BP  regulation of epithelial cell migration
## 3066     royalblue   85  6.4e-03  1.0e+00      11  BP           morphogenesis of an epithelium
## 3067     royalblue   85  6.8e-03  1.0e+00       3  BP    positive regulation of cell migration
## 3068     royalblue   85  6.8e-03  1.0e+00       3  BP               dorsal appendage formation
## 3069     royalblue   85  6.9e-03  1.0e+00       2  BP  cell-cell adhesion mediated by cadherin
## 3070     royalblue   85  7.0e-03  1.0e+00       6  BP establishment or maintenance of cell pol
## 3071     royalblue   85  7.1e-03  1.0e+00       7  MF DNA-binding transcription factor activit
## 3072     royalblue   85  7.8e-03  1.0e+00       2  BP                regulation of DNA binding
## 3073     royalblue   85  7.9e-03  1.0e+00       1  BP                 suture of dorsal opening
## 3074     royalblue   85  7.9e-03  1.0e+00       1  BP                   sternite morphogenesis
## 3075     royalblue   85  7.9e-03  1.0e+00       1  BP             mystery cell differentiation
## 3076     royalblue   85  7.9e-03  1.0e+00       1  BP                               defecation
## 3077     royalblue   85  7.9e-03  1.0e+00       1  BP conversion of ds siRNA to ss siRNA invol
## 3078     royalblue   85  7.9e-03  1.0e+00       1  BP               histone H3-K79 methylation
## 3079     royalblue   85  7.9e-03  1.0e+00       1  BP               nuclear cortical migration
## 3080     royalblue   85  7.9e-03  1.0e+00       1  BP               histone H3-K18 acetylation
## 3081     royalblue   85  7.9e-03  1.0e+00       1  BP                        R1/R6 development
## 3082     royalblue   85  7.9e-03  1.0e+00       1  BP  compound eye corneal lens morphogenesis
## 3083     royalblue   85  7.9e-03  1.0e+00       1  BP anterior mRNA localization involved in a
## 3084     royalblue   85  7.9e-03  1.0e+00       1  BP             histone H3-K36 dimethylation
## 3085     royalblue   85  7.9e-03  1.0e+00       1  BP protein localization to cell leading edg
## 3086     royalblue   85  7.9e-03  1.0e+00       1  BP positive regulation of receptor localiza
## 3087     royalblue   85  7.9e-03  1.0e+00       1  CC mitochondrial DNA-directed RNA polymeras
## 3088     royalblue   85  7.9e-03  1.0e+00       1  CC                      lateral cell cortex
## 3089     royalblue   85  7.9e-03  1.0e+00       1  MF mitochondrial promoter sequence-specific
## 3090     royalblue   85  7.9e-03  1.0e+00       1  MF UDP-glucose:glycoprotein glucosyltransfe
## 3091     royalblue   85  7.9e-03  1.0e+00       1  MF      elongation factor-2 kinase activity
## 3092     royalblue   85  7.9e-03  1.0e+00       1  MF histone methyltransferase activity (H3-K
## 3093     royalblue   85  7.9e-03  1.0e+00       1  MF DNA 5'-adenosine monophosphate hydrolase
## 3094     royalblue   85  7.9e-03  1.0e+00       1  MF histone acetyltransferase activity (H3-K
## 3095     royalblue   85  7.9e-03  1.0e+00       1  MF histone acetyltransferase activity (H3-K
## 3096     royalblue   85  7.9e-03  1.0e+00       1  MF single-strand break-containing DNA bindi
## 3097     royalblue   85  7.9e-03  1.0e+00       1  MF        histone methyltransferase binding
## 3098     royalblue   85  8.0e-03  1.0e+00       8  BP                     developmental growth
## 3099     royalblue   85  8.1e-03  1.0e+00       4  BP                      endosomal transport
## 3100     royalblue   85  8.3e-03  1.0e+00       7  BP proteolysis involved in cellular protein
## 3101     royalblue   85  8.4e-03  1.0e+00      22  BP                                signaling
## 3102     royalblue   85  8.8e-03  1.0e+00       2  BP   positive regulation of hippo signaling
## 3103     royalblue   85  9.0e-03  1.0e+00       7  BP       cellular protein catabolic process
## 3104     royalblue   85  9.1e-03  1.0e+00      27  BP                protein metabolic process
## 3105     royalblue   85  9.9e-03  1.0e+00       5  CC                            cell junction
## 3106        salmon  114  1.5e-04  1.0e+00       3  BP             anterograde axonal transport
## 3107        salmon  114  2.8e-04  1.0e+00       8  BP                         synapse assembly
## 3108        salmon  114  4.3e-04  1.0e+00       7  BP synaptic growth at neuromuscular junctio
## 3109        salmon  114  5.9e-04  1.0e+00       2  BP   anterograde synaptic vesicle transport
## 3110        salmon  114  8.9e-04  1.0e+00      22  BP               nervous system development
## 3111        salmon  114  1.6e-03  1.0e+00       9  BP imaginal disc-derived wing morphogenesis
## 3112        salmon  114  1.8e-03  1.0e+00       7  BP       neuromuscular junction development
## 3113        salmon  114  1.8e-03  1.0e+00       7  BP                     dendrite development
## 3114        salmon  114  2.0e-03  1.0e+00       9  BP                  wing disc morphogenesis
## 3115        salmon  114  2.2e-03  1.0e+00      36  BP       multicellular organism development
## 3116        salmon  114  2.4e-03  1.0e+00       9  BP                     synapse organization
## 3117        salmon  114  2.4e-03  1.0e+00       3  BP                         axonal transport
## 3118        salmon  114  2.7e-03  1.0e+00       2  BP   glycosphingolipid biosynthetic process
## 3119        salmon  114  2.8e-03  1.0e+00      16  BP               animal organ morphogenesis
## 3120        salmon  114  3.4e-03  1.0e+00       2  BP positive regulation of lamellipodium ass
## 3121        salmon  114  3.4e-03  1.0e+00       2  MF non-membrane spanning protein tyrosine k
## 3122        salmon  114  4.5e-03  1.0e+00       9  BP   post-embryonic appendage morphogenesis
## 3123        salmon  114  4.6e-03  1.0e+00       4  BP positive regulation of cell projection o
## 3124        salmon  114  5.2e-03  1.0e+00       2  BP      vesicle transport along microtubule
## 3125        salmon  114  5.4e-03  1.0e+00       9  BP imaginal disc-derived appendage morphoge
## 3126        salmon  114  5.7e-03  1.0e+00       9  BP                  appendage morphogenesis
## 3127        salmon  114  5.9e-03  1.0e+00      13  BP           morphogenesis of an epithelium
## 3128        salmon  114  6.0e-03  1.0e+00       3  MF         protein tyrosine kinase activity
## 3129        salmon  114  6.2e-03  1.0e+00       6  BP                   dendrite morphogenesis
## 3130        salmon  114  6.2e-03  1.0e+00       9  BP imaginal disc-derived appendage developm
## 3131        salmon  114  6.2e-03  1.0e+00       2  BP      retinal ganglion cell axon guidance
## 3132        salmon  114  7.3e-03  1.0e+00       2  BP                  axon target recognition
## 3133        salmon  114  8.0e-03  1.0e+00       3  BP                  axo-dendritic transport
## 3134        salmon  114  8.1e-03  1.0e+00       7  BP       regulation of developmental growth
## 3135        salmon  114  8.8e-03  1.0e+00      13  BP                       neuron development
## 3136       skyblue   44  5.9e-27  4.9e-23      18  CC                       cytosolic ribosome
## 3137       skyblue   44  1.2e-23  1.0e-19      17  BP                  cytoplasmic translation
## 3138       skyblue   44  7.7e-22  6.4e-18      18  CC                                 ribosome
## 3139       skyblue   44  5.7e-21  4.8e-17      17  MF       structural constituent of ribosome
## 3140       skyblue   44  1.2e-17  9.6e-14      21  BP                peptide metabolic process
## 3141       skyblue   44  3.1e-17  2.6e-13      19  BP                              translation
## 3142       skyblue   44  4.3e-15  3.6e-11      18  MF             structural molecule activity
## 3143       skyblue   44  3.0e-14  2.5e-10       9  CC        cytosolic small ribosomal subunit
## 3144       skyblue   44  3.6e-14  3.0e-10      19  CC                                  cytosol
## 3145       skyblue   44  1.1e-13  9.6e-10      18  CC                ribonucleoprotein complex
## 3146       skyblue   44  3.2e-13  2.7e-09       9  CC        cytosolic large ribosomal subunit
## 3147       skyblue   44  6.1e-12  5.1e-08       9  CC                  small ribosomal subunit
## 3148       skyblue   44  1.2e-10  1.0e-06       9  CC                  large ribosomal subunit
## 3149       skyblue   44  1.7e-09  1.4e-05      32  CC                                cytoplasm
## 3150       skyblue   44  9.9e-08  8.3e-04      25  CC               protein-containing complex
## 3151       skyblue   44  9.4e-07  7.9e-03      32  CC                  intracellular organelle
## 3152       skyblue   44  4.2e-06  3.5e-02      22  BP                     biosynthetic process
## 3153       skyblue   44  5.3e-06  4.4e-02      21  BP                          gene expression
## 3154       skyblue   44  9.0e-06  7.6e-02      34  CC                            intracellular
## 3155       skyblue   44  1.4e-05  1.2e-01       2  BP endonucleolytic cleavage to generate mat
## 3156       skyblue   44  3.1e-05  2.6e-01      30  BP               cellular metabolic process
## 3157       skyblue   44  3.3e-05  2.7e-01      20  BP                protein metabolic process
## 3158       skyblue   44  5.4e-05  4.5e-01       3  MF                         ribosome binding
## 3159       skyblue   44  2.9e-04  1.0e+00       2  BP endonucleolytic cleavage in ITS1 to sepa
## 3160       skyblue   44  7.2e-04  1.0e+00       5  BP                      ribosome biogenesis
## 3161       skyblue   44  8.1e-04  1.0e+00       3  BP                        ribosome assembly
## 3162       skyblue   44  8.1e-04  1.0e+00       3  MF        ribonucleoprotein complex binding
## 3163       skyblue   44  1.2e-03  1.0e+00       2  BP maturation of 5.8S rRNA from tricistroni
## 3164       skyblue   44  1.3e-03  1.0e+00       3  BP       ribosomal small subunit biogenesis
## 3165       skyblue   44  1.4e-03  1.0e+00       2  BP RNA phosphodiester bond hydrolysis, endo
## 3166       skyblue   44  1.6e-03  1.0e+00       2  BP     cleavage involved in rRNA processing
## 3167       skyblue   44  1.8e-03  1.0e+00       2  BP                  maturation of 5.8S rRNA
## 3168       skyblue   44  2.0e-03  1.0e+00       2  BP         ribosomal small subunit assembly
## 3169       skyblue   44  2.6e-03  1.0e+00       5  CC                       nuclear chromosome
## 3170       skyblue   44  3.8e-03  1.0e+00       1  BP                 rRNA export from nucleus
## 3171       skyblue   44  3.8e-03  1.0e+00       1  BP        S-adenosyl-L-methionine transport
## 3172       skyblue   44  3.8e-03  1.0e+00       1  BP positive regulation of translational fid
## 3173       skyblue   44  3.8e-03  1.0e+00       1  BP regulation of plasma membrane sterol dis
## 3174       skyblue   44  3.8e-03  1.0e+00       1  CC           cortical endoplasmic reticulum
## 3175       skyblue   44  3.8e-03  1.0e+00       1  MF S-adenosyl-L-methionine transmembrane tr
## 3176       skyblue   44  4.6e-03  1.0e+00       7  MF                              RNA binding
## 3177       skyblue   44  5.6e-03  1.0e+00       2  BP maturation of SSU-rRNA from tricistronic
## 3178       skyblue   44  5.6e-03  1.0e+00       2  BP                 translational elongation
## 3179       skyblue   44  6.1e-03  1.0e+00       4  CC                                nucleolus
## 3180       skyblue   44  6.4e-03  1.0e+00      33  BP                         cellular process
## 3181       skyblue   44  7.5e-03  1.0e+00       1  MF              fumarate hydratase activity
## 3182       skyblue   44  7.5e-03  1.0e+00       1  MF         inorganic diphosphatase activity
## 3183       skyblue   44  7.5e-03  1.0e+00       1  MF         phosphoglycerate kinase activity
## 3184       skyblue   44  7.5e-03  1.0e+00       1  MF oxidized purine nucleobase lesion DNA N-
## 3185       skyblue   44  7.5e-03  1.0e+00       1  MF            cysteine desulfurase activity
## 3186       skyblue   44  7.5e-03  1.0e+00       1  MF     large ribosomal subunit rRNA binding
## 3187       skyblue   44  9.4e-03  1.0e+00       2  BP                   maturation of SSU-rRNA
## 3188       skyblue   44  9.4e-03  1.0e+00       2  BP     regulation of hemocyte proliferation
## 3189       skyblue   44  9.4e-03  1.0e+00       2  MF                             rRNA binding
## 3190     steelblue   39  2.1e-06  1.8e-02       3  BP  protein localization to Golgi apparatus
## 3191     steelblue   39  3.5e-05  3.0e-01       2  BP     protein retention in Golgi apparatus
## 3192     steelblue   39  9.3e-05  7.8e-01      24  CC                                cytoplasm
## 3193     steelblue   39  1.2e-04  9.8e-01       2  BP                 perineurial glial growth
## 3194     steelblue   39  2.1e-04  1.0e+00       4  BP      regulation of Wnt signaling pathway
## 3195     steelblue   39  2.5e-04  1.0e+00       2  BP positive regulation of ion transmembrane
## 3196     steelblue   39  2.8e-04  1.0e+00       5  MF                       Ras GTPase binding
## 3197     steelblue   39  3.0e-04  1.0e+00       5  MF                     small GTPase binding
## 3198     steelblue   39  3.3e-04  1.0e+00       2  BP                        glial cell growth
## 3199     steelblue   39  4.1e-04  1.0e+00       3  BP positive regulation of Wnt signaling pat
## 3200     steelblue   39  4.6e-04  1.0e+00       5  MF                           GTPase binding
## 3201     steelblue   39  5.9e-04  1.0e+00       4  BP                    Wnt signaling pathway
## 3202     steelblue   39  6.6e-04  1.0e+00       9  BP        regulation of signal transduction
## 3203     steelblue   39  6.8e-04  1.0e+00       3  BP          Rho protein signal transduction
## 3204     steelblue   39  7.0e-04  1.0e+00       7  BP               regulation of localization
## 3205     steelblue   39  7.6e-04  1.0e+00       2  BP             protein targeting to vacuole
## 3206     steelblue   39  1.3e-03  1.0e+00       7  BP               vesicle-mediated transport
## 3207     steelblue   39  1.3e-03  1.0e+00       5  BP                  regulation of transport
## 3208     steelblue   39  1.5e-03  1.0e+00      16  MF                          protein binding
## 3209     steelblue   39  1.5e-03  1.0e+00       4  BP          Ras protein signal transduction
## 3210     steelblue   39  2.4e-03  1.0e+00       4  BP small GTPase mediated signal transductio
## 3211     steelblue   39  2.4e-03  1.0e+00       2  MF               channel regulator activity
## 3212     steelblue   39  2.4e-03  1.0e+00      28  CC                            intracellular
## 3213     steelblue   39  2.6e-03  1.0e+00       3  BP regulation of Ras protein signal transdu
## 3214     steelblue   39  2.7e-03  1.0e+00       3  MF                GTPase activator activity
## 3215     steelblue   39  3.4e-03  1.0e+00       3  BP regulation of small GTPase mediated sign
## 3216     steelblue   39  3.5e-03  1.0e+00       3  MF                       Rab GTPase binding
## 3217     steelblue   39  3.5e-03  1.0e+00       1  BP                 detection of calcium ion
## 3218     steelblue   39  3.5e-03  1.0e+00       1  BP activation of store-operated calcium cha
## 3219     steelblue   39  3.5e-03  1.0e+00       1  BP          locomotory exploration behavior
## 3220     steelblue   39  3.5e-03  1.0e+00       1  BP   negative regulation of gluconeogenesis
## 3221     steelblue   39  3.5e-03  1.0e+00       1  BP positive regulation of adenylate cyclase
## 3222     steelblue   39  3.5e-03  1.0e+00       1  BP positive regulation of protein desumoyla
## 3223     steelblue   39  3.5e-03  1.0e+00       1  BP positive regulation of potassium ion imp
## 3224     steelblue   39  3.5e-03  1.0e+00       1  BP negative regulation of non-canonical Wnt
## 3225     steelblue   39  3.5e-03  1.0e+00       1  BP positive regulation of sodium ion transm
## 3226     steelblue   39  3.5e-03  1.0e+00       1  MF      chloride channel inhibitor activity
## 3227     steelblue   39  3.5e-03  1.0e+00       1  MF     potassium channel inhibitor activity
## 3228     steelblue   39  3.5e-03  1.0e+00       1  MF      ubiquitin-dependent protein binding
## 3229     steelblue   39  3.6e-03  1.0e+00       3  BP                spermatid differentiation
## 3230     steelblue   39  3.7e-03  1.0e+00       6  MF                           enzyme binding
## 3231     steelblue   39  3.9e-03  1.0e+00       2  BP  retrograde transport, endosome to Golgi
## 3232     steelblue   39  4.0e-03  1.0e+00       4  BP                              endocytosis
## 3233     steelblue   39  4.5e-03  1.0e+00       2  MF Rho guanyl-nucleotide exchange factor ac
## 3234     steelblue   39  5.1e-03  1.0e+00       2  BP regulation of Rho protein signal transdu
## 3235     steelblue   39  5.3e-03  1.0e+00       4  MF   ubiquitin-protein transferase activity
## 3236     steelblue   39  5.5e-03  1.0e+00       2  BP  maintenance of protein location in cell
## 3237     steelblue   39  5.5e-03  1.0e+00       2  BP   regulation of hemocyte differentiation
## 3238     steelblue   39  6.5e-03  1.0e+00       4  MF ubiquitin-like protein transferase activ
## 3239     steelblue   39  7.0e-03  1.0e+00       1  BP          asparaginyl-tRNA aminoacylation
## 3240     steelblue   39  7.0e-03  1.0e+00       1  BP negative regulation of terminal cell fat
## 3241     steelblue   39  7.0e-03  1.0e+00       1  BP negative regulation of fusion cell fate 
## 3242     steelblue   39  7.0e-03  1.0e+00       1  BP negative regulation of inflammatory resp
## 3243     steelblue   39  7.0e-03  1.0e+00       1  BP                    oenocyte delamination
## 3244     steelblue   39  7.0e-03  1.0e+00       1  BP negative regulation of histone H2A K63-l
## 3245     steelblue   39  7.0e-03  1.0e+00       1  BP                     response to rotenone
## 3246     steelblue   39  7.0e-03  1.0e+00       1  BP ribosome-associated ubiquitin-dependent 
## 3247     steelblue   39  7.0e-03  1.0e+00       1  CC                              GID complex
## 3248     steelblue   39  7.0e-03  1.0e+00       1  CC                              RQC complex
## 3249     steelblue   39  7.0e-03  1.0e+00       1  MF          asparagine-tRNA ligase activity
## 3250     steelblue   39  7.0e-03  1.0e+00       1  MF calcium-induced calcium release activity
## 3251     steelblue   39  7.3e-03  1.0e+00       2  BP positive regulation of canonical Wnt sig
## 3252     steelblue   39  7.7e-03  1.0e+00       2  CC                                P granule
## 3253     steelblue   39  7.7e-03  1.0e+00       4  BP             muscle structure development
## 3254     steelblue   39  7.7e-03  1.0e+00       5  CC                      cytoplasmic vesicle
## 3255     steelblue   39  8.1e-03  1.0e+00       2  BP          maintenance of protein location
## 3256     steelblue   39  8.5e-03  1.0e+00       2  CC                               germ plasm
## 3257     steelblue   39  8.9e-03  1.0e+00       7  BP  cell surface receptor signaling pathway
## 3258           tan  123  1.1e-05  9.1e-02      17  BP               vesicle-mediated transport
## 3259           tan  123  3.9e-05  3.2e-01       4  BP          ommochrome biosynthetic process
## 3260           tan  123  3.9e-05  3.2e-01       4  BP     ocellus pigment biosynthetic process
## 3261           tan  123  4.6e-04  1.0e+00       3  CC    AP-type membrane coat adaptor complex
## 3262           tan  123  5.7e-04  1.0e+00       4  BP         eye pigment biosynthetic process
## 3263           tan  123  6.1e-04  1.0e+00       6  BP                      endosomal transport
## 3264           tan  123  6.4e-04  1.0e+00       2  CC                     AP-3 adaptor complex
## 3265           tan  123  7.0e-04  1.0e+00       4  BP            eye pigment metabolic process
## 3266           tan  123  1.1e-03  1.0e+00       5  CC  small nuclear ribonucleoprotein complex
## 3267           tan  123  1.1e-03  1.0e+00       3  BP                compound eye pigmentation
## 3268           tan  123  2.2e-03  1.0e+00       2  BP                          heart formation
## 3269           tan  123  2.7e-03  1.0e+00       3  BP  retrograde transport, endosome to Golgi
## 3270           tan  123  3.7e-03  1.0e+00       2  BP chromatin-mediated maintenance of transc
## 3271           tan  123  3.7e-03  1.0e+00       2  MF         nuclear hormone receptor binding
## 3272           tan  123  4.0e-03  1.0e+00       4  CC                             nuclear body
## 3273           tan  123  4.6e-03  1.0e+00       2  BP                   SNARE complex assembly
## 3274           tan  123  4.7e-03  1.0e+00       4  MF             protein transporter activity
## 3275           tan  123  5.6e-03  1.0e+00       2  BP     negative regulation of lipid storage
## 3276           tan  123  6.5e-03  1.0e+00       5  BP               developmental pigmentation
## 3277           tan  123  6.7e-03  1.0e+00      14  BP                  intracellular transport
## 3278           tan  123  7.1e-03  1.0e+00       7  BP                        secretion by cell
## 3279           tan  123  7.7e-03  1.0e+00       3  BP             meiotic spindle organization
## 3280           tan  123  8.9e-03  1.0e+00       3  BP              regulation of lipid storage
## 3281           tan  123  8.9e-03  1.0e+00      37  CC               protein-containing complex
## 3282     turquoise 3240  2.4e-47  2.0e-43     394  CC                     extracellular region
## 3283     turquoise 3240  3.7e-36  3.1e-32     233  CC                      extracellular space
## 3284     turquoise 3240  5.2e-31  4.3e-27     245  BP              oxidation-reduction process
## 3285     turquoise 3240  1.3e-30  1.1e-26     250  MF                  oxidoreductase activity
## 3286     turquoise 3240  2.2e-19  1.8e-15     161  MF                         cofactor binding
## 3287     turquoise 3240  6.1e-19  5.1e-15     107  BP                    response to bacterium
## 3288     turquoise 3240  1.7e-16  1.4e-12      81  MF                         iron ion binding
## 3289     turquoise 3240  7.3e-16  6.1e-12     152  BP                         defense response
## 3290     turquoise 3240  2.0e-15  1.7e-11     163  BP        carboxylic acid metabolic process
## 3291     turquoise 3240  3.1e-15  2.6e-11     168  BP           organic acid metabolic process
## 3292     turquoise 3240  6.4e-15  5.4e-11      71  MF                             heme binding
## 3293     turquoise 3240  1.6e-14  1.3e-10     112  BP           carbohydrate metabolic process
## 3294     turquoise 3240  1.7e-14  1.5e-10      40  BP defense response to Gram-positive bacter
## 3295     turquoise 3240  1.5e-13  1.2e-09      87  BP            defense response to bacterium
## 3296     turquoise 3240  1.6e-13  1.3e-09      80  MF oxidoreductase activity, acting on paire
## 3297     turquoise 3240  1.9e-13  1.6e-09     113  BP       defense response to other organism
## 3298     turquoise 3240  4.2e-12  3.5e-08      83  BP generation of precursor metabolites and 
## 3299     turquoise 3240  6.4e-12  5.3e-08      42  MF         endopeptidase inhibitor activity
## 3300     turquoise 3240  1.0e-11  8.7e-08      43  MF             peptidase inhibitor activity
## 3301     turquoise 3240  7.9e-10  6.6e-06      31  BP                      aerobic respiration
## 3302     turquoise 3240  9.1e-10  7.6e-06      45  BP defense response to Gram-negative bacter
## 3303     turquoise 3240  3.2e-09  2.7e-05      52  BP                     cellular respiration
## 3304     turquoise 3240  1.6e-08  1.4e-04      31  MF serine-type endopeptidase inhibitor acti
## 3305     turquoise 3240  3.5e-08  2.9e-04      46  MF                 receptor ligand activity
## 3306     turquoise 3240  6.2e-08  5.2e-04      33  CC                                myofibril
## 3307     turquoise 3240  7.2e-08  6.1e-04      28  BP           antibacterial humoral response
## 3308     turquoise 3240  7.6e-08  6.3e-04      17  BP aromatic amino acid family metabolic pro
## 3309     turquoise 3240  8.2e-08  6.9e-04      46  BP  cellular carbohydrate metabolic process
## 3310     turquoise 3240  1.8e-07  1.5e-03      11  BP               tyrosine metabolic process
## 3311     turquoise 3240  1.8e-07  1.5e-03     372  CC          intrinsic component of membrane
## 3312     turquoise 3240  2.2e-07  1.9e-03      19  BP                 tricarboxylic acid cycle
## 3313     turquoise 3240  3.4e-07  2.9e-03      36  CC          mitochondrial respiratory chain
## 3314     turquoise 3240  4.8e-07  4.0e-03      30  CC                                sarcomere
## 3315     turquoise 3240  5.0e-07  4.2e-03     364  CC           integral component of membrane
## 3316     turquoise 3240  5.8e-07  4.8e-03      19  BP                citrate metabolic process
## 3317     turquoise 3240  5.8e-07  4.8e-03     185  MF       transmembrane transporter activity
## 3318     turquoise 3240  7.4e-07  6.2e-03      89  MF               signaling receptor binding
## 3319     turquoise 3240  7.5e-07  6.3e-03      59  BP                  humoral immune response
## 3320     turquoise 3240  7.6e-07  6.4e-03      84  BP      purine nucleotide metabolic process
## 3321     turquoise 3240  1.1e-06  9.0e-03      86  BP       ribose phosphate metabolic process
## 3322     turquoise 3240  1.2e-06  9.9e-03      45  MF                enzyme inhibitor activity
## 3323     turquoise 3240  1.2e-06  1.0e-02      27  BP                       myofibril assembly
## 3324     turquoise 3240  1.4e-06  1.2e-02      33  BP                  muscle cell development
## 3325     turquoise 3240  1.9e-06  1.6e-02      21  BP             response to unfolded protein
## 3326     turquoise 3240  2.4e-06  2.0e-02      20  BP    cellular response to unfolded protein
## 3327     turquoise 3240  2.6e-06  2.2e-02      52  BP                    ATP metabolic process
## 3328     turquoise 3240  2.9e-06  2.4e-02      22  BP                    muscle system process
## 3329     turquoise 3240  3.0e-06  2.5e-02       9  BP             one-carbon metabolic process
## 3330     turquoise 3240  3.0e-06  2.5e-02      97  BP                          immune response
## 3331     turquoise 3240  3.2e-06  2.7e-02      82  BP         ribonucleotide metabolic process
## 3332     turquoise 3240  3.2e-06  2.7e-02      29  BP           neuropeptide signaling pathway
## 3333     turquoise 3240  3.9e-06  3.2e-02      86  CC                   mitochondrial membrane
## 3334     turquoise 3240  4.3e-06  3.6e-02      35  BP     respiratory electron transport chain
## 3335     turquoise 3240  4.7e-06  3.9e-02      36  BP                 electron transport chain
## 3336     turquoise 3240  6.6e-06  5.5e-02      72  CC             mitochondrial inner membrane
## 3337     turquoise 3240  6.6e-06  5.6e-02      99  BP   nucleoside phosphate metabolic process
## 3338     turquoise 3240  7.0e-06  5.8e-02     165  BP            response to organic substance
## 3339     turquoise 3240  7.1e-06  5.9e-02      98  BP             nucleotide metabolic process
## 3340     turquoise 3240  7.3e-06  6.1e-02      21  BP           triglyceride metabolic process
## 3341     turquoise 3240  7.3e-06  6.2e-02      23  BP cellular response to topologically incor
## 3342     turquoise 3240  7.4e-06  6.2e-02     210  CC                            mitochondrion
## 3343     turquoise 3240  8.0e-06  6.7e-02     204  MF                     transporter activity
## 3344     turquoise 3240  8.3e-06  7.0e-02      29  MF               electron transfer activity
## 3345     turquoise 3240  8.4e-06  7.0e-02      26  BP           carbohydrate catabolic process
## 3346     turquoise 3240  1.1e-05  9.6e-02      32  BP ATP synthesis coupled electron transport
## 3347     turquoise 3240  1.3e-05  1.1e-01     167  BP                  transmembrane transport
## 3348     turquoise 3240  1.3e-05  1.1e-01      19  BP negative regulation of peptidase activit
## 3349     turquoise 3240  1.4e-05  1.2e-01      35  MF oxidoreductase activity, acting on CH-OH
## 3350     turquoise 3240  1.4e-05  1.2e-01      70  BP    cellular amino acid metabolic process
## 3351     turquoise 3240  1.6e-05  1.3e-01      88  CC                   mitochondrial envelope
## 3352     turquoise 3240  1.7e-05  1.4e-01      84  BP                      cuticle development
## 3353     turquoise 3240  1.7e-05  1.4e-01      15  BP                        protein refolding
## 3354     turquoise 3240  1.9e-05  1.6e-01      25  BP cellular amino acid biosynthetic process
## 3355     turquoise 3240  2.0e-05  1.7e-01      35  BP                          lipid transport
## 3356     turquoise 3240  2.1e-05  1.7e-01      36  BP                oxidative phosphorylation
## 3357     turquoise 3240  2.1e-05  1.8e-01     142  MF   ion transmembrane transporter activity
## 3358     turquoise 3240  2.2e-05  1.8e-01      23  BP       chaperone-mediated protein folding
## 3359     turquoise 3240  2.2e-05  1.8e-01      23  MF              iron-sulfur cluster binding
## 3360     turquoise 3240  2.3e-05  1.9e-01      17  BP                       muscle contraction
## 3361     turquoise 3240  2.3e-05  1.9e-01      17  MF monocarboxylic acid transmembrane transp
## 3362     turquoise 3240  2.3e-05  2.0e-01      82  BP         open tracheal system development
## 3363     turquoise 3240  2.4e-05  2.0e-01      31  BP mitochondrial ATP synthesis coupled elec
## 3364     turquoise 3240  2.7e-05  2.2e-01      49  BP           antimicrobial humoral response
## 3365     turquoise 3240  2.8e-05  2.3e-01      28  MF oxidoreductase activity, acting on the C
## 3366     turquoise 3240  3.0e-05  2.5e-01      16  CC collagen-containing extracellular matrix
## 3367     turquoise 3240  3.0e-05  2.5e-01      16  MF                   growth factor activity
## 3368     turquoise 3240  3.0e-05  2.5e-01      11  MF                misfolded protein binding
## 3369     turquoise 3240  3.3e-05  2.8e-01      20  BP                        heart contraction
## 3370     turquoise 3240  3.3e-05  2.8e-01      18  BP negative regulation of endopeptidase act
## 3371     turquoise 3240  4.2e-05  3.6e-01      24  BP                            heart process
## 3372     turquoise 3240  5.0e-05  4.2e-01      22  BP       negative regulation of proteolysis
## 3373     turquoise 3240  5.6e-05  4.7e-01      64  CC                     extracellular matrix
## 3374     turquoise 3240  5.7e-05  4.8e-01      20  BP                   sarcomere organization
## 3375     turquoise 3240  6.1e-05  5.1e-01      18  MF            neuropeptide hormone activity
## 3376     turquoise 3240  7.0e-05  5.8e-01      60  BP         chitin-based cuticle development
## 3377     turquoise 3240  8.1e-05  6.8e-01      71  MF       serine-type endopeptidase activity
## 3378     turquoise 3240  8.3e-05  7.0e-01      17  BP            monocarboxylic acid transport
## 3379     turquoise 3240  8.6e-05  7.2e-01      56  BP                   innate immune response
## 3380     turquoise 3240  8.6e-05  7.2e-01       8  MF      sterol-transporting ATPase activity
## 3381     turquoise 3240  8.8e-05  7.4e-01      11  MF protein binding involved in protein fold
## 3382     turquoise 3240  9.9e-05  8.3e-01      40  MF organic anion transmembrane transporter 
## 3383     turquoise 3240  1.0e-04  8.7e-01      14  BP chaperone cofactor-dependent protein ref
## 3384     turquoise 3240  1.4e-04  1.0e+00      13  MF         4 iron, 4 sulfur cluster binding
## 3385     turquoise 3240  1.5e-04  1.0e+00     147  BP                  lipid metabolic process
## 3386     turquoise 3240  1.5e-04  1.0e+00      49  BP              muscle cell differentiation
## 3387     turquoise 3240  1.8e-04  1.0e+00     100  BP   cellular response to organic substance
## 3388     turquoise 3240  1.9e-04  1.0e+00      28  BP                       response to fungus
## 3389     turquoise 3240  2.1e-04  1.0e+00       6  BP              carnitine metabolic process
## 3390     turquoise 3240  2.1e-04  1.0e+00       6  MF imaginal disc growth factor receptor bin
## 3391     turquoise 3240  2.1e-04  1.0e+00     109  MF                   endopeptidase activity
## 3392     turquoise 3240  2.2e-04  1.0e+00      19  BP               pyruvate metabolic process
## 3393     turquoise 3240  2.3e-04  1.0e+00      79  BP             muscle structure development
## 3394     turquoise 3240  2.3e-04  1.0e+00      24  MF               lipid transporter activity
## 3395     turquoise 3240  2.5e-04  1.0e+00      65  MF                         coenzyme binding
## 3396     turquoise 3240  2.5e-04  1.0e+00      20  BP               acyl-CoA metabolic process
## 3397     turquoise 3240  2.7e-04  1.0e+00      10  BP      glycosaminoglycan catabolic process
## 3398     turquoise 3240  2.7e-04  1.0e+00      10  BP     central nervous system morphogenesis
## 3399     turquoise 3240  2.8e-04  1.0e+00      15  CC                       cytochrome complex
## 3400     turquoise 3240  3.0e-04  1.0e+00      56  BP        sulfur compound metabolic process
## 3401     turquoise 3240  3.1e-04  1.0e+00       9  MF                    transaminase activity
## 3402     turquoise 3240  3.1e-04  1.0e+00       9  MF         cholesterol transporter activity
## 3403     turquoise 3240  3.2e-04  1.0e+00       7  BP                glycine metabolic process
## 3404     turquoise 3240  3.2e-04  1.0e+00       7  BP       tetrahydrofolate metabolic process
## 3405     turquoise 3240  3.2e-04  1.0e+00       7  MF                          quinone binding
## 3406     turquoise 3240  3.5e-04  1.0e+00      19  CC                                   Z disc
## 3407     turquoise 3240  3.5e-04  1.0e+00      19  CC                                   I band
## 3408     turquoise 3240  3.5e-04  1.0e+00      16  BP               glycogen metabolic process
## 3409     turquoise 3240  3.7e-04  1.0e+00      12  MF signaling pattern recognition receptor a
## 3410     turquoise 3240  3.7e-04  1.0e+00      12  MF                    peptidoglycan binding
## 3411     turquoise 3240  3.8e-04  1.0e+00     949  MF                       catalytic activity
## 3412     turquoise 3240  3.8e-04  1.0e+00      14  BP                'de novo' protein folding
## 3413     turquoise 3240  3.9e-04  1.0e+00     118  BP         cellular lipid metabolic process
## 3414     turquoise 3240  4.7e-04  1.0e+00     623  CC                                 membrane
## 3415     turquoise 3240  5.0e-04  1.0e+00      72  MF active transmembrane transporter activit
## 3416     turquoise 3240  5.6e-04  1.0e+00      41  MF                     carbohydrate binding
## 3417     turquoise 3240  5.9e-04  1.0e+00      34  MF hydrolase activity, hydrolyzing O-glycos
## 3418     turquoise 3240  6.0e-04  1.0e+00      74  MF           serine-type peptidase activity
## 3419     turquoise 3240  6.1e-04  1.0e+00      39  BP                 muscle organ development
## 3420     turquoise 3240  6.9e-04  1.0e+00      74  MF                serine hydrolase activity
## 3421     turquoise 3240  7.6e-04  1.0e+00      34  BP regulation of tube architecture, open tr
## 3422     turquoise 3240  8.5e-04  1.0e+00       5  MF hydroxymethyl-, formyl- and related tran
## 3423     turquoise 3240  9.3e-04  1.0e+00      47  BP             response to oxidative stress
## 3424     turquoise 3240  1.1e-03  1.0e+00     135  CC   intrinsic component of plasma membrane
## 3425     turquoise 3240  1.1e-03  1.0e+00       7  BP               glycerol metabolic process
## 3426     turquoise 3240  1.1e-03  1.0e+00       7  BP              striated muscle contraction
## 3427     turquoise 3240  1.1e-03  1.0e+00       7  BP         central nervous system formation
## 3428     turquoise 3240  1.1e-03  1.0e+00       7  BP   pericardial nephrocyte differentiation
## 3429     turquoise 3240  1.1e-03  1.0e+00       7  MF cysteine-type endopeptidase inhibitor ac
## 3430     turquoise 3240  1.1e-03  1.0e+00      18  MF                   monooxygenase activity
## 3431     turquoise 3240  1.1e-03  1.0e+00       6  BP regulation of melanization defense respo
## 3432     turquoise 3240  1.1e-03  1.0e+00       6  CC                                 cytoneme
## 3433     turquoise 3240  1.2e-03  1.0e+00      39  MF        structural constituent of cuticle
## 3434     turquoise 3240  1.3e-03  1.0e+00      21  MF G protein-coupled peptide receptor activ
## 3435     turquoise 3240  1.3e-03  1.0e+00      22  BP regulation of tube size, open tracheal s
## 3436     turquoise 3240  1.4e-03  1.0e+00      44  BP             fatty acid metabolic process
## 3437     turquoise 3240  1.5e-03  1.0e+00      42  BP               developmental pigmentation
## 3438     turquoise 3240  1.6e-03  1.0e+00      17  BP regulation of tube length, open tracheal
## 3439     turquoise 3240  1.7e-03  1.0e+00      19  MF                         hormone activity
## 3440     turquoise 3240  1.7e-03  1.0e+00       9  BP negative regulation of natural killer ce
## 3441     turquoise 3240  1.7e-03  1.0e+00       9  BP                    response to hyperoxia
## 3442     turquoise 3240  1.7e-03  1.0e+00       9  CC             respiratory chain complex IV
## 3443     turquoise 3240  1.7e-03  1.0e+00       9  MF N-acetylmuramoyl-L-alanine amidase activ
## 3444     turquoise 3240  1.7e-03  1.0e+00       9  MF              sterol transporter activity
## 3445     turquoise 3240  1.7e-03  1.0e+00       9  MF          peptidoglycan receptor activity
## 3446     turquoise 3240  1.8e-03  1.0e+00      20  BP                 carbohydrate homeostasis
## 3447     turquoise 3240  1.8e-03  1.0e+00      20  MF           neuropeptide receptor activity
## 3448     turquoise 3240  1.8e-03  1.0e+00      23  BP                     cuticle pigmentation
## 3449     turquoise 3240  1.8e-03  1.0e+00      11  BP chitin-based embryonic cuticle biosynthe
## 3450     turquoise 3240  1.8e-03  1.0e+00      11  CC                            lipid droplet
## 3451     turquoise 3240  2.2e-03  1.0e+00      16  CC                                   fusome
## 3452     turquoise 3240  2.2e-03  1.0e+00      12  BP         regulation of female receptivity
## 3453     turquoise 3240  2.3e-03  1.0e+00       8  BP          peptidoglycan metabolic process
## 3454     turquoise 3240  2.3e-03  1.0e+00       8  BP          peptidoglycan catabolic process
## 3455     turquoise 3240  2.3e-03  1.0e+00       8  BP        triglyceride biosynthetic process
## 3456     turquoise 3240  2.3e-03  1.0e+00       8  CC  tricarboxylic acid cycle enzyme complex
## 3457     turquoise 3240  2.4e-03  1.0e+00      18  BP                        muscle attachment
## 3458     turquoise 3240  2.4e-03  1.0e+00      18  BP   negative regulation of immune response
## 3459     turquoise 3240  2.4e-03  1.0e+00      18  CC mitochondrial respiratory chain complex 
## 3460     turquoise 3240  2.4e-03  1.0e+00      18  CC               NADH dehydrogenase complex
## 3461     turquoise 3240  2.4e-03  1.0e+00      22  MF                peptide receptor activity
## 3462     turquoise 3240  2.5e-03  1.0e+00      10  MF            cytochrome-c oxidase activity
## 3463     turquoise 3240  2.5e-03  1.0e+00      10  MF                        cytokine activity
## 3464     turquoise 3240  2.5e-03  1.0e+00      96  MF cation transmembrane transporter activit
## 3465     turquoise 3240  2.7e-03  1.0e+00      36  CC                               peroxisome
## 3466     turquoise 3240  2.9e-03  1.0e+00     137  BP                            metamorphosis
## 3467     turquoise 3240  3.0e-03  1.0e+00       7  BP            UDP-glucose metabolic process
## 3468     turquoise 3240  3.0e-03  1.0e+00       7  BP                    hemolymph coagulation
## 3469     turquoise 3240  3.0e-03  1.0e+00       7  BP regulation of female receptivity, post-m
## 3470     turquoise 3240  3.0e-03  1.0e+00      34  MF hydrolase activity, acting on glycosyl b
## 3471     turquoise 3240  3.1e-03  1.0e+00      15  MF              NADH dehydrogenase activity
## 3472     turquoise 3240  3.1e-03  1.0e+00      11  BP        decapentaplegic signaling pathway
## 3473     turquoise 3240  3.3e-03  1.0e+00     158  BP       instar larval or pupal development
## 3474     turquoise 3240  3.3e-03  1.0e+00      36  BP                     mesoderm development
## 3475     turquoise 3240  3.3e-03  1.0e+00      21  BP                 genital disc development
## 3476     turquoise 3240  3.3e-03  1.0e+00      17  MF                  sodium channel activity
## 3477     turquoise 3240  3.4e-03  1.0e+00      18  BP epithelial cell migration, open tracheal
## 3478     turquoise 3240  3.4e-03  1.0e+00       9  BP              antifungal humoral response
## 3479     turquoise 3240  3.5e-03  1.0e+00       4  BP              skeletal muscle contraction
## 3480     turquoise 3240  3.5e-03  1.0e+00       4  BP  pyrimidine nucleobase catabolic process
## 3481     turquoise 3240  3.5e-03  1.0e+00       4  BP                glycine catabolic process
## 3482     turquoise 3240  3.5e-03  1.0e+00       4  BP melanin biosynthetic process from tyrosi
## 3483     turquoise 3240  3.5e-03  1.0e+00       4  BP                       pyruvate transport
## 3484     turquoise 3240  3.5e-03  1.0e+00       4  BP                  labial disc development
## 3485     turquoise 3240  3.5e-03  1.0e+00       4  BP         tetrahydrofolate interconversion
## 3486     turquoise 3240  3.5e-03  1.0e+00       4  BP           pheromone biosynthetic process
## 3487     turquoise 3240  3.5e-03  1.0e+00       4  BP           carnitine biosynthetic process
## 3488     turquoise 3240  3.5e-03  1.0e+00       4  BP positive regulation of cysteine-type end
## 3489     turquoise 3240  3.5e-03  1.0e+00       4  CC             larval serum protein complex
## 3490     turquoise 3240  3.5e-03  1.0e+00       4  CC                 glycine cleavage complex
## 3491     turquoise 3240  3.5e-03  1.0e+00       4  CC phosphatidylinositol 3-kinase complex, c
## 3492     turquoise 3240  3.5e-03  1.0e+00       4  MF pyruvate transmembrane transporter activ
## 3493     turquoise 3240  3.5e-03  1.0e+00     131  BP                            ion transport
## 3494     turquoise 3240  3.5e-03  1.0e+00      38  BP                  lipid catabolic process
## 3495     turquoise 3240  3.6e-03  1.0e+00     128  CC    integral component of plasma membrane
## 3496     turquoise 3240  3.6e-03  1.0e+00      12  BP                    cholesterol transport
## 3497     turquoise 3240  3.6e-03  1.0e+00      12  MF transmembrane receptor protein tyrosine 
## 3498     turquoise 3240  3.6e-03  1.0e+00      12  MF                   carboxy-lyase activity
## 3499     turquoise 3240  3.6e-03  1.0e+00       6  CC mitochondrial proton-transporting ATP sy
## 3500     turquoise 3240  3.6e-03  1.0e+00       6  CC proton-transporting ATP synthase complex
## 3501     turquoise 3240  3.6e-03  1.0e+00       6  MF                             Toll binding
## 3502     turquoise 3240  3.6e-03  1.0e+00       6  MF gamma-aminobutyric acid:sodium symporter
## 3503     turquoise 3240  3.7e-03  1.0e+00      43  BP            regulation of immune response
## 3504     turquoise 3240  3.7e-03  1.0e+00      26  MF            metalloendopeptidase activity
## 3505     turquoise 3240  3.9e-03  1.0e+00      91  MF inorganic cation transmembrane transport
## 3506     turquoise 3240  4.0e-03  1.0e+00      29  MF structural constituent of chitin-based l
## 3507     turquoise 3240  4.0e-03  1.0e+00      13  BP     regulation of Toll signaling pathway
## 3508     turquoise 3240  4.0e-03  1.0e+00      13  BP                         sterol transport
## 3509     turquoise 3240  4.0e-03  1.0e+00      13  BP regulation of developmental pigmentation
## 3510     turquoise 3240  4.1e-03  1.0e+00       5  BP              galactose metabolic process
## 3511     turquoise 3240  4.1e-03  1.0e+00       5  BP             tryptophan metabolic process
## 3512     turquoise 3240  4.1e-03  1.0e+00       5  BP                    equator specification
## 3513     turquoise 3240  4.1e-03  1.0e+00       5  CC            striated muscle thin filament
## 3514     turquoise 3240  4.1e-03  1.0e+00       5  MF                       ubiquinone binding
## 3515     turquoise 3240  4.1e-03  1.0e+00     101  BP                  appendage morphogenesis
## 3516     turquoise 3240  4.3e-03  1.0e+00     102  BP imaginal disc-derived appendage developm
## 3517     turquoise 3240  4.3e-03  1.0e+00      35  BP   purine nucleotide biosynthetic process
## 3518     turquoise 3240  4.3e-03  1.0e+00      14  BP          regulation of heart contraction
## 3519     turquoise 3240  4.3e-03  1.0e+00      14  MF oxidoreductase activity, acting on the a
## 3520     turquoise 3240  4.4e-03  1.0e+00      10  MF     amino acid:sodium symporter activity
## 3521     turquoise 3240  4.4e-03  1.0e+00     100  BP imaginal disc-derived appendage morphoge
## 3522     turquoise 3240  4.7e-03  1.0e+00      18  BP                 ventral cord development
## 3523     turquoise 3240  4.7e-03  1.0e+00       8  BP   glycerol-3-phosphate metabolic process
## 3524     turquoise 3240  4.7e-03  1.0e+00       8  BP         regulation of muscle contraction
## 3525     turquoise 3240  4.7e-03  1.0e+00       8  BP        wing and notum subfield formation
## 3526     turquoise 3240  4.7e-03  1.0e+00       8  CC mitochondrial respiratory chain complex 
## 3527     turquoise 3240  4.7e-03  1.0e+00       8  MF                   polysaccharide binding
## 3528     turquoise 3240  4.7e-03  1.0e+00     144  BP                       tube morphogenesis
## 3529     turquoise 3240  5.2e-03  1.0e+00      32  MF sodium ion transmembrane transporter act
## 3530     turquoise 3240  5.2e-03  1.0e+00      35  MF structural constituent of chitin-based c
## 3531     turquoise 3240  5.3e-03  1.0e+00     134  BP            epithelial tube morphogenesis
## 3532     turquoise 3240  6.0e-03  1.0e+00      20  BP                   Toll signaling pathway
## 3533     turquoise 3240  6.2e-03  1.0e+00       9  MF                        lysozyme activity
## 3534     turquoise 3240  6.2e-03  1.0e+00       9  MF                 neurotransmitter binding
## 3535     turquoise 3240  6.3e-03  1.0e+00      14  BP mitochondrial electron transport, NADH t
## 3536     turquoise 3240  6.4e-03  1.0e+00      17  MF                glycosaminoglycan binding
## 3537     turquoise 3240  6.4e-03  1.0e+00      15  MF             carbon-carbon lyase activity
## 3538     turquoise 3240  6.4e-03  1.0e+00      15  MF              pyridoxal phosphate binding
## 3539     turquoise 3240  6.4e-03  1.0e+00       7  BP                 oenocyte differentiation
## 3540     turquoise 3240  6.4e-03  1.0e+00       7  BP mitochondrial electron transport, cytoch
## 3541     turquoise 3240  6.4e-03  1.0e+00       7  BP                     fatty acid transport
## 3542     turquoise 3240  6.4e-03  1.0e+00       7  BP wing disc anterior/posterior pattern for
## 3543     turquoise 3240  6.4e-03  1.0e+00       7  BP regulation of epithelial cell migration,
## 3544     turquoise 3240  6.5e-03  1.0e+00      16  BP    cellular response to oxidative stress
## 3545     turquoise 3240  6.5e-03  1.0e+00      16  BP                      glucose homeostasis
## 3546     turquoise 3240  6.7e-03  1.0e+00      31  BP                         response to heat
## 3547     turquoise 3240  6.9e-03  1.0e+00      23  BP                    hindgut morphogenesis
## 3548     turquoise 3240  7.0e-03  1.0e+00      59  MF metal ion transmembrane transporter acti
## 3549     turquoise 3240  7.2e-03  1.0e+00      33  MF ATPase activity, coupled to transmembran
## 3550     turquoise 3240  7.2e-03  1.0e+00     174  BP               post-embryonic development
## 3551     turquoise 3240  7.3e-03  1.0e+00      10  BP apposition of dorsal and ventral imagina
## 3552     turquoise 3240  7.3e-03  1.0e+00      10  MF     amino acid:cation symporter activity
## 3553     turquoise 3240  8.0e-03  1.0e+00      11  BP proximal/distal pattern formation, imagi
## 3554     turquoise 3240  8.0e-03  1.0e+00      11  BP       nucleotide-sugar metabolic process
## 3555     turquoise 3240  8.2e-03  1.0e+00      41  BP                  organic anion transport
## 3556     turquoise 3240  8.5e-03  1.0e+00      12  BP larval midgut cell programmed cell death
## 3557     turquoise 3240  8.7e-03  1.0e+00       6  BP  long-chain fatty acid metabolic process
## 3558     turquoise 3240  8.7e-03  1.0e+00       6  BP  determination of muscle attachment site
## 3559     turquoise 3240  8.7e-03  1.0e+00       6  BP regulation of transforming growth factor
## 3560     turquoise 3240  8.7e-03  1.0e+00       6  BP heat shock-mediated polytene chromosome 
## 3561     turquoise 3240  8.7e-03  1.0e+00       6  BP                epithelial tube formation
## 3562     turquoise 3240  8.7e-03  1.0e+00       6  MF       phosphatidate phosphatase activity
## 3563     turquoise 3240  8.7e-03  1.0e+00       6  MF gamma-aminobutyric acid transmembrane tr
## 3564     turquoise 3240  8.7e-03  1.0e+00      17  BP                 ATP biosynthetic process
## 3565     turquoise 3240  8.7e-03  1.0e+00      17  BP                cellular response to heat
## 3566     turquoise 3240  8.8e-03  1.0e+00      13  BP      glycosaminoglycan metabolic process
## 3567     turquoise 3240  8.8e-03  1.0e+00      13  CC integral component of mitochondrial memb
## 3568     turquoise 3240  8.8e-03  1.0e+00      13  MF transmembrane receptor protein kinase ac
## 3569     turquoise 3240  9.1e-03  1.0e+00     137  MF                       peptidase activity
## 3570     turquoise 3240  9.6e-03  1.0e+00      34  BP                            wound healing
## 3571         white   54  5.8e-30  4.8e-26      19  MF structural constituent of chitin-based c
## 3572         white   54  2.0e-29  1.7e-25      19  MF        structural constituent of cuticle
## 3573         white   54  7.4e-27  6.2e-23      20  CC                     extracellular matrix
## 3574         white   54  1.3e-25  1.1e-21      19  BP         chitin-based cuticle development
## 3575         white   54  2.2e-22  1.8e-18      19  BP                      cuticle development
## 3576         white   54  2.0e-20  1.6e-16      26  CC                     extracellular region
## 3577         white   54  1.6e-19  1.4e-15      13  MF structural constituent of chitin-based l
## 3578         white   54  2.1e-18  1.8e-14      20  MF             structural molecule activity
## 3579         white   54  5.0e-09  4.2e-05       5  BP                       body morphogenesis
## 3580         white   54  2.2e-05  1.8e-01      20  BP       multicellular organism development
## 3581         white   54  1.5e-04  1.0e+00      20  BP         anatomical structure development
## 3582         white   54  2.8e-04  1.0e+00      20  BP                    developmental process
## 3583         white   54  5.3e-03  1.0e+00       6  CC                      extracellular space
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{subset}\NormalTok{(screenTab, module }\OperatorTok{==}\StringTok{ "black"} \OperatorTok{&}\StringTok{ }\NormalTok{termName }\OperatorTok{==}\StringTok{ "nucleus"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   module size    p-val     Bonf nInTerm ont termName
## 1  black 2886 3.9e-160 3.2e-156    1095  CC  nucleus
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{write.csv}\NormalTok{(screenTab, }\DataTypeTok{file =} \StringTok{"../../processed/DESEQ/Coexpression/resampMods/GOEnrichTab_Resamp_simple.csv"}\NormalTok{,}
              \DataTypeTok{row.names =} \OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{notes-for-each-module}{%
\section{Notes for each module}\label{notes-for-each-module}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ff <-}\StringTok{ }\KeywordTok{unique}\NormalTok{(screenTab}\OperatorTok{$}\NormalTok{module)}

\NormalTok{ff.list <-}\StringTok{ }\KeywordTok{vector}\NormalTok{(}\DataTypeTok{mode=}\StringTok{'list'}\NormalTok{, }\DataTypeTok{length=}\KeywordTok{length}\NormalTok{(ff))}
\KeywordTok{names}\NormalTok{(ff.list)  <-}\StringTok{ }\NormalTok{ff}

\ControlFlowTok{for}\NormalTok{ (ii }\ControlFlowTok{in} \DecValTok{1}\OperatorTok{:}\KeywordTok{length}\NormalTok{(ff)) }
\NormalTok{  \{}
  
\NormalTok{  ffc <-}\StringTok{ }\KeywordTok{subset}\NormalTok{(screenTab,screenTab}\OperatorTok{==}\NormalTok{ff[ii] )}

\NormalTok{  ff.list[[ii]] <-}\StringTok{ }\NormalTok{ffc}
  
\NormalTok{\}}

\CommentTok{# Note: # substitute 'screenTab' for 'tab' to see full details}

\CommentTok{#lapply(ff.list[[ii]], head)}
\end{Highlighting}
\end{Shaded}

\hypertarget{black-m1}{%
\subsection{Black (M1)}\label{black-m1}}

\begin{itemize}
\tightlist
\item
  2956 genes in module; 771 GO terms across all GO categories
\item
  BP: nucleobase metabolic process, gene expression, RNA processing,
  organelle organization, cellular metabolic process, cellular process,
  female gamete generation, germ cell development, reproduction,
  reproductive process, cellular protein modification process, cell
  communication, cell cycle, etc
\end{itemize}

\hypertarget{blue-m2}{%
\subsection{Blue (M2)}\label{blue-m2}}

\begin{itemize}
\tightlist
\item
  1747 genes, 311 GO terms
\item
  BP: signalling, cell communication, nervous system process, ion
  transport, signal transduction, locomotion, neuron differentiation,
  neuron development, nervous system development, neurogenesis, cell
  morphogenesis, etc
\end{itemize}

\hypertarget{brown-m3}{%
\subsection{Brown (M3)}\label{brown-m3}}

\begin{itemize}
\tightlist
\item
  1084 genes, 116 terms
\item
  BP: proteolysis, chitin metabolic process, fatty acid metabolic
  process, transmembrane transpoort, lipid metabolic process, lipid
  biosynthetic process, cellular lipid metabolic process, sulfur
  compound metabolic process, etc
\end{itemize}

\hypertarget{cyan-m4}{%
\subsection{Cyan (M4)}\label{cyan-m4}}

\begin{itemize}
\tightlist
\item
  507 genes, 98 terms
\item
  BP: transmembrane transport, chitin metabolic process, ion transport,
  homeostatic process, heart development, fatty acid metabolic process,
  digestive tract development, response to toxic substance, etc \#\#
  Darkgreen (M5)
\item
  199 genes, 273 terms
\item
  BP: multicellular organism development, anatomical structure
  development, signaling, cell communication, developmental process,
  signal transduction, neurogenesis, cell differentiation, organ
  development, etc
\end{itemize}

\hypertarget{darkgrey-m6}{%
\subsection{Darkgrey (M6)}\label{darkgrey-m6}}

\begin{itemize}
\tightlist
\item
  309 genes, 85 terms
\item
  BP: signaling, regulation of signal transduction, regulation of Ras
  protein signal transduction, regulation of small GTPase mediated
  signal transduction, cell communication, etc
\end{itemize}

\hypertarget{darkred-m7}{%
\subsection{Darkred (M7)}\label{darkred-m7}}

\begin{itemize}
\tightlist
\item
  84 genes, 71 terms
\item
  BP: cytoplasmic translation, translation, peptide metabolic process,
  cytosolic ribosome activity, cellular process, nitrogen compound
  metabolic process, cellular metabolic process, biosynthetic process,
  gene expression, protein metabolic process, ribosome biogenesis, etc
\end{itemize}

\hypertarget{darkturquoise-m8}{%
\subsection{Darkturquoise (M8)}\label{darkturquoise-m8}}

\begin{itemize}
\tightlist
\item
  79 genes, 22 terms
\item
  BP: L-amino acid transport, amino acid transport, proteasome assembly,
  etc
\end{itemize}

\hypertarget{greenyellow-m9}{%
\subsection{Greenyellow (M9)}\label{greenyellow-m9}}

\begin{itemize}
\tightlist
\item
  165 genes, 109 terms
\item
  BP: proteasome-mediated ubiquitin-dependent, proteasome catabolic
  process, ubiquitin processes, ubiquitin-dependent protein catabolic
  process, modification-dependent protein catabolic process, proteolysis
  involved in cellular protein, cellular protein catabolic process,
  proteasomal ubiquitin-independent protein catabolic process, protein
  catabolic process, organonitrogen compound catabolic proces, catabolic
  process, proteolysis, organic substance catabolic process,
  protein-containing complex assembly, etc
\end{itemize}

\hypertarget{grey-m10}{%
\subsection{Grey (M10)}\label{grey-m10}}

\begin{itemize}
\tightlist
\item
  45 + 1011 genes, 373 terms
\end{itemize}

\hypertarget{lightcyan-m11}{%
\subsection{Lightcyan (M11)}\label{lightcyan-m11}}

\begin{itemize}
\tightlist
\item
  117 genes, 209 terms
\item
  BP: cellular component organization, multicellular organism
  development, organelle organization, cell development, anatomical
  structure development, regulation of metabolic process, developmental
  process, cellular process, cellular metabolic process, etc
\end{itemize}

\hypertarget{lightgreen-m12}{%
\subsection{Lightgreen (M12)}\label{lightgreen-m12}}

\begin{itemize}
\tightlist
\item
  107 genes, 47 terms
\item
  BP: mitochondrial translation, translation, peptide metabolic process,
  protein metabolic process, biosynthetic process, gene expression,
  cellular metabolic process, nitrogen compound metabolic process, etc
\end{itemize}

\hypertarget{lightyellow-m13}{%
\subsection{Lightyellow (M13)}\label{lightyellow-m13}}

\begin{itemize}
\tightlist
\item
  107 genes, 99 terms
\item
  BP: Golgi vesicle transport, ER to Golgi vesicle-mediated transport,
  translation, peptide metabolic process,intracellular transport,
  intracellular protein transport, protein transport, establishment of
  protein localization, vesicle-mediated transport, biosythetic process,
  cellular macromolecule localization, organic substance transport, etc
\end{itemize}

\hypertarget{pink-m14}{%
\subsection{Pink (M14)}\label{pink-m14}}

\begin{itemize}
\tightlist
\item
  382 genes, 163 terms
\item
  BP: regulation of gene expression, regulation of metabolic process,
  regulation of RNA metabolic process, gene expression, RNA metabolic
  process, nucleobase-containing compound metabolic process, regulation
  of cellular metabolic process, transcription, nitrogen compound
  metabolic process, cellular process, cellular metabolic process,
  cellular protein modification process, cell differentiation, nervous
  system development, organelle organization, signaling, signal
  transduction, anatomical structure development, anatomical structure
  morphogenesis, gamete generation, etc, etc.
\end{itemize}

\hypertarget{red-m15}{%
\subsection{Red (M15)}\label{red-m15}}

\begin{itemize}
\tightlist
\item
  787 genes, 204 terms
\item
  BP: nervous system process, sensory perception of chemical stimulus,
  detection of stimulus involved in sensory perception, detection of
  chemical stimulus, olfactory and sensory perception of smell, sensory
  perception of smell, phototransduction, response to light stimulus,
  cell communication, signaling, signal transduction, etc, etc.
\end{itemize}

\hypertarget{royalblue-m16}{%
\subsection{Royalblue (M16)}\label{royalblue-m16}}

\begin{itemize}
\tightlist
\item
  132 genes, 154 terms
\item
  BP: regulation of metabolic process, regulation of gene expression,
  transcription, regulation of cellular metabolic process, regulation of
  transcription, regulation of RNA metabolic process, gene expression,
  germ cell development, RNA metabolic process, cellular metabolic
  process, nitrogen compound metabolic process, oogenesis, female gamete
  generation, ovarian follicle, etc, etc.
\end{itemize}

\hypertarget{salmon-m17}{%
\subsection{Salmon (M17)}\label{salmon-m17}}

\begin{itemize}
\tightlist
\item
  157 genes, 30 terms
\item
  BP: nervous system development, neuron development, synapse assembly,
  synaptic growth at neuromuscular junction, anterograde axonal
  transport, multicellular organism development, animal organ
  morphogenesis, morphogenesis of an epithelium, etc.
\end{itemize}

\hypertarget{skyblue-m18}{%
\subsection{Skyblue (M18)}\label{skyblue-m18}}

\begin{itemize}
\tightlist
\item
  52 genes, 54 tems
\item
  BP: cytoplasmic translation, peptide metabolic process, translation,
  biosynthetic process, gene expression, cellular metabolic process,
  protein metabolic process, ribosome biogenesis, cellular process, etc.
\end{itemize}

\hypertarget{steelblue-m19}{%
\subsection{Steelblue (M19)}\label{steelblue-m19}}

\begin{itemize}
\tightlist
\item
  44 genes, 68 terms
\item
  BP: protein localization Golgi apparatus, protein retention in Golgi
  apparatus, Wnt signaling pathway, regulation of signal transduction,
  endocytosis, cell surface receptor signaling pathway, etc.
\end{itemize}

\hypertarget{tan-m20}{%
\subsection{Tan (M20)}\label{tan-m20}}

\begin{itemize}
\tightlist
\item
  163 genes, 24 terms
\item
  BP: vesicle-mediated transport, ommochrome biosynthetic process,
  ocellus pigment biosynthetic process, eye pigment biosynthetic
  process, endosomal transport, eye pigment metabolic process, compound
  eye pigmentation, developmental pigmentation, intracellular transport,
  secretion by cell, regulation of lipid storage, etc.
\end{itemize}

\hypertarget{turquoise-m21}{%
\subsection{Turquoise (M21)}\label{turquoise-m21}}

\begin{itemize}
\tightlist
\item
  3337 genes, 289 terms
\item
  BP: oxidation-reduction process, response to bacterium, defence
  response, carboxylic acid metabolic - process, organic acid metabolic
  process, carbohydrate metabolic process, defense response to
  bacterium, defense response to other organism, cellular respiration,
  cellular carbohydrate metabolic process, ribose phosphate metabolic
  process, immune response, nucleoside phosphate metabolic process,
  nucleoside phosphate metabolic process, response to organic substance,
  nucleotide metabolic process, transmembrane transport, chitin-based
  cuticle development, lipid metabolic process, cellular lipid metabolic
  process, response to oxidative stress, fatty acid metabolic process,
  developmental pigmentation, ion transport, etc, etc, etc.
\end{itemize}

\hypertarget{white-m22}{%
\subsection{White (M22)}\label{white-m22}}

\begin{itemize}
\tightlist
\item
  54 genes, 13 terms
\item
  BP: chitin-based cuticle development, cuticle development, body
  morphogenesis, multicellular organism development, anatomical
  structure development, developmental process.
\end{itemize}


\end{document}
