---
title: "Set_up_array_jobs"
author: "Enoch Ng'oma"
date: "4/24/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
```

## Step 1: Align

```{r}
ll<-character(length=0)

runs<-c(1,2,3)
for(run in runs)
{
  
cat("", file=paste("S01_Align_cmd_run",run,".txt",sep=""))
  
ffs <- list.files(paste("/home/kingeg/Projects/RNA_SEQ_BasePop/Data/run",run,"/",sep=""), pattern="fastq.gz$")

for(ii in 1:length(ffs))
{
st.s <- "hisat2 --dta -q -x indexes/bdgp6_tran/genome_tran -U " 

st.p <- paste("/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/run",run,"/",sep="")

outf <- paste("/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotsams/",strsplit(ffs[ii], ".", fixed=TRUE)[[1]][1],"_run",run,sep="")

cat(paste(st.s, st.p, ffs[ii], " -S ", outf,".sam",sep=""),"\n", file=paste("S01_Align_cmd_run",run,".txt",sep=""),append=TRUE)

ll<- c(ll,paste(strsplit(ffs[ii], ".", fixed=TRUE)[[1]][1],"_run",run,sep=""))

}

}
```

## Step 2a: Samtools sort

```{r}
cat("", file="S02a_samtools_all.txt")
for(ii in ll)
{
   cat(paste("samtools sort -@ 8 -o /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/",ii,".bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotsams/", ii,".sam ",sep=""), "\n", file=paste("S02a_samtools_all",".txt",sep=""),append=TRUE)

}
```

## Step 2b: Samtools merge - combine transcripts from each run
## Step not in protocol

```{r}
set <- data.frame('code'=unlist(lapply(strsplit(ll, split='_S',fixed=TRUE), function(x) x[1])), 'sample' = ll, stringsAsFactors=FALSE)

un.set <- unique(set$code)

cat("", file="S02b_samtools_merge.txt")

for(ii in un.set)
{
  samp.set <- subset(set, code==ii)
   cat(paste("samtools merge -r  /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/",ii,"_merged.bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[1],".bam ","/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[2],".bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[3],".bam",sep=""), "\n", file=paste("S02b_samtools_merge",".txt",sep=""),append=TRUE)

}
```

# Step 3: StringTie Assemble and quantify expressed genes and transcripts

```{r}
# Assemble transcripts for each sample

cat("", file="S03_stringtie_assemble.txt")
cat("", file="assembled_file_list.txt")

indir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/" 
outdir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotgtfs/" 

base.cmd <- "stringtie -p 8 -G /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/genes/dmel-all-r6.18.gtf -o "

for(ff in un.set) 
{
  samp.set <- paste(base.cmd,outdir,ff,"_assembled.gtf ",indir,ff,"_merged.bam",sep="")
  cat(samp.set,"\n", file="S03_stringtie_assemble.txt",append=TRUE)
   
# For stringtie merge 
  cat(paste(outdir,ff,"_assembled_pl.gtf",sep=""), "\n",file="mergelist.txt",append=TRUE)
}
```

## Step 4: StringTie merge transcripts from all samples

```{r}
# Shell step - see projectNotes.md:
#perl mstrg_prep.pl stringtie_merged.gtf > stringtie_merged_pl.gtf

# `sbatch stingtie_merge.sh`
# stringtie --merge -p 8 -G /MyGithub/BasePop_RNAseq/base_pop/processed/dotgtfs/HS-6_O_assembled.gtf -o stringtie_merged_pl.gtf
```

## Step 5: Compare transcripts to reference genome (optional)

```{r}
# gffcompare
# step skipped
```

# Step 6: StringTie - transcript abundances and table of counts

```{r}
cat("", file="../assembly/S06_abundances_pl.txt")


indir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/" 
outdir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/scripts/assembly/" 


base.cmd <- "stringtie -e -B -p 8 -G /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/perl_postPrep_S04/stringtie_merged_pl.gtf -o "

for(ff in un.set) 
{
  cmd.cut <- paste(base.cmd,outdir,ff,"/",ff,"_ballgown.gtf ", indir,ff,"_merged.bam",sep="")
  cat(cmd.cut,"\n", file="S06_abundances_pl.txt",append=TRUE)
   
}
```

# Create a csv containing sample ids

```{r}
# Make phenotype data file (csv)

# ref. above where we make un.set

idd <- as.data.frame(un.set)
colnames(idd)[colnames(idd)=="un.set"] <- "id"

ids <- idd %>% 
  mutate(id = id,
         treatment = str_split(id, "-", simplify = TRUE)[, 1])%>%
  mutate(replic = str_split(id, "_", simplify = TRUE)[, 1],
         tissue = str_split(id, "_", simplify = TRUE)[, 2]) %>%
  unite(col="treat_tissue", c("treatment", "tissue"), sep="_", remove=FALSE)

write.csv(ids, file="../../processed/describe_samples_pl.csv", row.names=FALSE, quote=FALSE)
```

# Create a csv containing sample ids for DESeq

```{r}
# Make phenotype data file (csv)

# ref. above where we make un.set

idd <- as.data.frame(un.set)
colnames(idd)[colnames(idd)=="un.set"] <- "id"

ids <- idd %>% 
  mutate(id = id,
         treatment = str_split(id, "-", simplify = TRUE)[, 1])%>%
  mutate(replic = str_split(id, "_", simplify = TRUE)[, 1],
         tissue = str_split(id, "_", simplify = TRUE)[, 2])

ids <- ids %>%
  mutate(replic = replic,
         replicate = str_split(replic, "-", simplify = TRUE)[, 2]) %>%
  unite(col="treat_tissue", c("treatment", "replicate"), sep="", remove = FALSE) %>%
  unite(col="treat_tissue", c("treat_tissue", "tissue"), sep ="_", remove = FALSE)

# reorder columns and assign first colum to row names
ids <- ids[,c(2,1,3:6)]

#rownames(ids) <- ids[,1]
#ids[,1] <- NULL

write.csv(ids, file="../../processed/describe_samplesDEpl.csv", row.names=FALSE, quote=FALSE)
```
