---
title: "Prepare expression data for batch effect analysis"
author: "Enoch Ng'oma"
date: "6/15/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(genefilter)
library(RSkittleBrewer)
library(tidyverse)
library(genefilter)
library(devtools)
#library(calibrate)
```

# Extract expression data: https://github.com/alyssafrazee/ballgown

# Load data

```{r}
# load data filtered for low abundance
load("../processed/results/ballG_all_results/bg_ballG_filt.Rda")
bg_ballG

## Qu.1 Both sva and RUVseq tutorials filter out non-expressed genes by requiring >5 reads in at least 2 samples for each gene
# i.e. `filter <- apply(zfGenes, 1, function(x) lengthx[x>5]) >2`
```

# gene expression

```{r}
# get expression data
gfpkm_all = gexpr(bg_ballG_filt)

#get gene names
Gindicies <- match(rownames(gfpkm_all), texpr(bg_ballG_filt, 'all')$gene_id)
Gnames <- texpr(bg_ballG_filt, "all")$t_name[Gindicies]

rownames(gfpkm_all) <- Gnames

# log2 transform expr values, rename columns for sva
log_gfpkm_all <- log2(gfpkm_all+1)
colnames(log_gfpkm_all) <- phenDat$id

# Extract row names as a column
#log_gfpkm_all <- cbind(GeneName = rownames(log_gfpkm_all), log_gfpkm_all)

# Write results to csv
save(log_gfpkm_all, file = "../processed/results/ballG_all_results/log_gfpkm_all.Rda")
write.csv(log_gfpkm_all, 
          "../processed/results/ballG_all_results/log_gfpkm_all.csv",
          row.names=TRUE)



# For svaseq - untransformed
colnames(gfpkm_all) <- phenDat$id

# Extract row names as a column
#nolog_gfpkm_all <- cbind(GeneName = rownames(gfpkm_all), gfpkm_all)
#rownames(gfpkm_all) <- NULL

save(gfpkm_all, file = "../processed/results/ballG_all_results/nolog_gfpkm_all.Rda")
write.csv(gfpkm_all, 
          "../processed/results/ballG_all_results/nolog_gfpkm_all.csv",
          row.names=TRUE)
```










