---
title: "WGCNA clustering"
author: "Enoch Ng'oma"
date: "2/5/2019"
output: 
  html_document:
    toc: true
    toc_depth: 2 #defaults to 3 if not specified
    toc_float: true
    number_sections: true
    fig_width: 7
    fig_height: 6
    fig_caption: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(limma)
library(tidyverse)
library(cowplot)
library(AnnotationDbi)
library(org.Dm.eg.db)

# Tutorials
# https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-02-networkConstr-man.pdf
# https://wikis.utexas.edu/display/bioiteam/Clustering+using+WGCNA
# https://bioconductor.org/packages/release/bioc/vignettes/CVE/inst/doc/WGCNA_from_TCGA_RNAseq.html

library(flashClust)
library(WGCNA)
library(ape)

options(stringsAsFactors = FALSE)
allowWGCNAThreads()
#disableWGCNAThreads()

set.seed(341072)
```

# Prep sample data for WGCNA

```{r}
# Sample info
phenDat.sva <- read.csv("../../processed/DESEQ/sampleDat_with_SV.csv")
rownames(phenDat.sva) <- phenDat.sva[,1]
phenDat.sva[,1] <- NULL
phenDat.sva$batch <- as.factor(phenDat.sva$batch)
phenDat.sva$treat_tissue <- paste(phenDat.sva$treatment,"_",phenDat.sva$tissue, sep="")

# Sample data to numerical format
phenDat.sva$trt_tis <- ifelse(phenDat.sva$treat_tissue == "C_B",1, 
                          ifelse(phenDat.sva$treat_tissue=="C_H",2,
                          ifelse(phenDat.sva$treat_tissue=="C_O",3,
                          ifelse(phenDat.sva$treat_tissue=="DR_B",4,
                          ifelse(phenDat.sva$treat_tissue=="DR_H",5,
                          ifelse(phenDat.sva$treat_tissue=="DR_O",6,
                          ifelse(phenDat.sva$treat_tissue=="HS_B",7,
                          ifelse(phenDat.sva$treat_tissue=="HS_H",8, 9))))))))

# phenDat.sva$trtm <- ifelse(phenDat.sva$treatment == "C",1, 
#                           ifelse(phenDat.sva$treatment=="DR",2,3))
# 
# phenDat.sva$tiss <- ifelse(phenDat.sva$tissue == "B",1, 
#                           ifelse(phenDat.sva$tissue=="H",2,3))  

#trt <- phenDat.sva[,6:8]
#names(trt) = c("Diet-Tissue","Diet","Tissue")

trt <- phenDat.sva[,"treat_tissue"]

# Indicators for pairwise treatment contrasts
trt_prw <- binarizeCategoricalVariable(trt, 
                                       includePairwise = TRUE,
                                       includeLevelVsAll = TRUE,
                                       dropUninformative = TRUE)
trt_prw <- data.frame(trt, trt_prw)
trt_prw <- trt_prw[,2:37]

```

# Transformed DESeq objects
# Models created in DESeq_DExpr.Rmd

```{r}
# Interaction model
load("../../processed/DESEQ/dds_deseq.02.Rda")

# log transform null DESeq model: design ~ SV1 + batch + treatment + tissue
rlog.int <- rlog(dds_deseq.02, blind = FALSE, fitType = "parametric")
vst.int <- varianceStabilizingTransformation(dds_deseq.02,
                                            blind = FALSE, fitType = "parametric")

```

# Remove batches

```{r}
# check variation in size factors
hist(dds_deseq.02$sizeFactor)

# vst data
mat.int <- assay(vst.int)
mat.int <- limma::removeBatchEffect(mat.int, batch = phenDat.sva$SV1, batch2 = phenDat.sva$batch)
assay(vst.int) <- mat.int
print(plotPCA(vst.int, intgroup=c("treatment", "tissue"), returnData=FALSE))

# rlog data
rmat.int <- assay(rlog.int)
rmat.int <- limma::removeBatchEffect(rmat.int, batch = phenDat.sva$SV1, batch2 = phenDat.sva$batch)
assay(rlog.int) <- rmat.int
print(plotPCA(rlog.int, intgroup=c("treatment", "tissue"), returnData=FALSE))

```

# WGCNA Preliminaries

```{r}

rlogint <- rmat.int

# Get data into WGCNA format - genes as columns, samples as rows
rlogint = as.data.frame(t(rlogint)) 
dim(rlogint) 

# Check gene outliers or too many missing values
outls = goodSamplesGenes(rlogint, verbose = 3)
outls$allOK 

# Cluster samples to check outliers
sampleTree = hclust(dist(rlogint), method = "average");

sizeGrWindow(12,9) # inches
par(mfrow=c(2,1))
plot(sampleTree, 
     main = "Sample clustering to detect outliers", 
     sub="", xlab="", 
     cex.axis = 0.6)

# Visualize how traits relate to sample dendrogram
# (white = low, red = high, grey = missing)
#traitColors = numbers2colors(trt_prw, signed = FALSE);

traitColors = numbers2colors(trt_prw, signed = FALSE)

# Plot the sample dendrogram and the colors underneath
pdf(file="../../plots/sample_clustering.pdf")
#par(mfrow=c(2,1))
#par(mar = c(1,4,2,1))
#par(cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5)
#par(cex = 1.5)
plotDendroAndColors(sampleTree, traitColors,
                   groupLabels = names(trt_prw),
                   main = "Sample dendrogram and diet heatmap",
                   cex.colorLabels = 0.6,
                   cex.dendroLabels = 0.6,
                   cex.rowText = 0.2)
dev.off()

save(rlogint, trt_prw, file ="../../processed/DESEQ/Coexpression/trt-01-dataInput.RData")
load("../../processed/DESEQ/Coexpression/trt-01-dataInput.RData")
```

# Select soft power

```{r}
# Choose a soft threshold power
powers = c(c(1:10), seq(from = 10, to = 30, by=1)) 
sft = pickSoftThreshold(rlogint, powerVector=powers, 
                        #corFnc = cor, # see FAQ point 3
                        networkType="signed",
                        verbose = 5)
  
sizeGrWindow(9,5)
pdf(file="../../plots/softThreshold.pdf")
par(mfrow= c(1,2)) # one row, two columns panel
cex1=0.9

# Scale-free topology fit index as a function of the sft power
plot(sft$fitIndices[,1], 
     -sign(sft$fitIndices[,3])*sft$fitIndices[,2], 
     xlab= "Soft Threshold (power)", 
     ylab="Scale Free Topology Model Fit, signed R^2", 
     type= "n", main= paste("Scale independence"))

text(sft$fitIndices[,1], 
     -sign(sft$fitIndices[,3])*sft$fitIndices[,2], 
     labels=powers, cex=cex1, col="red")

# line corresponding to R^2 cut-off of h
abline(h=0.90, col="red")

# Mean connectivity as a function of the sft power
plot(sft$fitIndices[,1], sft$fitIndices[,5], 
     xlab= "Soft Threshold (power)", 
     ylab="Mean Connectivity", 
     type="n", main = paste("Mean connectivity"))

text(sft$fitIndices[,1], sft$fitIndices[,5], 
     labels=powers, cex=cex1, col="red")
dev.off()
# Choose the lowest power for which the scale free topology index reaches 0.90
```

# Construct a gene co-expression matrix and generate modules
# softPower threshold explained here: https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html

```{r}
# Build the adjacency "correlation" matrix
pe <- sft$powerEstimate
adjc = adjacency(rlogint, power=pe, type="signed")

# Topological overlap matrix (TOM) of the adjacency matrix
TOM = TOMsimilarity(adjc, TOMType="signed") # specify network type

# Calculate the corresponding dissimilarity
dissTOM = 1-TOM

save(sft, adjc, TOM, dissTOM, file= "../../processed/DESEQ/Coexpression/adjc_TOM.RData")
load("../../processed/DESEQ/Coexpression/adjc_TOM.RData")

# Note: dissTOM calc minimizes effects of noise and spurious associations
```

# Detect modules

```{r}
# Hierarchical clustering of the genes based on TOM dissimilarity
geneTree = flashClust(as.dist(dissTOM), method = "average")

sizeGrWindow(12,9)
plot(geneTree, xlab = "", sub = "", cex = 0.3, 
     main = "Gene Clustering on TOM-based dissimilarity", 
     labels = FALSE, hang = 0.04)

# Set minimum number of genes in a module
minModuleSize = 30

# Module identification using dynamic tree cut algorithm
dynamicMods = cutreeDynamic(dendro = geneTree,
                            distM = dissTOM, 
                            deepSplit = 2,
                            pamRespectsDendro = FALSE, 
                            minClusterSize = minModuleSize)
                            
# Modules identified
table(dynamicMods)

# Assign module colours
dynamicColors= labels2colors(dynamicMods)

table(dynamicColors)

# Plot dendrogram of igengenes
plotDendroAndColors(geneTree, dynamicColors, 'Modules', 
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05, main='')

# Set the diagonal of the dissimilarity to NA 
diag(dissTOM) = NA;

```

# Quantify module similarity by eigengene correlation

```{r}
# Calculate eigengenes
MEList = moduleEigengenes(rlogint, 
                         colors= dynamicColors,
                         softPower = pe)
                         
MEs = MEList$eigengenes;
#MET <- order(cbind(MEs, trt_prw$DR_H.Vs.C_B))

plotEigengeneNetworks(MEs, "", marDendro = c(0,4,1,2), marHeatmap = c(3,4,1,2))


# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);

# Cluster module eigengenes
METree = flashClust(as.dist(MEDiss), method= "average")

sizeGrWindow(7,6)
plot(METree, main= "Clustering of module eigengenes", xlab= "", sub= "")

datME=moduleEigengenes(rlogint,dynamicColors)$eigengenes
signif(cor(datME, use="p"), 2)

pdf(file="../../plots/pairwise_eigengene_plots.pdf", width = 9, height = 6)
par(mfrow=c(1,1))
par(cex = 0.6)
mar=c(1,1)
plotMEpairs(datME)
dev.off()

# See detected modules and their sizes
cat("\n\nNetwork Modules\n")
moduleColors = dynamicColors
modtable <- as.data.frame(table(moduleColors))
modtable <- modtable[match(gsub("ME", "", names(MEs)), modtable[,1]),] # same order of eigengenes as in MEs (needed for heatmap labels) 
print(modtable)
write.table(modtable, "../../processed/DESEQ/Coexpression/dynamicModules_Sizes.txt", quote=F, row.names=F, sep="\t")


modtable = rename(modtable, Module = moduleColors, Genes = Freq)

modtable$Genes <- as.numeric(modtable$Genes)
p <- ggplot(modtable, aes(x=Module, y=Genes, fill="Module")) + 
  geom_bar(stat = "identity") + 
  geom_text(aes(label=Genes), 
            size=3.5)+ annotate("text", label = "right") +
  
  coord_flip()
p + theme(legend.position = "none") 
```

# Merge at a selected threshold

```{r}
# Set threshold for merging modules 
# Can choose no merge i.e. MEDissThres=0.0)
# Merge modules that are 85% similar
MEDissThres = 0.10 # 0.05 is identical Dynamic Tree Cut

# Call automatic merging function
merge = mergeCloseModules(rlogint, 
                          dynamicColors, 
                          cutHeight= MEDissThres, 
                          verbose =3)

mergedColors = merge$colors;
mergedMEs = merge$newMEs;

table(mergedColors)

# Plot dendrogram with module colors below it
pdf(file="../../plots/dynamic_merged_clusters.pdf", width = 5, height = 3)
par(mfrow=c(2,2))
par(cex = 0.6)
plotDendroAndColors(geneTree, 
                    cbind(dynamicColors, mergedColors), 
                    c("Dynamic colors", "Module colors"),
                    main = NULL,
                    dendroLabels= FALSE, hang=0.03, 
                    addGuide= TRUE, guideHang=0.05)
dev.off()

# Rename to moduleColors
table(moduleColors)

# See merged modules and their sizes
cat("\n\nNetwork Modules\n")
moduleColors = mergedColors
module.table_merged <- as.data.frame(table(moduleColors))
module.table_merged <- module.table_merged[match(gsub("ME", "", names(MEs)), module.table_merged[,1]),] # same order of eigengenes as in MEs (needed for heatmap labels) 
print(module.table_merged)
write.table(module.table_merged, "../../processed/DESEQ/Coexpression/mergedModules_Sizes.txt", quote=F, row.names=F, sep="\t")

MEcount <- length(colnames(MEs))
rownames(MEs) <- rownames(trt_prw)
write.table(MEs, "../../processed/DESEQ/Coexpression/mergedModuleEigengenes_colorLabel.txt", row.names=F, quote=F, sep="\t")
modNames = substring(names(MEs), 3)  # remove "ME" at the beginning of module eigengene names
  
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50));
moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs;

save(MEs, moduleLabels, moduleColors, geneTree,
      file= "../../processed/DESEQ/Network_modules_signed_merge.RData")
load("../../processed/DESEQ/Network_modules_signed_merge.RData")

module.table_merged <- na.omit(module.table_merged)

```

# Relate gene expression modules to diet treatment
# https://peterlangfelder.com/2018/11/25/working-with-categorical-variables/

```{r}
# Define number of genes and samples
nGenes = ncol(rlogint)
nSamples = nrow(rlogint)

# Recalculate module eigengens 
MEs0 = moduleEigengenes(rlogint, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
write.table(MEs0, file="../../processed/DESEQ/Coexpression/WCGNA_eigengenes.txt", sep="\t", row.names = FALSE)


# Quantify module-trait association
#moduleTraitCor = cor(MEs, trt_prw, use = "p");
#moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);
#moduleTraitPvalueFish <- corPvalueFisher(moduleTraitCor, nSamples)

aov.mods1 <- aov(as.matrix(MEs) ~ phenDat.sva$treat_tissue) 

aov.mods2 <- aov(as.matrix(MEs) ~ phenDat.sva$treatment* phenDat.sva$tissue)

aov.mod_single <- aov(MEs[,19] ~ phenDat.sva$treat_tissue)
TukeyHSD(aov.mod_single)