---
title: "Stringtie short protocol output for visualizing expression"
author: "Enoch Ng'oma"
date: "October 28, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(genefilter)
library(tidyverse)
library(cowplot)
library(reshape)

source("../color_map.R")
source("../ggplot_theme.R")
```


```{r}
phenDat <- read.csv("../../processed/describe_samples.csv",
                    stringsAsFactors = FALSE)

#bg_ballG <- ballgown(dataDir = "../../processed/S03_short_ballG/ballgown", samplePattern = "", pData = phenDat)

# Save .Rda
#save(bg_ballG, file = "../../processed/S03_short_ballG/bgOut_shortP.Rda")

```

# Gene abundances across libraries, colored by treatment

```{r}
load("../../processed/S03_short_ballG/bgOut_shortP.Rda")
bg_ballG

# Filter out low-abundance transcripts 
short_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)
short_filt

# Get fpkm values
tfpkm_short<-texpr(short_filt, meas="FPKM")
tfpkm_short <-log2(tfpkm_short+1)
colnames(tfpkm_short)<-phenDat$id
save(tfpkm_short, file = "../../processed/S03_short_ballG/tfpkm_short.Rda")

tbg<- as.data.frame(tfpkm_short) %>% gather(id, fpkm)
tbg$treatment <- rep(c("C","DR","HS"), each=18*nrow(tfpkm_short))
tbg$tissue <- rep(c("B","H","O"),each=nrow(tfpkm_short),6)
tbg$replicate <- rep(c(1,2,3,4,5,6),each=nrow(tfpkm_short)*3)
tbg$prep <- paste(tbg$treatment,tbg$replicate,sep="_")
tbg$trep <- paste(tbg$tissue,tbg$replicate,sep="_")

b1 <- ggplot(tbg, aes(x=as.factor(prep), y=fpkm,color=treatment)) +
  geom_boxplot(alpha=0.8) +
          theme(axis.title.y = element_text(size = 12),
                axis.title.x=element_blank(),
                axis.text.x  = element_text(angle=90, vjust=0.5),
                text = element_text(size = 12))+
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +       
  #scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Samples")+
  facet_grid(tissue~.)
b1 + theme(legend.position="none")
  
ggsave(filename = "../../plots/Expression_by_sample1.pdf",width = 5,height=5)

b2 <- ggplot(tbg, aes(x=as.factor(trep), y=fpkm,color=tissue)) +
  geom_boxplot(alpha=0.8) +
          theme(axis.title.y = element_text(size = 12),
                axis.title.x=element_blank(),
                axis.text.x  = element_text(angle=90, vjust=0.5),
                text = element_text(size = 12)) +
  scale_color_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  #scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Tissue") +
  #scale_fill_discrete(name="Samples")+
  facet_grid(treatment~.)
b2 + theme(legend.position="none")

#ggsave(filename = "../../plots/Expression_by_sample2.pdf",width = 5,height=5)
```

# Number of transcripts per gene

```{r}
# Load transcript gene indexes from ballgown objects
t.genes_table_short <- indexes(short_filt)$t2g

# Plot # of transcripts per gene
counts.bg <- table(t.genes_table_short[,"g_id"])
bg_one <- length(which(counts.bg==1))
bg_more <- length(which(counts.bg>1))
bg_max <- max(counts.bg)
hist(counts.bg, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", bg_one), paste("Genes with more than one transcript =", bg_more), paste("Max transcripts for single gene = ", bg_max))
legend("topright", legend_text, lty = NULL)
```

# Distribution of transcript sizes

```{r}
full_table.bg<-texpr(short_filt, 'all')
hist(full_table.bg$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")
```

## PCA for global patterns

```{r}
gfpkm_bg <-gexpr(short_filt)
gfpkm_bg <-log2(gfpkm_bg +1)
colnames(gfpkm_bg)<-phenDat$id
#save(gfpkm_bg, file = "../processed/results/ballG_all_results/gfpkm_bg.Rda")


# all samples
X<- t(scale(t(gfpkm_bg),center=TRUE,scale=FALSE))
sv<- svd(t(X))
U<- sv$u
V<- sv$v
D<- sv$d
Z<- t(X)%*%V

# Plot of ratio of variance explained by each PC
plot(sv$d^2/sum(sv$d^2), xlim = c(0, 5), type = "b", pch = 16, xlab = "principal components", 
    ylab = "variance explained")
plot(sv$d^2/sum(sv$d^2), type = "b", pch = 16, xlab = "principal components", 
    ylab = "variance explained")

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc.dat<- data.frame(phenDat$id, PC1 = Z[,1], PC2 = Z[,2], PC3 = Z[,3],PC4 =Z[,4])
s.id <- strsplit(as.character(pc.dat$phenDat.id), split="-",fixed=TRUE)
pc.dat$Diet <-unlist(lapply(s.id, function(x) x[1]))
pc.dat$Tissue <- unlist(lapply(strsplit(as.character(phenDat$id), 
                                        split="_", fixed=TRUE), function(x) x[2]))

# PC1 & PC2
a <- ggplot(pc.dat, aes(x=PC1, y=PC2, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
a 

ggsave(filename = "../../plots/PCA_overall.pdf",width = 5,height=5)

# PC1 & 3
p <- ggplot(pc.dat, aes(x=PC1, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
p 

# PC2 & PC3
q <- ggplot(pc.dat, aes(x=PC2, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
q

# PC3 & PC4
r <- ggplot(pc.dat, aes(x=PC3, y=PC4, col=Diet, pch=Tissue),loadings=TRUE) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
r
which(pc.dat$PC3 < -60)
pc.dat[22,]
pc.dat <- pc.dat[-22, ] 
```

# PCA by tissue

```{r}
# Subset phenotype data
Bphen <- subset(phenDat, phenDat$tissue=="B")
Hphen <- subset(phenDat, phenDat$tissue=="H")
Ophen <- subset(phenDat, phenDat$tissue=="O")

# Subset data by tissue
k <- as.data.frame(gfpkm_bg)
ll <- k[,c(grep("_B",names(k))), drop=FALSE]
mm <- k[,c(grep("_H",names(k))), drop=FALSE]
nn <- k[,c(grep("_O",names(k))), drop=FALSE]
```

# Body

```{r}
B <- t(scale(t(ll),center=TRUE,scale=FALSE))
sv <- svd(t(B))
U <- sv$u
V <- sv$v
D <- sv$d
Z <- t(B)%*%V

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc.dat<- data.frame(Bphen$id, PC1 = Z[,1], PC2 = Z[,2], PC3 = Z[,3],PC4 =Z[,4])
s.id <- strsplit(as.character(pc.dat$Bphen.id), split="-",fixed=TRUE)
pc.dat$Diet <-unlist(lapply(s.id, function(x) x[1]))
pc.dat$Tissue <- unlist(lapply(strsplit(as.character(Bphen$id), 
                                        split="_", fixed=TRUE), function(x) x[2]))

# PC1 & PC2
B <- ggplot(pc.dat, aes(x=PC1, y=PC2, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
B 

ggsave(filename = "../../plots/PCA_overall.pdf",width = 5,height=5)

# PC1 & 3
C <- ggplot(pc.dat, aes(x=PC1, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
C 

# PC2 & PC3
D <- ggplot(pc.dat, aes(x=PC2, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
D

# PC3 & PC4
E <- ggplot(pc.dat, aes(x=PC3, y=PC4, col=Diet, pch=Tissue),loadings=TRUE) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
E
```

# Head
```{r}
H <- t(scale(t(mm),center=TRUE,scale=FALSE))
sv <- svd(t(H))
U <- sv$u
V <- sv$v
D <- sv$d
Z <- t(H)%*%V

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc.dat<- data.frame(Hphen$id, PC1 = Z[,1], PC2 = Z[,2], PC3 = Z[,3],PC4 =Z[,4])
s.id <- strsplit(as.character(pc.dat$Hphen.id), split="-",fixed=TRUE)
pc.dat$Diet <-unlist(lapply(s.id, function(x) x[1]))
pc.dat$Tissue <- unlist(lapply(strsplit(as.character(Hphen$id), 
                                        split="_", fixed=TRUE), function(x) x[2]))

# PC1 & PC2
Hh <- ggplot(pc.dat, aes(x=PC1, y=PC2, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Hh 

ggsave(filename = "../../plots/PCA_overall.pdf",width = 5,height=5)

# PC1 & 3
Hi <- ggplot(pc.dat, aes(x=PC1, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Hi 

# PC2 & PC3
Hj <- ggplot(pc.dat, aes(x=PC2, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Hj

# PC3 & PC4
Hk <- ggplot(pc.dat, aes(x=PC3, y=PC4, col=Diet, pch=Tissue),loadings=TRUE) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Hk
```

# Ovary
```{r}
O <- t(scale(t(nn),center=TRUE,scale=FALSE))
sv <- svd(t(O))
U <- sv$u
V <- sv$v
D <- sv$d
Z <- t(O)%*%V

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc.dat<- data.frame(Ophen$id, PC1 = Z[,1], PC2 = Z[,2], PC3 = Z[,3],PC4 =Z[,4])
s.id <- strsplit(as.character(pc.dat$Ophen.id), split="-",fixed=TRUE)
pc.dat$Diet <-unlist(lapply(s.id, function(x) x[1]))
pc.dat$Tissue <- unlist(lapply(strsplit(as.character(Ophen$id), 
                                        split="_", fixed=TRUE), function(x) x[2]))

# PC1 & PC2
Oo <- ggplot(pc.dat, aes(x=PC1, y=PC2, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Oo 

ggsave(filename = "../../plots/PCA_overall.pdf",width = 5,height=5)

# PC1 & 3
Op <- ggplot(pc.dat, aes(x=PC1, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Op 

# PC2 & PC3
Oq <- ggplot(pc.dat, aes(x=PC2, y=PC3, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Oq

# PC3 & PC4
Or <- ggplot(pc.dat, aes(x=PC3, y=PC4, col=Diet, pch=Tissue),loadings=TRUE) + 
        geom_point(alpha=0.8, size=3) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 16)) +
        theme(text = element_text(size = 12)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.5),
              axis.line.y = element_line(color="black", size = 0.5))
Or
```

