---
title: "Use stringtie short protocol output to analyze DE"
author: "Enoch Ng'oma"
date: "October 28, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(genefilter)
library(tidyverse)
library(cowplot)
library(reshape)
```


```{r}
phenDat <- read.csv("../../processed/describe_samples.csv",
                    stringsAsFactors = FALSE)

bg_ballG <- ballgown(dataDir = "../../processed/S03_short_ballG/ballgown", samplePattern = "", pData = phenDat)

# Save .Rda
save(bg_ballG, file = "../../processed/S03_short_ballG/bgOut_shortP.Rda")

```

# Gene abundances across libraries, colored by treatment

```{r}
load("../../processed/S03_short_ballG/bgOut_shortP.Rda")
bg_ballG

# Filter out low-abundance transcripts 
short_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)
short_filt

# Get fpkm values
tfpkm_short<-texpr(short_filt, meas="FPKM")
tfpkm_short <-log2(tfpkm_short+1)
colnames(tfpkm_short)<-phenDat$id
save(tfpkm_short, file = "../../processed/S03_short_ballG/tfpkm_short.Rda")

tbg<- as.data.frame(tfpkm_short) %>% gather(id, fpkm)
tbg$treatment <- rep(c("C","DR","HS"), each=18*nrow(tfpkm_short))
tbg$tissue <- rep(c("B","H","O"),each=nrow(tfpkm_short),6)
tbg$replicate <- rep(c(1,2,3,4,5,6),each=nrow(tfpkm_short)*3)
tbg$prep <- paste(tbg$treatment,tbg$replicate,sep="_")
tbg$trep <- paste(tbg$tissue,tbg$replicate,sep="_")

b1 <- ggplot(tbg, aes(x=as.factor(prep), y=fpkm,color=treatment)) +
  geom_boxplot(alpha=0.9) +
          theme(axis.title.y = element_text(size = 24),
                axis.title.x=element_blank(),
                axis.text.x  = element_text(angle=90, vjust=0.5),
                text = element_text(size = 20))+
                
  scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Samples")+
  facet_grid(tissue~.)
b1 + theme(legend.position="none")
  
ggsave(filename = "../../plots/Expression_by_sample1.pdf",width = 5,height=5)

b2 <- ggplot(tbg, aes(x=as.factor(trep), y=fpkm,color=tissue)) +
  geom_boxplot(alpha=0.9) +
          theme(axis.title.y = element_text(size = 24),
                axis.title.x=element_blank(),
                axis.text.x  = element_text(angle=90, vjust=0.5),
                text = element_text(size = 20)) +
  scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Tissue") +
  #scale_fill_discrete(name="Samples")+
  facet_grid(treatment~.)
b2 + theme(legend.position="none")

ggsave(filename = "../../plots/Expression_by_sample2.pdf",width = 5,height=5)
```

# Number of transcripts per gene

```{r}
# Load transcript gene indexes from ballgown objects
t.genes_table_short <- indexes(short_filt)$t2g

# Plot # of transcripts per gene
counts.bg <- table(t.genes_table_short[,"g_id"])
bg_one <- length(which(counts.bg==1))
bg_more <- length(which(counts.bg>1))
bg_max <- max(counts.bg)
hist(counts.bg, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", bg_one), paste("Genes with more than one transcript =", bg_more), paste("Max transcripts for single gene = ", bg_max))
legend("topright", legend_text, lty = NULL)
```

# Distribution of transcript sizes

```{r}
full_table.bg<-texpr(short_filt, 'all')
hist(full_table.bg$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")
```

# PCA

```{r}
# all samples
X<- t(scale(t(gfpkm_bg),center=TRUE,scale=FALSE))
sv<- svd(t(X))
U<- sv$u
V<- sv$v
D<- sv$d
Z<- t(X)%*%V

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc.dat<- data.frame(phenDat$id, PC1 = Z[,1], PC2 = Z[,2])

pc.dat$Diet <- unlist(lapply(strsplit(as.character(pc.dat$phenDat.id), split="-",fixed=TRUE),function(x) x[1]))

## make figure with ggplot2
library(ggthemes)
#e <- rep(c('B','H','O'),each=18)
name <- phenDat$tissue
a <- ggplot(pc.dat, aes(x=PC1, y=PC2, col=Diet, label=name)) + 
        geom_point() + 
        theme_bw() +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.6),
              axis.line.y = element_line(color="black", size = 0.6))
a + col 
#a + geom_label(label = rownames(phenDat$tissue))+
#a+geom_text()
```


