---
title: "ballgown_run"
author: "Enoch Ng'oma"
date: "5/2/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Differential expression analysis protocol

```{r}
#R
#load R version R version 3.2.2
#load R packagees:
library(ballgown)
library(genefilter)
library(tidyverse)
library(devtools)
library(cowplot)
library(reshape)


phenDat <- read.csv("describe_samples.csv", stringsAsFactors = FALSE)
```

```{r}
heads_phen <- subset(phenDat, phenDat$tissue=="H")
soma_phen <- subset(phenDat, phenDat$tissue=="B")
ovaries_phen <- subset(phenDat, phenDat$tissue=="O")
bg_ballG_heads <- ballgown(dataDir = "../processed/ballG/heads", samplePattern = "", pData = heads_phen)
bg_ballG_soma <- ballgown(dataDir = "../processed/ballG/soma", samplePattern = "", pData = soma_phen)
bg_ballG_ovaries <- ballgown(dataDir = "../processed/ballG/ovaries", samplePattern = "", pData = ovaries_phen)

# Dispplay a decsription of each tissue
bg_ballG_heads
bg_ballG_soma
bg_ballG_ovaries

# Save .Rda
save(bg_ballG_heads, file = "../processed/ballG/results/bg_ballG_heads.Rda")
save(bg_ballG_soma, file = "../processed/ballG/results/bg_ballG_soma.Rda")
save(bg_ballG_ovaries, file = "../processed/ballG/results/bg_ballG_ovaries.Rda")
```


```{r}
load("../processed/ballG/results/bg_ballG_heads.Rda")
load("../processed/ballG/results/bg_ballG_soma.Rda")
load("../processed/ballG/results/bg_ballG_ovaries.Rda")
```

# Filter to remove low-abundance genes (i.e. those with variance across samples of <=1)

```{r}
bg_heads_filt <- subset(bg_ballG_heads, "rowVars(texpr(bg_ballG_heads)) >1", genomesubset=TRUE)
bg_soma_filt <- subset(bg_ballG_soma, "rowVars(texpr(bg_ballG_soma)) >1", genomesubset=TRUE)
bg_ovaries_filt <- subset(bg_ballG_ovaries, "rowVars(texpr(bg_ballG_ovaries)) >1", genomesubset=TRUE)

# See counts for each tissue
bg_heads_filt
bg_soma_filt
bg_ovaries_filt
```

# Identify transcripts that show significant differences between groups

```{r}
results_heads_transcripts <- stattest(bg_heads_filt, feature="transcript",
        covariate="treatment", meas="FPKM") 
#which(results_heads_transcripts$qval<=0.05)
#hds.qval <- results_heads_transcripts[which(results_heads_transcripts$qval<=0.05),]

results_soma_transcripts <- stattest(bg_soma_filt, feature="transcript",
        covariate="treatment", meas="FPKM")
#which(results_soma_transcripts$qval<=0.05) 
#soma.qval <- results_soma_transcripts[which(results_soma_transcripts$qval<=0.05),]
  
results_ovaries_transcripts <- stattest(bg_ovaries_filt, feature="transcript",
        covariate="treatment", meas="FPKM")
#which(results_ovaries_transcripts$qval<=0.05) 
#ova.qval <- results_ovaries_transcripts[which(results_ovaries_transcripts$qval<=0.05),]
```

# Identify genes that show statistically significant differences between groups

```{r}
results_heads_genes <- stattest(bg_heads_filt, feature="gene", 
                                covariate="treatment",  meas="FPKM")
which(results_heads_genes$qval<0.05)

results_soma_genes <- stattest(bg_soma_filt, feature="gene", 
                               covariate="treatment",  meas="FPKM")
which(results_soma_transcripts$qval<0.05)

results_ovaries_genes <- stattest(bg_ovaries_filt, feature="gene", 
                                  covariate="treatment",  meas="FPKM")
which(results_ovaries_transcripts$qval<0.05)
```

# Add gene names and gene IDs to the results_transcripts data frame

```{r}
results_heads_transcripts <- data.frame(geneNames=ballgown::transcriptNames(bg_heads_filt), geneIDs=ballgown::geneIDs(bg_heads_filt), results_heads_transcripts)

results_soma_transcripts <- data.frame(geneNames=ballgown::transcriptNames(bg_soma_filt), geneIDs=ballgown::geneIDs(bg_soma_filt), results_soma_transcripts)

results_ovaries_transcripts <- data.frame(geneNames=ballgown::transcriptNames(bg_ovaries_filt), geneIDs=ballgown::geneIDs(bg_ovaries_filt), results_ovaries_transcripts)

full_table.hd<-texpr(bg_heads_filt, 'all')
full_table.sm<-texpr(bg_soma_filt, 'all')
full_table.ov<-texpr(bg_ovaries_filt, 'all')
```

```{r}
# pull gene expression data frame from the ballgown object
#xx.heads <- as.data.frame(gexpr(bg_heads_filt))
#xx.soma <- as.data.frame(gexpr(bg_soma_filt))
#xx.ovaries <- as.data.frame(gexpr(bg_ovaries_filt))
```


# Sort results from smallest P value to largest

```{r}
results_heads_transcripts = arrange(results_heads_transcripts,pval) 
results_heads_genes = arrange(results_heads_genes,pval)

results_soma_transcripts = arrange(results_soma_transcripts,pval) 
results_soma_genes = arrange(results_soma_genes,pval)

results_ovaries_transcripts = arrange(results_ovaries_transcripts,pval) 
results_ovaries_genes = arrange(results_ovaries_genes,pval)

# Transfer transcript names fron stringtie (protocol does not output gene names)

indicies.hds <- match(results_heads_genes$id, texpr(bg_heads_filt, "all")$gene_id)
gene_names_heads <- texpr(bg_heads_filt, "all")$t_name[indicies.hds]
results_heads_genes$tname <- gene_names_heads

indicies.som <- match(results_soma_genes$id, texpr(bg_soma_filt, "all")$gene_id)
gene_names_soma <- texpr(bg_soma_filt, "all")$t_name[indicies.som]
results_soma_genes$tname <- gene_names_soma

indicies.ov <- match(results_ovaries_genes$id, texpr(bg_ovaries_filt, "all")$gene_id)
gene_names_ovaries <- texpr(bg_ovaries_filt, "all")$t_name[indicies.ov]
results_ovaries_genes$tname <- gene_names_ovaries
```

# Write results to csv

```{r}
write.csv(results_heads_transcripts, 
          "../processed/ballG/results/results_bg_heads_transcript.csv", row.names=FALSE)
write.csv(results_heads_genes, 
          "../processed/ballG/results/results_bg_heads_gene.csv", row.names=FALSE)

write.csv(results_soma_transcripts, 
          "../processed/ballG/results/results_bg_soma_transcripts.csv", row.names=FALSE)
write.csv(results_soma_genes, 
          "../processed/ballG/results/results_bg_soma_gene.csv", row.names=FALSE)

write.csv(results_ovaries_transcripts, 
          "../processed/ballG/results/results_bg_ovaries_transcripts.csv", row.names=FALSE)
write.csv(results_ovaries_genes, 
          "../processed/ballG/results/results_bg_ovaries_gene.csv", row.names=FALSE)
```

# Identify transcripts and genes with a q value <0.05

```{r}
sigQval_heads_transcripts <- subset(results_heads_transcripts,results_heads_transcripts$qval<0.05)
sigQval_heads_gene <- subset(results_heads_genes,results_heads_genes$qval<0.05)
write.csv(sigQval_heads_gene, 
          "../processed/ballG/results/sigQval_heads_gene.csv", row.names=FALSE)

sigQval_soma_transcripts <- subset(results_soma_transcripts,results_soma_transcripts$qval<0.05)
sigQval_soma_gene <- subset(results_soma_genes,results_soma_genes$qval<0.05)
write.csv(sigQval_soma_gene, 
          "../processed/ballG/results/sigQval_soma_gene.csv", row.names=FALSE)

sigQval_ovaries_transcripts <- subset(results_ovaries_transcripts,results_ovaries_transcripts$qval<0.05)
sigQval_ovaries_gene <- subset(results_ovaries_genes,results_ovaries_genes$qval<0.05)
write.csv(sigQval_ovaries_gene, 
          "../processed/ballG/results/sigQval_ovaries_gene.csv", row.names=FALSE)

```

# Plotting

```{r}
#Set plot settings
tropical= c('darkorange', 'dodgerblue', 'hotpink', 'limegreen', 'yellow')
palette(tropical)
```

# Distribution of gene abundances (FPKM values) across samples (libraries), colored by treatment

```{r}
fpkm_heads = texpr(bg_ballG_heads,meas="FPKM")
fpkm_heads = log2(fpkm_heads+1) 
boxplot(fpkm_heads,col=as.numeric(phenDat$treatment),las=2,ylab='log2(FPKM+1: heads)')

fpkm_soma = texpr(bg_ballG_soma,meas="FPKM")
fpkm_soma = log2(fpkm_soma+1) 
boxplot(fpkm_soma,col=as.numeric(phenDat$treatment),las=2,ylab='log2(FPKM+1: soma)')

fpkm_ovaries = texpr(bg_ballG_ovaries,meas="FPKM")
fpkm_ovaries = log2(fpkm_ovaries+1) 
boxplot(fpkm_ovaries,col=as.numeric(phenDat$treatment),las=2,ylab='log2(FPKM+1: ovaries)')
```



# Number of transcripts per gene

```{r}

# Load transcript gene indexes from ballgown objects
t.genes_table.hds <- indexes(bg_ballG_heads)$t2g
t.genes_table.som <- indexes(bg_ballG_soma)$t2g
t.genes_table.ov <- indexes(bg_ballG_ovaries)$t2g

# Plot # of transcripts per gene

# heads
counts.h <- table(t.genes_table.hds[,"g_id"])
h_one <- length(which(counts.h==1))
h_more <- length(which(counts.h>1))
h_max <- max(counts.h)
hist(counts.h, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", h_one), paste("Genes with more than one transcript =", h_more), paste("Max transcripts for single gene = ", h_max))
legend("topright", legend_text, lty = NULL)

# soma
counts.s <- table(t.genes_table.som[,"g_id"])
s_one <- length(which(counts.s==1))
s_more <- length(which(counts.s>1))
s_max <- max(counts.s)
hist(counts.s, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", s_one), paste("Genes with more than one transcript =", s_more), paste("Max transcripts for single gene = ", s_max))
legend("topright", legend_text, lty = NULL)

# ovaries
counts.o <- table(t.genes_table.ov[,"g_id"])
o_one <- length(which(counts.o==1))
o_more <- length(which(counts.o>1))
o_max <- max(counts.o)
hist(counts.o, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", o_one), paste("Genes with more than one transcript =", o_more), paste("Max transcripts for single gene = ", o_max))
legend("topright", legend_text, lty = NULL)
```


# Distribution of transcript sizes

```{r}
full_table.hd<-texpr(bg_ballG_heads, 'all')
hist(full_table.hd$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")

full_table.sm<-texpr(bg_ballG_soma, 'all')
hist(full_table.sm$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")

full_table.ov<-texpr(bg_ballG_ovaries, 'all')
hist(full_table.ov$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")
```

# prepare RPKMs fo comparison across diets

```{r}
# heads
fpkm_h <- colnames(fpkm_heads)
ihs<-strsplit(fpkm_h, split=".", fixed=TRUE)
ihs<-unlist(lapply(ihs, function(x) x[2]))
ihs<-strsplit(ihs, split="-",fixed=TRUE)
ihs<-unlist(lapply(ihs, function(x) x[1]))
colnames(fpkm_heads) <- ihs
str(fpkm_heads)

# soma
fpkm_s <- colnames(fpkm_soma)
iss<-strsplit(fpkm_s, split=".", fixed=TRUE)
iss<-unlist(lapply(iss, function(x) x[2]))
iss<-strsplit(iss, split="-",fixed=TRUE)
iss<-unlist(lapply(iss, function(x) x[1]))
colnames(fpkm_soma) <- iss
str(fpkm_soma)

# ovaries
fpkm_o <- colnames(fpkm_ovaries)
ios<-strsplit(fpkm_o, split=".", fixed=TRUE)
ios<-unlist(lapply(ios, function(x) x[2]))
ios<-strsplit(ios, split="-",fixed=TRUE)
ios<-unlist(lapply(ios, function(x) x[1]))
colnames(fpkm_ovaries) <- ios
str(fpkm_ovaries)
```

# Visualize expression across libraries

```{r}
tfpkm_hds<-texpr(bg_heads_filt, meas="FPKM")
tfpkm_hds <-log2(tfpkm_hds+1)
colnames(tfpkm_hds)<-c("C1","C2","C3","C4","C5","C6","DR1","DR2",
                      "DR3","DR4","DR5","DR6","HS1","HS2","HS3","HS4","HS5","HS6")

tfpkm_som<-texpr(bg_soma_filt, meas="FPKM")
tfpkm_som <-log2(tfpkm_som+1)
colnames(tfpkm_som)<-c("C1","C2","C3","C4","C5","C6","DR1","DR2",
                      "DR3","DR4","DR5","DR6","HS1","HS2","HS3","HS4","HS5","HS6")

tfpkm_ov<-texpr(bg_ovaries_filt, meas="FPKM")
tfpkm_ov <-log2(tfpkm_ov+1)
colnames(tfpkm_ov)<-c("C1","C2","C3","C4","C5","C6","DR1","DR2",
                      "DR3","DR4","DR5","DR6","HS1","HS2","HS3","HS4","HS5","HS6")

gfpkm_hds <-gexpr(bg_heads_filt)
gfpkm_hds <-log2(gfpkm_hds +1)


gfpkm_som <-gexpr(bg_soma_filt)
gfpkm_som <-log2(gfpkm_som +1)
rownames(gfpkm_som)<-rownames(gfpkm_som)

gfpkm_ov <-gexpr(bg_ovaries_filt)
gfpkm_ov <-log2(gfpkm_ov +1)
rownames(gfpkm_ov)<-rownames(gfpkm_ov)

hh<- as.data.frame(tfpkm_hds) %>% gather(id, fpkm)
hh$treatment <- rep(c("C","DR","HS"), each=6*nrow(tfpkm_hds))

sm<- as.data.frame(tfpkm_som) %>% gather(id, fpkm)
sm$treatment <- rep(c("C","DR","HS"), each=6*nrow(tfpkm_som))

ov<- as.data.frame(tfpkm_ov) %>% gather(id, fpkm)
ov$treatment <- rep(c("C","DR","HS"), each=6*nrow(tfpkm_ov))

p1 <- ggplot(hh, aes(x=id, y=fpkm,fill=treatment)) +
  geom_boxplot() +
  ylab(expression("log"[2]*"(FPKM + 1)"))
p1 +labs(title="Distribution of FPKMs across head samples",
        x ="Diet treatment")  

p2 <- ggplot(sm, aes(x=id, y=fpkm,fill=treatment)) +
  geom_boxplot() +
  ylab(expression("log"[2]*"(FPKM + 1)"))
p2 +labs(title="Distribution of FPKMs across soma samples",
        x ="Diet treatment")  

p3 <- ggplot(ov, aes(x=id, y=fpkm,fill=treatment)) +
  geom_boxplot() +
  ylab(expression("log"[2]*"(FPKM + 1)"))
p3 +labs(title="Distribution of FPKMs across ovary samples",
        x ="Diet treatment")  

#ggsave(p1, filename="../Plots/FPKM_overall.pdf", width=6, height=4)
```



```{r}
# heads
rx <- which(results_heads_genes$qval<=0.05)
rowMeans(gfpkm_hds[rx,])

ff<-5
phh<-data.frame('FPKM'=gfpkm_hds[rx[ff],], 'Treatment' = rep(c('C','DR','HS'),each=6))

q1 <- ggplot(phh, aes(x=Treatment, y=FPKM)) +
  geom_point(position=position_jitter(width = 0.05),color='steelblue', alpha=0.7,size=3) +
  ylab(expression("log"[2]*"(FPKM + 1)"))
q1 +labs(title="Gene expression across diet treatments \n for heads",
        x ="Diet treatment")


# soma
ry <- which(results_soma_genes$qval<=0.05)
rowMeans(gfpkm_som[ry,])

gg<-5
pss<-data.frame('FPKM'=gfpkm_som[rx[gg],], 'Treatment' = rep(c('C','DR','HS'),each=6))

q2 <- ggplot(pss, aes(x=Treatment, y=FPKM)) +
  geom_point(position=position_jitter(width = 0.05),color='steelblue', alpha=0.7,size=3) +
  ylab(expression("log"[2]*"(FPKM + 1)"))
q2 +labs(title="Gene expression across diet treatments \n for soma",
        x ="Diet treatment")


# ovaries
rz <- which(results_ovaries_genes$qval<=0.05)
rowMeans(gfpkm_ov[rz,])

kk<-5
poo<-data.frame('FPKM'=gfpkm_ov[rz[kk],], 'Treatment' = rep(c('C','DR','HS'),each=6))

q3 <- ggplot(poo, aes(x=Treatment, y=FPKM)) +
  geom_point(position=position_jitter(width = 0.05),color='steelblue', alpha=0.7,size=3) +
  ylab(expression("log"[2]*"(FPKM + 1)"))
q3 +labs(title="Gene expression across diet treatments \n for ovaries",
        x ="Diet treatment")

```

# Heat maps

```{r}
rv<- t(scale(t(gfpkm_hds),center=TRUE,scale=FALSE))
rv<- rowVars(rv)
## select the top 250 most variable genes for clustering
idx<- order(-rv)[1:250]

ru<- t(scale(t(gfpkm_som),center=TRUE,scale=FALSE))
ru<- rowVars(ru)
idu<- order(-ru)[1:250]

ro<- t(scale(t(gfpkm_ov),center=TRUE,scale=FALSE))
ro<- rowVars(ro)
## select the top 250 most variable genes for clustering
ido<- order(-ro)[1:250]


library(ComplexHeatmap) 

Heatmap(gfpkm_hds[idx,], name = "log2(FPKM+1)", 
        show_row_names = FALSE, show_column_names = FALSE, 
        row_dend_reorder = TRUE, column_dend_reorder = TRUE)
        #clustering_distance_rows = "pearson",
        #clustering_distance_columns = "euclidean",
        #clustering_method_rows = "complete",
        #clustering_method_columns = "complete")

Heatmap(gfpkm_som[idu,], name = "log2(FPKM+1)", 
        show_row_names = FALSE, show_column_names = FALSE, 
        row_dend_reorder = TRUE, column_dend_reorder = TRUE)

Heatmap(gfpkm_ov[ido,], name = "log2(FPKM+1)", 
        show_row_names = FALSE, show_column_names = FALSE, 
        row_dend_reorder = TRUE, column_dend_reorder = TRUE)


###################
# ggplot simple heat map
# heads
rx <- as.character(results_heads_genes[which(results_heads_genes$qval<=0.05), 'id'])

gfpkm_hds_diff <-as.data.frame(gfpkm_hds[rx,])

gg <- rowMeans(gfpkm_hds_diff)
#gfpkm_hds_diff$gname<-rownames(gfpkm_hds_diff)
gfpkm_hds_diff<-gfpkm_hds_diff[order(gg),]
gfpkm_hds_diff <- gfpkm_hds_diff[800:900,]

gfpkm_hds_diff$gname<-seq(1,nrow(gfpkm_hds_diff))

hds.long<-melt(gfpkm_hds_diff,id='gname')
hds.long$variable <- unlist(lapply(strsplit(as.character(hds.long$variable), split=".",fixed=TRUE),function(x) x[2]))

ggplot(hds.long, aes(variable, gname)) +
  geom_tile(aes(fill = value), colour = "white") + 
  scale_fill_gradient(low = "red", high = "steelblue") +
  theme(#axis.text.x = element_blank(),
        axis.text.y=element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank())

####
# soma
ry <- as.character(results_soma_genes[which(results_soma_genes$qval<=0.05), 'id'])

gfpkm_som_diff <-as.data.frame(gfpkm_som[ry,])

yy <- rowMeans(gfpkm_som_diff)
#gfpkm_hds_diff$gname<-rownames(gfpkm_som_diff)
gfpkm_som_diff<-gfpkm_som_diff[order(yy),]
gfpkm_som_diff <- gfpkm_som_diff[1:200,]

gfpkm_som_diff$gname<-seq(1,nrow(gfpkm_som_diff))

som.long<-melt(gfpkm_som_diff,id='gname')
som.long$variable <- unlist(lapply(strsplit(as.character(som.long$variable), split=".",fixed=TRUE),function(x) x[2]))

ggplot(som.long, aes(variable, gname)) +
  geom_tile(aes(fill = value), colour = "white") + 
  scale_fill_gradient(low = "red", high = "steelblue") +
  theme(#axis.text.x = element_blank(),
        axis.text.y=element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank())


# ovaries
rz <- as.character(results_ovaries_genes[which(results_ovaries_genes$qval<=0.05), 'id'])

gfpkm_ov_diff <-as.data.frame(gfpkm_ov[rz,])

zz <- rowMeans(gfpkm_ov_diff)
#gfpkm_ov_diff$gname<-rownames(gfpkm_ov_diff)
gfpkm_ov_diff<-gfpkm_ov_diff[order(zz),]
gfpkm_ov_diff <- gfpkm_ov_diff[1:20,]

gfpkm_ov_diff$gname<-seq(1,nrow(gfpkm_ov_diff))

ov.long<-melt(gfpkm_ov_diff,id='gname')
ov.long$variable <- unlist(lapply(strsplit(as.character(ov.long$variable), split=".",fixed=TRUE),function(x) x[2]))

ggplot(ov.long, aes(variable, gname)) +
  geom_tile(aes(fill = value), colour = "white") + 
  scale_fill_gradient(low = "red", high = "steelblue") +
  theme(#axis.text.x = element_blank(),
        axis.text.y=element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank())
```

# Heatmap for whole experiment

```{r}
bg_ballG_heads <- ballgown(dataDir = "../processed/ballG/heads", samplePattern = "", pData = heads_phen)

```

# PCA

```{r}
# heads
X<- t(scale(t(gfpkm_hds),center=TRUE,scale=FALSE))
sv<- svd(t(X))
U<- sv$u
V<- sv$v
D<- sv$d
Z<- t(X)%*%V

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc_dat<- data.frame(heads_phen$id, PC1 = Z[,1], PC2 = Z[,2])

pc_dat$Diet <- unlist(lapply(strsplit(as.character(pc_dat$heads_phen.id), split="-",fixed=TRUE),function(x) x[1]))

## make figure with ggplot2
library(ggthemes)
ggplot(pc_dat, aes(x=PC1, y=PC2, col=Diet)) + 
        geom_point() + 
        theme_bw() +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.6),
              axis.line.y = element_line(color="black", size = 0.6))

# soma
Y<- t(scale(t(gfpkm_som),center=TRUE,scale=FALSE))
sk<- svd(t(Y))
A<- sk$u
B<- sk$v
C<- sk$d
E<-t(Y)%*%B

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc_dat.som<- data.frame(soma_phen$id, PC1 = E[,1], PC2 = E[,2])

pc_dat.som$Diet <- unlist(lapply(strsplit(as.character(pc_dat.som$soma_phen.id), split="-",fixed=TRUE),function(x) x[1]))

## make figure with ggplot2
ggplot(pc_dat.som, aes(x=PC1, y=PC2, col=Diet)) + 
        geom_point() + 
        theme_bw() +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.6),
              axis.line.y = element_line(color="black", size = 0.6))

# ovaries
W<- t(scale(t(gfpkm_ov),center=TRUE,scale=FALSE))
#W <- t(scale(t(log2(poo$FPKM+1)),center=TRUE,scale=FALSE))
sn<- svd(t(W))
F<- sn$u
G<- sn$v
H<- sn$d
I<-t(W)%*%G

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc_dat.ov<- data.frame(ovaries_phen$id, PC1 = I[,1], PC2 = I[,2])

pc_dat.ov$Diet <- unlist(lapply(strsplit(as.character(pc_dat.ov$ovaries_phen.id), split="-",fixed=TRUE),function(x) x[1]))

## make figure with ggplot2
ggplot(pc_dat.ov, aes(x=PC1, y=PC2, col=Diet)) + 
        geom_point() + 
        theme_bw() +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.6),
              axis.line.y = element_line(color="black", size = 0.6))

```

