---
title: "ballgown_run"
author: "Enoch Ng'oma"
date: "5/2/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Differential expression analysis protocol

```{r}
#R
#load R version R version 3.2.2
#load R packagees:
library(ballgown)
library(genefilter)
library(tidyverse)
library(devtools)

phenDat <- read.csv("describe_samples.csv", stringsAsFactors = FALSE)
```

```{r}
heads_phen <- subset(phenDat, phenDat$tissue=="H")
soma_phen <- subset(phenDat, phenDat$tissue=="B")
ovaries_phen <- subset(phenDat, phenDat$tissue=="O")
bg_ballG_heads <- ballgown(dataDir = "../processed/ballG/heads", samplePattern = "", pData = heads_phen)
bg_ballG_soma <- ballgown(dataDir = "../processed/ballG/soma", samplePattern = "", pData = soma_phen)
bg_ballG_ovaries <- ballgown(dataDir = "../processed/ballG/ovaries", samplePattern = "", pData = ovaries_phen)

# Dispplay a decsription of each tissue
bg_ballG_heads
bg_ballG_soma
bg_ballG_ovaries

# Save .Rda
save(bg_ballG_heads, file = "../processed/ballG/results/bg_ballG_heads.Rda")
save(bg_ballG_soma, file = "../processed/ballG/results/bg_ballG_soma.Rda")
save(bg_ballG_ovaries, file = "../processed/ballG/results/bg_ballG_ovaries.Rda")
```


```{r}
load("../processed/ballG/results/bg_ballG_heads.Rda")
load("../processed/ballG/results/bg_ballG_soma.Rda")
load("../processed/ballG/results/bg_ballG_ovaries.Rda")
```

# Filter to remove low-abundance genes (i.e. those with variance across samples of <=1)

```{r}
bg_heads_filt <- subset(bg_ballG_heads, "rowVars(texpr(bg_ballG_heads)) >1", genomesubset=TRUE)
bg_soma_filt <- subset(bg_ballG_soma, "rowVars(texpr(bg_ballG_soma)) >1", genomesubset=TRUE)
bg_ovaries_filt <- subset(bg_ballG_ovaries, "rowVars(texpr(bg_ballG_ovaries)) >1", genomesubset=TRUE)

# See counts for each tissue
bg_heads_filt
bg_soma_filt
bg_ovaries_filt
```

# Identify transcripts that show significant differences between groups

```{r}
results_heads_transcripts <- stattest(bg_heads_filt, feature="transcript",
        covariate="treatment", meas="FPKM") 
#which(results_heads_transcripts$qval<=0.05)
#hds.qval <- results_heads_transcripts[which(results_heads_transcripts$qval<=0.05),]

results_soma_transcripts <- stattest(bg_soma_filt, feature="transcript",
        covariate="treatment", meas="FPKM")
#which(results_soma_transcripts$qval<=0.05) 
#soma.qval <- results_soma_transcripts[which(results_soma_transcripts$qval<=0.05),]
  
results_ovaries_transcripts <- stattest(bg_ovaries_filt, feature="transcript",
        covariate="treatment", meas="FPKM")
#which(results_ovaries_transcripts$qval<=0.05) 
#ova.qval <- results_ovaries_transcripts[which(results_ovaries_transcripts$qval<=0.05),]
```

# Identify genes that show statistically significant differences between groups

```{r}
results_heads_genes <- stattest(bg_heads_filt, feature="gene", 
                                covariate="treatment",  meas="FPKM")
which(results_heads_genes$qval<=0.05)

results_soma_genes <- stattest(bg_soma_filt, feature="gene", 
                               covariate="treatment",  meas="FPKM")
which(results_soma_transcripts$qval<=0.05)

results_ovaries_genes <- stattest(bg_ovaries_filt, feature="gene", 
                                  covariate="treatment",  meas="FPKM")
which(results_ovaries_transcripts$qval<=0.05)
```

# Add gene names and gene IDs to the results_transcripts data frame

```{r}
results_heads_transcripts <- data.frame(geneNames=ballgown::transcriptNames(bg_heads_filt), geneIDs=ballgown::geneIDs(bg_heads_filt), results_heads_transcripts)

results_soma_transcripts <- data.frame(geneNames=ballgown::transcriptNames(bg_soma_filt), geneIDs=ballgown::geneIDs(bg_soma_filt), results_soma_transcripts)

results_ovaries_transcripts <- data.frame(geneNames=ballgown::transcriptNames(bg_ovaries_filt), geneIDs=ballgown::geneIDs(bg_ovaries_filt), results_ovaries_transcripts)

full_table.hd<-texpr(bg_heads_filt, 'all')
full_table.sm<-texpr(bg_soma_filt, 'all')
full_table.ov<-texpr(bg_ovaries_filt, 'all')
```

```{r}
# pull gene expression data frame from the ballgown object
#xx.heads <- as.data.frame(gexpr(bg_heads_filt))
#xx.soma <- as.data.frame(gexpr(bg_soma_filt))
#xx.ovaries <- as.data.frame(gexpr(bg_ovaries_filt))
```


# Sort results from smallest P value to largest

```{r}
results_heads_transcripts = arrange(results_heads_transcripts,pval) 
results_heads_genes = arrange(results_heads_genes,pval)

results_soma_transcripts = arrange(results_soma_transcripts,pval) 
results_soma_genes = arrange(results_soma_genes,pval)

results_ovaries_transcripts = arrange(results_ovaries_transcripts,pval) 
results_ovaries_genes = arrange(results_ovaries_genes,pval)

# Transfer transcript names fron stringtie (protocol does not output gene names)

indicies.hds <- match(results_heads_genes$id, texpr(bg_heads_filt, "all")$gene_id)
gene_names_heads <- texpr(bg_heads_filt, "all")$t_name[indicies.hds]
results_heads_genes$tname <- gene_names_heads

indicies.som <- match(results_soma_genes$id, texpr(bg_soma_filt, "all")$gene_id)
gene_names_soma <- texpr(bg_soma_filt, "all")$t_name[indicies.som]
results_soma_genes$tname <- gene_names_soma

indicies.ov <- match(results_ovaries_genes$id, texpr(bg_ovaries_filt, "all")$gene_id)
gene_names_ovaries <- texpr(bg_ovaries_filt, "all")$t_name[indicies.ov]
results_ovaries_genes$tname <- gene_names_ovaries
```

# Write results to csv

```{r}
write.csv(results_heads_transcripts, 
          "../processed/ballG/results/results_bg_heads_transcript.csv", row.names=FALSE)
write.csv(results_heads_genes, 
          "../processed/ballG/results/results_bg_heads_gene.csv", row.names=FALSE)

write.csv(results_soma_transcripts, 
          "../processed/ballG/results/results_bg_soma_transcripts.csv", row.names=FALSE)
write.csv(results_soma_genes, 
          "../processed/ballG/results/results_bg_soma_gene.csv", row.names=FALSE)

write.csv(results_ovaries_transcripts, 
          "../processed/ballG/results/results_bg_ovaries_transcripts.csv", row.names=FALSE)
write.csv(results_ovaries_genes, 
          "../processed/ballG/results/results_bg_ovaries_gene.csv", row.names=FALSE)
```

# Identify transcripts and genes with a q value <0.05

```{r}
sigQval_heads_transcripts <- subset(results_heads_transcripts,results_heads_transcripts$qval<0.05)
sigQval_heads_gene <- subset(results_heads_genes,results_heads_genes$qval<0.05)

sigQval_soma_transcripts <- subset(results_soma_transcripts,results_soma_transcripts$qval<0.05)
sigQval_soma_gene <- subset(results_soma_genes,results_soma_genes$qval<0.05)

sigQval_ovaries_transcripts <- subset(results_ovaries_transcripts,results_ovaries_transcripts$qval<0.05)
sigQval_ovaries_gene <- subset(results_ovaries_genes,results_ovaries_genes$qval<0.05)
```

# Plotting

```{r}
#Set plot settings
tropical= c('darkorange', 'dodgerblue', 'hotpink', 'limegreen', 'yellow')
palette(tropical)
```

# Distribution of gene abundances (FPKM values) across samples (libraries), colored by treatment

```{r}
fpkm_heads = texpr(bg_ballG_heads,meas="FPKM")
fpkm_heads = log2(fpkm_heads+1) 
boxplot(fpkm_heads,col=as.numeric(phenDat$treatment),las=2,ylab='log2(FPKM+1: heads)')

fpkm_soma = texpr(bg_ballG_soma,meas="FPKM")
fpkm_soma = log2(fpkm_soma+1) 
boxplot(fpkm_soma,col=as.numeric(phenDat$treatment),las=2,ylab='log2(FPKM+1: soma)')

fpkm_ovaries = texpr(bg_ballG_ovaries,meas="FPKM")
fpkm_ovaries = log2(fpkm_ovaries+1) 
boxplot(fpkm_ovaries,col=as.numeric(phenDat$treatment),las=2,ylab='log2(FPKM+1: ovaries)')
```



# Number of transcripts per gene

```{r}

# Load transcript gene indexes from ballgown objects
t.genes_table.hds <- indexes(bg_ballG_heads)$t2g
t.genes_table.som <- indexes(bg_ballG_soma)$t2g
t.genes_table.ov <- indexes(bg_ballG_ovaries)$t2g

# Plot # of transcripts per gene

# heads
counts.h <- table(t.genes_table.hds[,"g_id"])
h_one <- length(which(counts.h==1))
h_more <- length(which(counts.h>1))
h_max <- max(counts.h)
hist(counts.h, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", h_one), paste("Genes with more than one transcript =", h_more), paste("Max transcripts for single gene = ", h_max))
legend("topright", legend_text, lty = NULL)

# soma
counts.s <- table(t.genes_table.som[,"g_id"])
s_one <- length(which(counts.s==1))
s_more <- length(which(counts.s>1))
s_max <- max(counts.s)
hist(counts.s, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", s_one), paste("Genes with more than one transcript =", s_more), paste("Max transcripts for single gene = ", s_max))
legend("topright", legend_text, lty = NULL)

# ovaries
counts.o <- table(t.genes_table.ov[,"g_id"])
o_one <- length(which(counts.o==1))
o_more <- length(which(counts.o>1))
o_max <- max(counts.o)
hist(counts.o, breaks=50, col="bisque4", xlab="Transcripts per gene",
main="Distribution of transcript count per gene")
legend_text=c(paste("Genes with one transcript =", o_one), paste("Genes with more than one transcript =", o_more), paste("Max transcripts for single gene = ", o_max))
legend("topright", legend_text, lty = NULL)
```


# Distribution of transcript sizes

```{r}
full_table.hd<-texpr(bg_ballG_heads, 'all')
hist(full_table.hd$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")

full_table.sm<-texpr(bg_ballG_soma, 'all')
hist(full_table.sm$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")

full_table.ov<-texpr(bg_ballG_ovaries, 'all')
hist(full_table.ov$length, breaks=50, xlab="Transcript length (bp)", main="Distribution of transcript lengths", col="steelblue")
```

# prepare RPKMs fo comparison across diets

```{r}
# heads
fpkm_h <- colnames(fpkm_heads)
ihs<-strsplit(fpkm_h, split=".", fixed=TRUE)
ihs<-unlist(lapply(ihs, function(x) x[2]))
ihs<-strsplit(ihs, split="-",fixed=TRUE)
ihs<-unlist(lapply(ihs, function(x) x[1]))
colnames(fpkm_heads) <- ihs
str(fpkm_heads)

# soma
fpkm_s <- colnames(fpkm_soma)
iss<-strsplit(fpkm_s, split=".", fixed=TRUE)
iss<-unlist(lapply(iss, function(x) x[2]))
iss<-strsplit(iss, split="-",fixed=TRUE)
iss<-unlist(lapply(iss, function(x) x[1]))
colnames(fpkm_soma) <- iss
str(fpkm_soma)

# ovaries
fpkm_o <- colnames(fpkm_ovaries)
ios<-strsplit(fpkm_o, split=".", fixed=TRUE)
ios<-unlist(lapply(ios, function(x) x[2]))
ios<-strsplit(ios, split="-",fixed=TRUE)
ios<-unlist(lapply(ios, function(x) x[1]))
colnames(fpkm_ovaries) <- ios
str(fpkm_ovaries)
```

# Plot

```{r}
tfpkm_hds<-texpr(bg_heads_filt, meas="FPKM")
tfpkm_hds <-log2(tfpkm_hds+1)
colnames(tfpkm_hds)<-c("C1","C2","C3","C4","C5","C6","DR1","DR2",
                      "DR3","DR4","DR5","DR6","HS1","HS2","HS3","HS4","HS5","HS")

gfpkm <-gexpr(bg_heads_filt)
gfpkm <-log2(gfpkm +1)
rownames(gfpkm)<-rownames(gfpkm)

hh<- as.data.frame(fpkm_hds) %>% gather(id, fpkm)
hh$treatment <- rep(c("C","DR","HS"), each=6*nrow(fpkm_hds))

p1 <- ggplot(hh, aes(x=id, y=fpkm,fill=treatment)) +
  geom_boxplot() +
  ylab(expression("log"[2]*"(FPKM + 1)"))+
  main="Distribution of FPKMs for all 18 head samples" +
  xlab="Diet treatment"
p1
#ggsave(p1, filename="../Plots/FPKM_overall.pdf", width=6, height=4)
```

#Overall
```{r}
tfpkm<-texpr(bg_filt, meas="FPKM")
tlfpkm <-log2(tfpkm+1)
colnames(tlfpkm)<-c("C1","C2","C3","C4","DR1","DR2","DR3","DR4","HS1","HS2","HS3","HS4")

gfpkm <-gexpr(bg_filt)
glfpkm <-log2(gfpkm +1)
rownames(glfpkm)<-rownames(gfpkm)

tt<- as.data.frame(tlfpkm) %>% gather(id, fpkm)
tt$treatment <- rep(c("C","DR","HS"), each=4*nrow(tlfpkm))

p1 <- ggplot(tt, aes(x=id, y=fpkm,fill=treatment)) +
  geom_boxplot() +
  ylab(expression("log"[2]*"(FPKM + 1)"))
ggsave(p1, filename="../Plots/FPKM_overall.pdf", width=6, height=4)
```



