---
title: "Prepare expression data for batch effect analysis"
author: "Enoch Ng'oma"
date: "6/15/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(tidyverse)
library(genefilter)
```

# Data

```{r}
# phenotype data
phenDat <- read.csv("../processed/describe_samples_batch.csv", stringsAsFactors = FALSE)

#bg_ballG <- ballgown(dataDir = "../processed/ballG_all", samplePattern = "", pData = phenDat)

# Save all transcripts across tissues and diets as .Rda
#save(bg_ballG, file = "../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

load("../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

# Dispplay a decsription
bg_ballG
```

# Filter to remove low-abundance genes (i.e. those with variance across samples of <=1)

```{r}
bg_ballG_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)

save(bg_ballG_filt, file = "../processed/results/ballG_all_results/bg_ballG_filt.Rda")

## Qu.1 Both sva and RUVseq tutorials filter out non-expressed genes by requiring >5 reads in at least 2 samples for each gene
# i.e. `filter <- apply(zfGenes, 1, function(x) lengthx[x>5]) >2`
```

# Extract expression data
# Ref: https://github.com/alyssafrazee/ballgown

```{r}
load("../processed/results/ballG_all_results/bg_ballG_filt.Rda")

#results_all_transcripts = texpr(transcript_ballG_filt, meas = "FPKM")
results_all_genes = gexpr(bg_ballG_filt)

gfpkm_gene <-log2(results_all_genes +1)
colnames(results_all_genes)<-phenDat$id
save(results_all_genes, file = "../processed/results/ballG_all_results/gfpkm_gene_ballG_filt.csv")

```






