---
title: "Prepare expression data for batch effect analysis"
author: "Enoch Ng'oma"
date: "6/15/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(tidyverse)
library(genefilter)
```

# Data

```{r}
# phenotype data
phenDat <- read.csv("../processed/describe_samples_batch.csv", stringsAsFactors = FALSE)

#bg_ballG <- ballgown(dataDir = "../processed/ballG_all", samplePattern = "", pData = phenDat)

# Save all transcripts across tissues and diets as .Rda
#save(bg_ballG, file = "../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

load("../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

# Dispplay a decsription
bg_ballG
```

# Filter to remove low-abundance genes (i.e. those with variance across samples of <=1)

```{r}
bg_ballG_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)

save(bg_ballG_filt, file = "../processed/results/ballG_all_results/bg_ballG_filt.Rda")
```

# Extract expression data
# Ref: https://github.com/alyssafrazee/ballgown

```{r}
  # syntax: expr(ballgown_object_name, <EXPRESSION_MEASUREMENT>)

transcript_fpkm = texpr(bg_ballG_filt, meas = "FPKM")
log_transcript_fpkm <-log2(transcript_fpkm+1)
# colnames(transcript_fpkm)<-phenDat$id

#transcript_cov = texpr(bg, 'cov')
#whole_tx_table = texpr(bg_ballG_filt, 'all')
#exon_mcov = eexpr(bg, 'mcov')
#junction_rcount = iexpr(bg)
#whole_intron_table = iexpr(bg, 'all')
gene_expression = gexpr(bg_ballG_filt)

# Note: expr functions return matrices unless meas = 'all', 
# in which case some additional feature metadata is returned and the result is a data.frame
```

# Add gene names and gene IDs to log-transformed data

```{r}
log_transcript_fpkm <- data.frame(geneNames=ballgown::transcriptNames(bg_ballG_filt), geneIDs=ballgown::geneIDs(bg_ballG_filt), log_transcript_fpkm)

indicies.bg <- match(log_transcript_fpkm$geneIDs, texpr(bg_ballG_filt, "all")$gene_id)
gene_names_bg <- texpr(bg_ballG_filt, "all")$t_name[indicies.bg]
log_transcript_fpkm$tname <- gene_names_bg

write.csv(log_transcript_fpkm, 
          "../processed/results/ballG_all_results/logGene_associated_texpr.csv", row.names=FALSE)

log_transcript_fpkm[,c(1:2,57)]
```

# Add gene names and gene IDs to non log-transformed data

```{r}
load("../processed/results/ballG_all_results/bg_ballG_filt.Rda")
# filtered for low abundance transcripts

# Extract expression data with no transformation
transcript_fpkm = texpr(bg_ballG_filt, meas = "FPKM")

# Add and match gene names
transcript_fpkm <- data.frame(geneNames=ballgown::transcriptNames(bg_ballG_filt), geneIDs=ballgown::geneIDs(bg_ballG_filt), transcript_fpkm)

indicies.bg <- match(transcript_fpkm$geneIDs, texpr(bg_ballG_filt, "all")$gene_id)
gene_names_bg <- texpr(bg_ballG_filt, "all")$t_name[indicies.bg]
transcript_fpkm$tname <- gene_names_bg

write.csv(transcript_fpkm, 
          "../processed/results/ballG_all_results/gene_associated_texpr.csv", row.names=FALSE)

transcript_fpkm[,c(1:2,57)]
```

