---
title: "Prepare expression data for batch effect analysis"
author: "Enoch Ng'oma"
date: "6/15/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ballgown)
library(genefilter)
library(RSkittleBrewer)
library(tidyverse)
library(genefilter)
library(devtools)
#library(calibrate)
```

# Extract expression data: https://github.com/alyssafrazee/ballgown

# Load data

```{r}
# phenotype data
phenDat <- read.csv("../processed/describe_samples_batch.csv", stringsAsFactors = FALSE)

#bg_ballG <- ballgown(dataDir = "../processed/ballG_all", samplePattern = "", pData = phenDat)

# Save all transcripts across tissues and diets as .Rda
#save(bg_ballG, file = "../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

load("../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

# Dispplay a decsription
bg_ballG
```

# Filter to remove low-abundance genes (i.e. those with variance across samples of <=1)

```{r}
bg_ballG_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE) 

save(bg_ballG_filt, file = "../processed/results/ballG_all_results/bg_ballG_filt.Rda")

## Qu.1 Both sva and RUVseq tutorials filter out non-expressed genes by requiring >5 reads in at least 2 samples for each gene
# i.e. `filter <- apply(zfGenes, 1, function(x) lengthx[x>5]) >2`
```


```{r}
load("../processed/results/ballG_all_results/bg_ballG_filt.Rda")

gfpkm_all = gexpr(bg_ballG_filt)

#get gene names
Gindicies <- match(rownames(gfpkm_all), texpr(bg_ballG_filt, 'all')$gene_id)
Gnames <- texpr(bg_ballG_filt, "all")$t_name[Gindicies]
#gfpkm_all$tname <- Gnames

rownames(gfpkm_all) <- Gnames

# log2 transform expr values, rename columns for sva
log_gfpkm_all <- log2(gfpkm_all+1)
colnames(log_gfpkm_all) <- phenDat$id

# Extract row names as a column
log_gfpkm_all <- cbind(GeneName = rownames(log_gfpkm_all), log_gfpkm_all)

# Write results to csv
write.csv(log_gfpkm_all, 
          "../processed/results/ballG_all_results/log_gfpkm_all.csv",
          row.names=FALSE)



# For svaseq - untransformed
colnames(gfpkm_all) <- phenDat$id

# Extract row names as a column
nolog_gfpkm_all <- cbind(GeneName = rownames(gfpkm_all), gfpkm_all)
#rownames(gfpkm_all) <- NULL

write.csv(nolog_gfpkm_all, 
          "../processed/results/ballG_all_results/nolog_gfpkm_all.csv",
          row.names=FALSE)
```










