---
title: "DiffExpr"
author: "Enoch Ng'oma"
date: "5/7/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
#R
#load R version R version 3.2.2
#load R packagees:
library(ballgown)
library(genefilter)
library(tidyverse)
library(devtools)
library(cowplot)
library(reshape)
#library(ggthemes)

```

# Differential expression analysis protocol

```{r}
phenDat <- read.csv("describe_samples.csv", stringsAsFactors = FALSE)

#bg_ballG <- ballgown(dataDir = "../processed/ballG_all", samplePattern = "", pData = phenDat)

# Save .Rda
#save(bg_ballG, file = "../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

load("../processed/results/ballG_all_results/bg_ballG_all_results.Rda")

# Dispplay a decsription
bg_ballG

```

# Filter to remove low-abundance genes (i.e. those with variance across samples of <=1)

```{r}
bg_ballG_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)

# See counts
bg_ballG_filt

save(bg_ballG_filt, file = "../processed/results/ballG_all_results/bg_ballG_filt.Rda")
```

## Overall expression numbers
# Distribution of gene abundances (FPKM values) across samples (libraries), colored by treatment or tissue and faceted 

```{r}
load(file = "../processed/results/ballG_all_results/bg_ballG_filt.Rda")

tfpkm_bg<-texpr(bg_ballG_filt, meas="FPKM")
tfpkm_bg <-log2(tfpkm_bg+1)
colnames(tfpkm_bg)<-phenDat$id
save(tfpkm_bg, file = "../processed/results/ballG_all_results/tfpkm_bg_ballG_filt.Rda")


tbg<- as.data.frame(tfpkm_bg) %>% gather(id, fpkm)
tbg$treatment <- rep(c("C","DR","HS"), each=18*nrow(tfpkm_bg))
tbg$tissue <- rep(c("B","H","O"),each=nrow(tfpkm_bg),6)
tbg$replicate <- rep(c(1,2,3,4,5,6),each=nrow(tfpkm_bg)*3)
tbg$prep <- paste(tbg$treatment,tbg$replicate,sep="_")
tbg$trep <- paste(tbg$tissue,tbg$replicate,sep="_")

b1 <- ggplot(tbg, aes(x=as.factor(prep), y=fpkm,color=treatment)) +
  geom_boxplot(alpha=0.9) +
          theme(axis.title.y = element_text(size = 24),
                axis.title.x=element_blank(),
                axis.text.x  = element_text(angle=90, vjust=0.5),
                text = element_text(size = 20))+
                
  scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Samples")+
  facet_grid(tissue~.)
b1 + theme(legend.position="none")
  
ggsave(filename = "../plots/Expression_by_sample1.pdf",width = 5,height=5)

b2 <- ggplot(tbg, aes(x=as.factor(trep), y=fpkm,color=tissue)) +
  geom_boxplot(alpha=0.9) +
          theme(axis.title.y = element_text(size = 24),
                axis.title.x=element_blank(),
                axis.text.x  = element_text(angle=90, vjust=0.5),
                text = element_text(size = 20)) +
  scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
  ylab(expression("log"[2]*"(FPKM + 1)")) +
  xlab("Tissue") +
  #scale_fill_discrete(name="Samples")+
  facet_grid(treatment~.)
b2 + theme(legend.position="none")

ggsave(filename = "../plots/Expression_by_sample2.pdf",width = 5,height=5)
```

## PCA for global patterns

```{r}

gfpkm_bg <-gexpr(bg_ballG_filt)
gfpkm_bg <-log2(gfpkm_bg +1)
colnames(gfpkm_bg)<-phenDat$id
save(gfpkm_bg, file = "../processed/results/ballG_all_results/gfpkm_bg_ballG_filt.Rda")


# all samples
X<- t(scale(t(gfpkm_bg),center=TRUE,scale=FALSE))
sv<- svd(t(X))
U<- sv$u
V<- sv$v
D<- sv$d
Z<- t(X)%*%V

# put PC1 and PC2 into a dataframe and add the phenotpe info side by side
pc.dat<- data.frame(phenDat$id, PC1 = Z[,1], PC2 = Z[,2], PC3 = Z[,3],PC4 =Z[,4])
s.id <- strsplit(as.character(pc.dat$phenDat.id), split="-",fixed=TRUE)
pc.dat$Diet <-unlist(lapply(s.id, function(x) x[1]))
pc.dat$Tissue <- unlist(lapply(strsplit(as.character(phenDat$id), 
                                        split="_", fixed=TRUE), function(x) x[2]))

# plot with ggplot2
a <- ggplot(pc.dat, aes(x=PC1, y=PC2, col=Diet, pch=Tissue)) + 
        geom_point(alpha=0.5, size=5) + 
        scale_color_manual(values=c("#CC6666", "#9999CC", "#66CC99")) +
        theme_bw() +
        theme(axis.title = element_text(size = 24)) +
        theme(text = element_text(size = 20)) +
        theme(panel.border = element_blank(), 
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(), 
              axis.line.x = element_line(color="black", size = 0.8),
              axis.line.y = element_line(color="black", size = 0.8))
a 

ggsave(filename = "../plots/PCA_overall.pdf",width = 5,height=5)
```

## Differential expression visualization

```{r}
load("/home/engoma/MyGithub/BasePop_RNAseq/base_pop/processed/results/bg_results_tissues/heads.Rda")
bg_ballG_heads_filt <- subset(bg_ballG_heads, "rowVars(texpr(bg_ballG_heads)) >1", genomesubset=TRUE)

load("/home/engoma/MyGithub/BasePop_RNAseq/base_pop/processed/results/bg_results_tissues/soma.Rda")
bg_ballG_soma_filt <- subset(bg_ballG_soma, "rowVars(texpr(bg_ballG_soma)) >1", genomesubset=TRUE)

load("/home/engoma/MyGithub/BasePop_RNAseq/base_pop/processed/results/bg_results_tissues/ovaries.Rda")
bg_ballG_ovaries_filt <- subset(bg_ballG_ovaries, "rowVars(texpr(bg_ballG_ovaries)) >1", genomesubset=TRUE)

phenDat <- read.csv("describe_samples.csv", stringsAsFactors = FALSE)

heads_phen <- subset(phenDat, phenDat$tissue=="H")
soma_phen <- subset(phenDat, phenDat$tissue=="B")
ovaries_phen <- subset(phenDat, phenDat$tissue=="O")

h_gfpkm_bg <-gexpr(bg_ballG_heads_filt)
h_gfpkm_bg <-log2(h_gfpkm_bg +1)
colnames(h_gfpkm_bg)<-heads_phen$id

o_gfpkm_bg <-gexpr(bg_ballG_ovaries_filt)
o_gfpkm_bg <-log2(o_gfpkm_bg +1)
colnames(o_gfpkm_bg)<-ovaries_phen$id

s_gfpkm_bg <-gexpr(bg_ballG_soma_filt)
s_gfpkm_bg <-log2(s_gfpkm_bg +1)
colnames(s_gfpkm_bg)<-soma_phen$id

#read in diff exp objects
hd.r <- read.csv(file="../processed/results/bg_results_tissues/results_bg_heads_gene.csv", header=TRUE)

ov.r <- read.csv(file="../processed/results/bg_results_tissues/results_bg_ovaries_gene.csv", header=TRUE)

so.r <- read.csv(file="../processed/results/bg_results_tissues/results_bg_soma_gene.csv", header=TRUE)


rx <- as.character(hd.r[which(hd.r$qval<=0.05), 'id'])
h_ord <- h_gfpkm_bg[rx,]
res.sub <- hd.r[which(hd.r$qval<=0.05),]
t1 <- rowMeans(h_ord[rx,1:6])
t2<-rowMeans(h_ord[rx,7:12])
t3<-rowMeans(h_ord[rx,13:18])
tt <- abs(t1-t2)
v.set <-rowVars(t(scale(t(cbind(t1,t2,t3)),center=TRUE,scale=FALSE)))
m.set <- rowMeans(cbind(t1,t2,t3))
#& m.set <7 & m.set >3
ss <- which(res.sub$qval<0.009)
h_ord<-h_ord[ss,]
h_ord <- h_ord[order(rowMeans(h_ord[,1:6])),]
write.csv(res.sub, 
          "../processed/results/bg_results_tissues/DEs_sig_q05_heads.csv", row.names=FALSE)

#h_ord <- h_ord[order(res.sub$qval),]
#h_ord <- h_ord[1:20,]

h_ord <- as.data.frame(h_ord)
h_ord$gname<-seq(1,nrow(h_ord))

hds.long<-melt(h_ord,id='gname')
#hds.long$variable <- unlist(lapply(strsplit(as.character(hds.long$variable), split="-",fixed=TRUE),function(x) x[1]))

ids_h <- strsplit(as.character(hds.long$variable), split=".",fixed=TRUE)
hds.long$sampl <-unlist(lapply(ids_h, function(x) x[2]))

ggplot(hds.long, aes(sampl, gname)) +
  geom_tile(aes(fill = value), colour = "white") + 
  scale_fill_gradient(low = "white", high = "steelblue") +
  scale_y_continuous(expand = c(0, 0)) +
  theme(#axis.text.x = element_blank(),
        axis.text.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x  = element_text(angle=90, vjust=0.5),
        axis.title.y = element_blank())
ggsave(filename = "../plots/Heatmap_heads.pdf",width = 5,height=5)

######################

#SOMA

rx <- as.character(so.r[which(so.r$qval<=0.05), 'id'])
s_ord <- s_gfpkm_bg[rx,]
res.sub <- so.r[which(so.r$qval<=0.05),]
t1 <- rowMeans(s_ord[rx,1:6])
t2<-rowMeans(s_ord[rx,7:12])
t3<-rowMeans(s_ord[rx,13:18])
tt <- abs(t1-t2)
v.set <-rowVars(t(scale(t(cbind(t1,t2,t3)),center=TRUE,scale=FALSE)))
m.set <- rowMeans(cbind(t1,t2,t3))
#& m.set <7 & m.set >3
ss <- which(res.sub$qval<0.009 & m.set <7 & m.set >3)
s_ord<-s_ord[ss,]
s_ord <- s_ord[order(rowMeans(s_ord[,1:6])),]

write.csv(res.sub, 
          "../processed/results/bg_results_tissues/DEs_sig_q05_soma.csv", row.names=FALSE)

#s_ord <- s_ord[order(res.sub$qval),]
#s_ord <- s_ord[1:20,]

s_ord <- as.data.frame(s_ord)
s_ord$gname<-seq(1,nrow(s_ord))

so.long<-melt(s_ord,id='gname')
#so.long$variable <- unlist(lapply(strsplit(as.character(so.long$variable), split="-",fixed=TRUE),function(x) x[1]))

ids_so <- strsplit(as.character(so.long$variable), split=".",fixed=TRUE)
so.long$sampl <-unlist(lapply(ids_so, function(x) x[2]))

ggplot(so.long, aes(sampl, gname)) +
  geom_tile(aes(fill = value), colour = "white") + 
  scale_fill_gradient(low = "white", high = "steelblue") +
  scale_y_continuous(expand = c(0, 0)) +
  theme(#axis.text.x = element_blank(),
        axis.text.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x  = element_text(angle=90, vjust=0.5),
        axis.title.y = element_blank())
ggsave(filename = "../plots/Heatmap_soma.pdf",width = 5,height=5)

######################
#OVARY


rx <- as.character(ov.r[which(ov.r$qval<=0.05), 'id'])
o_ord <- o_gfpkm_bg[rx,]
res.sub <- ov.r[which(ov.r$qval<=0.05),]
t1 <- rowMeans(o_ord[rx,1:6])
t2<-rowMeans(o_ord[rx,7:12])
t3<-rowMeans(o_ord[rx,13:18])
tt <- abs(t1-t2)
v.set <-rowVars(t(scale(t(cbind(t1,t2,t3)),center=TRUE,scale=FALSE)))
m.set <- rowMeans(cbind(t1,t2,t3))

#ss <- which(res.sub$qval<0.009 & m.set <7 & m.set >3)
#o_ord<-o_ord[ss,]
o_ord <- o_ord[order(rowMeans(o_ord[,1:6])),]
o_ord<-o_ord[1:(nrow(o_ord)-1),]
#o_ord <- o_ord[order(res.sub$qval),]
#o_ord <- o_ord[1:20,]

o_ord <- as.data.frame(o_ord)
o_ord$gname<-seq(1,nrow(o_ord))

ov.long<-melt(o_ord,id='gname')
#ov.long$variable <- unlist(lapply(strsplit(as.character(ov.long$variable), split="-",fixed=TRUE),function(x) x[1]))

write.csv(res.sub, 
          "../processed/results/bg_results_tissues/DEs_sig_q05_ovaries.csv", row.names=FALSE)

ids_ov <- strsplit(as.character(ov.long$variable), split=".",fixed=TRUE)
ov.long$sampl <-unlist(lapply(ids_ov, function(x) x[2]))

ggplot(ov.long, aes(sampl, gname)) +
  geom_tile(aes(fill = value), colour = "white") + 
  scale_fill_gradient(low = "white", high = "steelblue") +
  scale_y_continuous(expand = c(0, 0)) +
  theme(#axis.text.x = element_blank(),
        axis.text.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x  = element_text(angle=90, vjust=0.5),
        axis.title.y = element_blank())
ggsave(filename = "../plots/Heatmap_ovaries.pdf",width = 5,height=5)

```

# Potential candidate genes under past QTL

```{r}
# Lifespan difference (C vs DR) QTL in Stanley et al (2017)
# Position coordinates FB6: 3L 17.35 - 3R 10.74 Mb
# 3L: 10.76 Mbp from 3L:17,350,000..28,110,227
# 3R: 10.74 Mbp from 3R:1..10,740,000

# coordinates across
low3L <- 17350000
high3R <- 10740000

# read in CSVs with converted gene ids (DEGs @ qval=0.05)
q05_heads <- read.csv(file="../processed/results/bg_results_tissues/DEs_sig_q05_IDsConverted_heads.csv", header=TRUE, stringsAsFactors = FALSE)

q05_soma <- read.csv(file="../processed/results/bg_results_tissues/DEs_sig_q05_IDsConverted_soma.csv", header=TRUE, stringsAsFactors = FALSE)

q05_ovaries <- read.csv(file="../processed/results/bg_results_tissues/DEs_sig_q05_IDsConverted_ovaries.csv", header=TRUE, stringsAsFactors = FALSE)

# heads
qh <- q05_heads%>%
  select(X.SUBMITTED.ID,ANNOTATION_SYMBOL,LOCATION_ARM,LOCATION_MAX,LOCATION_MIN,SYMBOL)

sub.3L_hds <- subset(qh, qh$LOCATION_ARM=="3L")
sub.3R_hds <- subset(qh, qh$LOCATION_ARM=="3R")

qtl3l_heads <- subset(sub.3L_hds, sub.3L_hds$LOCATION_MIN >= low3L)
qtl3r_heads <- subset(sub.3R_hds, sub.3R_hds$LOCATION_MIN <= high3R)
#99 genes total!!!

qtl3l_heads
qtl3r_heads
qtl_heads <- rbind(qtl3l_heads,qtl3r_heads)
write.csv(qtl_ova, 
          "../processed/results/bg_results_tissues/qtlDiff_genes_heads.csv", row.names=FALSE)


# soma
qs <- q05_soma%>%
  select(X.SUBMITTED.ID,ANNOTATION_SYMBOL,LOCATION_ARM,LOCATION_MAX,LOCATION_MIN,SYMBOL)

sub.3L_som <- subset(qs, qs$LOCATION_ARM=="3L")
sub.3R_som <- subset(qs, qs$LOCATION_ARM=="3R")

qtl3l_soma <- subset(sub.3L_som, sub.3L_som$LOCATION_MIN >= low3L)
qtl3r_soma <- subset(sub.3R_som, sub.3R_som$LOCATION_MIN <= high3R)

qtl3l_soma
qtl3r_soma
qtl_soma <- rbind(qtl3l_soma,qtl3r_soma)
write.csv(qtl_soma, 
          "../processed/results/bg_results_tissues/qtlDiff_genes_soma.csv", row.names=FALSE)


# ovaries
qo <- q05_ovaries%>%
  select(X.SUBMITTED.ID,ANNOTATION_SYMBOL,LOCATION_ARM,LOCATION_MAX,LOCATION_MIN,SYMBOL)

sub.3L_ova <- subset(qo, qo$LOCATION_ARM=="3L")
sub.3R_ova <- subset(qo, qo$LOCATION_ARM=="3R")

qtl3l_ova <- subset(sub.3L_ova, sub.3L_ova$LOCATION_MIN >= low3L)
qtl3r_ova <- subset(sub.3R_ova, sub.3R_ova$LOCATION_MIN <= high3R)

qtl3l_ova
qtl3r_ova
qtl_ova <- rbind(qtl3l_ova,qtl3r_ova)
write.csv(qtl_ova, 
          "../processed/results/bg_results_tissues/qtlDiff_genes_ovaries.csv", row.names=FALSE)

# combine the 3 data frames
qtl_heads$coder <- "heads"
qtl_soma$coder <- "soma"
qtl_ova$coder <- "ovaries"
qtl.genes <- rbind(qtl_heads, qtl_soma, qtl_ova)  

# Find the rows which have duplicates in a different group.
n_occur <- data.frame(table(qtl.genes$SYMBOL))
n_occur[n_occur$Freq > 1,]
qtl.genes_tissues <- qtl.genes[qtl.genes$SYMBOL %in% n_occur$Var1[n_occur$Freq > 1],]

write.csv(qtl.genes_tissues, 
          "../processed/results/bg_results_tissues/qtlDiff_genes_shared_between_tissues.csv", row.names=FALSE)
```



