---
title: "Set_up_array_jobs"
author: "Enoch Ng'oma"
date: "4/24/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Step 1 Align

```{r}
ll<-character(length=0)

runs<-c(1,2,3)
for(run in runs)
{
  
cat("", file=paste("S01_Align_cmd_run",run,".txt",sep=""))
  
ffs <- list.files(paste("/home/kingeg/Projects/RNA_SEQ_BasePop/Data/run",run,"/",sep=""), pattern="fastq.gz$")

for(ii in 1:length(ffs))
{
st.s <- "hisat2 --dta -q -x indexes/bdgp6_tran/genome_tran -U " 

st.p <- paste("/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/run",run,"/",sep="")

outf <- paste("/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotsams/",strsplit(ffs[ii], ".", fixed=TRUE)[[1]][1],"_run",run,sep="")

cat(paste(st.s, st.p, ffs[ii], " -S ", outf,".sam",sep=""),"\n", file=paste("S01_Align_cmd_run",run,".txt",sep=""),append=TRUE)

ll<- c(ll,paste(strsplit(ffs[ii], ".", fixed=TRUE)[[1]][1],"_run",run,sep=""))

}

}


```

## Samtools sort

```{r}
cat("", file="S02_samtools_all.txt")
for(ii in ll)
{
   cat(paste("samtools sort -@ 8 -o /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/",ii,".bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotsams/", ii,".sam ",sep=""), "\n", file=paste("S02_samtools_all",".txt",sep=""),append=TRUE)

}
```


## Samtools merge: combine transcripts from each run

```{r}
set <- data.frame('code'=unlist(lapply(strsplit(ll, split='_S',fixed=TRUE), function(x) x[1])), 'sample' = ll, stringsAsFactors=FALSE)

un.set <- unique(set$code)

cat("", file="S03_samtools_merge.txt")


for(ii in un.set)
{
  samp.set <- subset(set, code==ii)
   cat(paste("samtools merge -r  /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/",ii,"_merged.bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[1],".bam ","/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[2],".bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[3],".bam",sep=""), "\n", file=paste("S03_samtools_merge",".txt",sep=""),append=TRUE)

}
```

# StringTie: Assemble and quantify expressed genes and transcripts

```{r}
# Assemble transcripts for each sample

cat("", file="S04_stringtie_assemble.txt")
cat("", file="assembled_file_list.txt")

indir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/" 
outdir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotgtfs/" 

base.cmd <- "stringtie -p 8 -G /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/genes/dmel-all-r6.18.gtf -o "

for(ff in un.set) 
{
  samp.set <- paste(base.cmd,outdir,ff,"_assembled.gtf ",indir,ff,"_merged.bam",sep="")
  cat(samp.set,"\n", file="S04_stringtie_assemble.txt",append=TRUE)
   
# For stringtie merge 
  cat(paste(outdir,ff,"_assembled.gtf",sep=""), "\n",file="mergelist.txt",append=TRUE)
}
```

# Compare transcripts to reference genome (optional)

```{r}
# gffcompare
# step skipped
```

# StringTie: transcript abundances and table of counts

```{r}
cat("", file="S06_Abundances.txt")


indir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/" 
outdir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/ballG/" 


base.cmd <- "stringtie -e -B -p 8 -G /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotgtfs/stringtie_merged.gtf -o "

for(ff in un.set) 
{
  cmd.cut <- paste(base.cmd,outdir,ff,"_ballgown.gtf ", indir,ff,"_merged.bam",sep="")
  cat(cmd.cut,"\n", file="S06_abundances.txt",append=TRUE)
   
}

```

# Differential expression analysis protocol

```{r}
# Make phenotype data file (csv)

# ref. above where we make un.set

descr.samp <- data.frame('id'=character(length=length(un.set)),
                        'treatment'=character(length=length(un.set)),
                        'replicate'=character(length=length(un.set)),stringsAsFactors = FALSE)

descr.samp$id <- sample.ids 
descr.samp$treatment <- substr(descr.samp$id, 1,(nchar(descr.samp$id)-2))
descr.samp$replicate <- substr(descr.samp$id, (nchar(descr.samp$id)),(nchar(descr.samp$id)))

write.csv(descr.samp, file="describe_sample.csv", row.names=FALSE, quote=FALSE)

#Ballgown step in the tuxedo pipeline,  
#load R version R version 3.2.2
#load R packagees:
library(ballgown)
library(genefilter)
library(dplyr)
library(devtools)

```



