---
title: "Set_up_array_jobs"
author: "Enoch Ng'oma"
date: "4/24/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
```

## Step 1 Align

```{r}
ll<-character(length=0)

runs<-c(1,2,3)
for(run in runs)
{
  
cat("", file=paste("S01_Align_cmd_run",run,".txt",sep=""))
  
ffs <- list.files(paste("/home/kingeg/Projects/RNA_SEQ_BasePop/Data/run",run,"/",sep=""), pattern="fastq.gz$")

for(ii in 1:length(ffs))
{
st.s <- "hisat2 --dta -q -x indexes/bdgp6_tran/genome_tran -U " 

st.p <- paste("/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/run",run,"/",sep="")

outf <- paste("/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotsams/",strsplit(ffs[ii], ".", fixed=TRUE)[[1]][1],"_run",run,sep="")

cat(paste(st.s, st.p, ffs[ii], " -S ", outf,".sam",sep=""),"\n", file=paste("S01_Align_cmd_run",run,".txt",sep=""),append=TRUE)

ll<- c(ll,paste(strsplit(ffs[ii], ".", fixed=TRUE)[[1]][1],"_run",run,sep=""))

}

}
```

## Samtools sort

```{r}
cat("", file="S02_samtools_all.txt")
for(ii in ll)
{
   cat(paste("samtools sort -@ 8 -o /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/",ii,".bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotsams/", ii,".sam ",sep=""), "\n", file=paste("S02_samtools_all",".txt",sep=""),append=TRUE)

}
```


## Samtools merge: combine transcripts from each run

```{r}
set <- data.frame('code'=unlist(lapply(strsplit(ll, split='_S',fixed=TRUE), function(x) x[1])), 'sample' = ll, stringsAsFactors=FALSE)

un.set <- unique(set$code)

cat("", file="S03_samtools_merge.txt")

for(ii in un.set)
{
  samp.set <- subset(set, code==ii)
   cat(paste("samtools merge -r  /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/",ii,"_merged.bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[1],".bam ","/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[2],".bam ", "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/", samp.set$sample[3],".bam",sep=""), "\n", file=paste("S03_samtools_merge",".txt",sep=""),append=TRUE)

}
```

# StringTie: Assemble and quantify expressed genes and transcripts

```{r}
# Assemble transcripts for each sample

cat("", file="S04_stringtie_assemble.txt")
cat("", file="assembled_file_list.txt")

indir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/" 
outdir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotgtfs/" 

base.cmd <- "stringtie -p 8 -G /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/genes/dmel-all-r6.18.gtf -o "

for(ff in un.set) 
{
  samp.set <- paste(base.cmd,outdir,ff,"_assembled.gtf ",indir,ff,"_merged.bam",sep="")
  cat(samp.set,"\n", file="S04_stringtie_assemble.txt",append=TRUE)
   
# For stringtie merge 
  cat(paste(outdir,ff,"_assembled.gtf",sep=""), "\n",file="mergelist.txt",append=TRUE)
}
```

# Compare transcripts to reference genome (optional)

```{r}
# gffcompare
# step skipped
```

# StringTie: transcript abundances and table of counts

```{r}
cat("", file="S06_abundances.txt")


indir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotbams/" 
outdir <- "/group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/ballG/" 


base.cmd <- "stringtie -e -B -p 8 -G /group/kinglab/enoch/MyGithub/BasePop_RNAseq/base_pop/processed/dotgtfs/stringtie_merged.gtf -o "

for(ff in un.set) 
{
  cmd.cut <- paste(base.cmd,outdir,ff,"/",ff,"_ballgown.gtf ", indir,ff,"_merged.bam",sep="")
  cat(cmd.cut,"\n", file="S06_abundances.txt",append=TRUE)
   
}
```

# Create a csv containing sample ids

```{r}
# Make phenotype data file (csv)

# ref. above where we make un.set

idd <- as.data.frame(un.set)
colnames(idd)[colnames(idd)=="un.set"] <- "id"

ids <- idd %>% 
  mutate(id = id,
         treatment = str_split(id, "-", simplify = TRUE)[, 1])%>%
  mutate(replic = str_split(id, "_", simplify = TRUE)[, 1],
         tissue = str_split(id, "_", simplify = TRUE)[, 2]) %>%
  select(-replic)

write.csv(ids, file="describe_samples.csv", row.names=FALSE, quote=FALSE)
```

# Differential expression analysis protocol

```{r}
#R
#load R version R version 3.2.2
#load R packagees:
library(ballgown)
library(genefilter)
library(tidyverse)
library(devtools)

phenDat <- read.csv("describe_samples.csv")
```

```{r}
bg_ballG <- ballgown(dataDir = "../processed/ballG", samplePattern = "", pData = phenDat)
```

# Filter to remove low-abundance genes

```{r}
bg_ballG_filt <- subset(bg_ballG, "rowVars(texpr(bg_ballG)) >1", genomesubset=TRUE)
```

# Identify transcripts that show significant differences between groups

```{r}
results_transcripts <- stattest(bg_ballG_filt, feature="transcript",covariate="treatment",adjustvars = c("tissue"), meas="FPKM")
```

# Identify transcripts that show significant differences between groups

```{r}
results_genes <- stattest(bg_ballG_filt, feature="gene", covariate="treatment", adjustvars = c("tissue"), meas="FPKM")
```

# Add gene names and gene IDs to the results_transcripts data frame

```{r}
results_transcripts <- data.frame(geneNames=ballgown::geneNames(bg_ballG_filt), geneIDs=ballgown::geneIDs(bg_ballG_filt), results_transcripts)
```

# Sort results from smallest P value to largest

```{r}
results_transcripts = arrange(results_transcripts,pval) 

results_genes = arrange(results_genes,pval)
```

# Write results to csv

```{r}
write.csv(results_transcripts, "bg_ballG_transcript_results.csv", row.names=FALSE)

write.csv(results_genes, "bg_ballG_gene_results.csv", row.names=FALSE)
```

# Identify transcripts and genes with a q value <0.05

```{r}
subset(results_transcripts,results_transcripts$qval<0.05)

subset(results_genes,results_genes$qval<0.05)
```

# Plotting

```{r}
#Set plot settings
tropical= c('darkorange', 'dodgerblue', 'hotpink', 'limegreen', 'yellow')
palette(tropical)
```

# Show the distribution of gene abundances (measured as FPKM values) across samples, colored by treatment

```{r}
fpkm = texpr(bg_ballG,meas="FPKM")

fpkm = log2(fpkm+1) 

boxplot(fpkm,col=as.numeric(phenDat$treatment),las=2,ylab='log2(FPKM+1)')
```

# Make plots of individual transcripts across samples

```{r}
ballgown::transcriptNames(bg_ballG)[249] ## 12

plot(fpkm[249,] ~ phenDat$treatment, border=c(1,2), main=paste(ballgown::geneNames(bg_ballG)[249],' : ', ballgown::transcriptNames(bg_ballG)[249]),pch=19, xlab="Treatment", ylab='log2(FPKM+1)')
points(fpkm[249,] ~ jitter(as.numeric(phenDat$treatment)), col=as.numeric(phenDat$treatment))
```

# Plot the structure and expression levels in a sample of all transcripts that share the same gene locus

```{r}
plotTranscripts(ballgown::geneIDs(bg_ballG)[128], bg_ballG, main=c('Gene blah-blah in sample DR-4_H'), sample=c('DR-4_H'))
```

# We can also plot the average expression levels for all transcripts of a gene within different groups using the plotMeans function

```{r}
plotMeans('MSTRG.100', bg_ballG_filt,groupvar="treatment",legend=FALSE)
```

